{
  "cells": [
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "view-in-github",
        "colab_type": "text"
      },
      "source": [
        "<a href=\"https://colab.research.google.com/github/azamatarmanuly99/docs/blob/main/notebooks/12.02-RosettaAntibodyDesign-RAbD.ipynb\" target=\"_parent\"><img src=\"https://colab.research.google.com/assets/colab-badge.svg\" alt=\"Open In Colab\"/></a>"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "vHMaQwgS2LMj"
      },
      "source": [
        "<!--NOTEBOOK_HEADER-->\n",
        "*This notebook contains material from [PyRosetta](https://RosettaCommons.github.io/PyRosetta.notebooks);\n",
        "content is available [on Github](https://github.com/RosettaCommons/PyRosetta.notebooks.git).*"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "eNmNWrHL2LMo"
      },
      "source": [
        "<!--NAVIGATION-->\n",
        "< [RosettaAntibody Framework](http://nbviewer.jupyter.org/github/RosettaCommons/PyRosetta.notebooks/blob/master/notebooks/12.01-RosettaAntibody-Framework-and-SimpleMetrics.ipynb) | [Contents](toc.ipynb) | [Index](index.ipynb) | [RosettaCarbohydrates](http://nbviewer.jupyter.org/github/RosettaCommons/PyRosetta.notebooks/blob/master/notebooks/13.00-RosettaCarbohydrates-Working-with-Glycans.ipynb) ><p><a href=\"https://colab.research.google.com/github/RosettaCommons/PyRosetta.notebooks/blob/master/notebooks/12.02-RosettaAntibodyDesign-RAbD.ipynb\"><img align=\"left\" src=\"https://colab.research.google.com/assets/colab-badge.svg\" alt=\"Open in Colab\" title=\"Open in Google Colaboratory\"></a>"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "qLsL9erL2LMp"
      },
      "source": [
        "# RosettaAntibodyDesign\n",
        "## Notes\n",
        "This tutorial will walk you through how to use `RosettaAntibodyDesign` in PyRosetta.  You should also go through the parellel distribution workshop as you will most likely need to create many decoys for some of these design tasks. Note that we are using the XML interface to the code here for simplicity (and because I had a C++ workshop I am converting - truth be told).  The code-level interface is as robust as the XML - but will require more knowledge use.  You are welcome to play around with it - all functions have descriptions and all options are possible to change through code.\n",
        "\n",
        "Grab a coffee, take a breath, and lets learn how to design some antibodies!\n",
        "\n",
        "## Citation\n",
        "\n",
        "\n",
        "[Rosetta Antibody Design (RAbD): A General Framework for Computational Antibody Design, PLOS Computational Biology, 4/27/2018](http://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1006112)\n",
        "\n",
        "Jared Adolf-Bryfogle, Oleks Kalyuzhniy, Michael Kubitz, Brian D. Weitzner, Xiaozhen Hu, Yumiko Adachi, William R. Schief, Roland L. Dunbrack Jr.\n",
        "\n",
        "## Manual\n",
        "The full RAbD manual can be found here: https://www.rosettacommons.org/docs/latest/application_documentation/antibody/RosettaAntibodyDesign\n"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "3q3Z9wUH2LMp"
      },
      "source": [
        "# Overview\n",
        "\n",
        "__RosettaAntibodyDesign (RAbD)__ is a generalized framework for the design of antibodies, in which a user can easily tailor the run to their project needs.  __The algorithm is meant to sample the diverse sequence, structure, and binding space of an antibody-antigen complex.__  An app is available, and all components can be used within RosettaScripts for easy scripting of antibody design and incorporation into other Rosetta protocols.\n",
        "\n",
        "The framework is based on rigorous bioinformatic analysis and rooted very much on our [recent clustering](https://www.ncbi.nlm.nih.gov/pubmed/21035459) of antibody CDR regions.  It uses the __North/Dunbrack CDR definition__ as outlined in the North/Dunbrack clustering paper. A new clustering paper will be out in the next year, and this new analysis will be incorporated into RAbD.\n",
        "\n",
        "The supplemental methods section of the published paper has all details of the RosettaAntibodyDesign method.  This manual serves to get you started running RAbD in typical use fashions.\n",
        "\n",
        "\n",
        "# Algorithm\n",
        "  \n",
        "Broadly, the RAbD protocol consists of alternating outer and inner Monte Carlo cycles. Each outer cycle consists of randomly choosing a CDR (L1, L2, etc.) from those CDRs set to design, randomly choosing a cluster and then a structure from that cluster from the database according to the input instructions, and optionally grafting that CDR's structure onto the antibody framework in place of the existing CDR (__GraftDesign__). The program then performs N rounds of the inner cycle, consisting of sequence design (__SeqDesign__) using cluster-based sequence profiles and structural constraints, energy minimization, and optional docking. Each inner cycle structurally optimizes the backbone and repacks side chains of the CDR chosen in the outer cycle as well as optional neighbors in order to optimize interactions of the CDR with the antigen and other CDRs.\n",
        "\n",
        "__Backbone dihedral angle (CircularHarmonic) constraints__ derived from the cluster data are applied to each CDR to limit deleterious structural perturbations. Amino acid changes are typically sampled from __profiles derived for each CDR cluster in PyIgClassify__. Conservative amino acid substitutions (according to the BLOSUM62 substitution matrix) may be performed when too few sequences are available to produce a profile (e.g., for H3). After each inner cycle is completed, the new sequence and structure are accepted according to the Metropolis Monte Carlo criterion. After N rounds within the inner cycle, the program returns to the outer cycle, at which point the energy of the resulting design is compared to the previous design in the outer cycle. The new design is accepted or rejected according to the Monte Carlo criterion.\n",
        "\n",
        "If optimizing the antibody-antigen orientation during the design (dock), SiteConstraints are automatically used to keep the CDRs (paratope) facing the antigen surface.  These are termed __ParatopeSiteConstraints__.   Optionally, one can enable constraints that keep the paratope of the antibody around a target epitope (antigen binding site).  These are called __ParatopeEpitopeSiteConstraints__ as the constraints are between the paratope and the epitope. The epitope is automatically determined as the interface residues around the paratope on input into the program, however, any residue(s) can be set as the epitope to limit unwanted movement and sampling of the antibody.  See the examples and options below.\n",
        "\n",
        "More detail on the algorithm can be found in the published paper.\n",
        "\n",
        "\n",
        "# General Setup and Inputs\n",
        "\n",
        "1.  Antibody Design Database\n",
        "\n",
        "\tThis app requires the Rosetta Antibody Design Database.  A database of antibodies from the original North Clustering paper is included in Rosetta and is used as the default .  An updated database (which is currently updated bi-yearly) can be downloaded here: <http://dunbrack2.fccc.edu/PyIgClassify/>.  \n",
        "\n",
        "\tFor C++, It should be placed in `Rosetta/main/database/sampling/antibodies/`. For PyRosetta, use the cmd-line option `antibody_database` and set it to the full path of the downloaded database within the `init()` function as you have done in the past. It is recommended to use this up-to-date database for production runs. For this tutorial, we will use the database within Rosetta.\n",
        "\n",
        "\n",
        "2.  Starting Structure\n",
        "\n",
        "\tThe protocol begins with the three-dimensional structure of an antibody-antigen complex. Designs should start with an antibody bound to a target antigen (however optimizing just the antibody without the complex is also possible).  Camelid antibodies are fully supported.  This structure may be an experimental structure of an existing antibody in complex with its antigen, a predicted structure of an existing antibody docked computationally to its antigen, or even the best scoring result of low-resolution docking a large number of unrelated antibodies to a desired epitope on the structure of a target antigen as a prelude to de novo design.\n",
        "\n",
        "\tThe program CAN computationally design an antibody to anywhere on the target protein, but it is recommended to place the antibody at the target epitope.  It is beyond the scope of this program to determine potential epitopes for binding, however servers and programs exist to predict these. Automatic SiteConstraints can be used to further limit the design to target regions.\n",
        "\n",
        "\n",
        "3.  Model Numbering and Light Chain identification\n",
        "\n",
        "\tThe input PDB file must be renumbered to the AHo Scheme and the light chain gene must be identified.  This can be done through the [PyIgClassify Server](http://dunbrack2.fccc.edu/pyigclassify/).\n",
        "\n",
        "\tOn input into the program, Rosetta assigns our CDR clusters using the same methodology as PyIgClassify. The RosettaAntibodyDesign protocol is then driven by a set of command-line options and a set of design instructions provided as an input file that controls which CDR(s) are designed and how. Details and example command lines and instruction files are provided below.\n",
        "\n",
        "\tThe gene of the light chain should always be set on the command-line using the option `-light_chain`, these are either lamda or kappa.  PyIgClassify will identify the gene of the light chain.\n",
        "\n",
        "\tFor this tutorial, the starting antibody is renumbered for you.\n",
        "\n",
        "4.  Notes for Tutorial Shortening\n",
        "\n",
        "\tAlways set the option, `-outer_cycle_rounds` to 5 in order to run these examples quickly.  The default is 25.  We include this in our common options file that is read in by Rosetta at the start. We will only be outputting a single structure, but typical use of the protocol is with default settings of `-outer_cycle_rounds` and an `nstruct` of at least 1000, with 5000-10000 recommended for jobs that are doing a lot of grafting.  For De-novo design runs, one would want to go even higher. Note that the Docking stage increases runtime significantly as well.\n",
        "\n",
        "\tThe total number of rounds is outer_cycle_rounds * nstruct.  \n",
        "\n",
        "\n",
        "5.  General Notes\n",
        "\n",
        "\t\t\tsetenv PATH ${PATH}:${HOME}/rosetta_workshop/rosetta/main/source/tools\n",
        "\n",
        "\tWe will be using JSON output of the scorefile, as this is much easier to work with in python and pandas.\n",
        "\tWe use the option `-scorefile_format json`\n",
        "\n",
        "\tAll of our common options for the tutorial are in the common file that you will copy to your working directory.\n",
        "\tRosetta/PyRosetta will look for this file in your working directory or your home folder in the directory `$HOME/.rosetta/flags`.\n",
        "\tSee this page for more info on using rosetta with custom config files: <https://www.rosettacommons.org/docs/latest/rosetta_basics/running-rosetta-with-options#common-options-and-default-user-configuration>\n",
        "\n",
        "\tAll tutorials have generated output in `outputs/rabd` and their approximate time to finish on a single (core i7) processor.\n",
        "\n"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 1,
      "metadata": {
        "id": "JJOrVjcf2LMq",
        "outputId": "87e33dcf-8a9f-460a-ab50-13d59eee0206",
        "colab": {
          "base_uri": "https://localhost:8080/"
        }
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Collecting pyrosettacolabsetup\n",
            "  Downloading pyrosettacolabsetup-1.0.9-py3-none-any.whl (4.9 kB)\n",
            "Installing collected packages: pyrosettacolabsetup\n",
            "Successfully installed pyrosettacolabsetup-1.0.9\n",
            "Mounted at /content/google_drive\n",
            "\n",
            "Note that USE OF PyRosetta FOR COMMERCIAL PURPOSES REQUIRE PURCHASE OF A LICENSE.\n",
            "See https://github.com/RosettaCommons/rosetta/blob/main/LICENSE.md or email license@uw.edu for details.\n",
            "\n",
            "Looking for compatible PyRosetta wheel file at google-drive/PyRosetta/colab.bin//wheels...\n",
            "Found compatible wheel: /content/google_drive/MyDrive/PyRosetta/colab.bin/wheels//content/google_drive/MyDrive/PyRosetta/colab.bin/wheels/pyrosetta-2024.1+release.00b79147e63-cp310-cp310-linux_x86_64.whl\n",
            "\n",
            "\n",
            "PyRosetta-4 2023 [Rosetta PyRosetta4.MinSizeRel.python310.ubuntu 2024.01+release.00b79147e63be743438188f93a3f069ca75106d6 2023-12-25T16:35:48] retrieved from: http://www.pyrosetta.org\n",
            "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n",
            "core.init: Checking for fconfig files in pwd and ./rosetta/flags\n",
            "core.init: Rosetta version: PyRosetta4.MinSizeRel.python310.ubuntu r366 2024.01+release.00b79147e63 00b79147e63be743438188f93a3f069ca75106d6 http://www.pyrosetta.org 2023-12-25T16:35:48\n",
            "core.init: command: PyRosetta -ex1 -ex2aro -database /usr/local/lib/python3.10/dist-packages/pyrosetta/database\n",
            "basic.random.init_random_generator: 'RNG device' seed mode, using '/dev/urandom', seed=-735151591 seed_offset=0 real_seed=-735151591\n",
            "basic.random.init_random_generator: RandomGenerator:init: Normal mode, seed=-735151591 RG_type=mt19937\n"
          ]
        }
      ],
      "source": [
        "!pip install pyrosettacolabsetup\n",
        "import pyrosettacolabsetup; pyrosettacolabsetup.install_pyrosetta()\n",
        "import pyrosetta; pyrosetta.init()\n"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "W8zLBiAN2LMr"
      },
      "source": [
        "**Make sure you are in the directory with the pdb files:**\n",
        "\n",
        "`cd google_drive/MyDrive/student-notebooks/`"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 2,
      "metadata": {
        "id": "UlhYz8cp2LMs"
      },
      "outputs": [],
      "source": [
        "from typing import *\n",
        "import pandas\n",
        "from pathlib import Path\n",
        "import json\n",
        "import re\n",
        "\n",
        "#Functions we will be using. I like to collect any extra functions at the top of my notebook.\n",
        "def load_json_scorefile(file_path: Path, sort_by: str=\"dG_separated\") -> pandas.DataFrame:\n",
        "        \"\"\"\n",
        "        Read scorefile lines as a dataframe, sorted by total_score with Nan's correctly replaced.\n",
        "        \"\"\"\n",
        "\n",
        "        local_lines = open(file_path, 'r').readlines()\n",
        "        decoys=[]\n",
        "        for line in local_lines:\n",
        "                o = json.loads(line.replace(\"nan\", \"NaN\"))\n",
        "                # print o[self.decoy_field_name]\n",
        "                # print repr(o)\n",
        "                decoys.append(o)\n",
        "        local_df = pandas.DataFrame.from_dict(decoys)\n",
        "        local_df = local_df.infer_objects()\n",
        "        # df.to_csv(\"debugging.csv\", sep=\",\")\n",
        "\n",
        "        local_df = local_df.sort_values(sort_by, ascending=True)\n",
        "\n",
        "        return local_df\n",
        "\n",
        "def drop_cluster_columns(local_df: pandas.DataFrame, keep_cdrs: List[str]=None) -> pandas.DataFrame:\n",
        "        \"\"\"\n",
        "        Drop cluster columns that RAbD outputs to make it easier to work with the dataframe.\n",
        "        \"\"\"\n",
        "        to_drop = []\n",
        "        for column in local_df.columns:\n",
        "            if re.search(\"cdr_cluster\", column):\n",
        "                skip=False\n",
        "                if (keep_cdrs):\n",
        "                    for cdr in keep_cdrs:\n",
        "                        if re.search(cdr, column):\n",
        "                            skip=True\n",
        "                            break\n",
        "                if not skip:\n",
        "                    to_drop.append(column)\n",
        "        return local_df.drop(columns=to_drop)"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "uLxGQivz2LMs"
      },
      "source": [
        "## Imports"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 3,
      "metadata": {
        "id": "c1AwpzG72LMt",
        "outputId": "3b97f8e0-9500-4e0f-ecdb-101d17b4a068",
        "colab": {
          "base_uri": "https://localhost:8080/"
        }
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "<ipython-input-3-b52d39fe3732>:8: UserWarning: Import of 'rosetta' as a top-level module is deprecated and may be removed in 2018, import via 'pyrosetta.rosetta'.\n",
            "  from rosetta.protocols.rosetta_scripts import *\n"
          ]
        }
      ],
      "source": [
        "#Python\n",
        "from pyrosetta import *\n",
        "from pyrosetta.rosetta import *\n",
        "from pyrosetta.teaching import *\n",
        "import os\n",
        "\n",
        "#Core Includes\n",
        "from rosetta.protocols.rosetta_scripts import *\n",
        "from rosetta.protocols.antibody import *\n",
        "from rosetta.protocols.antibody.design import *\n",
        "from rosetta.utility import *"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "ePxCsDFq2LMt"
      },
      "source": [
        "## Intitlialization\n",
        "Since we are sharing the working directory with all other notebooks, instead of using the common-configuration we spoke about in the introduction, we will be using the flags file located in the inputs directory."
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 13,
      "metadata": {
        "id": "ZX1llPPx2LMu",
        "outputId": "51f554a2-2557-4763-9165-16aa62aba594",
        "colab": {
          "base_uri": "https://localhost:8080/"
        }
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "PyRosetta-4 2023 [Rosetta PyRosetta4.MinSizeRel.python310.ubuntu 2024.01+release.00b79147e63be743438188f93a3f069ca75106d6 2023-12-25T16:35:48] retrieved from: http://www.pyrosetta.org\n",
            "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n",
            "core.init: Rosetta version: PyRosetta4.MinSizeRel.python310.ubuntu r366 2024.01+release.00b79147e63 00b79147e63be743438188f93a3f069ca75106d6 http://www.pyrosetta.org 2023-12-25T16:35:48\n",
            "core.init: command: PyRosetta -no_fconfig @rabd/common.txt -database /usr/local/lib/python3.10/dist-packages/pyrosetta/database\n",
            "basic.random.init_random_generator: 'RNG device' seed mode, using '/dev/urandom', seed=679289324 seed_offset=0 real_seed=679289324\n",
            "basic.random.init_random_generator: RandomGenerator:init: Normal mode, seed=679289324 RG_type=mt19937\n"
          ]
        }
      ],
      "source": [
        "init('-no_fconfig @rabd/common.txt')"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 14,
      "metadata": {
        "id": "ZFVy8z_T2LMu",
        "outputId": "52271cca-407f-48e1-b58a-634d1ef6cc61",
        "colab": {
          "base_uri": "https://localhost:8080/"
        }
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "core.chemical.GlobalResidueTypeSet: Finished initializing fa_standard residue type set.  Created 985 residue types\n",
            "core.chemical.GlobalResidueTypeSet: Total time to initialize 1.0453 seconds.\n",
            "core.import_pose.import_pose: File 'rabd/my_ab.pdb' automatically determined to be of type PDB\n",
            "core.io.pdb.pdb_reader: Parsing 993 .pdb records with unknown format to search for Rosetta-specific comments.\n",
            "core.conformation.Conformation: Found disulfide between residues 771 845\n",
            "core.conformation.Conformation: Found disulfide between residues 891 956\n"
          ]
        }
      ],
      "source": [
        "#Import a pose\n",
        "pose = pose_from_pdb(\"rabd/my_ab.pdb\")\n",
        "original_pose = pose.clone()"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "zBbq8h_P2LMu"
      },
      "source": [
        "# Tutorial\n",
        "\n",
        "\n",
        "## Tutorial A: General Design\n",
        "\n",
        "In many of these examples, we will use the xml interface to PyRosetta for simplicity with the AntibodyDesignMover - which is  the actual C++ application as a mover. <https://www.rosettacommons.org/docs/latest/scripting_documentation/RosettaScripts/Movers/movers_pages/antibodies/AntibodyDesignMover>\n",
        "\n",
        "Lets copy the files we need first:\n",
        "\n",
        "\t\tcp ../inputs/rabd/color_cdrs.pml .\n",
        "\t\tcp ../inputs/rabd/rabd.xml .\n",
        "\n",
        "You are starting design on a new antibody that is not bound to the antigen in the crystal. This is difficult and risky, but we review how one could go about this anyway.  We start by selecting a framework.  Here, we use the trastuzumab framework as it expresses well, is thermodynamically stable with a Tm of 69.5 degrees, and has been shown repeatedly that it can tolerate CDRs of different sequence and structure.  Note that the energy of the complex is high as we are starting from a manual placement of the antibody to antigen.  If we relax the structure too much, we will fall into an energy well that is hard to escape without significant sampling.\n",
        "\n",
        "We are using an arbitrary protein at an arbitrary site for design.  The PDB of our target is 1qaw.  1qaw is an oligomer of the TRP RNA-Binding Attenuation Protein from Bacillus Stearothermophilus.  It is usually a monomer/dimer, but at its multimeric interface is a tryptophan residue by itself.  \n",
        "\n",
        "It's a beautiful protein, with a cool mechanism.  We will attempt to build an antibody to bind to two subunits to stabilize the dimeric state of the complex in the absence of TRP. Note that denovo design currently takes a large amount of processing power. Each tutorial below is more complex than the one before it.  The examples we have for this tutorial are short runs to show HOW it can be done, but more outer_cycle_rounds and nstruct would produce far better models than the ones you will see here - as we will need to sample the relative orientation of the antibody-antigen complex through docking, the CDR clusters and lengths, the internal backbone degrees of freedom of the CDRs, as well as the sequence of the CDRs and possibly the framework.  As you can tell, just the sampling problem alone is difficult. However, this will give you a basis for using RAbD on your own.\n"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "IuqULqVR2LMv"
      },
      "source": [
        "### Tut A1. Sequence Design\n",
        "\n",
        "Using the application is as simple as setting the `-seq_design_cdrs` option.\n",
        "This simply designs the CDRs of the heavy chain using cdr profiles if they exist for those clusters during flexible-backbone design.  If the clusters do not exist (as is the case for H3 at the moment), we use conservative design by default.  Note that InterfaceAnalyzer is run on each output decoy in the RAbD mover. Note that you can also set `light_chain` on the command line if you are only working on a single PDB through the rosetta run.\n",
        "\n",
        "\n",
        "\t<AntibodyDesignMover name=\"RAbD\" seq_design_cdrs=\"L1,L3\" light_chain=\"kappa\"/>\n",
        "            \n",
        "This will take a about a minute (50 seconds on my laptop).  Output structures and scores are in `outputs/rabd` if you wish to copy them over - these include 4 more structures."
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 15,
      "metadata": {
        "id": "OPuZKzXf2LMv",
        "outputId": "014ba0a6-2f10-4bd3-d419-c9bcde5e62ae",
        "colab": {
          "base_uri": "https://localhost:8080/"
        }
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "protocols.rosetta_scripts.RosettaScriptsParser: Validating input script...\n",
            "protocols.rosetta_scripts.RosettaScriptsSchemaValidator: Generating XML Schema for rosetta_scripts...\n",
            "protocols.rosetta_scripts.RosettaScriptsSchemaValidator: ...done\n",
            "protocols.rosetta_scripts.RosettaScriptsSchemaValidator: Initializing schema validator...\n",
            "protocols.rosetta_scripts.RosettaScriptsSchemaValidator: ...done\n",
            "protocols.rosetta_scripts.RosettaScriptsParser: ...done\n",
            "protocols.rosetta_scripts.RosettaScriptsParser: Parsed script:\n",
            "<ROSETTASCRIPTS>\n",
            "\t<MOVERS>\n",
            "\t\t<AntibodyDesignMover light_chain=\"kappa\" name=\"RAbD\" seq_design_cdrs=\"L1,L3\"/>\n",
            "\t</MOVERS>\n",
            "\t<PROTOCOLS/>\n",
            "</ROSETTASCRIPTS>\n",
            "core.scoring.ScoreFunctionFactory: SCOREFUNCTION: ref2015\n",
            "core.scoring.etable: Starting energy table calculation\n",
            "core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
            "core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
            "core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
            "core.scoring.etable: Finished calculating energy tables.\n",
            "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBPoly1D.csv\n",
            "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBFadeIntervals.csv\n",
            "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/HBEval.csv\n",
            "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/DonStrength.csv\n",
            "basic.io.database: Database file opened: scoring/score_functions/hbonds/ref2015_params/AccStrength.csv\n",
            "basic.io.database: Database file opened: scoring/score_functions/rama/fd/all.ramaProb\n",
            "basic.io.database: Database file opened: scoring/score_functions/rama/fd/prepro.ramaProb\n",
            "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.all.txt\n",
            "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.gly.txt\n",
            "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.pro.txt\n",
            "basic.io.database: Database file opened: scoring/score_functions/omega/omega_ppdep.valile.txt\n",
            "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA\n",
            "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/P_AA_n\n",
            "core.scoring.P_AA: shapovalov_lib::shap_p_aa_pp_smooth_level of 1( aka low_smooth ) got activated.\n",
            "basic.io.database: Database file opened: scoring/score_functions/P_AA_pp/shapovalov/10deg/kappa131/a20.prop\n",
            "core.scoring.etable: Starting energy table calculation\n",
            "core.scoring.etable: smooth_etable: changing atr/rep split to bottom of energy well\n",
            "core.scoring.etable: smooth_etable: spline smoothing lj etables (maxdis = 6)\n",
            "core.scoring.etable: smooth_etable: spline smoothing solvation etables (max_dis = 6)\n",
            "core.scoring.etable: Finished calculating energy tables.\n",
            "basic.io.database: Database file opened: scoring/score_functions/PairEPotential/pdb_pair_stats_fine\n",
            "basic.io.database: Database file opened: scoring/score_functions/InterchainPotential/interchain_env_log.txt\n",
            "basic.io.database: Database file opened: scoring/score_functions/InterchainPotential/interchain_pair_log.txt\n",
            "basic.io.database: Database file opened: scoring/score_functions/EnvPairPotential/env_log.txt\n",
            "basic.io.database: Database file opened: scoring/score_functions/EnvPairPotential/cbeta_den.txt\n",
            "basic.io.database: Database file opened: scoring/score_functions/EnvPairPotential/pair_log.txt\n",
            "basic.io.database: Database file opened: scoring/score_functions/EnvPairPotential/cenpack_log.txt\n",
            "core.scoring.ramachandran: shapovalov_lib::shap_rama_smooth_level of 4( aka highest_smooth ) got activated.\n",
            "basic.io.database: Database file opened: scoring/score_functions/rama/shapovalov/kappa25/all.ramaProb\n",
            "core.scoring.ScoreFunctionFactory: SCOREFUNCTION: ref2015\n",
            "core.scoring.ScoreFunctionFactory: SCOREFUNCTION: ref2015\n",
            "protocols.rosetta_scripts.RosettaScriptsParser: Defined mover named \"RAbD\" of type AntibodyDesignMover\n",
            "protocols.rosetta_scripts.ParsedProtocol: ParsedProtocol mover with the following settings\n",
            "basic.io.database: Database file opened: sampling/antibodies/cluster_center_dihedrals.txt\n",
            "protocols.antibody.AntibodyNumberingParser: Antibody numbering scheme definitions read successfully\n",
            "protocols.antibody.AntibodyNumberingParser: Antibody CDR definition read successfully\n",
            "antibody.AntibodyInfo: Successfully finished the CDR definition\n",
            "antibody.AntibodyInfo: AC Detecting Regular CDR H3 Stem Type\n",
            "antibody.AntibodyInfo: SRWGGDGFYAMDYW\n",
            "antibody.AntibodyInfo: AC Finished Detecting Regular CDR H3 Stem Type: KINKED\n",
            "antibody.AntibodyInfo: AC Finished Detecting Regular CDR H3 Stem Type: Kink: 1 Extended: 0\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H1\n",
            "protocols.antibody.cluster.CDRClusterMatcher: Length: 13 Omega: TTTTTTTTTTTTT\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H2\n",
            "protocols.antibody.cluster.CDRClusterMatcher: Length: 10 Omega: TTTTTTTTTT\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H3\n",
            "protocols.antibody.cluster.CDRClusterMatcher: Length: 13 Omega: TTTTTTTTTTTTT\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L1\n",
            "protocols.antibody.cluster.CDRClusterMatcher: Length: 11 Omega: TTTTTTTTTTT\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L2\n",
            "protocols.antibody.cluster.CDRClusterMatcher: Length: 8 Omega: TTTTTTTT\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L3\n",
            "protocols.antibody.cluster.CDRClusterMatcher: Length: 9 Omega: TTTTTTCTT\n",
            "core.energy_methods.CartesianBondedEnergy: Initializing IdealParametersDatabase with default Ks=300 , 80 , 80 , 10 , 80\n",
            "basic.io.database: Database file opened: scoring/score_functions/bondlength_bondangle/default-lengths.txt\n",
            "core.energy_methods.CartesianBondedEnergy: Read 759 bb-independent lengths.\n",
            "basic.io.database: Database file opened: scoring/score_functions/bondlength_bondangle/default-angles.txt\n",
            "core.energy_methods.CartesianBondedEnergy: Read 1434 bb-independent angles.\n",
            "basic.io.database: Database file opened: scoring/score_functions/bondlength_bondangle/default-torsions.txt\n",
            "core.energy_methods.CartesianBondedEnergy: Read 1 bb-independent torsions.\n",
            "basic.io.database: Database file opened: scoring/score_functions/bondlength_bondangle/default-improper.txt\n",
            "core.energy_methods.CartesianBondedEnergy: Read 529 bb-independent improper tors.\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H1\n",
            "protocols.antibody.cluster.CDRClusterMatcher: Length: 13 Omega: TTTTTTTTTTTTT\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H2\n",
            "protocols.antibody.cluster.CDRClusterMatcher: Length: 10 Omega: TTTTTTTTTT\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H3\n",
            "protocols.antibody.cluster.CDRClusterMatcher: Length: 13 Omega: TTTTTTTTTTTTT\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L1\n",
            "protocols.antibody.cluster.CDRClusterMatcher: Length: 11 Omega: TTTTTTTTTTT\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L2\n",
            "protocols.antibody.cluster.CDRClusterMatcher: Length: 8 Omega: TTTTTTTT\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L3\n",
            "protocols.antibody.cluster.CDRClusterMatcher: Length: 9 Omega: TTTTTTCTT\n",
            "protocols.antibody.clusters.CDRClusterSet: Adding cacheable cluster data to pose\n",
            "protocols.antibody.design.util: Reading CDRSetOptions from: /sampling/antibodies/design/default_instructions.txt\n",
            "protocols.antibody.design.util: Reading CDRGraftDesignOptions from: /sampling/antibodies/design/default_instructions.txt\n",
            "protocols.antibody.design.util: Reading CDRSeqDesignOptions from: /sampling/antibodies/design/default_instructions.txt\n",
            "core.scoring.ScoreFunctionFactory: SCOREFUNCTION: ref2015\n",
            "core.scoring.ScoreFunctionFactory: SCOREFUNCTION: ref2015\n",
            "core.scoring.ScoreFunctionFactory: SCOREFUNCTION: ref2015\n",
            "basic.io.database: [ WARNING ] Unable to locate database file /sampling/antibodies/antibody_database_rosetta_design.db\n",
            "antibody.database.AntibodyDatabaseManager: The antibody design database is not specified by the user.\n",
            "antibody.database.AntibodyDatabaseManager: Reading from: /usr/local/lib/python3.10/dist-packages/pyrosetta/database//sampling/antibodies/antibody_database_rosetta_design_north_paper.db\n",
            "antibody.database.AntibodyDatabaseManager: Checking loadable structures\n",
            "antibody.database.AntibodyDatabaseManager: Skipping 12 structures from reading\n",
            "core.scoring.ScoreFunctionFactory: SCOREFUNCTION: ref2015\n",
            "core.scoring.ScoreFunctionFactory: SCOREFUNCTION: ref2015\n",
            "antibody.database.AntibodyDatabaseManager: 0 Cached\n",
            "antibody.database.AntibodyDatabaseManager: cdrs loaded from database.\n",
            "protocols.antibody.design.AntibodyDesignMover: ///// Total CDRs in set /////\n",
            "protocols.antibody.design.AntibodyDesignMover: Total possible CDR combinations: 1\n",
            "basic.io.database: Database file opened: scoring/score_functions/elec_cp_reps.dat\n",
            "core.scoring.elec.util: Read 40 countpair representative atoms\n",
            "core.pack.dunbrack.RotamerLibrary: shapovalov_lib_fixes_enable option is true.\n",
            "core.pack.dunbrack.RotamerLibrary: shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
            "core.pack.dunbrack.RotamerLibrary: Binary rotamer library selected: /usr/local/lib/python3.10/dist-packages/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
            "core.pack.dunbrack.RotamerLibrary: Using Dunbrack library binary file '/usr/local/lib/python3.10/dist-packages/pyrosetta/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
            "core.pack.dunbrack.RotamerLibrary: Dunbrack 2010 library took 0.544593 seconds to load from binary\n",
            "core.scoring.ScoreFunction: \n",
            "------------------------------------------------------------\n",
            " Scores                       Weight   Raw Score Wghtd.Score\n",
            "------------------------------------------------------------\n",
            " fa_atr                       1.000   -5695.317   -5695.317\n",
            " fa_rep                       0.550   24719.870   13595.929\n",
            " fa_sol                       1.000    3501.170    3501.170\n",
            " fa_intra_rep                 0.005    2661.048      13.305\n",
            " fa_intra_sol_xover4          1.000     175.028     175.028\n",
            " lk_ball_wtd                  1.000     -57.980     -57.980\n",
            " fa_elec                      1.000   -1588.297   -1588.297\n",
            " pro_close                    1.250      10.290      12.862\n",
            " hbond_sr_bb                  1.000     -74.000     -74.000\n",
            " hbond_lr_bb                  1.000    -511.787    -511.787\n",
            " hbond_bb_sc                  1.000    -130.727    -130.727\n",
            " hbond_sc                     1.000     -78.766     -78.766\n",
            " dslf_fa13                    1.250      -1.572      -1.965\n",
            " atom_pair_constraint         0.010       0.000       0.000\n",
            " omega                        0.400     474.068     189.627\n",
            " fa_dun                       0.700    5380.790    3766.553\n",
            " p_aa_pp                      0.600    -289.096    -173.458\n",
            " yhh_planarity                0.625       0.000       0.000\n",
            " ref                          1.000     372.765     372.765\n",
            " rama_prepro                  0.450     191.068      85.981\n",
            "---------------------------------------------------\n",
            " Total weighted score:                    13400.923\n",
            "protocols.antibody.design.AntibodyDesignMover: Running basic monte carlo algorithm\n",
            "protocols.antibody.design.AntibodyDesignMover: Outer round: 1\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H1\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H2\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H3\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L1\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L2\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L3\n",
            "protocols.antibody.clusters.CDRClusterSet: Adding cacheable cluster data to pose\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: H1-13-1 data: 583 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/H1-13-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 26 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: H2-10-1 data: 328 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/H2-10-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 20 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: Adding general dihedral constraints for H3\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L1-11-1 data: 188 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/L1-11-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 22 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L2-8-1 data: 632 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/L2-8-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 16 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L3-9-cis7-1 data: 419 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/L3-9-cis7-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 18 constraints\n",
            "protocols.antibody.design.AntibodyDesignMover: Beginning MinProtocol\n",
            "protocols.antibody.design.AntibodyDesignMover: Inner round: 1\n",
            "protocols.antibody.design.AntibodyDesignMover: Add min neighbors : L2\n",
            "protocols.antibody.design.AntibodyDesignMover: Add min neighbors : L3\n",
            "basic.io.database: [ WARNING ] Unable to locate database file /sampling/antibodies/antibody_database_rosetta_design.db\n",
            "antibody.database.AntibodyDatabaseManager: The antibody design database is not specified by the user.\n",
            "antibody.database.AntibodyDatabaseManager: Reading from: /usr/local/lib/python3.10/dist-packages/pyrosetta/database//sampling/antibodies/antibody_database_rosetta_design_north_paper.db\n",
            "antibody.database.AntibodyDatabaseManager: Loaded L1-11-1 with 55 datapoints.\n",
            "antibody.database.AntibodyDatabaseManager: Loaded L3-9-cis7-1 with 179 datapoints.\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Data pre-loaded.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: applying prob task op\n",
            "core.pack.pack_rotamers: built 966 rotamers at 78 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating PDInteractionGraph\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs start: 13400.9\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs end: 7648.21\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 897\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 898\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 920\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 921\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 961\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 962\n",
            "antibody.design.GeneralAntibodyModeler: CDR min start: 7648.21\n",
            "antibody.design.GeneralAntibodyModeler: CDR min end: 4140.33\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: applying prob task op\n",
            "core.pack.pack_rotamers: built 1313 rotamers at 79 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating PDInteractionGraph\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs start: 4140.33\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs end: 6020.02\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 897\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 898\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 920\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 921\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 961\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 962\n",
            "antibody.design.GeneralAntibodyModeler: CDR min start: 4140.33\n",
            "antibody.design.GeneralAntibodyModeler: CDR min end: 4129.83\n",
            "protocols.antibody.design.AntibodyDesignMover: Min Cycle complete\n",
            "protocols.antibody.design.AntibodyDesignMover:\n",
            "protocols.antibody.design.AntibodyDesignMover: SCORE 1 4129.83 1\n",
            "protocols.antibody.design.AntibodyDesignMover: FINAL 1 4129.83 1\n",
            "protocols.antibody.design.AntibodyDesignMover:\n",
            "protocols.antibody.design.AntibodyDesignMover: Last Accepted Score: 4129.83\n",
            "protocols.antibody.design.AntibodyDesignMover: Lowest        Score:4129.83\n",
            "protocols.antibody.design.AntibodyDesignMover:\n",
            "protocols.antibody.design.AntibodyDesignMover: Outer round: 2\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H1\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H2\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H3\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L1\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L2\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L3\n",
            "protocols.antibody.clusters.CDRClusterSet: Adding cacheable cluster data to pose\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: H1-13-1 data: 583 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/H1-13-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 26 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: H2-10-1 data: 328 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/H2-10-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 20 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: Adding general dihedral constraints for H3\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L1-11-1 data: 188 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/L1-11-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 22 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L2-8-1 data: 632 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/L2-8-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 16 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L3-9-cis7-1 data: 419 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/L3-9-cis7-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 18 constraints\n",
            "protocols.antibody.design.AntibodyDesignMover: Beginning MinProtocol\n",
            "protocols.antibody.design.AntibodyDesignMover: Inner round: 1\n",
            "protocols.antibody.design.AntibodyDesignMover: Add min neighbors : L2\n",
            "protocols.antibody.design.AntibodyDesignMover: Add min neighbors : L3\n",
            "basic.io.database: [ WARNING ] Unable to locate database file /sampling/antibodies/antibody_database_rosetta_design.db\n",
            "antibody.database.AntibodyDatabaseManager: The antibody design database is not specified by the user.\n",
            "antibody.database.AntibodyDatabaseManager: Reading from: /usr/local/lib/python3.10/dist-packages/pyrosetta/database//sampling/antibodies/antibody_database_rosetta_design_north_paper.db\n",
            "antibody.database.AntibodyDatabaseManager: Loaded L1-11-1 with 55 datapoints.\n",
            "antibody.database.AntibodyDatabaseManager: Loaded L3-9-cis7-1 with 179 datapoints.\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Data pre-loaded.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: applying prob task op\n",
            "core.pack.pack_rotamers: built 1193 rotamers at 79 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating PDInteractionGraph\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs start: 4129.83\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs end: 6655.43\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 897\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 898\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 920\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 921\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 961\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 962\n",
            "antibody.design.GeneralAntibodyModeler: CDR min start: 4129.83\n",
            "antibody.design.GeneralAntibodyModeler: CDR min end: 4126.81\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: applying prob task op\n",
            "core.pack.pack_rotamers: built 1184 rotamers at 79 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating PDInteractionGraph\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs start: 4126.81\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs end: 6235.2\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 897\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 898\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 920\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 921\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 961\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 962\n",
            "antibody.design.GeneralAntibodyModeler: CDR min start: 4126.81\n",
            "antibody.design.GeneralAntibodyModeler: CDR min end: 4123.78\n",
            "protocols.antibody.design.AntibodyDesignMover: Min Cycle complete\n",
            "protocols.antibody.design.AntibodyDesignMover:\n",
            "protocols.antibody.design.AntibodyDesignMover: SCORE 2 4123.78 1\n",
            "protocols.antibody.design.AntibodyDesignMover: FINAL 2 4123.78 1\n",
            "protocols.antibody.design.AntibodyDesignMover:\n",
            "protocols.antibody.design.AntibodyDesignMover: Last Accepted Score: 4123.78\n",
            "protocols.antibody.design.AntibodyDesignMover: Lowest        Score:4123.78\n",
            "protocols.antibody.design.AntibodyDesignMover:\n",
            "protocols.antibody.design.AntibodyDesignMover: Outer round: 3\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H1\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H2\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H3\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L1\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L2\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L3\n",
            "protocols.antibody.clusters.CDRClusterSet: Adding cacheable cluster data to pose\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: H1-13-1 data: 583 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/H1-13-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 26 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: H2-10-1 data: 328 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/H2-10-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 20 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: Adding general dihedral constraints for H3\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L1-11-1 data: 188 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/L1-11-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 22 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L2-8-1 data: 632 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/L2-8-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 16 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L3-9-cis7-1 data: 419 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/L3-9-cis7-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 18 constraints\n",
            "protocols.antibody.design.AntibodyDesignMover: Beginning MinProtocol\n",
            "protocols.antibody.design.AntibodyDesignMover: Inner round: 1\n",
            "protocols.antibody.design.AntibodyDesignMover: Add min neighbors : L2\n",
            "protocols.antibody.design.AntibodyDesignMover: Add min neighbors : L3\n",
            "basic.io.database: [ WARNING ] Unable to locate database file /sampling/antibodies/antibody_database_rosetta_design.db\n",
            "antibody.database.AntibodyDatabaseManager: The antibody design database is not specified by the user.\n",
            "antibody.database.AntibodyDatabaseManager: Reading from: /usr/local/lib/python3.10/dist-packages/pyrosetta/database//sampling/antibodies/antibody_database_rosetta_design_north_paper.db\n",
            "antibody.database.AntibodyDatabaseManager: Loaded L1-11-1 with 55 datapoints.\n",
            "antibody.database.AntibodyDatabaseManager: Loaded L3-9-cis7-1 with 179 datapoints.\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Data pre-loaded.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: applying prob task op\n",
            "core.pack.pack_rotamers: built 1253 rotamers at 79 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating PDInteractionGraph\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs start: 4123.78\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs end: 6153.4\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 897\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 898\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 920\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 921\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 961\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 962\n",
            "antibody.design.GeneralAntibodyModeler: CDR min start: 4123.78\n",
            "antibody.design.GeneralAntibodyModeler: CDR min end: 4121.78\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: applying prob task op\n",
            "core.pack.pack_rotamers: built 1081 rotamers at 78 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating PDInteractionGraph\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs start: 4121.78\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs end: 6358.36\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 897\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 898\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 920\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 921\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 961\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 962\n",
            "antibody.design.GeneralAntibodyModeler: CDR min start: 4121.78\n",
            "antibody.design.GeneralAntibodyModeler: CDR min end: 4118.35\n",
            "protocols.antibody.design.AntibodyDesignMover: Min Cycle complete\n",
            "protocols.antibody.design.AntibodyDesignMover:\n",
            "protocols.antibody.design.AntibodyDesignMover: SCORE 3 4118.35 1\n",
            "protocols.antibody.design.AntibodyDesignMover: FINAL 3 4118.35 1\n",
            "protocols.antibody.design.AntibodyDesignMover:\n",
            "protocols.antibody.design.AntibodyDesignMover: Last Accepted Score: 4118.35\n",
            "protocols.antibody.design.AntibodyDesignMover: Lowest        Score:4118.35\n",
            "protocols.antibody.design.AntibodyDesignMover:\n",
            "protocols.antibody.design.AntibodyDesignMover: Outer round: 4\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H1\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H2\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H3\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L1\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L2\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L3\n",
            "protocols.antibody.clusters.CDRClusterSet: Adding cacheable cluster data to pose\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: H1-13-1 data: 583 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/H1-13-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 26 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: H2-10-1 data: 328 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/H2-10-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 20 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: Adding general dihedral constraints for H3\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L1-11-1 data: 188 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/L1-11-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 22 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L2-8-1 data: 632 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/L2-8-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 16 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L3-9-cis7-1 data: 419 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/L3-9-cis7-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 18 constraints\n",
            "protocols.antibody.design.AntibodyDesignMover: Beginning MinProtocol\n",
            "protocols.antibody.design.AntibodyDesignMover: Inner round: 1\n",
            "protocols.antibody.design.AntibodyDesignMover: Add min neighbors : L1\n",
            "protocols.antibody.design.AntibodyDesignMover: Add min neighbors : H3\n",
            "basic.io.database: [ WARNING ] Unable to locate database file /sampling/antibodies/antibody_database_rosetta_design.db\n",
            "antibody.database.AntibodyDatabaseManager: The antibody design database is not specified by the user.\n",
            "antibody.database.AntibodyDatabaseManager: Reading from: /usr/local/lib/python3.10/dist-packages/pyrosetta/database//sampling/antibodies/antibody_database_rosetta_design_north_paper.db\n",
            "antibody.database.AntibodyDatabaseManager: Loaded L1-11-1 with 55 datapoints.\n",
            "antibody.database.AntibodyDatabaseManager: Loaded L3-9-cis7-1 with 179 datapoints.\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Data pre-loaded.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: applying prob task op\n",
            "core.pack.pack_rotamers: built 1090 rotamers at 93 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating PDInteractionGraph\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs start: 4118.35\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs end: 5555.12\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 852\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 853\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 897\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 898\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 961\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 962\n",
            "antibody.design.GeneralAntibodyModeler: CDR min start: 4118.35\n",
            "antibody.design.GeneralAntibodyModeler: CDR min end: 3455.56\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: applying prob task op\n",
            "core.pack.pack_rotamers: built 1239 rotamers at 95 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating PDInteractionGraph\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs start: 3455.56\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs end: 5643.6\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 852\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 853\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 897\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 898\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 961\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 962\n",
            "antibody.design.GeneralAntibodyModeler: CDR min start: 3455.56\n",
            "antibody.design.GeneralAntibodyModeler: CDR min end: 3440.9\n",
            "protocols.antibody.design.AntibodyDesignMover: Min Cycle complete\n",
            "protocols.antibody.design.AntibodyDesignMover:\n",
            "protocols.antibody.design.AntibodyDesignMover: SCORE 4 3440.9 1\n",
            "protocols.antibody.design.AntibodyDesignMover: FINAL 4 3440.9 1\n",
            "protocols.antibody.design.AntibodyDesignMover:\n",
            "protocols.antibody.design.AntibodyDesignMover: Last Accepted Score: 3440.9\n",
            "protocols.antibody.design.AntibodyDesignMover: Lowest        Score:3440.9\n",
            "protocols.antibody.design.AntibodyDesignMover:\n",
            "protocols.antibody.design.AntibodyDesignMover: Outer round: 5\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H1\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H2\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H3\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L1\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L2\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L3\n",
            "protocols.antibody.clusters.CDRClusterSet: Adding cacheable cluster data to pose\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: H1-13-1 data: 583 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/H1-13-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 26 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: H2-10-1 data: 328 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/H2-10-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 20 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: Adding general dihedral constraints for H3\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L1-11-1 data: 188 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/L1-11-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 22 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L2-8-1 data: 632 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/L2-8-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 16 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L3-9-cis7-1 data: 419 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/L3-9-cis7-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 18 constraints\n",
            "protocols.antibody.design.AntibodyDesignMover: Beginning MinProtocol\n",
            "protocols.antibody.design.AntibodyDesignMover: Inner round: 1\n",
            "protocols.antibody.design.AntibodyDesignMover: Add min neighbors : L2\n",
            "protocols.antibody.design.AntibodyDesignMover: Add min neighbors : L3\n",
            "basic.io.database: [ WARNING ] Unable to locate database file /sampling/antibodies/antibody_database_rosetta_design.db\n",
            "antibody.database.AntibodyDatabaseManager: The antibody design database is not specified by the user.\n",
            "antibody.database.AntibodyDatabaseManager: Reading from: /usr/local/lib/python3.10/dist-packages/pyrosetta/database//sampling/antibodies/antibody_database_rosetta_design_north_paper.db\n",
            "antibody.database.AntibodyDatabaseManager: Loaded L1-11-1 with 55 datapoints.\n",
            "antibody.database.AntibodyDatabaseManager: Loaded L3-9-cis7-1 with 179 datapoints.\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Data pre-loaded.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: applying prob task op\n",
            "core.pack.pack_rotamers: built 1125 rotamers at 78 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating PDInteractionGraph\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs start: 3440.9\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs end: 5083.9\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 897\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 898\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 920\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 921\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 961\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 962\n",
            "antibody.design.GeneralAntibodyModeler: CDR min start: 3440.9\n",
            "antibody.design.GeneralAntibodyModeler: CDR min end: 3441.68\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: applying prob task op\n",
            "core.pack.pack_rotamers: built 1091 rotamers at 78 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating PDInteractionGraph\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs start: 3440.9\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs end: 4728.24\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 897\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 898\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 920\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 921\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 961\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 962\n",
            "antibody.design.GeneralAntibodyModeler: CDR min start: 3440.9\n",
            "antibody.design.GeneralAntibodyModeler: CDR min end: 3441.68\n",
            "protocols.antibody.design.AntibodyDesignMover: Min Cycle complete\n",
            "protocols.antibody.design.AntibodyDesignMover:\n",
            "protocols.antibody.design.AntibodyDesignMover: SCORE 5 3440.9 1\n",
            "protocols.antibody.design.AntibodyDesignMover: FINAL 5 3440.9 1\n",
            "protocols.antibody.design.AntibodyDesignMover:\n",
            "protocols.antibody.design.AntibodyDesignMover: Last Accepted Score: 3440.9\n",
            "protocols.antibody.design.AntibodyDesignMover: Lowest        Score:3440.9\n",
            "protocols.antibody.design.AntibodyDesignMover:\n",
            "protocols.antibody.design.AntibodyDesignMover: Outer cycles attempted: 5 Outer cycles completed: 5\n",
            "protocols.antibody.design.AntibodyDesignMover:\n",
            "protocols.antibody.design.AntibodyDesignMover: Native Pose (MC): 13400.9\n",
            "protocols.antibody.design.AntibodyDesignMover: Final Pose (MC): 3440.9\n",
            "protocols.antibody.design.AntibodyDesignMover:\n",
            "protocols.moves.TrialCounter:              unk trials=      5;  accepts= 1.0000;  energy_drop/trial= -1992.00556\n",
            "protocols.antibody.design.AntibodyDesignMover:\n",
            "protocols.antibody.design.AntibodyDesignMover: Native Pose (Total): 13400.9\n",
            "protocols.antibody.design.AntibodyDesignMover: Final Pose (Total): 3440.9\n",
            "protocols.antibody.design.AntibodyDesignMover: \n",
            "------------------------------------------------------------\n",
            " Scores                       Weight   Raw Score Wghtd.Score\n",
            "------------------------------------------------------------\n",
            " fa_atr                       1.000   -5639.367   -5639.367\n",
            " fa_rep                       0.550    7017.840    3859.812\n",
            " fa_sol                       1.000    3452.608    3452.608\n",
            " fa_intra_rep                 0.005    2601.744      13.009\n",
            " fa_intra_sol_xover4          1.000     174.910     174.910\n",
            " lk_ball_wtd                  1.000     -63.070     -63.070\n",
            " fa_elec                      1.000   -1607.836   -1607.836\n",
            " pro_close                    1.250       7.731       9.664\n",
            " hbond_sr_bb                  1.000     -73.010     -73.010\n",
            " hbond_lr_bb                  1.000    -508.795    -508.795\n",
            " hbond_bb_sc                  1.000    -134.850    -134.850\n",
            " hbond_sc                     1.000     -79.486     -79.486\n",
            " dslf_fa13                    1.250      -1.042      -1.303\n",
            " atom_pair_constraint         0.010       0.000       0.000\n",
            " omega                        0.400     496.120     198.448\n",
            " fa_dun                       0.700    5091.216    3563.851\n",
            " p_aa_pp                      0.600    -290.769    -174.462\n",
            " yhh_planarity                0.625       0.046       0.029\n",
            " ref                          1.000     371.893     371.893\n",
            " rama_prepro                  0.450     175.222      78.850\n",
            "---------------------------------------------------\n",
            " Total weighted score:                     3440.895\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H1\n",
            "protocols.antibody.cluster.CDRClusterMatcher: Length: 13 Omega: TTTTTTTTTTTTT\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H2\n",
            "protocols.antibody.cluster.CDRClusterMatcher: Length: 10 Omega: TTTTTTTTTT\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H3\n",
            "protocols.antibody.cluster.CDRClusterMatcher: Length: 13 Omega: TTTTTTTTTTTTT\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L1\n",
            "protocols.antibody.cluster.CDRClusterMatcher: Length: 11 Omega: TTTTTTTTTTT\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L2\n",
            "protocols.antibody.cluster.CDRClusterMatcher: Length: 8 Omega: TTTTTTTT\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L3\n",
            "protocols.antibody.cluster.CDRClusterMatcher: Length: 9 Omega: TTTTTTCTT\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using explicit constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using interface constructor\n",
            "protocols.evaluation.ChiWellRmsdEvaluatorCreator: Evaluation Creator active ...\n",
            "protocols.analysis.InterfaceAnalyzerMover: Making an interface set with fixed chains\n",
            "protocols.analysis.InterfaceAnalyzerMover: Interface set residues total: 99\n",
            "protocols.analysis.InterfaceAnalyzerMover: NOT detecting disulfides in the separated pose.\n",
            "protocols.analysis.InterfaceAnalyzerMover: Calculating dSASA\n",
            "protocols.analysis.InterfaceAnalyzerMover: Calculating per-res dSASA data\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 1270 rotamers at 99 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.analysis.InterfaceAnalyzerMover: included_nres: 975\n",
            "protocols.analysis.InterfaceAnalyzerMover: Making hbond ScoreFunction with bb-bb energy method options\n",
            "core.scoring.ScoreFunctionFactory: SCOREFUNCTION: ref2015\n",
            "protocols.analysis.InterfaceAnalyzerMover: Found Hbond between chains: 9 and 13\n",
            "protocols.analysis.InterfaceAnalyzerMover: Found Hbond between chains: 12 and 9\n",
            "protocols.analysis.InterfaceAnalyzerMover: Found Hbond between chains: 9 and 13\n",
            "protocols.analysis.InterfaceAnalyzerMover: Found Hbond between chains: 13 and 10\n",
            "protocols.analysis.InterfaceAnalyzerMover: Computing delta unsat polar residues...\n",
            "basic.io.database: Database file opened: scoring/score_functions/hbonds/sp2_elec_params/HBPoly1D.csv\n",
            "basic.io.database: Database file opened: scoring/score_functions/hbonds/sp2_elec_params/HBFadeIntervals.csv\n",
            "basic.io.database: Database file opened: scoring/score_functions/hbonds/sp2_elec_params/HBEval.csv\n",
            "basic.io.database: Database file opened: scoring/score_functions/sc/sc_radii.lib\n",
            "protocols.analysis.InterfaceAnalyzerMover: Computing Shape Complementarity Score...\n",
            "protocols.analysis.InterfaceAnalyzerMover: Upstream chain(s) numbers: 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11,\n",
            "protocols.analysis.InterfaceAnalyzerMover: Downstream chain(s) numbers: 12, 13,\n"
          ]
        }
      ],
      "source": [
        "rabd = XmlObjects.static_get_mover('<AntibodyDesignMover name=\"RAbD\" seq_design_cdrs=\"L1,L3\" light_chain=\"kappa\"/>')\n",
        "if not os.getenv(\"DEBUG\"):\n",
        "    rabd.apply(pose)"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "sI8_CSkg2LMv"
      },
      "source": [
        "Now, for the sake of learning how to do this - how would we do this in code instead of the XML - we just need to use setters."
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 16,
      "metadata": {
        "id": "Py0tCFYn2LMv",
        "outputId": "947ac8bc-b049-4828-beba-d878d153d7ec",
        "colab": {
          "base_uri": "https://localhost:8080/"
        }
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "basic.io.database: Database file opened: sampling/antibodies/cluster_center_dihedrals.txt\n",
            "protocols.antibody.AntibodyNumberingParser: Antibody numbering scheme definitions read successfully\n",
            "protocols.antibody.AntibodyNumberingParser: Antibody CDR definition read successfully\n",
            "antibody.AntibodyInfo: Successfully finished the CDR definition\n",
            "antibody.AntibodyInfo: AC Detecting Regular CDR H3 Stem Type\n",
            "antibody.AntibodyInfo: SRWGGDGFYAMDYW\n",
            "antibody.AntibodyInfo: AC Finished Detecting Regular CDR H3 Stem Type: KINKED\n",
            "antibody.AntibodyInfo: AC Finished Detecting Regular CDR H3 Stem Type: Kink: 1 Extended: 0\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H1\n",
            "protocols.antibody.cluster.CDRClusterMatcher: Length: 13 Omega: TTTTTTTTTTTTT\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H2\n",
            "protocols.antibody.cluster.CDRClusterMatcher: Length: 10 Omega: TTTTTTTTTT\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H3\n",
            "protocols.antibody.cluster.CDRClusterMatcher: Length: 13 Omega: TTTTTTTTTTTTT\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L1\n",
            "protocols.antibody.cluster.CDRClusterMatcher: Length: 11 Omega: TTTTTTTTTTT\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L2\n",
            "protocols.antibody.cluster.CDRClusterMatcher: Length: 8 Omega: TTTTTTTT\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L3\n",
            "protocols.antibody.cluster.CDRClusterMatcher: Length: 9 Omega: TTTTTTCTT\n",
            "core.scoring.ScoreFunctionFactory: SCOREFUNCTION: ref2015\n",
            "core.scoring.ScoreFunctionFactory: SCOREFUNCTION: ref2015\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H1\n",
            "protocols.antibody.cluster.CDRClusterMatcher: Length: 13 Omega: TTTTTTTTTTTTT\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H2\n",
            "protocols.antibody.cluster.CDRClusterMatcher: Length: 10 Omega: TTTTTTTTTT\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H3\n",
            "protocols.antibody.cluster.CDRClusterMatcher: Length: 13 Omega: TTTTTTTTTTTTT\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L1\n",
            "protocols.antibody.cluster.CDRClusterMatcher: Length: 11 Omega: TTTTTTTTTTT\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L2\n",
            "protocols.antibody.cluster.CDRClusterMatcher: Length: 8 Omega: TTTTTTTT\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L3\n",
            "protocols.antibody.cluster.CDRClusterMatcher: Length: 9 Omega: TTTTTTCTT\n",
            "protocols.antibody.clusters.CDRClusterSet: Adding cacheable cluster data to pose\n",
            "protocols.antibody.design.util: Reading CDRSetOptions from: /sampling/antibodies/design/default_instructions.txt\n",
            "protocols.antibody.design.util: Reading CDRGraftDesignOptions from: /sampling/antibodies/design/default_instructions.txt\n",
            "protocols.antibody.design.util: Reading CDRSeqDesignOptions from: /sampling/antibodies/design/default_instructions.txt\n",
            "core.scoring.ScoreFunctionFactory: SCOREFUNCTION: ref2015\n",
            "core.scoring.ScoreFunctionFactory: SCOREFUNCTION: ref2015\n",
            "core.scoring.ScoreFunctionFactory: SCOREFUNCTION: ref2015\n",
            "basic.io.database: [ WARNING ] Unable to locate database file /sampling/antibodies/antibody_database_rosetta_design.db\n",
            "antibody.database.AntibodyDatabaseManager: The antibody design database is not specified by the user.\n",
            "antibody.database.AntibodyDatabaseManager: Reading from: /usr/local/lib/python3.10/dist-packages/pyrosetta/database//sampling/antibodies/antibody_database_rosetta_design_north_paper.db\n",
            "antibody.database.AntibodyDatabaseManager: Checking loadable structures\n",
            "antibody.database.AntibodyDatabaseManager: Skipping 12 structures from reading\n",
            "core.scoring.ScoreFunctionFactory: SCOREFUNCTION: ref2015\n",
            "core.scoring.ScoreFunctionFactory: SCOREFUNCTION: ref2015\n",
            "antibody.database.AntibodyDatabaseManager: 0 Cached\n",
            "antibody.database.AntibodyDatabaseManager: cdrs loaded from database.\n",
            "protocols.antibody.design.AntibodyDesignMover: ///// Total CDRs in set /////\n",
            "protocols.antibody.design.AntibodyDesignMover: Total possible CDR combinations: 1\n",
            "core.scoring.ScoreFunction: \n",
            "------------------------------------------------------------\n",
            " Scores                       Weight   Raw Score Wghtd.Score\n",
            "------------------------------------------------------------\n",
            " fa_atr                       1.000   -5695.317   -5695.317\n",
            " fa_rep                       0.550   24719.870   13595.929\n",
            " fa_sol                       1.000    3501.170    3501.170\n",
            " fa_intra_rep                 0.005    2661.048      13.305\n",
            " fa_intra_sol_xover4          1.000     175.028     175.028\n",
            " lk_ball_wtd                  1.000     -57.980     -57.980\n",
            " fa_elec                      1.000   -1588.297   -1588.297\n",
            " pro_close                    1.250      10.290      12.862\n",
            " hbond_sr_bb                  1.000     -74.000     -74.000\n",
            " hbond_lr_bb                  1.000    -511.787    -511.787\n",
            " hbond_bb_sc                  1.000    -130.727    -130.727\n",
            " hbond_sc                     1.000     -78.766     -78.766\n",
            " dslf_fa13                    1.250      -1.572      -1.965\n",
            " atom_pair_constraint         0.010       0.000       0.000\n",
            " omega                        0.400     474.068     189.627\n",
            " fa_dun                       0.700    5380.790    3766.553\n",
            " p_aa_pp                      0.600    -289.096    -173.458\n",
            " yhh_planarity                0.625       0.000       0.000\n",
            " ref                          1.000     372.765     372.765\n",
            " rama_prepro                  0.450     191.068      85.981\n",
            "---------------------------------------------------\n",
            " Total weighted score:                    13400.923\n",
            "protocols.antibody.design.AntibodyDesignMover: Running basic monte carlo algorithm\n",
            "protocols.antibody.design.AntibodyDesignMover: Outer round: 1\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H1\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H2\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H3\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L1\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L2\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L3\n",
            "protocols.antibody.clusters.CDRClusterSet: Adding cacheable cluster data to pose\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: H1-13-1 data: 583 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/H1-13-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 26 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: H2-10-1 data: 328 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/H2-10-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 20 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: Adding general dihedral constraints for H3\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L1-11-1 data: 188 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/L1-11-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 22 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L2-8-1 data: 632 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/L2-8-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 16 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L3-9-cis7-1 data: 419 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/L3-9-cis7-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 18 constraints\n",
            "protocols.antibody.design.AntibodyDesignMover: Beginning MinProtocol\n",
            "protocols.antibody.design.AntibodyDesignMover: Inner round: 1\n",
            "protocols.antibody.design.AntibodyDesignMover: Add min neighbors : L1\n",
            "protocols.antibody.design.AntibodyDesignMover: Add min neighbors : H3\n",
            "basic.io.database: [ WARNING ] Unable to locate database file /sampling/antibodies/antibody_database_rosetta_design.db\n",
            "antibody.database.AntibodyDatabaseManager: The antibody design database is not specified by the user.\n",
            "antibody.database.AntibodyDatabaseManager: Reading from: /usr/local/lib/python3.10/dist-packages/pyrosetta/database//sampling/antibodies/antibody_database_rosetta_design_north_paper.db\n",
            "antibody.database.AntibodyDatabaseManager: Loaded L1-11-1 with 55 datapoints.\n",
            "antibody.database.AntibodyDatabaseManager: Loaded L3-9-cis7-1 with 179 datapoints.\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Data pre-loaded.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: applying prob task op\n",
            "core.pack.pack_rotamers: built 1029 rotamers at 94 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating PDInteractionGraph\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs start: 13400.9\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs end: 7622.11\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 852\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 853\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 897\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 898\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 961\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 962\n",
            "antibody.design.GeneralAntibodyModeler: CDR min start: 7622.11\n",
            "antibody.design.GeneralAntibodyModeler: CDR min end: 2515.13\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: applying prob task op\n",
            "core.pack.pack_rotamers: built 1179 rotamers at 94 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating PDInteractionGraph\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs start: 2515.13\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs end: 4716.57\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 852\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 853\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 897\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 898\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 961\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 962\n",
            "antibody.design.GeneralAntibodyModeler: CDR min start: 2515.13\n",
            "antibody.design.GeneralAntibodyModeler: CDR min end: 2468.56\n",
            "protocols.antibody.design.AntibodyDesignMover: Min Cycle complete\n",
            "protocols.antibody.design.AntibodyDesignMover:\n",
            "protocols.antibody.design.AntibodyDesignMover: SCORE 1 2468.56 1\n",
            "protocols.antibody.design.AntibodyDesignMover: FINAL 1 2468.56 1\n",
            "protocols.antibody.design.AntibodyDesignMover:\n",
            "protocols.antibody.design.AntibodyDesignMover: Last Accepted Score: 2468.56\n",
            "protocols.antibody.design.AntibodyDesignMover: Lowest        Score:2468.56\n",
            "protocols.antibody.design.AntibodyDesignMover:\n",
            "protocols.antibody.design.AntibodyDesignMover: Outer round: 2\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H1\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H2\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H3\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L1\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L2\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L3\n",
            "protocols.antibody.clusters.CDRClusterSet: Adding cacheable cluster data to pose\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: H1-13-1 data: 583 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/H1-13-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 26 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: H2-10-1 data: 328 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/H2-10-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 20 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: Adding general dihedral constraints for H3\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L1-11-1 data: 188 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/L1-11-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 22 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L2-8-1 data: 632 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/L2-8-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 16 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L3-9-cis7-1 data: 419 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/L3-9-cis7-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 18 constraints\n",
            "protocols.antibody.design.AntibodyDesignMover: Beginning MinProtocol\n",
            "protocols.antibody.design.AntibodyDesignMover: Inner round: 1\n",
            "protocols.antibody.design.AntibodyDesignMover: Add min neighbors : L1\n",
            "protocols.antibody.design.AntibodyDesignMover: Add min neighbors : H3\n",
            "basic.io.database: [ WARNING ] Unable to locate database file /sampling/antibodies/antibody_database_rosetta_design.db\n",
            "antibody.database.AntibodyDatabaseManager: The antibody design database is not specified by the user.\n",
            "antibody.database.AntibodyDatabaseManager: Reading from: /usr/local/lib/python3.10/dist-packages/pyrosetta/database//sampling/antibodies/antibody_database_rosetta_design_north_paper.db\n",
            "antibody.database.AntibodyDatabaseManager: Loaded L1-11-1 with 55 datapoints.\n",
            "antibody.database.AntibodyDatabaseManager: Loaded L3-9-cis7-1 with 179 datapoints.\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Data pre-loaded.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: applying prob task op\n",
            "core.pack.pack_rotamers: built 1186 rotamers at 93 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating PDInteractionGraph\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs start: 2468.56\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs end: 6038.86\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 852\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 853\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 897\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 898\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 961\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 962\n",
            "antibody.design.GeneralAntibodyModeler: CDR min start: 2468.56\n",
            "antibody.design.GeneralAntibodyModeler: CDR min end: 2467.49\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: applying prob task op\n",
            "core.pack.pack_rotamers: built 1185 rotamers at 94 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating PDInteractionGraph\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs start: 2467.49\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs end: 4854.71\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 852\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 853\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 897\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 898\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 961\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 962\n",
            "antibody.design.GeneralAntibodyModeler: CDR min start: 2467.49\n",
            "antibody.design.GeneralAntibodyModeler: CDR min end: 2460.33\n",
            "protocols.antibody.design.AntibodyDesignMover: Min Cycle complete\n",
            "protocols.antibody.design.AntibodyDesignMover:\n",
            "protocols.antibody.design.AntibodyDesignMover: SCORE 2 2460.33 1\n",
            "protocols.antibody.design.AntibodyDesignMover: FINAL 2 2460.33 1\n",
            "protocols.antibody.design.AntibodyDesignMover:\n",
            "protocols.antibody.design.AntibodyDesignMover: Last Accepted Score: 2460.33\n",
            "protocols.antibody.design.AntibodyDesignMover: Lowest        Score:2460.33\n",
            "protocols.antibody.design.AntibodyDesignMover:\n",
            "protocols.antibody.design.AntibodyDesignMover: Outer round: 3\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H1\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H2\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H3\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L1\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L2\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L3\n",
            "protocols.antibody.clusters.CDRClusterSet: Adding cacheable cluster data to pose\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: H1-13-1 data: 583 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/H1-13-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 26 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: H2-10-1 data: 328 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/H2-10-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 20 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: Adding general dihedral constraints for H3\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L1-11-1 data: 188 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/L1-11-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 22 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L2-8-1 data: 632 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/L2-8-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 16 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L3-9-cis7-1 data: 419 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/L3-9-cis7-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 18 constraints\n",
            "protocols.antibody.design.AntibodyDesignMover: Beginning MinProtocol\n",
            "protocols.antibody.design.AntibodyDesignMover: Inner round: 1\n",
            "protocols.antibody.design.AntibodyDesignMover: Add min neighbors : L2\n",
            "protocols.antibody.design.AntibodyDesignMover: Add min neighbors : L3\n",
            "basic.io.database: [ WARNING ] Unable to locate database file /sampling/antibodies/antibody_database_rosetta_design.db\n",
            "antibody.database.AntibodyDatabaseManager: The antibody design database is not specified by the user.\n",
            "antibody.database.AntibodyDatabaseManager: Reading from: /usr/local/lib/python3.10/dist-packages/pyrosetta/database//sampling/antibodies/antibody_database_rosetta_design_north_paper.db\n",
            "antibody.database.AntibodyDatabaseManager: Loaded L1-11-1 with 55 datapoints.\n",
            "antibody.database.AntibodyDatabaseManager: Loaded L3-9-cis7-1 with 179 datapoints.\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Data pre-loaded.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: applying prob task op\n",
            "core.pack.pack_rotamers: built 1250 rotamers at 79 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating PDInteractionGraph\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs start: 2460.33\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs end: 4288.03\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 897\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 898\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 920\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 921\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 961\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 962\n",
            "antibody.design.GeneralAntibodyModeler: CDR min start: 2460.33\n",
            "antibody.design.GeneralAntibodyModeler: CDR min end: 2420.16\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: applying prob task op\n",
            "core.pack.pack_rotamers: built 1087 rotamers at 78 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating PDInteractionGraph\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs start: 2420.16\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs end: 4273.72\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 897\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 898\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 920\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 921\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 961\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 962\n",
            "antibody.design.GeneralAntibodyModeler: CDR min start: 2420.16\n",
            "antibody.design.GeneralAntibodyModeler: CDR min end: 2419.73\n",
            "protocols.antibody.design.AntibodyDesignMover: Min Cycle complete\n",
            "protocols.antibody.design.AntibodyDesignMover:\n",
            "protocols.antibody.design.AntibodyDesignMover: SCORE 3 2419.73 1\n",
            "protocols.antibody.design.AntibodyDesignMover: FINAL 3 2419.73 1\n",
            "protocols.antibody.design.AntibodyDesignMover:\n",
            "protocols.antibody.design.AntibodyDesignMover: Last Accepted Score: 2419.73\n",
            "protocols.antibody.design.AntibodyDesignMover: Lowest        Score:2419.73\n",
            "protocols.antibody.design.AntibodyDesignMover:\n",
            "protocols.antibody.design.AntibodyDesignMover: Outer round: 4\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H1\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H2\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H3\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L1\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L2\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L3\n",
            "protocols.antibody.clusters.CDRClusterSet: Adding cacheable cluster data to pose\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: H1-13-1 data: 583 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/H1-13-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 26 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: H2-10-1 data: 328 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/H2-10-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 20 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: Adding general dihedral constraints for H3\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L1-11-1 data: 188 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/L1-11-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 22 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L2-8-1 data: 632 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/L2-8-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 16 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L3-9-cis7-1 data: 419 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/L3-9-cis7-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 18 constraints\n",
            "protocols.antibody.design.AntibodyDesignMover: Beginning MinProtocol\n",
            "protocols.antibody.design.AntibodyDesignMover: Inner round: 1\n",
            "protocols.antibody.design.AntibodyDesignMover: Add min neighbors : L2\n",
            "protocols.antibody.design.AntibodyDesignMover: Add min neighbors : L3\n",
            "basic.io.database: [ WARNING ] Unable to locate database file /sampling/antibodies/antibody_database_rosetta_design.db\n",
            "antibody.database.AntibodyDatabaseManager: The antibody design database is not specified by the user.\n",
            "antibody.database.AntibodyDatabaseManager: Reading from: /usr/local/lib/python3.10/dist-packages/pyrosetta/database//sampling/antibodies/antibody_database_rosetta_design_north_paper.db\n",
            "antibody.database.AntibodyDatabaseManager: Loaded L1-11-1 with 55 datapoints.\n",
            "antibody.database.AntibodyDatabaseManager: Loaded L3-9-cis7-1 with 179 datapoints.\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Data pre-loaded.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: applying prob task op\n",
            "core.pack.pack_rotamers: built 1049 rotamers at 78 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating PDInteractionGraph\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs start: 2419.73\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs end: 4400.82\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 897\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 898\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 920\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 921\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 961\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 962\n",
            "antibody.design.GeneralAntibodyModeler: CDR min start: 2419.73\n",
            "antibody.design.GeneralAntibodyModeler: CDR min end: 2419.72\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: applying prob task op\n",
            "core.pack.pack_rotamers: built 1154 rotamers at 78 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating PDInteractionGraph\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs start: 2419.72\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs end: 4270.95\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 897\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 898\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 920\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 921\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 961\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 962\n",
            "antibody.design.GeneralAntibodyModeler: CDR min start: 2419.72\n",
            "antibody.design.GeneralAntibodyModeler: CDR min end: 2419.64\n",
            "protocols.antibody.design.AntibodyDesignMover: Min Cycle complete\n",
            "protocols.antibody.design.AntibodyDesignMover:\n",
            "protocols.antibody.design.AntibodyDesignMover: SCORE 4 2419.64 1\n",
            "protocols.antibody.design.AntibodyDesignMover: FINAL 4 2419.64 1\n",
            "protocols.antibody.design.AntibodyDesignMover:\n",
            "protocols.antibody.design.AntibodyDesignMover: Last Accepted Score: 2419.64\n",
            "protocols.antibody.design.AntibodyDesignMover: Lowest        Score:2419.64\n",
            "protocols.antibody.design.AntibodyDesignMover:\n",
            "protocols.antibody.design.AntibodyDesignMover: Outer round: 5\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H1\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H2\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H3\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L1\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L2\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L3\n",
            "protocols.antibody.clusters.CDRClusterSet: Adding cacheable cluster data to pose\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: H1-13-1 data: 583 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/H1-13-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 26 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: H2-10-1 data: 328 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/H2-10-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 20 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: Adding general dihedral constraints for H3\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L1-11-1 data: 188 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/L1-11-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 22 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L2-8-1 data: 632 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/L2-8-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 16 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L3-9-cis7-1 data: 419 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/L3-9-cis7-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 18 constraints\n",
            "protocols.antibody.design.AntibodyDesignMover: Beginning MinProtocol\n",
            "protocols.antibody.design.AntibodyDesignMover: Inner round: 1\n",
            "protocols.antibody.design.AntibodyDesignMover: Add min neighbors : L2\n",
            "protocols.antibody.design.AntibodyDesignMover: Add min neighbors : L3\n",
            "basic.io.database: [ WARNING ] Unable to locate database file /sampling/antibodies/antibody_database_rosetta_design.db\n",
            "antibody.database.AntibodyDatabaseManager: The antibody design database is not specified by the user.\n",
            "antibody.database.AntibodyDatabaseManager: Reading from: /usr/local/lib/python3.10/dist-packages/pyrosetta/database//sampling/antibodies/antibody_database_rosetta_design_north_paper.db\n",
            "antibody.database.AntibodyDatabaseManager: Loaded L1-11-1 with 55 datapoints.\n",
            "antibody.database.AntibodyDatabaseManager: Loaded L3-9-cis7-1 with 179 datapoints.\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Data pre-loaded.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: applying prob task op\n",
            "core.pack.pack_rotamers: built 1169 rotamers at 78 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating PDInteractionGraph\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs start: 2419.64\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs end: 4147.91\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 897\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 898\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 920\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 921\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 961\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 962\n",
            "antibody.design.GeneralAntibodyModeler: CDR min start: 2419.64\n",
            "antibody.design.GeneralAntibodyModeler: CDR min end: 2419.6\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: applying prob task op\n",
            "core.pack.pack_rotamers: built 1048 rotamers at 78 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating PDInteractionGraph\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs start: 2419.6\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs end: 4651.21\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 897\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 898\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 920\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 921\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 961\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 962\n",
            "antibody.design.GeneralAntibodyModeler: CDR min start: 2419.6\n",
            "antibody.design.GeneralAntibodyModeler: CDR min end: 2419.62\n",
            "protocols.antibody.design.AntibodyDesignMover: Min Cycle complete\n",
            "protocols.antibody.design.AntibodyDesignMover:\n",
            "protocols.antibody.design.AntibodyDesignMover: SCORE 5 2419.6 1\n",
            "protocols.antibody.design.AntibodyDesignMover: FINAL 5 2419.6 1\n",
            "protocols.antibody.design.AntibodyDesignMover:\n",
            "protocols.antibody.design.AntibodyDesignMover: Last Accepted Score: 2419.6\n",
            "protocols.antibody.design.AntibodyDesignMover: Lowest        Score:2419.6\n",
            "protocols.antibody.design.AntibodyDesignMover:\n",
            "protocols.antibody.design.AntibodyDesignMover: Outer cycles attempted: 5 Outer cycles completed: 5\n",
            "protocols.antibody.design.AntibodyDesignMover:\n",
            "protocols.antibody.design.AntibodyDesignMover: Native Pose (MC): 13400.9\n",
            "protocols.antibody.design.AntibodyDesignMover: Final Pose (MC): 2419.6\n",
            "protocols.antibody.design.AntibodyDesignMover:\n",
            "protocols.moves.TrialCounter:              unk trials=      5;  accepts= 1.0000;  energy_drop/trial= -2196.26513\n",
            "protocols.antibody.design.AntibodyDesignMover:\n",
            "protocols.antibody.design.AntibodyDesignMover: Native Pose (Total): 13400.9\n",
            "protocols.antibody.design.AntibodyDesignMover: Final Pose (Total): 2419.6\n",
            "protocols.antibody.design.AntibodyDesignMover: \n",
            "------------------------------------------------------------\n",
            " Scores                       Weight   Raw Score Wghtd.Score\n",
            "------------------------------------------------------------\n",
            " fa_atr                       1.000   -5629.500   -5629.500\n",
            " fa_rep                       0.550    5119.910    2815.951\n",
            " fa_sol                       1.000    3439.751    3439.751\n",
            " fa_intra_rep                 0.005    2676.875      13.384\n",
            " fa_intra_sol_xover4          1.000     174.662     174.662\n",
            " lk_ball_wtd                  1.000     -67.837     -67.837\n",
            " fa_elec                      1.000   -1588.952   -1588.952\n",
            " pro_close                    1.250       8.630      10.787\n",
            " hbond_sr_bb                  1.000     -72.530     -72.530\n",
            " hbond_lr_bb                  1.000    -505.144    -505.144\n",
            " hbond_bb_sc                  1.000    -136.768    -136.768\n",
            " hbond_sc                     1.000     -73.167     -73.167\n",
            " dslf_fa13                    1.250      -0.958      -1.198\n",
            " atom_pair_constraint         0.010       0.000       0.000\n",
            " omega                        0.400     500.073     200.029\n",
            " fa_dun                       0.700    5068.562    3547.993\n",
            " p_aa_pp                      0.600    -286.747    -172.048\n",
            " yhh_planarity                0.625       0.247       0.154\n",
            " ref                          1.000     375.926     375.926\n",
            " rama_prepro                  0.450     195.787      88.104\n",
            "---------------------------------------------------\n",
            " Total weighted score:                     2419.597\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H1\n",
            "protocols.antibody.cluster.CDRClusterMatcher: Length: 13 Omega: TTTTTTTTTTTTT\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H2\n",
            "protocols.antibody.cluster.CDRClusterMatcher: Length: 10 Omega: TTTTTTTTTT\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H3\n",
            "protocols.antibody.cluster.CDRClusterMatcher: Length: 13 Omega: TTTTTTTTTTTTT\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L1\n",
            "protocols.antibody.cluster.CDRClusterMatcher: Length: 11 Omega: TTTTTTTTTTT\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L2\n",
            "protocols.antibody.cluster.CDRClusterMatcher: Length: 8 Omega: TTTTTTTT\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L3\n",
            "protocols.antibody.cluster.CDRClusterMatcher: Length: 9 Omega: TTTTTTCTT\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using explicit constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using interface constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: [ WARNING ] In InterfaceAnalyzerMover, calculator Sasa_1 already exists, this is hopefully correct for your purposes\n",
            "protocols.analysis.InterfaceAnalyzerMover: [ WARNING ] In InterfaceAnalyzerMover, calculator InterfaceNeighborDefinition_1 already exists, this is hopefully correct for your purposes\n",
            "protocols.analysis.InterfaceAnalyzerMover: [ WARNING ] In InterfaceAnalyzerMover, calculator InterfaceDeltaEnergetics_1 already exists, this is hopefully correct for your purposes\n",
            "protocols.analysis.InterfaceAnalyzerMover: [ WARNING ] In InterfaceAnalyzerMover, calculator NumberHBonds_1 already exists, this is hopefully correct for your purposes\n",
            "protocols.analysis.InterfaceAnalyzerMover: [ WARNING ] In InterfaceAnalyzerMover, calculator BuriedUnsatisfiedPolars_1 already exists, this is hopefully correct for your purposes\n",
            "protocols.analysis.InterfaceAnalyzerMover: Making an interface set with fixed chains\n",
            "protocols.analysis.InterfaceAnalyzerMover: Interface set residues total: 97\n",
            "protocols.analysis.InterfaceAnalyzerMover: NOT detecting disulfides in the separated pose.\n",
            "protocols.analysis.InterfaceAnalyzerMover: Calculating dSASA\n",
            "protocols.analysis.InterfaceAnalyzerMover: Calculating per-res dSASA data\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 1181 rotamers at 97 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.analysis.InterfaceAnalyzerMover: included_nres: 975\n",
            "protocols.analysis.InterfaceAnalyzerMover: Making hbond ScoreFunction with bb-bb energy method options\n",
            "core.scoring.ScoreFunctionFactory: SCOREFUNCTION: ref2015\n",
            "protocols.analysis.InterfaceAnalyzerMover: Found Hbond between chains: 9 and 13\n",
            "protocols.analysis.InterfaceAnalyzerMover: Found Hbond between chains: 12 and 9\n",
            "protocols.analysis.InterfaceAnalyzerMover: Found Hbond between chains: 13 and 10\n",
            "protocols.analysis.InterfaceAnalyzerMover: Computing delta unsat polar residues...\n",
            "protocols.analysis.InterfaceAnalyzerMover: Computing Shape Complementarity Score...\n",
            "protocols.analysis.InterfaceAnalyzerMover: Upstream chain(s) numbers: 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11,\n",
            "protocols.analysis.InterfaceAnalyzerMover: Downstream chain(s) numbers: 12, 13,\n"
          ]
        }
      ],
      "source": [
        "pose = original_pose.clone()\n",
        "rabd2 = AntibodyDesignMover()\n",
        "\n",
        "cdrs = vector1_protocols_antibody_CDRNameEnum()\n",
        "cdrs.append(l1)\n",
        "cdrs.append(l3)\n",
        "\n",
        "rabd2.set_seq_design_cdrs(cdrs)\n",
        "rabd2.set_light_chain(\"kappa\")\n",
        "if not os.getenv(\"DEBUG\"):\n",
        "    rabd2.apply(pose)"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "jzur4rKm2LMw"
      },
      "source": [
        "Score the input pose using the InterfaceAnalayzerMover"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 22,
      "metadata": {
        "id": "BQQOFTYQ2LMw",
        "outputId": "b9468ec9-a99f-44ed-c67e-294ddf77b51a",
        "colab": {
          "base_uri": "https://localhost:8080/"
        }
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "protocols.analysis.InterfaceAnalyzerMover: Repacking separated Pose before running calculations\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using explicit constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using interface constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: [ WARNING ] In InterfaceAnalyzerMover, calculator Sasa_1 already exists, this is hopefully correct for your purposes\n",
            "protocols.analysis.InterfaceAnalyzerMover: [ WARNING ] In InterfaceAnalyzerMover, calculator InterfaceNeighborDefinition_1 already exists, this is hopefully correct for your purposes\n",
            "protocols.analysis.InterfaceAnalyzerMover: [ WARNING ] In InterfaceAnalyzerMover, calculator InterfaceDeltaEnergetics_1 already exists, this is hopefully correct for your purposes\n",
            "protocols.analysis.InterfaceAnalyzerMover: [ WARNING ] In InterfaceAnalyzerMover, calculator NumberHBonds_1 already exists, this is hopefully correct for your purposes\n",
            "protocols.analysis.InterfaceAnalyzerMover: [ WARNING ] In InterfaceAnalyzerMover, calculator BuriedUnsatisfiedPolars_1 already exists, this is hopefully correct for your purposes\n",
            "protocols.analysis.InterfaceAnalyzerMover: Making an interface set with fixed chains\n",
            "protocols.analysis.InterfaceAnalyzerMover: Interface set residues total: 97\n",
            "protocols.analysis.InterfaceAnalyzerMover: NULL scorefunction. Initialize from cmd line.\n",
            "core.scoring.ScoreFunctionFactory: SCOREFUNCTION: ref2015\n",
            "protocols.analysis.InterfaceAnalyzerMover: Detecting disulfides in the separated pose.\n",
            "core.conformation.Conformation: Found disulfide between residues 771 845\n",
            "core.conformation.Conformation: Found disulfide between residues 891 956\n",
            "protocols.analysis.InterfaceAnalyzerMover: Calculating dSASA\n",
            "protocols.analysis.InterfaceAnalyzerMover: Calculating per-res dSASA data\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 1181 rotamers at 97 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.analysis.InterfaceAnalyzerMover: included_nres: 975\n",
            "protocols.analysis.InterfaceAnalyzerMover: Making hbond ScoreFunction with bb-bb energy method options\n",
            "core.scoring.ScoreFunctionFactory: SCOREFUNCTION: ref2015\n",
            "protocols.analysis.InterfaceAnalyzerMover: Found Hbond between chains: 9 and 13\n",
            "protocols.analysis.InterfaceAnalyzerMover: Found Hbond between chains: 12 and 9\n",
            "protocols.analysis.InterfaceAnalyzerMover: Found Hbond between chains: 13 and 10\n",
            "protocols.analysis.InterfaceAnalyzerMover: Computing delta unsat polar residues...\n",
            "protocols.analysis.InterfaceAnalyzerMover: Computing Shape Complementarity Score...\n",
            "protocols.analysis.InterfaceAnalyzerMover: Upstream chain(s) numbers: 12, 13,\n",
            "protocols.analysis.InterfaceAnalyzerMover: Downstream chain(s) numbers: 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11,\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using explicit constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using interface constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Making an interface set with fixed chains\n",
            "protocols.analysis.InterfaceAnalyzerMover: Interface set residues total: 100\n",
            "protocols.analysis.InterfaceAnalyzerMover: Detecting disulfides in the separated pose.\n",
            "core.conformation.Conformation: Found disulfide between residues 771 845\n",
            "core.conformation.Conformation: Found disulfide between residues 891 956\n",
            "protocols.analysis.InterfaceAnalyzerMover: Calculating dSASA\n",
            "protocols.analysis.InterfaceAnalyzerMover: Calculating per-res dSASA data\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 1268 rotamers at 100 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.analysis.InterfaceAnalyzerMover: included_nres: 975\n",
            "protocols.analysis.InterfaceAnalyzerMover: Making hbond ScoreFunction with bb-bb energy method options\n",
            "core.scoring.ScoreFunctionFactory: SCOREFUNCTION: ref2015\n",
            "protocols.analysis.InterfaceAnalyzerMover: Found Hbond between chains: 12 and 9\n",
            "protocols.analysis.InterfaceAnalyzerMover: Found Hbond between chains: 9 and 13\n",
            "protocols.analysis.InterfaceAnalyzerMover: Found Hbond between chains: 12 and 9\n",
            "protocols.analysis.InterfaceAnalyzerMover: Computing delta unsat polar residues...\n",
            "protocols.analysis.InterfaceAnalyzerMover: Computing Shape Complementarity Score...\n",
            "protocols.analysis.InterfaceAnalyzerMover: Upstream chain(s) numbers: 12, 13,\n",
            "protocols.analysis.InterfaceAnalyzerMover: Downstream chain(s) numbers: 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11,\n",
            "dG Diff: -10796.484985351562\n"
          ]
        }
      ],
      "source": [
        "from rosetta.protocols.analysis import InterfaceAnalyzerMover\n",
        "\n",
        "\n",
        "if not os.getenv(\"DEBUG\"):\n",
        "    iam = InterfaceAnalyzerMover(\"LH_ABCDEFGIJKZ\")\n",
        "    iam.set_pack_separated(True)\n",
        "    iam.apply(pose)\n",
        "    iam.apply(original_pose)\n",
        "\n",
        "    #dg_term = \"dG_separated\"\n",
        "    #print(\"dG Diff:\", pose.scores(dg_term) - original_pose(dg_term))\n",
        "\n",
        "    dg_term = \"dG_separated\"\n",
        "    # Correctly access score values using square brackets\n",
        "    dG_diff = pose.scores[dg_term] - original_pose.scores[dg_term]\n",
        "    print(\"dG Diff:\", dG_diff)\n"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "Os-U7koB2LMw"
      },
      "source": [
        "Has the energy gone down after our sequence design?  The `dG_separated` is calculated by scoring the complex, separating the antigen from the antibody, repacking side-chains at the interface, and then taking the difference in score - i.e. the dG.\n",
        "    \n",
        "Lets take a look at scores from a previous run of 5 antibodies.  The scorefiles are in json format, so it will be easy to turn them into pandas Dataframes and do some cool stuff.  We'll do this often as the runtimes increase for our protocol - but all the scores in them can be accessed using the pose.scores attribute (which is PyRosetta-specific functionality.)\n",
        "\n",
        "Are any of these better than our input pose?"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 24,
      "metadata": {
        "id": "sXiTIz3S2LMw",
        "outputId": "81b82af0-7c66-4e46-a4e6-0a84a1bb4eb0",
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 255
        }
      },
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "              decoy  atom_pair_constraint  cdr_cluster_DIS_L1  \\\n",
              "0  tutA1_my_ab_0001                   0.0           10.244154   \n",
              "3  tutA1_my_ab_0004                   0.0            9.604385   \n",
              "1  tutA1_my_ab_0002                   0.0            9.939879   \n",
              "4  tutA1_my_ab_0005                   0.0           13.124744   \n",
              "2  tutA1_my_ab_0003                   0.0            9.693962   \n",
              "\n",
              "   cdr_cluster_DIS_L3  cdr_cluster_LEN_L1  cdr_cluster_LEN_L3  \\\n",
              "0           14.169669                11.0                 9.0   \n",
              "3           10.078285                11.0                 9.0   \n",
              "1           14.834209                11.0                 9.0   \n",
              "4           14.362237                11.0                 9.0   \n",
              "2           12.150679                11.0                 9.0   \n",
              "\n",
              "   complex_normalized  dG_cross  dG_cross/dSASAx100  dG_separated  ...  \\\n",
              "0            2.494409       0.0                 0.0   1784.542480  ...   \n",
              "3            2.459319       0.0                 0.0   1820.213013  ...   \n",
              "1            2.494796       0.0                 0.0   1849.303833  ...   \n",
              "4            2.667474       0.0                 0.0   1934.344604  ...   \n",
              "2            3.530113       0.0                 0.0   2854.750488  ...   \n",
              "\n",
              "         ref  sc_value  side1_normalized  side1_score  side2_normalized  \\\n",
              "0  374.81054  0.491761         14.690318   837.348145         20.825792   \n",
              "3  374.09441  0.596517         14.805477   858.717651         20.484030   \n",
              "1  372.58994  0.468819         17.130129   907.896790         20.831068   \n",
              "4  374.83686  0.415013         17.054668   920.952087         22.980696   \n",
              "2  376.96998  0.464472         24.664927  1381.235962         33.255428   \n",
              "\n",
              "   side2_score  total_score  yhh_planarity  cdr_cluster_ID_L1  \\\n",
              "0   874.683289  2432.049006       0.134353            L1-11-1   \n",
              "3   860.329285  2397.836108       0.420196            L1-11-1   \n",
              "1   874.904846  2432.425954       0.061565            L1-11-1   \n",
              "4   965.189209  2600.787151       0.044588            L1-11-1   \n",
              "2  1396.727905  3441.859648       0.075642            L1-11-1   \n",
              "\n",
              "   cdr_cluster_ID_L3  \n",
              "0        L3-9-cis7-1  \n",
              "3        L3-9-cis7-1  \n",
              "1        L3-9-cis7-1  \n",
              "4        L3-9-cis7-1  \n",
              "2        L3-9-cis7-1  \n",
              "\n",
              "[5 rows x 48 columns]"
            ],
            "text/html": [
              "\n",
              "  <div id=\"df-0aac0e27-151c-40a7-b13c-f0e5f9d362f9\" class=\"colab-df-container\">\n",
              "    <div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>decoy</th>\n",
              "      <th>atom_pair_constraint</th>\n",
              "      <th>cdr_cluster_DIS_L1</th>\n",
              "      <th>cdr_cluster_DIS_L3</th>\n",
              "      <th>cdr_cluster_LEN_L1</th>\n",
              "      <th>cdr_cluster_LEN_L3</th>\n",
              "      <th>complex_normalized</th>\n",
              "      <th>dG_cross</th>\n",
              "      <th>dG_cross/dSASAx100</th>\n",
              "      <th>dG_separated</th>\n",
              "      <th>...</th>\n",
              "      <th>ref</th>\n",
              "      <th>sc_value</th>\n",
              "      <th>side1_normalized</th>\n",
              "      <th>side1_score</th>\n",
              "      <th>side2_normalized</th>\n",
              "      <th>side2_score</th>\n",
              "      <th>total_score</th>\n",
              "      <th>yhh_planarity</th>\n",
              "      <th>cdr_cluster_ID_L1</th>\n",
              "      <th>cdr_cluster_ID_L3</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>tutA1_my_ab_0001</td>\n",
              "      <td>0.0</td>\n",
              "      <td>10.244154</td>\n",
              "      <td>14.169669</td>\n",
              "      <td>11.0</td>\n",
              "      <td>9.0</td>\n",
              "      <td>2.494409</td>\n",
              "      <td>0.0</td>\n",
              "      <td>0.0</td>\n",
              "      <td>1784.542480</td>\n",
              "      <td>...</td>\n",
              "      <td>374.81054</td>\n",
              "      <td>0.491761</td>\n",
              "      <td>14.690318</td>\n",
              "      <td>837.348145</td>\n",
              "      <td>20.825792</td>\n",
              "      <td>874.683289</td>\n",
              "      <td>2432.049006</td>\n",
              "      <td>0.134353</td>\n",
              "      <td>L1-11-1</td>\n",
              "      <td>L3-9-cis7-1</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3</th>\n",
              "      <td>tutA1_my_ab_0004</td>\n",
              "      <td>0.0</td>\n",
              "      <td>9.604385</td>\n",
              "      <td>10.078285</td>\n",
              "      <td>11.0</td>\n",
              "      <td>9.0</td>\n",
              "      <td>2.459319</td>\n",
              "      <td>0.0</td>\n",
              "      <td>0.0</td>\n",
              "      <td>1820.213013</td>\n",
              "      <td>...</td>\n",
              "      <td>374.09441</td>\n",
              "      <td>0.596517</td>\n",
              "      <td>14.805477</td>\n",
              "      <td>858.717651</td>\n",
              "      <td>20.484030</td>\n",
              "      <td>860.329285</td>\n",
              "      <td>2397.836108</td>\n",
              "      <td>0.420196</td>\n",
              "      <td>L1-11-1</td>\n",
              "      <td>L3-9-cis7-1</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>tutA1_my_ab_0002</td>\n",
              "      <td>0.0</td>\n",
              "      <td>9.939879</td>\n",
              "      <td>14.834209</td>\n",
              "      <td>11.0</td>\n",
              "      <td>9.0</td>\n",
              "      <td>2.494796</td>\n",
              "      <td>0.0</td>\n",
              "      <td>0.0</td>\n",
              "      <td>1849.303833</td>\n",
              "      <td>...</td>\n",
              "      <td>372.58994</td>\n",
              "      <td>0.468819</td>\n",
              "      <td>17.130129</td>\n",
              "      <td>907.896790</td>\n",
              "      <td>20.831068</td>\n",
              "      <td>874.904846</td>\n",
              "      <td>2432.425954</td>\n",
              "      <td>0.061565</td>\n",
              "      <td>L1-11-1</td>\n",
              "      <td>L3-9-cis7-1</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4</th>\n",
              "      <td>tutA1_my_ab_0005</td>\n",
              "      <td>0.0</td>\n",
              "      <td>13.124744</td>\n",
              "      <td>14.362237</td>\n",
              "      <td>11.0</td>\n",
              "      <td>9.0</td>\n",
              "      <td>2.667474</td>\n",
              "      <td>0.0</td>\n",
              "      <td>0.0</td>\n",
              "      <td>1934.344604</td>\n",
              "      <td>...</td>\n",
              "      <td>374.83686</td>\n",
              "      <td>0.415013</td>\n",
              "      <td>17.054668</td>\n",
              "      <td>920.952087</td>\n",
              "      <td>22.980696</td>\n",
              "      <td>965.189209</td>\n",
              "      <td>2600.787151</td>\n",
              "      <td>0.044588</td>\n",
              "      <td>L1-11-1</td>\n",
              "      <td>L3-9-cis7-1</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>tutA1_my_ab_0003</td>\n",
              "      <td>0.0</td>\n",
              "      <td>9.693962</td>\n",
              "      <td>12.150679</td>\n",
              "      <td>11.0</td>\n",
              "      <td>9.0</td>\n",
              "      <td>3.530113</td>\n",
              "      <td>0.0</td>\n",
              "      <td>0.0</td>\n",
              "      <td>2854.750488</td>\n",
              "      <td>...</td>\n",
              "      <td>376.96998</td>\n",
              "      <td>0.464472</td>\n",
              "      <td>24.664927</td>\n",
              "      <td>1381.235962</td>\n",
              "      <td>33.255428</td>\n",
              "      <td>1396.727905</td>\n",
              "      <td>3441.859648</td>\n",
              "      <td>0.075642</td>\n",
              "      <td>L1-11-1</td>\n",
              "      <td>L3-9-cis7-1</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "<p>5 rows  48 columns</p>\n",
              "</div>\n",
              "    <div class=\"colab-df-buttons\">\n",
              "\n",
              "  <div class=\"colab-df-container\">\n",
              "    <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-0aac0e27-151c-40a7-b13c-f0e5f9d362f9')\"\n",
              "            title=\"Convert this dataframe to an interactive table.\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\" viewBox=\"0 -960 960 960\">\n",
              "    <path d=\"M120-120v-720h720v720H120Zm60-500h600v-160H180v160Zm220 220h160v-160H400v160Zm0 220h160v-160H400v160ZM180-400h160v-160H180v160Zm440 0h160v-160H620v160ZM180-180h160v-160H180v160Zm440 0h160v-160H620v160Z\"/>\n",
              "  </svg>\n",
              "    </button>\n",
              "\n",
              "  <style>\n",
              "    .colab-df-container {\n",
              "      display:flex;\n",
              "      gap: 12px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert {\n",
              "      background-color: #E8F0FE;\n",
              "      border: none;\n",
              "      border-radius: 50%;\n",
              "      cursor: pointer;\n",
              "      display: none;\n",
              "      fill: #1967D2;\n",
              "      height: 32px;\n",
              "      padding: 0 0 0 0;\n",
              "      width: 32px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert:hover {\n",
              "      background-color: #E2EBFA;\n",
              "      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "      fill: #174EA6;\n",
              "    }\n",
              "\n",
              "    .colab-df-buttons div {\n",
              "      margin-bottom: 4px;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert {\n",
              "      background-color: #3B4455;\n",
              "      fill: #D2E3FC;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert:hover {\n",
              "      background-color: #434B5C;\n",
              "      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
              "      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
              "      fill: #FFFFFF;\n",
              "    }\n",
              "  </style>\n",
              "\n",
              "    <script>\n",
              "      const buttonEl =\n",
              "        document.querySelector('#df-0aac0e27-151c-40a7-b13c-f0e5f9d362f9 button.colab-df-convert');\n",
              "      buttonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "\n",
              "      async function convertToInteractive(key) {\n",
              "        const element = document.querySelector('#df-0aac0e27-151c-40a7-b13c-f0e5f9d362f9');\n",
              "        const dataTable =\n",
              "          await google.colab.kernel.invokeFunction('convertToInteractive',\n",
              "                                                    [key], {});\n",
              "        if (!dataTable) return;\n",
              "\n",
              "        const docLinkHtml = 'Like what you see? Visit the ' +\n",
              "          '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n",
              "          + ' to learn more about interactive tables.';\n",
              "        element.innerHTML = '';\n",
              "        dataTable['output_type'] = 'display_data';\n",
              "        await google.colab.output.renderOutput(dataTable, element);\n",
              "        const docLink = document.createElement('div');\n",
              "        docLink.innerHTML = docLinkHtml;\n",
              "        element.appendChild(docLink);\n",
              "      }\n",
              "    </script>\n",
              "  </div>\n",
              "\n",
              "\n",
              "<div id=\"df-394ae007-88ad-4179-94a8-fdd842b31ff0\">\n",
              "  <button class=\"colab-df-quickchart\" onclick=\"quickchart('df-394ae007-88ad-4179-94a8-fdd842b31ff0')\"\n",
              "            title=\"Suggest charts\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "<svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
              "     width=\"24px\">\n",
              "    <g>\n",
              "        <path d=\"M19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zM9 17H7v-7h2v7zm4 0h-2V7h2v10zm4 0h-2v-4h2v4z\"/>\n",
              "    </g>\n",
              "</svg>\n",
              "  </button>\n",
              "\n",
              "<style>\n",
              "  .colab-df-quickchart {\n",
              "      --bg-color: #E8F0FE;\n",
              "      --fill-color: #1967D2;\n",
              "      --hover-bg-color: #E2EBFA;\n",
              "      --hover-fill-color: #174EA6;\n",
              "      --disabled-fill-color: #AAA;\n",
              "      --disabled-bg-color: #DDD;\n",
              "  }\n",
              "\n",
              "  [theme=dark] .colab-df-quickchart {\n",
              "      --bg-color: #3B4455;\n",
              "      --fill-color: #D2E3FC;\n",
              "      --hover-bg-color: #434B5C;\n",
              "      --hover-fill-color: #FFFFFF;\n",
              "      --disabled-bg-color: #3B4455;\n",
              "      --disabled-fill-color: #666;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart {\n",
              "    background-color: var(--bg-color);\n",
              "    border: none;\n",
              "    border-radius: 50%;\n",
              "    cursor: pointer;\n",
              "    display: none;\n",
              "    fill: var(--fill-color);\n",
              "    height: 32px;\n",
              "    padding: 0;\n",
              "    width: 32px;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart:hover {\n",
              "    background-color: var(--hover-bg-color);\n",
              "    box-shadow: 0 1px 2px rgba(60, 64, 67, 0.3), 0 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "    fill: var(--button-hover-fill-color);\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart-complete:disabled,\n",
              "  .colab-df-quickchart-complete:disabled:hover {\n",
              "    background-color: var(--disabled-bg-color);\n",
              "    fill: var(--disabled-fill-color);\n",
              "    box-shadow: none;\n",
              "  }\n",
              "\n",
              "  .colab-df-spinner {\n",
              "    border: 2px solid var(--fill-color);\n",
              "    border-color: transparent;\n",
              "    border-bottom-color: var(--fill-color);\n",
              "    animation:\n",
              "      spin 1s steps(1) infinite;\n",
              "  }\n",
              "\n",
              "  @keyframes spin {\n",
              "    0% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "      border-left-color: var(--fill-color);\n",
              "    }\n",
              "    20% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    30% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    40% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    60% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    80% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "    90% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "  }\n",
              "</style>\n",
              "\n",
              "  <script>\n",
              "    async function quickchart(key) {\n",
              "      const quickchartButtonEl =\n",
              "        document.querySelector('#' + key + ' button');\n",
              "      quickchartButtonEl.disabled = true;  // To prevent multiple clicks.\n",
              "      quickchartButtonEl.classList.add('colab-df-spinner');\n",
              "      try {\n",
              "        const charts = await google.colab.kernel.invokeFunction(\n",
              "            'suggestCharts', [key], {});\n",
              "      } catch (error) {\n",
              "        console.error('Error during call to suggestCharts:', error);\n",
              "      }\n",
              "      quickchartButtonEl.classList.remove('colab-df-spinner');\n",
              "      quickchartButtonEl.classList.add('colab-df-quickchart-complete');\n",
              "    }\n",
              "    (() => {\n",
              "      let quickchartButtonEl =\n",
              "        document.querySelector('#df-394ae007-88ad-4179-94a8-fdd842b31ff0 button');\n",
              "      quickchartButtonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "    })();\n",
              "  </script>\n",
              "</div>\n",
              "\n",
              "  <div id=\"id_9ee97160-4e4d-4801-b5b4-ea52c0506a02\">\n",
              "    <style>\n",
              "      .colab-df-generate {\n",
              "        background-color: #E8F0FE;\n",
              "        border: none;\n",
              "        border-radius: 50%;\n",
              "        cursor: pointer;\n",
              "        display: none;\n",
              "        fill: #1967D2;\n",
              "        height: 32px;\n",
              "        padding: 0 0 0 0;\n",
              "        width: 32px;\n",
              "      }\n",
              "\n",
              "      .colab-df-generate:hover {\n",
              "        background-color: #E2EBFA;\n",
              "        box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "        fill: #174EA6;\n",
              "      }\n",
              "\n",
              "      [theme=dark] .colab-df-generate {\n",
              "        background-color: #3B4455;\n",
              "        fill: #D2E3FC;\n",
              "      }\n",
              "\n",
              "      [theme=dark] .colab-df-generate:hover {\n",
              "        background-color: #434B5C;\n",
              "        box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
              "        filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
              "        fill: #FFFFFF;\n",
              "      }\n",
              "    </style>\n",
              "    <button class=\"colab-df-generate\" onclick=\"generateWithVariable('df')\"\n",
              "            title=\"Generate code using this dataframe.\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
              "       width=\"24px\">\n",
              "    <path d=\"M7,19H8.4L18.45,9,17,7.55,7,17.6ZM5,21V16.75L18.45,3.32a2,2,0,0,1,2.83,0l1.4,1.43a1.91,1.91,0,0,1,.58,1.4,1.91,1.91,0,0,1-.58,1.4L9.25,21ZM18.45,9,17,7.55Zm-12,3A5.31,5.31,0,0,0,4.9,8.1,5.31,5.31,0,0,0,1,6.5,5.31,5.31,0,0,0,4.9,4.9,5.31,5.31,0,0,0,6.5,1,5.31,5.31,0,0,0,8.1,4.9,5.31,5.31,0,0,0,12,6.5,5.46,5.46,0,0,0,6.5,12Z\"/>\n",
              "  </svg>\n",
              "    </button>\n",
              "    <script>\n",
              "      (() => {\n",
              "      const buttonEl =\n",
              "        document.querySelector('#id_9ee97160-4e4d-4801-b5b4-ea52c0506a02 button.colab-df-generate');\n",
              "      buttonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "\n",
              "      buttonEl.onclick = () => {\n",
              "        google.colab.notebook.generateWithVariable('df');\n",
              "      }\n",
              "      })();\n",
              "    </script>\n",
              "  </div>\n",
              "\n",
              "    </div>\n",
              "  </div>\n"
            ],
            "application/vnd.google.colaboratory.intrinsic+json": {
              "type": "dataframe",
              "variable_name": "df"
            }
          },
          "metadata": {},
          "execution_count": 24
        }
      ],
      "source": [
        "df = load_json_scorefile(\"expected_outputs/tutA1_score.sc\")\n",
        "df = drop_cluster_columns(df, keep_cdrs=[\"L1\", \"L3\"])\n",
        "df"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "3jjh5WB82LMx"
      },
      "source": [
        "### Tut A2. Graft Design\n",
        "\n",
        "Now we will be enabling graft design AND sequence design on L1 and L3 loops.  With an nstruct (n decoys) of 5, we are doing 25 design trials total - IE 25 actual grafts.\n",
        "\n",
        "\n",
        "    <AntibodyDesignMover name=\"RAbD\" seq_design_cdrs=\"L1,L3\" graft_design_cdrs=\"L1,L3\">\n",
        "\n",
        "\n",
        "This will take a about 2-3 times as long as sequence design, as grafting a non-breaking loop takes time.  This was 738 seconds on my laptop to generate 5. Here, you will generate 1 at about 150 seconds  Ouptut structures and scores are in ../expected_outputs/rabd.\n",
        "\n",
        "\n",
        "\n"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "upAG9MfW2LMx"
      },
      "source": [
        "Typically, we require a much higher `-outer_cycle_rounds` and number of decoys to see anything significant.  Did this improve energies in your single antibody?  How about our pre-generated ones? Load and take a look at the scorefile as a pandas DataFrame as we did above (`expected_outputs/rabd/tutA2_score.sc`).\n"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 25,
      "metadata": {
        "nbgrader": {
          "grade": true,
          "grade_id": "cell-6e0d61e14fe45905",
          "locked": false,
          "points": 0,
          "schema_version": 3,
          "solution": true
        },
        "id": "T3Y3RjLn2LMx",
        "outputId": "b83e05d9-00cd-4074-ede2-806f85f9a72f",
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 255
        }
      },
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "              decoy  atom_pair_constraint  cdr_cluster_DIS_L1  \\\n",
              "4  tutA2_my_ab_0005                   0.0           11.192533   \n",
              "2  tutA2_my_ab_0003                   0.0           20.217281   \n",
              "0  tutA2_my_ab_0001                   0.0            8.599037   \n",
              "1  tutA2_my_ab_0002                   0.0           29.971132   \n",
              "3  tutA2_my_ab_0004                   0.0           12.153269   \n",
              "\n",
              "   cdr_cluster_DIS_L3  cdr_cluster_LEN_L1  cdr_cluster_LEN_L3  \\\n",
              "4           31.677544                11.0                 7.0   \n",
              "2           24.178181                15.0                11.0   \n",
              "0           27.562218                11.0                10.0   \n",
              "1           45.571491                17.0                 9.0   \n",
              "3           17.002886                10.0                 8.0   \n",
              "\n",
              "   complex_normalized  dG_cross  dG_cross/dSASAx100  dG_separated  ...  \\\n",
              "4            2.450019       0.0                 0.0   1751.548462  ...   \n",
              "2            2.760204       0.0                 0.0   1843.231445  ...   \n",
              "0            3.348572       0.0                 0.0   2484.706299  ...   \n",
              "1            3.352178       0.0                 0.0   2587.653564  ...   \n",
              "3            5.869139       0.0                 0.0   5082.230469  ...   \n",
              "\n",
              "         ref  sc_value  side1_normalized  side1_score  side2_normalized  \\\n",
              "4  364.65868  0.564960         15.000569   825.031311         21.473850   \n",
              "2  375.17336  0.537087         13.518681   851.676880         21.286449   \n",
              "0  374.05882  0.486206         22.621260  1198.926758         29.544832   \n",
              "1  373.00912  0.620726         18.605259  1265.157593         28.719591   \n",
              "3  376.47676  0.571551         42.399551  2501.573486         62.364841   \n",
              "\n",
              "   side2_score  total_score  yhh_planarity  cdr_cluster_ID_L1  \\\n",
              "4   837.480164  2383.868655       0.107882            L1-11-1   \n",
              "2   979.176697  2707.759790       0.221997            L1-15-1   \n",
              "0  1270.427734  3268.206329       0.361406            L1-11-1   \n",
              "1  1292.381592  3288.486156       0.126786            L1-17-1   \n",
              "3  2494.593750  5710.672501       0.100791            L1-10-2   \n",
              "\n",
              "   cdr_cluster_ID_L3  \n",
              "4             L3-7-1  \n",
              "2            L3-11-1  \n",
              "0     L3-10-cis7,8-1  \n",
              "1             L3-9-1  \n",
              "3             L3-8-1  \n",
              "\n",
              "[5 rows x 48 columns]"
            ],
            "text/html": [
              "\n",
              "  <div id=\"df-a92597af-241b-4abd-af2a-1e63ef6a8333\" class=\"colab-df-container\">\n",
              "    <div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>decoy</th>\n",
              "      <th>atom_pair_constraint</th>\n",
              "      <th>cdr_cluster_DIS_L1</th>\n",
              "      <th>cdr_cluster_DIS_L3</th>\n",
              "      <th>cdr_cluster_LEN_L1</th>\n",
              "      <th>cdr_cluster_LEN_L3</th>\n",
              "      <th>complex_normalized</th>\n",
              "      <th>dG_cross</th>\n",
              "      <th>dG_cross/dSASAx100</th>\n",
              "      <th>dG_separated</th>\n",
              "      <th>...</th>\n",
              "      <th>ref</th>\n",
              "      <th>sc_value</th>\n",
              "      <th>side1_normalized</th>\n",
              "      <th>side1_score</th>\n",
              "      <th>side2_normalized</th>\n",
              "      <th>side2_score</th>\n",
              "      <th>total_score</th>\n",
              "      <th>yhh_planarity</th>\n",
              "      <th>cdr_cluster_ID_L1</th>\n",
              "      <th>cdr_cluster_ID_L3</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>4</th>\n",
              "      <td>tutA2_my_ab_0005</td>\n",
              "      <td>0.0</td>\n",
              "      <td>11.192533</td>\n",
              "      <td>31.677544</td>\n",
              "      <td>11.0</td>\n",
              "      <td>7.0</td>\n",
              "      <td>2.450019</td>\n",
              "      <td>0.0</td>\n",
              "      <td>0.0</td>\n",
              "      <td>1751.548462</td>\n",
              "      <td>...</td>\n",
              "      <td>364.65868</td>\n",
              "      <td>0.564960</td>\n",
              "      <td>15.000569</td>\n",
              "      <td>825.031311</td>\n",
              "      <td>21.473850</td>\n",
              "      <td>837.480164</td>\n",
              "      <td>2383.868655</td>\n",
              "      <td>0.107882</td>\n",
              "      <td>L1-11-1</td>\n",
              "      <td>L3-7-1</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>tutA2_my_ab_0003</td>\n",
              "      <td>0.0</td>\n",
              "      <td>20.217281</td>\n",
              "      <td>24.178181</td>\n",
              "      <td>15.0</td>\n",
              "      <td>11.0</td>\n",
              "      <td>2.760204</td>\n",
              "      <td>0.0</td>\n",
              "      <td>0.0</td>\n",
              "      <td>1843.231445</td>\n",
              "      <td>...</td>\n",
              "      <td>375.17336</td>\n",
              "      <td>0.537087</td>\n",
              "      <td>13.518681</td>\n",
              "      <td>851.676880</td>\n",
              "      <td>21.286449</td>\n",
              "      <td>979.176697</td>\n",
              "      <td>2707.759790</td>\n",
              "      <td>0.221997</td>\n",
              "      <td>L1-15-1</td>\n",
              "      <td>L3-11-1</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>tutA2_my_ab_0001</td>\n",
              "      <td>0.0</td>\n",
              "      <td>8.599037</td>\n",
              "      <td>27.562218</td>\n",
              "      <td>11.0</td>\n",
              "      <td>10.0</td>\n",
              "      <td>3.348572</td>\n",
              "      <td>0.0</td>\n",
              "      <td>0.0</td>\n",
              "      <td>2484.706299</td>\n",
              "      <td>...</td>\n",
              "      <td>374.05882</td>\n",
              "      <td>0.486206</td>\n",
              "      <td>22.621260</td>\n",
              "      <td>1198.926758</td>\n",
              "      <td>29.544832</td>\n",
              "      <td>1270.427734</td>\n",
              "      <td>3268.206329</td>\n",
              "      <td>0.361406</td>\n",
              "      <td>L1-11-1</td>\n",
              "      <td>L3-10-cis7,8-1</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>tutA2_my_ab_0002</td>\n",
              "      <td>0.0</td>\n",
              "      <td>29.971132</td>\n",
              "      <td>45.571491</td>\n",
              "      <td>17.0</td>\n",
              "      <td>9.0</td>\n",
              "      <td>3.352178</td>\n",
              "      <td>0.0</td>\n",
              "      <td>0.0</td>\n",
              "      <td>2587.653564</td>\n",
              "      <td>...</td>\n",
              "      <td>373.00912</td>\n",
              "      <td>0.620726</td>\n",
              "      <td>18.605259</td>\n",
              "      <td>1265.157593</td>\n",
              "      <td>28.719591</td>\n",
              "      <td>1292.381592</td>\n",
              "      <td>3288.486156</td>\n",
              "      <td>0.126786</td>\n",
              "      <td>L1-17-1</td>\n",
              "      <td>L3-9-1</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3</th>\n",
              "      <td>tutA2_my_ab_0004</td>\n",
              "      <td>0.0</td>\n",
              "      <td>12.153269</td>\n",
              "      <td>17.002886</td>\n",
              "      <td>10.0</td>\n",
              "      <td>8.0</td>\n",
              "      <td>5.869139</td>\n",
              "      <td>0.0</td>\n",
              "      <td>0.0</td>\n",
              "      <td>5082.230469</td>\n",
              "      <td>...</td>\n",
              "      <td>376.47676</td>\n",
              "      <td>0.571551</td>\n",
              "      <td>42.399551</td>\n",
              "      <td>2501.573486</td>\n",
              "      <td>62.364841</td>\n",
              "      <td>2494.593750</td>\n",
              "      <td>5710.672501</td>\n",
              "      <td>0.100791</td>\n",
              "      <td>L1-10-2</td>\n",
              "      <td>L3-8-1</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "<p>5 rows  48 columns</p>\n",
              "</div>\n",
              "    <div class=\"colab-df-buttons\">\n",
              "\n",
              "  <div class=\"colab-df-container\">\n",
              "    <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-a92597af-241b-4abd-af2a-1e63ef6a8333')\"\n",
              "            title=\"Convert this dataframe to an interactive table.\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\" viewBox=\"0 -960 960 960\">\n",
              "    <path d=\"M120-120v-720h720v720H120Zm60-500h600v-160H180v160Zm220 220h160v-160H400v160Zm0 220h160v-160H400v160ZM180-400h160v-160H180v160Zm440 0h160v-160H620v160ZM180-180h160v-160H180v160Zm440 0h160v-160H620v160Z\"/>\n",
              "  </svg>\n",
              "    </button>\n",
              "\n",
              "  <style>\n",
              "    .colab-df-container {\n",
              "      display:flex;\n",
              "      gap: 12px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert {\n",
              "      background-color: #E8F0FE;\n",
              "      border: none;\n",
              "      border-radius: 50%;\n",
              "      cursor: pointer;\n",
              "      display: none;\n",
              "      fill: #1967D2;\n",
              "      height: 32px;\n",
              "      padding: 0 0 0 0;\n",
              "      width: 32px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert:hover {\n",
              "      background-color: #E2EBFA;\n",
              "      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "      fill: #174EA6;\n",
              "    }\n",
              "\n",
              "    .colab-df-buttons div {\n",
              "      margin-bottom: 4px;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert {\n",
              "      background-color: #3B4455;\n",
              "      fill: #D2E3FC;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert:hover {\n",
              "      background-color: #434B5C;\n",
              "      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
              "      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
              "      fill: #FFFFFF;\n",
              "    }\n",
              "  </style>\n",
              "\n",
              "    <script>\n",
              "      const buttonEl =\n",
              "        document.querySelector('#df-a92597af-241b-4abd-af2a-1e63ef6a8333 button.colab-df-convert');\n",
              "      buttonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "\n",
              "      async function convertToInteractive(key) {\n",
              "        const element = document.querySelector('#df-a92597af-241b-4abd-af2a-1e63ef6a8333');\n",
              "        const dataTable =\n",
              "          await google.colab.kernel.invokeFunction('convertToInteractive',\n",
              "                                                    [key], {});\n",
              "        if (!dataTable) return;\n",
              "\n",
              "        const docLinkHtml = 'Like what you see? Visit the ' +\n",
              "          '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n",
              "          + ' to learn more about interactive tables.';\n",
              "        element.innerHTML = '';\n",
              "        dataTable['output_type'] = 'display_data';\n",
              "        await google.colab.output.renderOutput(dataTable, element);\n",
              "        const docLink = document.createElement('div');\n",
              "        docLink.innerHTML = docLinkHtml;\n",
              "        element.appendChild(docLink);\n",
              "      }\n",
              "    </script>\n",
              "  </div>\n",
              "\n",
              "\n",
              "<div id=\"df-05631be8-bf1b-4d94-b334-cf701c5f07d9\">\n",
              "  <button class=\"colab-df-quickchart\" onclick=\"quickchart('df-05631be8-bf1b-4d94-b334-cf701c5f07d9')\"\n",
              "            title=\"Suggest charts\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "<svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
              "     width=\"24px\">\n",
              "    <g>\n",
              "        <path d=\"M19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zM9 17H7v-7h2v7zm4 0h-2V7h2v10zm4 0h-2v-4h2v4z\"/>\n",
              "    </g>\n",
              "</svg>\n",
              "  </button>\n",
              "\n",
              "<style>\n",
              "  .colab-df-quickchart {\n",
              "      --bg-color: #E8F0FE;\n",
              "      --fill-color: #1967D2;\n",
              "      --hover-bg-color: #E2EBFA;\n",
              "      --hover-fill-color: #174EA6;\n",
              "      --disabled-fill-color: #AAA;\n",
              "      --disabled-bg-color: #DDD;\n",
              "  }\n",
              "\n",
              "  [theme=dark] .colab-df-quickchart {\n",
              "      --bg-color: #3B4455;\n",
              "      --fill-color: #D2E3FC;\n",
              "      --hover-bg-color: #434B5C;\n",
              "      --hover-fill-color: #FFFFFF;\n",
              "      --disabled-bg-color: #3B4455;\n",
              "      --disabled-fill-color: #666;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart {\n",
              "    background-color: var(--bg-color);\n",
              "    border: none;\n",
              "    border-radius: 50%;\n",
              "    cursor: pointer;\n",
              "    display: none;\n",
              "    fill: var(--fill-color);\n",
              "    height: 32px;\n",
              "    padding: 0;\n",
              "    width: 32px;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart:hover {\n",
              "    background-color: var(--hover-bg-color);\n",
              "    box-shadow: 0 1px 2px rgba(60, 64, 67, 0.3), 0 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "    fill: var(--button-hover-fill-color);\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart-complete:disabled,\n",
              "  .colab-df-quickchart-complete:disabled:hover {\n",
              "    background-color: var(--disabled-bg-color);\n",
              "    fill: var(--disabled-fill-color);\n",
              "    box-shadow: none;\n",
              "  }\n",
              "\n",
              "  .colab-df-spinner {\n",
              "    border: 2px solid var(--fill-color);\n",
              "    border-color: transparent;\n",
              "    border-bottom-color: var(--fill-color);\n",
              "    animation:\n",
              "      spin 1s steps(1) infinite;\n",
              "  }\n",
              "\n",
              "  @keyframes spin {\n",
              "    0% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "      border-left-color: var(--fill-color);\n",
              "    }\n",
              "    20% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    30% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    40% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    60% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    80% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "    90% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "  }\n",
              "</style>\n",
              "\n",
              "  <script>\n",
              "    async function quickchart(key) {\n",
              "      const quickchartButtonEl =\n",
              "        document.querySelector('#' + key + ' button');\n",
              "      quickchartButtonEl.disabled = true;  // To prevent multiple clicks.\n",
              "      quickchartButtonEl.classList.add('colab-df-spinner');\n",
              "      try {\n",
              "        const charts = await google.colab.kernel.invokeFunction(\n",
              "            'suggestCharts', [key], {});\n",
              "      } catch (error) {\n",
              "        console.error('Error during call to suggestCharts:', error);\n",
              "      }\n",
              "      quickchartButtonEl.classList.remove('colab-df-spinner');\n",
              "      quickchartButtonEl.classList.add('colab-df-quickchart-complete');\n",
              "    }\n",
              "    (() => {\n",
              "      let quickchartButtonEl =\n",
              "        document.querySelector('#df-05631be8-bf1b-4d94-b334-cf701c5f07d9 button');\n",
              "      quickchartButtonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "    })();\n",
              "  </script>\n",
              "</div>\n",
              "\n",
              "  <div id=\"id_a679375a-c666-411e-95de-9f9b8ccf3d53\">\n",
              "    <style>\n",
              "      .colab-df-generate {\n",
              "        background-color: #E8F0FE;\n",
              "        border: none;\n",
              "        border-radius: 50%;\n",
              "        cursor: pointer;\n",
              "        display: none;\n",
              "        fill: #1967D2;\n",
              "        height: 32px;\n",
              "        padding: 0 0 0 0;\n",
              "        width: 32px;\n",
              "      }\n",
              "\n",
              "      .colab-df-generate:hover {\n",
              "        background-color: #E2EBFA;\n",
              "        box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "        fill: #174EA6;\n",
              "      }\n",
              "\n",
              "      [theme=dark] .colab-df-generate {\n",
              "        background-color: #3B4455;\n",
              "        fill: #D2E3FC;\n",
              "      }\n",
              "\n",
              "      [theme=dark] .colab-df-generate:hover {\n",
              "        background-color: #434B5C;\n",
              "        box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
              "        filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
              "        fill: #FFFFFF;\n",
              "      }\n",
              "    </style>\n",
              "    <button class=\"colab-df-generate\" onclick=\"generateWithVariable('df_a2')\"\n",
              "            title=\"Generate code using this dataframe.\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
              "       width=\"24px\">\n",
              "    <path d=\"M7,19H8.4L18.45,9,17,7.55,7,17.6ZM5,21V16.75L18.45,3.32a2,2,0,0,1,2.83,0l1.4,1.43a1.91,1.91,0,0,1,.58,1.4,1.91,1.91,0,0,1-.58,1.4L9.25,21ZM18.45,9,17,7.55Zm-12,3A5.31,5.31,0,0,0,4.9,8.1,5.31,5.31,0,0,0,1,6.5,5.31,5.31,0,0,0,4.9,4.9,5.31,5.31,0,0,0,6.5,1,5.31,5.31,0,0,0,8.1,4.9,5.31,5.31,0,0,0,12,6.5,5.46,5.46,0,0,0,6.5,12Z\"/>\n",
              "  </svg>\n",
              "    </button>\n",
              "    <script>\n",
              "      (() => {\n",
              "      const buttonEl =\n",
              "        document.querySelector('#id_a679375a-c666-411e-95de-9f9b8ccf3d53 button.colab-df-generate');\n",
              "      buttonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "\n",
              "      buttonEl.onclick = () => {\n",
              "        google.colab.notebook.generateWithVariable('df_a2');\n",
              "      }\n",
              "      })();\n",
              "    </script>\n",
              "  </div>\n",
              "\n",
              "    </div>\n",
              "  </div>\n"
            ],
            "application/vnd.google.colaboratory.intrinsic+json": {
              "type": "dataframe",
              "variable_name": "df_a2"
            }
          },
          "metadata": {},
          "execution_count": 25
        }
      ],
      "source": [
        "### BEGIN SOLUTION\n",
        "\n",
        "df_a2 = load_json_scorefile(\"expected_outputs/tutA2_score.sc\")\n",
        "df_a2 = drop_cluster_columns(df_a2, keep_cdrs=[\"L1\", \"L3\"])\n",
        "df_a2\n",
        "\n",
        "### END SOLUTION"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "NLFl5umK2LMy"
      },
      "source": [
        "Lets merge these dataframes, sort by dG_separated, and see if any of our graft-design models did better."
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 26,
      "metadata": {
        "id": "LZll-0yX2LMy",
        "outputId": "d9c6e56f-a053-4e12-8ad1-449786294aa8",
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 412
        }
      },
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "              decoy  atom_pair_constraint  cdr_cluster_DIS_L1  \\\n",
              "5  tutA2_my_ab_0005                   0.0           11.192533   \n",
              "0  tutA1_my_ab_0001                   0.0           10.244154   \n",
              "1  tutA1_my_ab_0004                   0.0            9.604385   \n",
              "6  tutA2_my_ab_0003                   0.0           20.217281   \n",
              "2  tutA1_my_ab_0002                   0.0            9.939879   \n",
              "3  tutA1_my_ab_0005                   0.0           13.124744   \n",
              "7  tutA2_my_ab_0001                   0.0            8.599037   \n",
              "8  tutA2_my_ab_0002                   0.0           29.971132   \n",
              "4  tutA1_my_ab_0003                   0.0            9.693962   \n",
              "9  tutA2_my_ab_0004                   0.0           12.153269   \n",
              "\n",
              "   cdr_cluster_DIS_L3  cdr_cluster_LEN_L1  cdr_cluster_LEN_L3  \\\n",
              "5           31.677544                11.0                 7.0   \n",
              "0           14.169669                11.0                 9.0   \n",
              "1           10.078285                11.0                 9.0   \n",
              "6           24.178181                15.0                11.0   \n",
              "2           14.834209                11.0                 9.0   \n",
              "3           14.362237                11.0                 9.0   \n",
              "7           27.562218                11.0                10.0   \n",
              "8           45.571491                17.0                 9.0   \n",
              "4           12.150679                11.0                 9.0   \n",
              "9           17.002886                10.0                 8.0   \n",
              "\n",
              "   complex_normalized  dG_cross  dG_cross/dSASAx100  dG_separated  ...  \\\n",
              "5            2.450019       0.0                 0.0   1751.548462  ...   \n",
              "0            2.494409       0.0                 0.0   1784.542480  ...   \n",
              "1            2.459319       0.0                 0.0   1820.213013  ...   \n",
              "6            2.760204       0.0                 0.0   1843.231445  ...   \n",
              "2            2.494796       0.0                 0.0   1849.303833  ...   \n",
              "3            2.667474       0.0                 0.0   1934.344604  ...   \n",
              "7            3.348572       0.0                 0.0   2484.706299  ...   \n",
              "8            3.352178       0.0                 0.0   2587.653564  ...   \n",
              "4            3.530113       0.0                 0.0   2854.750488  ...   \n",
              "9            5.869139       0.0                 0.0   5082.230469  ...   \n",
              "\n",
              "         ref  sc_value  side1_normalized  side1_score  side2_normalized  \\\n",
              "5  364.65868  0.564960         15.000569   825.031311         21.473850   \n",
              "0  374.81054  0.491761         14.690318   837.348145         20.825792   \n",
              "1  374.09441  0.596517         14.805477   858.717651         20.484030   \n",
              "6  375.17336  0.537087         13.518681   851.676880         21.286449   \n",
              "2  372.58994  0.468819         17.130129   907.896790         20.831068   \n",
              "3  374.83686  0.415013         17.054668   920.952087         22.980696   \n",
              "7  374.05882  0.486206         22.621260  1198.926758         29.544832   \n",
              "8  373.00912  0.620726         18.605259  1265.157593         28.719591   \n",
              "4  376.96998  0.464472         24.664927  1381.235962         33.255428   \n",
              "9  376.47676  0.571551         42.399551  2501.573486         62.364841   \n",
              "\n",
              "   side2_score  total_score  yhh_planarity  cdr_cluster_ID_L1  \\\n",
              "5   837.480164  2383.868655       0.107882            L1-11-1   \n",
              "0   874.683289  2432.049006       0.134353            L1-11-1   \n",
              "1   860.329285  2397.836108       0.420196            L1-11-1   \n",
              "6   979.176697  2707.759790       0.221997            L1-15-1   \n",
              "2   874.904846  2432.425954       0.061565            L1-11-1   \n",
              "3   965.189209  2600.787151       0.044588            L1-11-1   \n",
              "7  1270.427734  3268.206329       0.361406            L1-11-1   \n",
              "8  1292.381592  3288.486156       0.126786            L1-17-1   \n",
              "4  1396.727905  3441.859648       0.075642            L1-11-1   \n",
              "9  2494.593750  5710.672501       0.100791            L1-10-2   \n",
              "\n",
              "   cdr_cluster_ID_L3  \n",
              "5             L3-7-1  \n",
              "0        L3-9-cis7-1  \n",
              "1        L3-9-cis7-1  \n",
              "6            L3-11-1  \n",
              "2        L3-9-cis7-1  \n",
              "3        L3-9-cis7-1  \n",
              "7     L3-10-cis7,8-1  \n",
              "8             L3-9-1  \n",
              "4        L3-9-cis7-1  \n",
              "9             L3-8-1  \n",
              "\n",
              "[10 rows x 48 columns]"
            ],
            "text/html": [
              "\n",
              "  <div id=\"df-1f6a5ec5-6cff-4e88-a0c3-616d9cb503fb\" class=\"colab-df-container\">\n",
              "    <div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>decoy</th>\n",
              "      <th>atom_pair_constraint</th>\n",
              "      <th>cdr_cluster_DIS_L1</th>\n",
              "      <th>cdr_cluster_DIS_L3</th>\n",
              "      <th>cdr_cluster_LEN_L1</th>\n",
              "      <th>cdr_cluster_LEN_L3</th>\n",
              "      <th>complex_normalized</th>\n",
              "      <th>dG_cross</th>\n",
              "      <th>dG_cross/dSASAx100</th>\n",
              "      <th>dG_separated</th>\n",
              "      <th>...</th>\n",
              "      <th>ref</th>\n",
              "      <th>sc_value</th>\n",
              "      <th>side1_normalized</th>\n",
              "      <th>side1_score</th>\n",
              "      <th>side2_normalized</th>\n",
              "      <th>side2_score</th>\n",
              "      <th>total_score</th>\n",
              "      <th>yhh_planarity</th>\n",
              "      <th>cdr_cluster_ID_L1</th>\n",
              "      <th>cdr_cluster_ID_L3</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>5</th>\n",
              "      <td>tutA2_my_ab_0005</td>\n",
              "      <td>0.0</td>\n",
              "      <td>11.192533</td>\n",
              "      <td>31.677544</td>\n",
              "      <td>11.0</td>\n",
              "      <td>7.0</td>\n",
              "      <td>2.450019</td>\n",
              "      <td>0.0</td>\n",
              "      <td>0.0</td>\n",
              "      <td>1751.548462</td>\n",
              "      <td>...</td>\n",
              "      <td>364.65868</td>\n",
              "      <td>0.564960</td>\n",
              "      <td>15.000569</td>\n",
              "      <td>825.031311</td>\n",
              "      <td>21.473850</td>\n",
              "      <td>837.480164</td>\n",
              "      <td>2383.868655</td>\n",
              "      <td>0.107882</td>\n",
              "      <td>L1-11-1</td>\n",
              "      <td>L3-7-1</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>tutA1_my_ab_0001</td>\n",
              "      <td>0.0</td>\n",
              "      <td>10.244154</td>\n",
              "      <td>14.169669</td>\n",
              "      <td>11.0</td>\n",
              "      <td>9.0</td>\n",
              "      <td>2.494409</td>\n",
              "      <td>0.0</td>\n",
              "      <td>0.0</td>\n",
              "      <td>1784.542480</td>\n",
              "      <td>...</td>\n",
              "      <td>374.81054</td>\n",
              "      <td>0.491761</td>\n",
              "      <td>14.690318</td>\n",
              "      <td>837.348145</td>\n",
              "      <td>20.825792</td>\n",
              "      <td>874.683289</td>\n",
              "      <td>2432.049006</td>\n",
              "      <td>0.134353</td>\n",
              "      <td>L1-11-1</td>\n",
              "      <td>L3-9-cis7-1</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>tutA1_my_ab_0004</td>\n",
              "      <td>0.0</td>\n",
              "      <td>9.604385</td>\n",
              "      <td>10.078285</td>\n",
              "      <td>11.0</td>\n",
              "      <td>9.0</td>\n",
              "      <td>2.459319</td>\n",
              "      <td>0.0</td>\n",
              "      <td>0.0</td>\n",
              "      <td>1820.213013</td>\n",
              "      <td>...</td>\n",
              "      <td>374.09441</td>\n",
              "      <td>0.596517</td>\n",
              "      <td>14.805477</td>\n",
              "      <td>858.717651</td>\n",
              "      <td>20.484030</td>\n",
              "      <td>860.329285</td>\n",
              "      <td>2397.836108</td>\n",
              "      <td>0.420196</td>\n",
              "      <td>L1-11-1</td>\n",
              "      <td>L3-9-cis7-1</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>6</th>\n",
              "      <td>tutA2_my_ab_0003</td>\n",
              "      <td>0.0</td>\n",
              "      <td>20.217281</td>\n",
              "      <td>24.178181</td>\n",
              "      <td>15.0</td>\n",
              "      <td>11.0</td>\n",
              "      <td>2.760204</td>\n",
              "      <td>0.0</td>\n",
              "      <td>0.0</td>\n",
              "      <td>1843.231445</td>\n",
              "      <td>...</td>\n",
              "      <td>375.17336</td>\n",
              "      <td>0.537087</td>\n",
              "      <td>13.518681</td>\n",
              "      <td>851.676880</td>\n",
              "      <td>21.286449</td>\n",
              "      <td>979.176697</td>\n",
              "      <td>2707.759790</td>\n",
              "      <td>0.221997</td>\n",
              "      <td>L1-15-1</td>\n",
              "      <td>L3-11-1</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>tutA1_my_ab_0002</td>\n",
              "      <td>0.0</td>\n",
              "      <td>9.939879</td>\n",
              "      <td>14.834209</td>\n",
              "      <td>11.0</td>\n",
              "      <td>9.0</td>\n",
              "      <td>2.494796</td>\n",
              "      <td>0.0</td>\n",
              "      <td>0.0</td>\n",
              "      <td>1849.303833</td>\n",
              "      <td>...</td>\n",
              "      <td>372.58994</td>\n",
              "      <td>0.468819</td>\n",
              "      <td>17.130129</td>\n",
              "      <td>907.896790</td>\n",
              "      <td>20.831068</td>\n",
              "      <td>874.904846</td>\n",
              "      <td>2432.425954</td>\n",
              "      <td>0.061565</td>\n",
              "      <td>L1-11-1</td>\n",
              "      <td>L3-9-cis7-1</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3</th>\n",
              "      <td>tutA1_my_ab_0005</td>\n",
              "      <td>0.0</td>\n",
              "      <td>13.124744</td>\n",
              "      <td>14.362237</td>\n",
              "      <td>11.0</td>\n",
              "      <td>9.0</td>\n",
              "      <td>2.667474</td>\n",
              "      <td>0.0</td>\n",
              "      <td>0.0</td>\n",
              "      <td>1934.344604</td>\n",
              "      <td>...</td>\n",
              "      <td>374.83686</td>\n",
              "      <td>0.415013</td>\n",
              "      <td>17.054668</td>\n",
              "      <td>920.952087</td>\n",
              "      <td>22.980696</td>\n",
              "      <td>965.189209</td>\n",
              "      <td>2600.787151</td>\n",
              "      <td>0.044588</td>\n",
              "      <td>L1-11-1</td>\n",
              "      <td>L3-9-cis7-1</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>7</th>\n",
              "      <td>tutA2_my_ab_0001</td>\n",
              "      <td>0.0</td>\n",
              "      <td>8.599037</td>\n",
              "      <td>27.562218</td>\n",
              "      <td>11.0</td>\n",
              "      <td>10.0</td>\n",
              "      <td>3.348572</td>\n",
              "      <td>0.0</td>\n",
              "      <td>0.0</td>\n",
              "      <td>2484.706299</td>\n",
              "      <td>...</td>\n",
              "      <td>374.05882</td>\n",
              "      <td>0.486206</td>\n",
              "      <td>22.621260</td>\n",
              "      <td>1198.926758</td>\n",
              "      <td>29.544832</td>\n",
              "      <td>1270.427734</td>\n",
              "      <td>3268.206329</td>\n",
              "      <td>0.361406</td>\n",
              "      <td>L1-11-1</td>\n",
              "      <td>L3-10-cis7,8-1</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>8</th>\n",
              "      <td>tutA2_my_ab_0002</td>\n",
              "      <td>0.0</td>\n",
              "      <td>29.971132</td>\n",
              "      <td>45.571491</td>\n",
              "      <td>17.0</td>\n",
              "      <td>9.0</td>\n",
              "      <td>3.352178</td>\n",
              "      <td>0.0</td>\n",
              "      <td>0.0</td>\n",
              "      <td>2587.653564</td>\n",
              "      <td>...</td>\n",
              "      <td>373.00912</td>\n",
              "      <td>0.620726</td>\n",
              "      <td>18.605259</td>\n",
              "      <td>1265.157593</td>\n",
              "      <td>28.719591</td>\n",
              "      <td>1292.381592</td>\n",
              "      <td>3288.486156</td>\n",
              "      <td>0.126786</td>\n",
              "      <td>L1-17-1</td>\n",
              "      <td>L3-9-1</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4</th>\n",
              "      <td>tutA1_my_ab_0003</td>\n",
              "      <td>0.0</td>\n",
              "      <td>9.693962</td>\n",
              "      <td>12.150679</td>\n",
              "      <td>11.0</td>\n",
              "      <td>9.0</td>\n",
              "      <td>3.530113</td>\n",
              "      <td>0.0</td>\n",
              "      <td>0.0</td>\n",
              "      <td>2854.750488</td>\n",
              "      <td>...</td>\n",
              "      <td>376.96998</td>\n",
              "      <td>0.464472</td>\n",
              "      <td>24.664927</td>\n",
              "      <td>1381.235962</td>\n",
              "      <td>33.255428</td>\n",
              "      <td>1396.727905</td>\n",
              "      <td>3441.859648</td>\n",
              "      <td>0.075642</td>\n",
              "      <td>L1-11-1</td>\n",
              "      <td>L3-9-cis7-1</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>9</th>\n",
              "      <td>tutA2_my_ab_0004</td>\n",
              "      <td>0.0</td>\n",
              "      <td>12.153269</td>\n",
              "      <td>17.002886</td>\n",
              "      <td>10.0</td>\n",
              "      <td>8.0</td>\n",
              "      <td>5.869139</td>\n",
              "      <td>0.0</td>\n",
              "      <td>0.0</td>\n",
              "      <td>5082.230469</td>\n",
              "      <td>...</td>\n",
              "      <td>376.47676</td>\n",
              "      <td>0.571551</td>\n",
              "      <td>42.399551</td>\n",
              "      <td>2501.573486</td>\n",
              "      <td>62.364841</td>\n",
              "      <td>2494.593750</td>\n",
              "      <td>5710.672501</td>\n",
              "      <td>0.100791</td>\n",
              "      <td>L1-10-2</td>\n",
              "      <td>L3-8-1</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "<p>10 rows  48 columns</p>\n",
              "</div>\n",
              "    <div class=\"colab-df-buttons\">\n",
              "\n",
              "  <div class=\"colab-df-container\">\n",
              "    <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-1f6a5ec5-6cff-4e88-a0c3-616d9cb503fb')\"\n",
              "            title=\"Convert this dataframe to an interactive table.\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\" viewBox=\"0 -960 960 960\">\n",
              "    <path d=\"M120-120v-720h720v720H120Zm60-500h600v-160H180v160Zm220 220h160v-160H400v160Zm0 220h160v-160H400v160ZM180-400h160v-160H180v160Zm440 0h160v-160H620v160ZM180-180h160v-160H180v160Zm440 0h160v-160H620v160Z\"/>\n",
              "  </svg>\n",
              "    </button>\n",
              "\n",
              "  <style>\n",
              "    .colab-df-container {\n",
              "      display:flex;\n",
              "      gap: 12px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert {\n",
              "      background-color: #E8F0FE;\n",
              "      border: none;\n",
              "      border-radius: 50%;\n",
              "      cursor: pointer;\n",
              "      display: none;\n",
              "      fill: #1967D2;\n",
              "      height: 32px;\n",
              "      padding: 0 0 0 0;\n",
              "      width: 32px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert:hover {\n",
              "      background-color: #E2EBFA;\n",
              "      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "      fill: #174EA6;\n",
              "    }\n",
              "\n",
              "    .colab-df-buttons div {\n",
              "      margin-bottom: 4px;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert {\n",
              "      background-color: #3B4455;\n",
              "      fill: #D2E3FC;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert:hover {\n",
              "      background-color: #434B5C;\n",
              "      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
              "      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
              "      fill: #FFFFFF;\n",
              "    }\n",
              "  </style>\n",
              "\n",
              "    <script>\n",
              "      const buttonEl =\n",
              "        document.querySelector('#df-1f6a5ec5-6cff-4e88-a0c3-616d9cb503fb button.colab-df-convert');\n",
              "      buttonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "\n",
              "      async function convertToInteractive(key) {\n",
              "        const element = document.querySelector('#df-1f6a5ec5-6cff-4e88-a0c3-616d9cb503fb');\n",
              "        const dataTable =\n",
              "          await google.colab.kernel.invokeFunction('convertToInteractive',\n",
              "                                                    [key], {});\n",
              "        if (!dataTable) return;\n",
              "\n",
              "        const docLinkHtml = 'Like what you see? Visit the ' +\n",
              "          '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n",
              "          + ' to learn more about interactive tables.';\n",
              "        element.innerHTML = '';\n",
              "        dataTable['output_type'] = 'display_data';\n",
              "        await google.colab.output.renderOutput(dataTable, element);\n",
              "        const docLink = document.createElement('div');\n",
              "        docLink.innerHTML = docLinkHtml;\n",
              "        element.appendChild(docLink);\n",
              "      }\n",
              "    </script>\n",
              "  </div>\n",
              "\n",
              "\n",
              "<div id=\"df-b090fd4c-a99f-48c7-99e7-98209534d928\">\n",
              "  <button class=\"colab-df-quickchart\" onclick=\"quickchart('df-b090fd4c-a99f-48c7-99e7-98209534d928')\"\n",
              "            title=\"Suggest charts\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "<svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
              "     width=\"24px\">\n",
              "    <g>\n",
              "        <path d=\"M19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zM9 17H7v-7h2v7zm4 0h-2V7h2v10zm4 0h-2v-4h2v4z\"/>\n",
              "    </g>\n",
              "</svg>\n",
              "  </button>\n",
              "\n",
              "<style>\n",
              "  .colab-df-quickchart {\n",
              "      --bg-color: #E8F0FE;\n",
              "      --fill-color: #1967D2;\n",
              "      --hover-bg-color: #E2EBFA;\n",
              "      --hover-fill-color: #174EA6;\n",
              "      --disabled-fill-color: #AAA;\n",
              "      --disabled-bg-color: #DDD;\n",
              "  }\n",
              "\n",
              "  [theme=dark] .colab-df-quickchart {\n",
              "      --bg-color: #3B4455;\n",
              "      --fill-color: #D2E3FC;\n",
              "      --hover-bg-color: #434B5C;\n",
              "      --hover-fill-color: #FFFFFF;\n",
              "      --disabled-bg-color: #3B4455;\n",
              "      --disabled-fill-color: #666;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart {\n",
              "    background-color: var(--bg-color);\n",
              "    border: none;\n",
              "    border-radius: 50%;\n",
              "    cursor: pointer;\n",
              "    display: none;\n",
              "    fill: var(--fill-color);\n",
              "    height: 32px;\n",
              "    padding: 0;\n",
              "    width: 32px;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart:hover {\n",
              "    background-color: var(--hover-bg-color);\n",
              "    box-shadow: 0 1px 2px rgba(60, 64, 67, 0.3), 0 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "    fill: var(--button-hover-fill-color);\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart-complete:disabled,\n",
              "  .colab-df-quickchart-complete:disabled:hover {\n",
              "    background-color: var(--disabled-bg-color);\n",
              "    fill: var(--disabled-fill-color);\n",
              "    box-shadow: none;\n",
              "  }\n",
              "\n",
              "  .colab-df-spinner {\n",
              "    border: 2px solid var(--fill-color);\n",
              "    border-color: transparent;\n",
              "    border-bottom-color: var(--fill-color);\n",
              "    animation:\n",
              "      spin 1s steps(1) infinite;\n",
              "  }\n",
              "\n",
              "  @keyframes spin {\n",
              "    0% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "      border-left-color: var(--fill-color);\n",
              "    }\n",
              "    20% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    30% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    40% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    60% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    80% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "    90% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "  }\n",
              "</style>\n",
              "\n",
              "  <script>\n",
              "    async function quickchart(key) {\n",
              "      const quickchartButtonEl =\n",
              "        document.querySelector('#' + key + ' button');\n",
              "      quickchartButtonEl.disabled = true;  // To prevent multiple clicks.\n",
              "      quickchartButtonEl.classList.add('colab-df-spinner');\n",
              "      try {\n",
              "        const charts = await google.colab.kernel.invokeFunction(\n",
              "            'suggestCharts', [key], {});\n",
              "      } catch (error) {\n",
              "        console.error('Error during call to suggestCharts:', error);\n",
              "      }\n",
              "      quickchartButtonEl.classList.remove('colab-df-spinner');\n",
              "      quickchartButtonEl.classList.add('colab-df-quickchart-complete');\n",
              "    }\n",
              "    (() => {\n",
              "      let quickchartButtonEl =\n",
              "        document.querySelector('#df-b090fd4c-a99f-48c7-99e7-98209534d928 button');\n",
              "      quickchartButtonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "    })();\n",
              "  </script>\n",
              "</div>\n",
              "\n",
              "  <div id=\"id_7ab4f651-ea38-49d6-b3d3-9ebc76a0f0ba\">\n",
              "    <style>\n",
              "      .colab-df-generate {\n",
              "        background-color: #E8F0FE;\n",
              "        border: none;\n",
              "        border-radius: 50%;\n",
              "        cursor: pointer;\n",
              "        display: none;\n",
              "        fill: #1967D2;\n",
              "        height: 32px;\n",
              "        padding: 0 0 0 0;\n",
              "        width: 32px;\n",
              "      }\n",
              "\n",
              "      .colab-df-generate:hover {\n",
              "        background-color: #E2EBFA;\n",
              "        box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "        fill: #174EA6;\n",
              "      }\n",
              "\n",
              "      [theme=dark] .colab-df-generate {\n",
              "        background-color: #3B4455;\n",
              "        fill: #D2E3FC;\n",
              "      }\n",
              "\n",
              "      [theme=dark] .colab-df-generate:hover {\n",
              "        background-color: #434B5C;\n",
              "        box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
              "        filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
              "        fill: #FFFFFF;\n",
              "      }\n",
              "    </style>\n",
              "    <button class=\"colab-df-generate\" onclick=\"generateWithVariable('df_tut_a12')\"\n",
              "            title=\"Generate code using this dataframe.\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
              "       width=\"24px\">\n",
              "    <path d=\"M7,19H8.4L18.45,9,17,7.55,7,17.6ZM5,21V16.75L18.45,3.32a2,2,0,0,1,2.83,0l1.4,1.43a1.91,1.91,0,0,1,.58,1.4,1.91,1.91,0,0,1-.58,1.4L9.25,21ZM18.45,9,17,7.55Zm-12,3A5.31,5.31,0,0,0,4.9,8.1,5.31,5.31,0,0,0,1,6.5,5.31,5.31,0,0,0,4.9,4.9,5.31,5.31,0,0,0,6.5,1,5.31,5.31,0,0,0,8.1,4.9,5.31,5.31,0,0,0,12,6.5,5.46,5.46,0,0,0,6.5,12Z\"/>\n",
              "  </svg>\n",
              "    </button>\n",
              "    <script>\n",
              "      (() => {\n",
              "      const buttonEl =\n",
              "        document.querySelector('#id_7ab4f651-ea38-49d6-b3d3-9ebc76a0f0ba button.colab-df-generate');\n",
              "      buttonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "\n",
              "      buttonEl.onclick = () => {\n",
              "        google.colab.notebook.generateWithVariable('df_tut_a12');\n",
              "      }\n",
              "      })();\n",
              "    </script>\n",
              "  </div>\n",
              "\n",
              "    </div>\n",
              "  </div>\n"
            ],
            "application/vnd.google.colaboratory.intrinsic+json": {
              "type": "dataframe",
              "variable_name": "df_tut_a12"
            }
          },
          "metadata": {},
          "execution_count": 26
        }
      ],
      "source": [
        "df_tut_a12 = pandas.concat([df, df_a2], ignore_index=True).sort_values(\"dG_separated\", ascending=True)\n",
        "df_tut_a12"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "Jq-vfdWK2LMy"
      },
      "source": [
        "Take a look at the lowest (dG) scoring pose in pymol - do you see any difference in L1 and L3 loops there?  Do they make better contact than what we had before?\n",
        "\n",
        "\tLets take a look in pymol.   \n",
        "\n",
        "\t\t\tpymol inputs/rabd/my_ab.pdb inputs/rabd/tutA2_*\n",
        "\t\t\t@color_cdrs.pml\n",
        "\t\t\tcenter full_epitope\n",
        "\n",
        " How different are the L1 and L3 loops? Have any changed length?\n",
        "\n",
        " Lets take a look at the clusters in our dataframe.  Have they changed from the native?"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 34,
      "metadata": {
        "id": "q5RhjIWr2LMy",
        "outputId": "eb7a94b6-2725-4e73-f670-62b5f56690f1",
        "colab": {
          "base_uri": "https://localhost:8080/"
        }
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "One or both CDR cluster IDs are missing.\n"
          ]
        }
      ],
      "source": [
        "if not os.getenv(\"DEBUG\"):\n",
        "    # Using the .get method to safely access the scores with a default value of None\n",
        "    L1 = original_pose.scores.get(\"cdr_cluster_ID_L1\", None)\n",
        "    L3 = original_pose.scores.get(\"cdr_cluster_ID_L3\", None)\n",
        "\n",
        "    # Now you can check if L1 and L3 were found before proceeding\n",
        "    if L1 is not None and L3 is not None:\n",
        "        # Proceed with your logic if both scores are found\n",
        "        print(\"L1 Cluster ID:\", L1)\n",
        "        print(\"L3 Cluster ID:\", L3)\n",
        "    else:\n",
        "        # Handle the case where one or both scores are missing\n",
        "        print(\"One or both CDR cluster IDs are missing.\")"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "QltreyyR2LMy"
      },
      "source": [
        "### Tut A3.  Basic De-novo run\n",
        "\n",
        "Here, we want to do a denovo-run (without docking), starting with random CDRs grafted in - instead of whatever we have in the antibody to start with (only for the CDRs that are actually undergoing graft-design).  This is useful, as we start the design with very high energy and work our way down.  Note that since this is an entirely new interface for our model protein, this interface is already at a very high energy - and so this is less needed, but it should be noted how to do this. (139 seconds on my laptop). Do this below as you have done in other tutorials - either through code or XML.\n",
        "\n",
        "\n",
        "\t<AntibodyDesignMover name=\"RAbD\" graft_design_cdrs=\"L1,L3\" seq_design_cdrs=\"L1,L3\"\n",
        "\t\t\t                                                                  random_start=\"1\"/>  "
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 36,
      "metadata": {
        "nbgrader": {
          "grade": true,
          "grade_id": "cell-b7dcc3797eed573e",
          "locked": false,
          "points": 0,
          "schema_version": 3,
          "solution": true
        },
        "id": "xwjsuCAC2LMz",
        "outputId": "1188aec7-ab83-48ef-b0da-decde6e45de0",
        "colab": {
          "base_uri": "https://localhost:8080/"
        }
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "protocols.rosetta_scripts.RosettaScriptsParser: Validating input script...\n",
            "protocols.rosetta_scripts.RosettaScriptsParser: ...done\n",
            "protocols.rosetta_scripts.RosettaScriptsParser: Parsed script:\n",
            "<ROSETTASCRIPTS>\n",
            "\t<MOVERS>\n",
            "\t\t<AntibodyDesignMover graft_design_cdrs=\"L1,L3\" name=\"RAbD\" random_start=\"1\" seq_design_cdrs=\"L1,L3\"/>\n",
            "\t</MOVERS>\n",
            "\t<PROTOCOLS/>\n",
            "</ROSETTASCRIPTS>\n",
            "core.scoring.ScoreFunctionFactory: SCOREFUNCTION: ref2015\n",
            "core.scoring.ScoreFunctionFactory: SCOREFUNCTION: ref2015\n",
            "core.scoring.ScoreFunctionFactory: SCOREFUNCTION: ref2015\n",
            "protocols.rosetta_scripts.RosettaScriptsParser: Defined mover named \"RAbD\" of type AntibodyDesignMover\n",
            "protocols.rosetta_scripts.ParsedProtocol: ParsedProtocol mover with the following settings\n",
            "basic.io.database: Database file opened: sampling/antibodies/cluster_center_dihedrals.txt\n",
            "protocols.antibody.AntibodyNumberingParser: Antibody numbering scheme definitions read successfully\n",
            "protocols.antibody.AntibodyNumberingParser: Antibody CDR definition read successfully\n",
            "antibody.AntibodyInfo: Successfully finished the CDR definition\n",
            "antibody.AntibodyInfo: AC Detecting Regular CDR H3 Stem Type\n",
            "antibody.AntibodyInfo: SRWGGDGFYAMDYW\n",
            "antibody.AntibodyInfo: AC Finished Detecting Regular CDR H3 Stem Type: KINKED\n",
            "antibody.AntibodyInfo: AC Finished Detecting Regular CDR H3 Stem Type: Kink: 1 Extended: 0\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H1\n",
            "protocols.antibody.cluster.CDRClusterMatcher: Length: 13 Omega: TTTTTTTTTTTTT\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H2\n",
            "protocols.antibody.cluster.CDRClusterMatcher: Length: 10 Omega: TTTTTTTTTT\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H3\n",
            "protocols.antibody.cluster.CDRClusterMatcher: Length: 13 Omega: TTTTTTTTTTTTT\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L1\n",
            "protocols.antibody.cluster.CDRClusterMatcher: Length: 11 Omega: TTTTTTTTTTT\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L2\n",
            "protocols.antibody.cluster.CDRClusterMatcher: Length: 8 Omega: TTTTTTTT\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L3\n",
            "protocols.antibody.cluster.CDRClusterMatcher: Length: 9 Omega: TTTTTTCTT\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H1\n",
            "protocols.antibody.cluster.CDRClusterMatcher: Length: 13 Omega: TTTTTTTTTTTTT\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H2\n",
            "protocols.antibody.cluster.CDRClusterMatcher: Length: 10 Omega: TTTTTTTTTT\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H3\n",
            "protocols.antibody.cluster.CDRClusterMatcher: Length: 13 Omega: TTTTTTTTTTTTT\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L1\n",
            "protocols.antibody.cluster.CDRClusterMatcher: Length: 11 Omega: TTTTTTTTTTT\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L2\n",
            "protocols.antibody.cluster.CDRClusterMatcher: Length: 8 Omega: TTTTTTTT\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L3\n",
            "protocols.antibody.cluster.CDRClusterMatcher: Length: 9 Omega: TTTTTTCTT\n",
            "protocols.antibody.clusters.CDRClusterSet: Adding cacheable cluster data to pose\n",
            "protocols.antibody.design.util: Reading CDRSetOptions from: /sampling/antibodies/design/default_instructions.txt\n",
            "protocols.antibody.design.util: Reading CDRGraftDesignOptions from: /sampling/antibodies/design/default_instructions.txt\n",
            "protocols.antibody.design.util: Reading CDRSeqDesignOptions from: /sampling/antibodies/design/default_instructions.txt\n",
            "core.scoring.ScoreFunctionFactory: SCOREFUNCTION: ref2015\n",
            "core.scoring.ScoreFunctionFactory: SCOREFUNCTION: ref2015\n",
            "core.scoring.ScoreFunctionFactory: SCOREFUNCTION: ref2015\n",
            "basic.io.database: [ WARNING ] Unable to locate database file /sampling/antibodies/antibody_database_rosetta_design.db\n",
            "antibody.database.AntibodyDatabaseManager: The antibody design database is not specified by the user.\n",
            "antibody.database.AntibodyDatabaseManager: Reading from: /usr/local/lib/python3.10/dist-packages/pyrosetta/database//sampling/antibodies/antibody_database_rosetta_design_north_paper.db\n",
            "antibody.database.AntibodyDatabaseManager: Checking loadable structures\n",
            "antibody.database.AntibodyDatabaseManager: Skipping 12 structures from reading\n",
            "core.scoring.ScoreFunctionFactory: SCOREFUNCTION: ref2015\n",
            "core.scoring.ScoreFunctionFactory: SCOREFUNCTION: ref2015\n",
            "antibody.database.AntibodyDatabaseManager: Total L1 in CDRSet: 191\n",
            "antibody.database.AntibodyDatabaseManager: Total L3 in CDRSet: 230\n",
            "antibody.database.AntibodyDatabaseManager: 300 Cached\n",
            "antibody.database.AntibodyDatabaseManager: cdrs loaded from database.\n",
            "protocols.antibody.design.AntibodyDesignMover: ///// Total CDRs in set /////\n",
            "protocols.antibody.design.AntibodyDesignMover: /// L1 191\n",
            "protocols.antibody.design.AntibodyDesignMover: /// L3 230\n",
            "protocols.antibody.design.AntibodyDesignMover: Total possible CDR combinations: 43930\n",
            "core.scoring.ScoreFunction: \n",
            "------------------------------------------------------------\n",
            " Scores                       Weight   Raw Score Wghtd.Score\n",
            "------------------------------------------------------------\n",
            " fa_atr                       1.000   -5695.317   -5695.317\n",
            " fa_rep                       0.550   24719.870   13595.929\n",
            " fa_sol                       1.000    3501.170    3501.170\n",
            " fa_intra_rep                 0.005    2661.048      13.305\n",
            " fa_intra_sol_xover4          1.000     175.028     175.028\n",
            " lk_ball_wtd                  1.000     -57.980     -57.980\n",
            " fa_elec                      1.000   -1588.297   -1588.297\n",
            " pro_close                    1.250      10.290      12.862\n",
            " hbond_sr_bb                  1.000     -74.000     -74.000\n",
            " hbond_lr_bb                  1.000    -511.787    -511.787\n",
            " hbond_bb_sc                  1.000    -130.727    -130.727\n",
            " hbond_sc                     1.000     -78.766     -78.766\n",
            " dslf_fa13                    1.250      -1.572      -1.965\n",
            " atom_pair_constraint         0.010       0.000       0.000\n",
            " omega                        0.400     474.068     189.627\n",
            " fa_dun                       0.700    5380.790    3766.553\n",
            " p_aa_pp                      0.600    -289.096    -173.458\n",
            " yhh_planarity                0.625       0.000       0.000\n",
            " ref                          1.000     372.765     372.765\n",
            " rama_prepro                  0.450     191.068      85.981\n",
            "---------------------------------------------------\n",
            " Total weighted score:                    13400.923\n",
            "protocols.antibody.design.AntibodyDesignMover: Randomizing starting CDRs\n",
            "protocols.antibody.design.AntibodyDesignMover: Grafting CDR from cluster L1-16-1 fragment 1zeaL_L1_0001\n",
            "protocols.grafting.CCDEndsGraftMover: Start: 891 End: 903 NterO: 3 CterO: 3\n",
            "protocols.grafting.util: Superimposing overhang residues\n",
            "core.chemical.AtomICoor: [ WARNING ] IcoorAtomID::atom_id(): Cannot get atom_id for POLYMER_LOWER of residue ILE 1.  Returning BOGUS ID instead.\n",
            "core.conformation.Residue: [ WARNING ] missing an atom: 1  H   that depends on a nonexistent polymer connection!\n",
            "core.conformation.Residue: [ WARNING ]  --> generating it using idealized coordinates.\n",
            "protocols.grafting.util: Deleting 11 residues from 892 to 902\n",
            "core.conformation.Conformation: Found disulfide between residues 771 845\n",
            "core.conformation.Conformation: Found disulfide between residues 891 945\n",
            "protocols.grafting.util: insert_point 891\n",
            "protocols.grafting.util: insert_start 892\n",
            "protocols.grafting.util: insert_end 907\n",
            "protocols.grafting.util: insert_length 16\n",
            "protocols.grafting.util: FOLD_TREE  EDGE 1 891 -1  EDGE 891 892 1  EDGE 892 964 -1\n",
            "protocols.grafting.GraftMoverBase: Insertion complete.\n",
            "protocols.grafting.AnchoredGraftMover: Setting default movemap\n",
            "protocols.grafting.CCDEndsGraftMover: Start: 891\n",
            "protocols.grafting.CCDEndsGraftMover: Original End: 903\n",
            "protocols.grafting.CCDEndsGraftMover: End: 908\n",
            "protocols.grafting.CCDEndsGraftMover: Insert Length: 16\n",
            "basic.io.database: Database file opened: scoring/score_functions/centroid_smooth/cen_smooth_params.txt\n",
            "protocols.grafting.CCDEndsGraftMover: LOOP start: 890  stop: 893  cut: 891  size: 4  skip rate: 0  extended?: False\n",
            "protocols.grafting.CCDEndsGraftMover:\n",
            "protocols.grafting.CCDEndsGraftMover: LOOP start: 906  stop: 909  cut: 907  size: 4  skip rate: 0  extended?: False\n",
            "protocols.grafting.CCDEndsGraftMover:\n",
            "protocols.loops.FoldTreeFromLoopsWrapper: old foldtree FOLD_TREE  EDGE 1 891 -1  EDGE 891 908 1  EDGE 908 980 -1  EDGE 891 892 -2  C    N    EDGE 892 893 -2  C    N    EDGE 893 894 -2  C    N    EDGE 894 895 -2  C    N    EDGE 895 896 -2  C    N    EDGE 896 897 -2  C    N    EDGE 897 898 -2  C    N    EDGE 898 899 -2  C    N    EDGE 899 900 -2  C    N    EDGE 900 901 -2  C    N    EDGE 901 902 -2  C    N    EDGE 902 903 -2  C    N    EDGE 903 904 -2  C    N    EDGE 904 905 -2  C    N    EDGE 905 906 -2  C    N    EDGE 906 907 -2  C    N   \n",
            "New foldtree FOLD_TREE  EDGE 1 889 -1  EDGE 889 891 -1  EDGE 889 894 1  EDGE 894 892 -1  EDGE 894 905 -1  EDGE 905 907 -1  EDGE 905 910 2  EDGE 910 908 -1  EDGE 910 980 -1\n",
            "protocols.grafting.util: Loop: 1\n",
            "protocols.grafting.util: Add variant to: 891\n",
            "protocols.grafting.util: Loop: 2\n",
            "protocols.grafting.util: Add variant to: 907\n",
            "protocols.grafting.util: idealized  890\n",
            "protocols.grafting.util: idealized  891\n",
            "protocols.grafting.util: ideal 892\n",
            "protocols.grafting.util: ideal 907\n",
            "protocols.grafting.util: idealized 908\n",
            "protocols.grafting.util: idealized 909\n",
            "core.chemical.GlobalResidueTypeSet: Finished initializing centroid residue type set.  Created 69 residue types\n",
            "core.chemical.GlobalResidueTypeSet: Total time to initialize 0.030189 seconds.\n",
            "basic.io.database: Database file opened: scoring/score_functions/disulfides/centroid_distance_score\n",
            "basic.io.database: Database file opened: scoring/score_functions/disulfides/centroid_CaCbCb_angle_score\n",
            "basic.io.database: Database file opened: scoring/score_functions/disulfides/centroid_CaCbCbCa_dihedral_score\n",
            "basic.io.database: Database file opened: scoring/score_functions/disulfides/centroid_backbone_dihedral_score\n",
            "protocols.grafting.CCDEndsGraftMover: start 1849.31\n",
            "protocols.grafting.CCDEndsGraftMover: round 1\n",
            "protocols.grafting.CCDEndsGraftMover: Graft Closed early - returning\n",
            "protocols.grafting.CCDEndsGraftMover: 1 1395.99\n",
            "protocols.grafting.CCDEndsGraftMover: finish 1395.99\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 703 rotamers at 56 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.grafting.CCDEndsGraftMover: Graft meets ideal geometry true\n",
            "protocols.grafting.CCDEndsGraftMover: Complete\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L1-16-1 data: 117 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/L1-16-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 32 constraints\n",
            "protocols.antibody.design.AntibodyDesignMover: Checking Geometry\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H1\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H2\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H3\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L1\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L2\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L3\n",
            "protocols.antibody.clusters.CDRClusterSet: Adding cacheable cluster data to pose\n",
            "protocols.antibody.design.util: Setting CDR sequence in NativeAntibodySeq\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: H1-13-1 data: 583 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/H1-13-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 26 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: H2-10-1 data: 328 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/H2-10-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 20 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: Adding general dihedral constraints for H3\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L1-16-1 data: 117 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/L1-16-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 32 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L2-8-1 data: 632 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/L2-8-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 16 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L3-9-cis7-1 data: 419 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/L3-9-cis7-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 18 constraints\n",
            "protocols.antibody.design.AntibodyDesignMover: Min Cycle complete\n",
            "protocols.antibody.design.AntibodyDesignMover: Attempts 1 L1\n",
            "protocols.antibody.design.AntibodyDesignMover: Grafting CDR from cluster L3-9-cis7-1 fragment 1cloL_L3_0001\n",
            "protocols.grafting.CCDEndsGraftMover: Start: 961 End: 971 NterO: 3 CterO: 3\n",
            "protocols.grafting.util: Superimposing overhang residues\n",
            "core.chemical.AtomICoor: [ WARNING ] IcoorAtomID::atom_id(): Cannot get atom_id for POLYMER_LOWER of residue TYR 1.  Returning BOGUS ID instead.\n",
            "core.conformation.Residue: [ WARNING ] missing an atom: 1  H   that depends on a nonexistent polymer connection!\n",
            "core.conformation.Residue: [ WARNING ]  --> generating it using idealized coordinates.\n",
            "protocols.grafting.util: Deleting 9 residues from 962 to 970\n",
            "core.conformation.Conformation: Found disulfide between residues 771 845\n",
            "core.conformation.Conformation: Found disulfide between residues 891 961\n",
            "protocols.grafting.util: insert_point 961\n",
            "protocols.grafting.util: insert_start 962\n",
            "protocols.grafting.util: insert_end 970\n",
            "protocols.grafting.util: insert_length 9\n",
            "protocols.grafting.util: FOLD_TREE  EDGE 1 961 -1  EDGE 961 962 1  EDGE 962 971 -1\n",
            "protocols.grafting.GraftMoverBase: Insertion complete.\n",
            "protocols.grafting.AnchoredGraftMover: Setting default movemap\n",
            "protocols.grafting.CCDEndsGraftMover: Start: 961\n",
            "protocols.grafting.CCDEndsGraftMover: Original End: 971\n",
            "protocols.grafting.CCDEndsGraftMover: End: 971\n",
            "protocols.grafting.CCDEndsGraftMover: Insert Length: 9\n",
            "protocols.grafting.CCDEndsGraftMover: LOOP start: 960  stop: 963  cut: 961  size: 4  skip rate: 0  extended?: False\n",
            "protocols.grafting.CCDEndsGraftMover:\n",
            "protocols.grafting.CCDEndsGraftMover: LOOP start: 969  stop: 972  cut: 970  size: 4  skip rate: 0  extended?: False\n",
            "protocols.grafting.CCDEndsGraftMover:\n",
            "protocols.loops.FoldTreeFromLoopsWrapper: old foldtree FOLD_TREE  EDGE 1 961 -1  EDGE 961 971 1  EDGE 971 980 -1  EDGE 961 962 -2  C    N    EDGE 962 963 -2  C    N    EDGE 963 964 -2  C    N    EDGE 964 965 -2  C    N    EDGE 965 966 -2  C    N    EDGE 966 967 -2  C    N    EDGE 967 968 -2  C    N    EDGE 968 969 -2  C    N    EDGE 969 970 -2  C    N   \n",
            "New foldtree FOLD_TREE  EDGE 1 959 -1  EDGE 959 961 -1  EDGE 959 964 1  EDGE 964 962 -1  EDGE 964 968 -1  EDGE 968 970 -1  EDGE 968 973 2  EDGE 973 971 -1  EDGE 973 980 -1\n",
            "protocols.grafting.util: Loop: 1\n",
            "protocols.grafting.util: Add variant to: 961\n",
            "protocols.grafting.util: Loop: 2\n",
            "protocols.grafting.util: Add variant to: 970\n",
            "protocols.grafting.util: idealized  960\n",
            "protocols.grafting.util: idealized  961\n",
            "protocols.grafting.util: ideal 962\n",
            "protocols.grafting.util: ideal 970\n",
            "protocols.grafting.util: idealized 971\n",
            "protocols.grafting.util: idealized 972\n",
            "protocols.grafting.CCDEndsGraftMover: start 1773.81\n",
            "protocols.grafting.CCDEndsGraftMover: round 1\n",
            "protocols.loops.util: PepBondGeom: 138 L  C-N --  1.58346 max: 1.5\n",
            "protocols.grafting.CCDEndsGraftMover: 1 1410.83\n",
            "protocols.grafting.CCDEndsGraftMover: round 2\n",
            "protocols.grafting.CCDEndsGraftMover: Graft Closed early - returning\n",
            "protocols.grafting.CCDEndsGraftMover: 2 1380.32\n",
            "protocols.grafting.CCDEndsGraftMover: finish 1380.32\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 292 rotamers at 40 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.grafting.CCDEndsGraftMover: Graft meets ideal geometry true\n",
            "protocols.grafting.CCDEndsGraftMover: Complete\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L3-9-cis7-1 data: 419 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/L3-9-cis7-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 18 constraints\n",
            "protocols.antibody.design.AntibodyDesignMover: Checking Geometry\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H1\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H2\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H3\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L1\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L2\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L3\n",
            "protocols.antibody.clusters.CDRClusterSet: Adding cacheable cluster data to pose\n",
            "protocols.antibody.design.util: Setting CDR sequence in NativeAntibodySeq\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: H1-13-1 data: 583 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/H1-13-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 26 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: H2-10-1 data: 328 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/H2-10-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 20 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: Adding general dihedral constraints for H3\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L1-16-1 data: 117 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/L1-16-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 32 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L2-8-1 data: 632 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/L2-8-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 16 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L3-9-cis7-1 data: 419 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/L3-9-cis7-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 18 constraints\n",
            "protocols.antibody.design.AntibodyDesignMover: Min Cycle complete\n",
            "protocols.antibody.design.AntibodyDesignMover: Attempts 1 L3\n",
            "protocols.antibody.design.AntibodyDesignMover: Running basic monte carlo algorithm\n",
            "protocols.antibody.design.AntibodyDesignMover: Outer round: 1\n",
            "protocols.antibody.design.AntibodyDesignMover: Grafting CDR from cluster L3-9-cis7-2 fragment 1a2yA_L3_0001\n",
            "protocols.grafting.CCDEndsGraftMover: Start: 961 End: 971 NterO: 3 CterO: 3\n",
            "protocols.grafting.util: Superimposing overhang residues\n",
            "core.chemical.AtomICoor: [ WARNING ] IcoorAtomID::atom_id(): Cannot get atom_id for POLYMER_LOWER of residue TYR 1.  Returning BOGUS ID instead.\n",
            "core.conformation.Residue: [ WARNING ] missing an atom: 1  H   that depends on a nonexistent polymer connection!\n",
            "core.conformation.Residue: [ WARNING ]  --> generating it using idealized coordinates.\n",
            "protocols.grafting.util: Deleting 9 residues from 962 to 970\n",
            "core.conformation.Conformation: Found disulfide between residues 771 845\n",
            "core.conformation.Conformation: Found disulfide between residues 891 961\n",
            "protocols.grafting.util: insert_point 961\n",
            "protocols.grafting.util: insert_start 962\n",
            "protocols.grafting.util: insert_end 970\n",
            "protocols.grafting.util: insert_length 9\n",
            "protocols.grafting.util: FOLD_TREE  EDGE 1 961 -1  EDGE 961 962 1  EDGE 962 971 -1\n",
            "protocols.grafting.GraftMoverBase: Insertion complete.\n",
            "protocols.grafting.AnchoredGraftMover: Setting default movemap\n",
            "protocols.grafting.CCDEndsGraftMover: Start: 961\n",
            "protocols.grafting.CCDEndsGraftMover: Original End: 971\n",
            "protocols.grafting.CCDEndsGraftMover: End: 971\n",
            "protocols.grafting.CCDEndsGraftMover: Insert Length: 9\n",
            "protocols.grafting.CCDEndsGraftMover: LOOP start: 960  stop: 963  cut: 961  size: 4  skip rate: 0  extended?: False\n",
            "protocols.grafting.CCDEndsGraftMover:\n",
            "protocols.grafting.CCDEndsGraftMover: LOOP start: 969  stop: 972  cut: 970  size: 4  skip rate: 0  extended?: False\n",
            "protocols.grafting.CCDEndsGraftMover:\n",
            "protocols.loops.FoldTreeFromLoopsWrapper: old foldtree FOLD_TREE  EDGE 1 961 -1  EDGE 961 971 1  EDGE 971 980 -1  EDGE 961 962 -2  C    N    EDGE 962 963 -2  C    N    EDGE 963 964 -2  C    N    EDGE 964 965 -2  C    N    EDGE 965 966 -2  C    N    EDGE 966 967 -2  C    N    EDGE 967 968 -2  C    N    EDGE 968 969 -2  C    N    EDGE 969 970 -2  C    N   \n",
            "New foldtree FOLD_TREE  EDGE 1 959 -1  EDGE 959 961 -1  EDGE 959 964 1  EDGE 964 962 -1  EDGE 964 968 -1  EDGE 968 970 -1  EDGE 968 973 2  EDGE 973 971 -1  EDGE 973 980 -1\n",
            "protocols.grafting.util: Loop: 1\n",
            "protocols.grafting.util: Add variant to: 961\n",
            "protocols.grafting.util: Loop: 2\n",
            "protocols.grafting.util: Add variant to: 970\n",
            "protocols.grafting.util: idealized  960\n",
            "protocols.grafting.util: idealized  961\n",
            "protocols.grafting.util: ideal 962\n",
            "protocols.grafting.util: ideal 970\n",
            "protocols.grafting.util: idealized 971\n",
            "protocols.grafting.util: idealized 972\n",
            "protocols.grafting.CCDEndsGraftMover: start 1754.71\n",
            "protocols.grafting.CCDEndsGraftMover: round 1\n",
            "protocols.loops.util: PepBondGeom: 970 - 138 L  Ca-C-N --  99.5 (min): 160.005 : 134.5 (max)\n",
            "protocols.grafting.CCDEndsGraftMover: 1 1564.27\n",
            "protocols.grafting.CCDEndsGraftMover: round 2\n",
            "protocols.loops.util: PepBondGeom: 138 L  C-N --  2.08687 max: 1.5\n",
            "protocols.grafting.CCDEndsGraftMover: round 3\n",
            "protocols.loops.util: PepBondGeom: 138 L  C-N --  2.14747 max: 1.5\n",
            "protocols.grafting.CCDEndsGraftMover: round 4\n",
            "protocols.loops.util: PepBondGeom: 138 L  C-N --  2.12891 max: 1.5\n",
            "protocols.grafting.CCDEndsGraftMover: round 5\n",
            "core.optimization.LineMinimizer: Inaccurate G in step= 1.90735e-06 Deriv= -11.8771 Finite Diff= -0.281276\n",
            "protocols.loops.util: PepBondGeom: 138 L  C-N --  2.07797 max: 1.5\n",
            "protocols.grafting.CCDEndsGraftMover: round 6\n",
            "protocols.loops.util: PepBondGeom: 138 L  C-N --  2.0963 max: 1.5\n",
            "protocols.grafting.CCDEndsGraftMover: round 7\n",
            "protocols.loops.util: PepBondGeom: 138 L  C-N --  2.06187 max: 1.5\n",
            "protocols.grafting.CCDEndsGraftMover: round 8\n",
            "protocols.loops.util: PepBondGeom: 138 L  C-N --  2.15466 max: 1.5\n",
            "protocols.grafting.CCDEndsGraftMover: round 9\n",
            "protocols.loops.util: PepBondGeom: 138 L  C-N --  2.12378 max: 1.5\n",
            "protocols.grafting.CCDEndsGraftMover: round 10\n",
            "protocols.loops.util: PepBondGeom: 138 L  C-N --  2.16254 max: 1.5\n",
            "protocols.grafting.CCDEndsGraftMover: round 11\n",
            "protocols.loops.util: PepBondGeom: 138 L  C-N --  2.11038 max: 1.5\n",
            "protocols.grafting.CCDEndsGraftMover: round 12\n",
            "protocols.loops.util: PepBondGeom: 138 L  C-N --  2.11401 max: 1.5\n",
            "protocols.grafting.CCDEndsGraftMover: round 13\n",
            "protocols.loops.util: PepBondGeom: 138 L  C-N --  2.18548 max: 1.5\n",
            "protocols.grafting.CCDEndsGraftMover: round 14\n",
            "protocols.loops.util: PepBondGeom: 138 L  C-N --  2.09543 max: 1.5\n",
            "protocols.grafting.CCDEndsGraftMover: round 15\n",
            "protocols.loops.util: PepBondGeom: 138 L  C-N --  2.10063 max: 1.5\n",
            "protocols.grafting.CCDEndsGraftMover: round 16\n",
            "protocols.loops.util: PepBondGeom: 138 L  C-N --  2.12634 max: 1.5\n",
            "protocols.grafting.CCDEndsGraftMover: round 17\n",
            "protocols.loops.util: PepBondGeom: 138 L  C-N --  2.0816 max: 1.5\n",
            "protocols.grafting.CCDEndsGraftMover: round 18\n",
            "core.optimization.LineMinimizer: Inaccurate G in step= 3.8147e-06 Deriv= -2.22344 Finite Diff= 6872.98\n",
            "protocols.loops.util: PepBondGeom: 138 L  C-N --  2.07109 max: 1.5\n",
            "protocols.grafting.CCDEndsGraftMover: round 19\n",
            "protocols.loops.util: PepBondGeom: 138 L  C-N --  2.1172 max: 1.5\n",
            "protocols.grafting.CCDEndsGraftMover: round 20\n",
            "protocols.loops.util: PepBondGeom: 138 L  C-N --  2.08924 max: 1.5\n",
            "protocols.grafting.CCDEndsGraftMover: round 21\n",
            "protocols.loops.util: PepBondGeom: 138 L  C-N --  2.0908 max: 1.5\n",
            "protocols.grafting.CCDEndsGraftMover: round 22\n",
            "protocols.loops.util: PepBondGeom: 138 L  C-N --  2.10001 max: 1.5\n",
            "protocols.grafting.CCDEndsGraftMover: round 23\n",
            "protocols.loops.util: PepBondGeom: 138 L  C-N --  2.04216 max: 1.5\n",
            "protocols.grafting.CCDEndsGraftMover: round 24\n",
            "protocols.loops.util: PepBondGeom: 138 L  C-N --  2.1658 max: 1.5\n",
            "protocols.grafting.CCDEndsGraftMover: round 25\n",
            "protocols.loops.util: PepBondGeom: 138 L  C-N --  2.13987 max: 1.5\n",
            "protocols.grafting.CCDEndsGraftMover: round 26\n",
            "protocols.loops.util: PepBondGeom: 138 L  C-N --  2.14132 max: 1.5\n",
            "protocols.grafting.CCDEndsGraftMover: round 27\n",
            "protocols.loops.util: PepBondGeom: 138 L  C-N --  2.10787 max: 1.5\n",
            "protocols.grafting.CCDEndsGraftMover: round 28\n",
            "protocols.loops.util: PepBondGeom: 138 L  C-N --  2.0505 max: 1.5\n",
            "protocols.grafting.CCDEndsGraftMover: round 29\n",
            "protocols.loops.util: PepBondGeom: 138 L  C-N --  2.11613 max: 1.5\n",
            "protocols.grafting.CCDEndsGraftMover: round 30\n",
            "protocols.loops.util: PepBondGeom: 138 L  C-N --  2.07137 max: 1.5\n",
            "protocols.grafting.CCDEndsGraftMover: round 31\n",
            "core.optimization.LineMinimizer: Inaccurate G in step= 3.8147e-06 Deriv= -1.13897 Finite Diff= 9280.28\n",
            "protocols.loops.util: PepBondGeom: 138 L  C-N --  2.09287 max: 1.5\n",
            "protocols.grafting.CCDEndsGraftMover: round 32\n",
            "protocols.loops.util: PepBondGeom: 138 L  C-N --  2.09329 max: 1.5\n",
            "protocols.grafting.CCDEndsGraftMover: round 33\n",
            "protocols.loops.util: PepBondGeom: 138 L  C-N --  2.11823 max: 1.5\n",
            "protocols.grafting.CCDEndsGraftMover: round 34\n",
            "protocols.loops.util: PepBondGeom: 138 L  C-N --  2.10119 max: 1.5\n",
            "protocols.grafting.CCDEndsGraftMover: round 35\n",
            "protocols.loops.util: PepBondGeom: 138 L  C-N --  2.14725 max: 1.5\n",
            "protocols.grafting.CCDEndsGraftMover: round 36\n",
            "protocols.loops.util: PepBondGeom: 138 L  C-N --  2.1009 max: 1.5\n",
            "protocols.grafting.CCDEndsGraftMover: round 37\n",
            "protocols.loops.util: PepBondGeom: 138 L  C-N --  2.07909 max: 1.5\n",
            "protocols.grafting.CCDEndsGraftMover: round 38\n",
            "protocols.loops.util: PepBondGeom: 138 L  C-N --  2.12357 max: 1.5\n",
            "protocols.grafting.CCDEndsGraftMover: round 39\n",
            "protocols.loops.util: PepBondGeom: 138 L  C-N --  2.20687 max: 1.5\n",
            "protocols.grafting.CCDEndsGraftMover: round 40\n",
            "protocols.loops.util: PepBondGeom: 138 L  C-N --  2.1075 max: 1.5\n",
            "protocols.grafting.CCDEndsGraftMover: round 41\n",
            "protocols.loops.util: PepBondGeom: 138 L  C-N --  2.24215 max: 1.5\n",
            "protocols.grafting.CCDEndsGraftMover: round 42\n",
            "protocols.loops.util: PepBondGeom: 138 L  C-N --  2.16058 max: 1.5\n",
            "protocols.grafting.CCDEndsGraftMover: round 43\n",
            "protocols.loops.util: PepBondGeom: 138 L  C-N --  2.10946 max: 1.5\n",
            "protocols.grafting.CCDEndsGraftMover: round 44\n",
            "protocols.loops.util: PepBondGeom: 138 L  C-N --  2.07861 max: 1.5\n",
            "protocols.grafting.CCDEndsGraftMover: round 45\n",
            "protocols.loops.util: PepBondGeom: 138 L  C-N --  2.10722 max: 1.5\n",
            "protocols.grafting.CCDEndsGraftMover: round 46\n",
            "protocols.loops.util: PepBondGeom: 138 L  C-N --  2.14808 max: 1.5\n",
            "protocols.grafting.CCDEndsGraftMover: round 47\n",
            "protocols.loops.util: PepBondGeom: 138 L  C-N --  2.17843 max: 1.5\n",
            "protocols.grafting.CCDEndsGraftMover: round 48\n",
            "protocols.loops.util: PepBondGeom: 138 L  C-N --  2.1447 max: 1.5\n",
            "protocols.grafting.CCDEndsGraftMover: round 49\n",
            "protocols.loops.util: PepBondGeom: 138 L  C-N --  2.09984 max: 1.5\n",
            "protocols.grafting.CCDEndsGraftMover: round 50\n",
            "protocols.loops.util: PepBondGeom: 138 L  C-N --  2.12918 max: 1.5\n",
            "protocols.grafting.CCDEndsGraftMover: finish 1564.27\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 260 rotamers at 41 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.loops.util: PepBondGeom: 970 - 138 L  Ca-C-N --  99.5 (min): 160.005 : 134.5 (max)\n",
            "protocols.grafting.CCDEndsGraftMover: Graft meets ideal geometry true\n",
            "protocols.grafting.CCDEndsGraftMover: Complete\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L3-9-cis7-2 data: 11 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/L3-9-cis7-2.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 18 constraints\n",
            "protocols.antibody.design.AntibodyDesignMover: Checking Geometry\n",
            "protocols.loops.util: PepBondGeom: 970 - 138 L  Ca-C-N --  99.5 (min): 160.005 : 134.5 (max)\n",
            "protocols.antibody.design.AntibodyDesignMover: Graft not closed. Adapting graft closure\n",
            "protocols.grafting.util: Deleting 9 residues from 962 to 970\n",
            "core.conformation.Conformation: Found disulfide between residues 771 845\n",
            "core.conformation.Conformation: Found disulfide between residues 891 961\n",
            "protocols.grafting.util: insert_point 961\n",
            "protocols.grafting.util: insert_start 962\n",
            "protocols.grafting.util: insert_end 970\n",
            "protocols.grafting.util: insert_length 9\n",
            "protocols.grafting.util: FOLD_TREE  EDGE 1 961 -1  EDGE 961 962 1  EDGE 962 971 -1\n",
            "protocols.grafting.GraftMoverBase: Insertion complete.\n",
            "protocols.grafting.AnchoredGraftMover: Setting default movemap\n",
            "protocols.grafting.util: FOLD_TREE  EDGE 1 959 -1  EDGE 959 971 -1  EDGE 959 973 1  EDGE 973 972 -1  EDGE 973 980 -1\n",
            "protocols.grafting.AnchoredGraftMover: Loops: LOOP  begin  end  cut  skip_rate  extended\n",
            "protocols.grafting.AnchoredGraftMover: LOOP start: 960  stop: 972  cut: 971  size: 13  skip rate: 0  extended?: False\n",
            "protocols.grafting.AnchoredGraftMover:\n",
            "protocols.grafting.AnchoredGraftMover:\n",
            "protocols.grafting.util: Loop: 1\n",
            "protocols.grafting.util: Add variant to: 971\n",
            "protocols.grafting.util: idealized  960\n",
            "protocols.grafting.util: idealized  961\n",
            "protocols.grafting.util: ideal 962\n",
            "protocols.grafting.util: ideal 970\n",
            "protocols.grafting.util: idealized 971\n",
            "protocols.grafting.util: idealized 972\n",
            "protocols.grafting.AnchoredGraftMover: start 2964.31\n",
            "protocols.grafting.AnchoredGraftMover: round 1\n",
            "protocols.grafting.AnchoredGraftMover: Graft Closed early - returning\n",
            "protocols.grafting.AnchoredGraftMover: 1 1424.81\n",
            "protocols.grafting.AnchoredGraftMover: finish 1424.81\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 260 rotamers at 41 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.grafting.AnchoredGraftMover: Graft meets ideal geometry true\n",
            "protocols.grafting.AnchoredGraftMover: Complete\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L3-9-cis7-2 data: 11 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/L3-9-cis7-2.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 18 constraints\n",
            "protocols.antibody.design.AntibodyDesignMover: Checking Geometry\n",
            "protocols.antibody.design.AntibodyDesignMover: DATA GRAFT_CLOSURE 1 rabd/my_ab.pdb L3 L3-9-cis7-2 1a2yA_L3_0001\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H1\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H2\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H3\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L1\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L2\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L3\n",
            "protocols.antibody.clusters.CDRClusterSet: Adding cacheable cluster data to pose\n",
            "protocols.antibody.design.util: Setting CDR sequence in NativeAntibodySeq\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: H1-13-1 data: 583 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/H1-13-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 26 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: H2-10-1 data: 328 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/H2-10-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 20 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: Adding general dihedral constraints for H3\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L1-16-1 data: 117 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/L1-16-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 32 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L2-8-1 data: 632 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/L2-8-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 16 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L3-9-cis7-2 data: 11 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/L3-9-cis7-2.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 18 constraints\n",
            "protocols.antibody.design.AntibodyDesignMover: Beginning MinProtocol\n",
            "protocols.antibody.design.AntibodyDesignMover: Inner round: 1\n",
            "protocols.antibody.design.AntibodyDesignMover: Add min neighbors : L1\n",
            "protocols.antibody.design.AntibodyDesignMover: Add min neighbors : H3\n",
            "basic.io.database: [ WARNING ] Unable to locate database file /sampling/antibodies/antibody_database_rosetta_design.db\n",
            "antibody.database.AntibodyDatabaseManager: The antibody design database is not specified by the user.\n",
            "antibody.database.AntibodyDatabaseManager: Reading from: /usr/local/lib/python3.10/dist-packages/pyrosetta/database//sampling/antibodies/antibody_database_rosetta_design_north_paper.db\n",
            "antibody.database.AntibodyDatabaseManager: Loaded L1-16-1 with 47 datapoints.\n",
            "antibody.database.AntibodyDatabaseManager: Skipped L3-9-cis7-2 with 7 data point\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Data pre-loaded.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: applying prob task op\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Using conservative op for L3\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Using original bb sequence for conservative design.\n",
            "protocols.task_operations.ConservativeDesignOperation: Loading conservative mutational data\n",
            "core.pack.pack_rotamers: built 1417 rotamers at 103 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating PDInteractionGraph\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs start: 38117.7\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs end: 29407.3\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 852\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 853\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 899\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 900\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 966\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 967\n",
            "core.optimization.LineMinimizer: Inaccurate G in step= 3.72529e-09 Deriv= -119.672 Finite Diff= 1.78367e+09\n",
            "core.optimization.LineMinimizer: Inaccurate G in step= 6.19379e-09 Deriv= -623.159 Finite Diff= 1.07261e+09\n",
            "core.optimization.LineMinimizer: Inaccurate G in step= 1.98369e-07 Deriv= -1479.22 Finite Diff= 3.34891e+07\n",
            "core.optimization.Minimizer: Line search failed even after resetting Hessian; aborting at iter#889\n",
            "antibody.design.GeneralAntibodyModeler: CDR min start: 29407.3\n",
            "antibody.design.GeneralAntibodyModeler: CDR min end: 9862.49\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: applying prob task op\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Using conservative op for L3\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Using original bb sequence for conservative design.\n",
            "core.pack.pack_rotamers: built 1820 rotamers at 109 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating PDInteractionGraph\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs start: 9862.49\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs end: 9928.24\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 852\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 853\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 899\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 900\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 966\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 967\n",
            "core.optimization.LineMinimizer: Inaccurate G in step= 3.72529e-09 Deriv= -7.78412 Finite Diff= 3.08697e+09\n",
            "core.optimization.LineMinimizer: Inaccurate G in step= 6.42974e-09 Deriv= -15.5698 Finite Diff= 1.7946e+09\n",
            "core.optimization.LineMinimizer: Inaccurate G in step= 6.82062e-07 Deriv= -125.122 Finite Diff= 1.69174e+07\n",
            "core.optimization.Minimizer: Line search failed even after resetting Hessian; aborting at iter#223\n",
            "antibody.design.GeneralAntibodyModeler: CDR min start: 9862.49\n",
            "antibody.design.GeneralAntibodyModeler: CDR min end: 4458.95\n",
            "protocols.antibody.design.AntibodyDesignMover: Min Cycle complete\n",
            "protocols.antibody.design.AntibodyDesignMover:\n",
            "protocols.antibody.design.AntibodyDesignMover: SCORE 1 4458.95 1\n",
            "protocols.antibody.design.AntibodyDesignMover: FINAL 1 4458.95 1\n",
            "protocols.antibody.design.AntibodyDesignMover:\n",
            "protocols.antibody.design.AntibodyDesignMover: Last Accepted Score: 4458.95\n",
            "protocols.antibody.design.AntibodyDesignMover: Lowest        Score:4458.95\n",
            "protocols.antibody.design.AntibodyDesignMover:\n",
            "protocols.antibody.design.AntibodyDesignMover: Outer round: 2\n",
            "protocols.antibody.design.AntibodyDesignMover: Grafting CDR from cluster L3-9-2 fragment 1a4kL_L3_0001\n",
            "protocols.grafting.CCDEndsGraftMover: Start: 961 End: 971 NterO: 3 CterO: 3\n",
            "protocols.grafting.util: Superimposing overhang residues\n",
            "core.chemical.AtomICoor: [ WARNING ] IcoorAtomID::atom_id(): Cannot get atom_id for POLYMER_LOWER of residue TYR 1.  Returning BOGUS ID instead.\n",
            "core.conformation.Residue: [ WARNING ] missing an atom: 1  H   that depends on a nonexistent polymer connection!\n",
            "core.conformation.Residue: [ WARNING ]  --> generating it using idealized coordinates.\n",
            "protocols.grafting.util: Deleting 9 residues from 962 to 970\n",
            "core.conformation.Conformation: Found disulfide between residues 771 845\n",
            "core.conformation.Conformation: Found disulfide between residues 891 961\n",
            "protocols.grafting.util: insert_point 961\n",
            "protocols.grafting.util: insert_start 962\n",
            "protocols.grafting.util: insert_end 970\n",
            "protocols.grafting.util: insert_length 9\n",
            "protocols.grafting.util: FOLD_TREE  EDGE 1 961 -1  EDGE 961 962 1  EDGE 962 971 -1\n",
            "protocols.grafting.GraftMoverBase: Insertion complete.\n",
            "protocols.grafting.AnchoredGraftMover: Setting default movemap\n",
            "protocols.grafting.CCDEndsGraftMover: Start: 961\n",
            "protocols.grafting.CCDEndsGraftMover: Original End: 971\n",
            "protocols.grafting.CCDEndsGraftMover: End: 971\n",
            "protocols.grafting.CCDEndsGraftMover: Insert Length: 9\n",
            "protocols.grafting.CCDEndsGraftMover: LOOP start: 960  stop: 963  cut: 961  size: 4  skip rate: 0  extended?: False\n",
            "protocols.grafting.CCDEndsGraftMover:\n",
            "protocols.grafting.CCDEndsGraftMover: LOOP start: 969  stop: 972  cut: 970  size: 4  skip rate: 0  extended?: False\n",
            "protocols.grafting.CCDEndsGraftMover:\n",
            "protocols.loops.FoldTreeFromLoopsWrapper: old foldtree FOLD_TREE  EDGE 1 961 -1  EDGE 961 971 1  EDGE 971 980 -1  EDGE 961 962 -2  C    N    EDGE 962 963 -2  C    N    EDGE 963 964 -2  C    N    EDGE 964 965 -2  C    N    EDGE 965 966 -2  C    N    EDGE 966 967 -2  C    N    EDGE 967 968 -2  C    N    EDGE 968 969 -2  C    N    EDGE 969 970 -2  C    N   \n",
            "New foldtree FOLD_TREE  EDGE 1 959 -1  EDGE 959 961 -1  EDGE 959 964 1  EDGE 964 962 -1  EDGE 964 968 -1  EDGE 968 970 -1  EDGE 968 973 2  EDGE 973 971 -1  EDGE 973 980 -1\n",
            "protocols.grafting.util: Loop: 1\n",
            "protocols.grafting.util: Add variant to: 961\n",
            "protocols.grafting.util: Loop: 2\n",
            "protocols.grafting.util: Add variant to: 970\n",
            "protocols.grafting.util: idealized  960\n",
            "protocols.grafting.util: idealized  961\n",
            "protocols.grafting.util: ideal 962\n",
            "protocols.grafting.util: ideal 970\n",
            "protocols.grafting.util: idealized 971\n",
            "protocols.grafting.util: idealized 972\n",
            "protocols.grafting.CCDEndsGraftMover: start 2802.72\n",
            "protocols.grafting.CCDEndsGraftMover: round 1\n",
            "protocols.grafting.CCDEndsGraftMover: Graft Closed early - returning\n",
            "protocols.grafting.CCDEndsGraftMover: 1 2343.02\n",
            "protocols.grafting.CCDEndsGraftMover: finish 2343.02\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 184 rotamers at 32 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.grafting.CCDEndsGraftMover: Graft meets ideal geometry true\n",
            "protocols.grafting.CCDEndsGraftMover: Complete\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L3-9-2 data: 25 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/L3-9-2.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 18 constraints\n",
            "protocols.antibody.design.AntibodyDesignMover: Checking Geometry\n",
            "protocols.antibody.design.AntibodyDesignMover: DATA GRAFT_CLOSURE 1 rabd/my_ab.pdb L3 L3-9-2 1a4kL_L3_0001\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H1\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H2\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H3\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L1\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L2\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L3\n",
            "protocols.antibody.clusters.CDRClusterSet: Adding cacheable cluster data to pose\n",
            "protocols.antibody.design.util: Setting CDR sequence in NativeAntibodySeq\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: H1-13-1 data: 583 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/H1-13-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 26 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: H2-10-1 data: 328 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/H2-10-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 20 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: Adding general dihedral constraints for H3\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L1-16-1 data: 117 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/L1-16-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 32 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L2-8-1 data: 632 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/L2-8-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 16 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L3-9-2 data: 25 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/L3-9-2.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 18 constraints\n",
            "protocols.antibody.design.AntibodyDesignMover: Beginning MinProtocol\n",
            "protocols.antibody.design.AntibodyDesignMover: Inner round: 1\n",
            "protocols.antibody.design.AntibodyDesignMover: Add min neighbors : L1\n",
            "protocols.antibody.design.AntibodyDesignMover: Add min neighbors : H3\n",
            "basic.io.database: [ WARNING ] Unable to locate database file /sampling/antibodies/antibody_database_rosetta_design.db\n",
            "antibody.database.AntibodyDatabaseManager: The antibody design database is not specified by the user.\n",
            "antibody.database.AntibodyDatabaseManager: Reading from: /usr/local/lib/python3.10/dist-packages/pyrosetta/database//sampling/antibodies/antibody_database_rosetta_design_north_paper.db\n",
            "antibody.database.AntibodyDatabaseManager: Loaded L1-16-1 with 47 datapoints.\n",
            "antibody.database.AntibodyDatabaseManager: Loaded L3-9-2 with 11 datapoints.\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Data pre-loaded.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: applying prob task op\n",
            "core.pack.pack_rotamers: built 1345 rotamers at 107 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating PDInteractionGraph\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs start: 8474.1\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs end: 13665.4\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 852\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 853\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 899\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 900\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 966\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 967\n",
            "antibody.design.GeneralAntibodyModeler: CDR min start: 8474.1\n",
            "antibody.design.GeneralAntibodyModeler: CDR min end: 3839.85\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: applying prob task op\n",
            "core.pack.pack_rotamers: built 1501 rotamers at 108 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating PDInteractionGraph\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs start: 3839.85\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs end: 9970.56\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 852\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 853\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 899\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 900\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 966\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 967\n",
            "antibody.design.GeneralAntibodyModeler: CDR min start: 3839.85\n",
            "antibody.design.GeneralAntibodyModeler: CDR min end: 3831.2\n",
            "protocols.antibody.design.AntibodyDesignMover: Min Cycle complete\n",
            "protocols.antibody.design.AntibodyDesignMover:\n",
            "protocols.antibody.design.AntibodyDesignMover: SCORE 2 3831.2 1\n",
            "protocols.antibody.design.AntibodyDesignMover: FINAL 2 3831.2 1\n",
            "protocols.antibody.design.AntibodyDesignMover:\n",
            "protocols.antibody.design.AntibodyDesignMover: Last Accepted Score: 3831.2\n",
            "protocols.antibody.design.AntibodyDesignMover: Lowest        Score:3831.2\n",
            "protocols.antibody.design.AntibodyDesignMover:\n",
            "protocols.antibody.design.AntibodyDesignMover: Outer round: 3\n",
            "protocols.antibody.design.AntibodyDesignMover: Grafting CDR from cluster L3-8-cis6-1 fragment 1e6oL_L3_0001\n",
            "protocols.grafting.CCDEndsGraftMover: Start: 961 End: 971 NterO: 3 CterO: 3\n",
            "protocols.grafting.util: Superimposing overhang residues\n",
            "core.chemical.AtomICoor: [ WARNING ] IcoorAtomID::atom_id(): Cannot get atom_id for POLYMER_LOWER of residue TYR 1.  Returning BOGUS ID instead.\n",
            "core.conformation.Residue: [ WARNING ] missing an atom: 1  H   that depends on a nonexistent polymer connection!\n",
            "core.conformation.Residue: [ WARNING ]  --> generating it using idealized coordinates.\n",
            "protocols.grafting.util: Deleting 9 residues from 962 to 970\n",
            "core.conformation.Conformation: Found disulfide between residues 771 845\n",
            "core.conformation.Conformation: Found disulfide between residues 891 961\n",
            "protocols.grafting.util: insert_point 961\n",
            "protocols.grafting.util: insert_start 962\n",
            "protocols.grafting.util: insert_end 969\n",
            "protocols.grafting.util: insert_length 8\n",
            "protocols.grafting.util: FOLD_TREE  EDGE 1 961 -1  EDGE 961 962 1  EDGE 962 971 -1\n",
            "protocols.grafting.GraftMoverBase: Insertion complete.\n",
            "protocols.grafting.AnchoredGraftMover: Setting default movemap\n",
            "protocols.grafting.CCDEndsGraftMover: Start: 961\n",
            "protocols.grafting.CCDEndsGraftMover: Original End: 971\n",
            "protocols.grafting.CCDEndsGraftMover: End: 970\n",
            "protocols.grafting.CCDEndsGraftMover: Insert Length: 8\n",
            "protocols.grafting.CCDEndsGraftMover: LOOP start: 960  stop: 963  cut: 961  size: 4  skip rate: 0  extended?: False\n",
            "protocols.grafting.CCDEndsGraftMover:\n",
            "protocols.grafting.CCDEndsGraftMover: LOOP start: 968  stop: 971  cut: 969  size: 4  skip rate: 0  extended?: False\n",
            "protocols.grafting.CCDEndsGraftMover:\n",
            "protocols.loops.FoldTreeFromLoopsWrapper: old foldtree FOLD_TREE  EDGE 1 961 -1  EDGE 961 970 1  EDGE 970 979 -1  EDGE 961 962 -2  C    N    EDGE 962 963 -2  C    N    EDGE 963 964 -2  C    N    EDGE 964 965 -2  C    N    EDGE 965 966 -2  C    N    EDGE 966 967 -2  C    N    EDGE 967 968 -2  C    N    EDGE 968 969 -2  C    N   \n",
            "New foldtree FOLD_TREE  EDGE 1 959 -1  EDGE 959 961 -1  EDGE 959 964 1  EDGE 964 962 -1  EDGE 964 967 -1  EDGE 967 969 -1  EDGE 967 972 2  EDGE 972 970 -1  EDGE 972 979 -1\n",
            "protocols.grafting.util: Loop: 1\n",
            "protocols.grafting.util: Add variant to: 961\n",
            "protocols.grafting.util: Loop: 2\n",
            "protocols.grafting.util: Add variant to: 969\n",
            "protocols.grafting.util: idealized  960\n",
            "protocols.grafting.util: idealized  961\n",
            "protocols.grafting.util: ideal 962\n",
            "protocols.grafting.util: ideal 969\n",
            "protocols.grafting.util: idealized 970\n",
            "protocols.grafting.util: idealized 971\n",
            "protocols.grafting.CCDEndsGraftMover: start 2852.6\n",
            "protocols.grafting.CCDEndsGraftMover: round 1\n",
            "protocols.grafting.CCDEndsGraftMover: Graft Closed early - returning\n",
            "protocols.grafting.CCDEndsGraftMover: 1 2365.93\n",
            "protocols.grafting.CCDEndsGraftMover: finish 2365.93\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 253 rotamers at 34 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.grafting.CCDEndsGraftMover: Graft meets ideal geometry true\n",
            "protocols.grafting.CCDEndsGraftMover: Complete\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L3-8-cis6-1 data: 4 cutoff: 10\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: Adding general dihedral constraints for L3\n",
            "protocols.antibody.design.AntibodyDesignMover: Checking Geometry\n",
            "protocols.antibody.design.AntibodyDesignMover: DATA GRAFT_CLOSURE 1 rabd/my_ab.pdb L3 L3-8-cis6-1 1e6oL_L3_0001\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H1\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H2\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H3\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L1\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L2\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L3\n",
            "protocols.antibody.clusters.CDRClusterSet: Adding cacheable cluster data to pose\n",
            "protocols.antibody.design.util: Setting CDR sequence in NativeAntibodySeq\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: H1-13-1 data: 583 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/H1-13-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 26 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: H2-10-1 data: 328 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/H2-10-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 20 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: Adding general dihedral constraints for H3\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L1-16-1 data: 117 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/L1-16-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 32 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L2-8-1 data: 632 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/L2-8-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 16 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L3-8-cis6-1 data: 4 cutoff: 10\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: Adding general dihedral constraints for L3\n",
            "protocols.antibody.design.AntibodyDesignMover: Beginning MinProtocol\n",
            "protocols.antibody.design.AntibodyDesignMover: Inner round: 1\n",
            "protocols.antibody.design.AntibodyDesignMover: Add min neighbors : L1\n",
            "protocols.antibody.design.AntibodyDesignMover: Add min neighbors : H3\n",
            "basic.io.database: [ WARNING ] Unable to locate database file /sampling/antibodies/antibody_database_rosetta_design.db\n",
            "antibody.database.AntibodyDatabaseManager: The antibody design database is not specified by the user.\n",
            "antibody.database.AntibodyDatabaseManager: Reading from: /usr/local/lib/python3.10/dist-packages/pyrosetta/database//sampling/antibodies/antibody_database_rosetta_design_north_paper.db\n",
            "antibody.database.AntibodyDatabaseManager: Loaded L1-16-1 with 47 datapoints.\n",
            "antibody.database.AntibodyDatabaseManager: Skipped L3-8-cis6-1 with 3 data point\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Data pre-loaded.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: applying prob task op\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Using conservative op for L3\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Using original bb sequence for conservative design.\n",
            "protocols.task_operations.ConservativeDesignOperation: Loading conservative mutational data\n",
            "core.pack.pack_rotamers: built 1618 rotamers at 104 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating PDInteractionGraph\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs start: 12511.6\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs end: 13242.2\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 852\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 853\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 899\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 900\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 965\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 966\n",
            "antibody.design.GeneralAntibodyModeler: CDR min start: 12511.6\n",
            "antibody.design.GeneralAntibodyModeler: CDR min end: 3892.29\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: applying prob task op\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Using conservative op for L3\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Using original bb sequence for conservative design.\n",
            "core.pack.pack_rotamers: built 1718 rotamers at 107 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating PDInteractionGraph\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs start: 3892.29\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs end: 9567.69\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 852\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 853\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 899\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 900\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 965\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 966\n",
            "antibody.design.GeneralAntibodyModeler: CDR min start: 3892.29\n",
            "antibody.design.GeneralAntibodyModeler: CDR min end: 3868.77\n",
            "protocols.antibody.design.AntibodyDesignMover: Min Cycle complete\n",
            "protocols.antibody.design.AntibodyDesignMover:\n",
            "protocols.antibody.design.AntibodyDesignMover: SCORE 3 3868.77 0\n",
            "protocols.antibody.design.AntibodyDesignMover: FINAL 3 3831.2 1\n",
            "protocols.antibody.design.AntibodyDesignMover:\n",
            "protocols.antibody.design.AntibodyDesignMover: Last Accepted Score: 3831.2\n",
            "protocols.antibody.design.AntibodyDesignMover: Lowest        Score:3831.2\n",
            "protocols.antibody.design.AntibodyDesignMover:\n",
            "protocols.antibody.design.AntibodyDesignMover: Outer round: 4\n",
            "protocols.antibody.design.AntibodyDesignMover: Grafting CDR from cluster L3-9-cis6-1 fragment 2fbjL_L3_0001\n",
            "core.scoring.ScoreFunctionFactory: SCOREFUNCTION: ref2015\n",
            "core.scoring.ScoreFunctionFactory: SCOREFUNCTION: ref2015\n",
            "protocols.grafting.CCDEndsGraftMover: Start: 961 End: 971 NterO: 3 CterO: 3\n",
            "protocols.grafting.util: Superimposing overhang residues\n",
            "core.chemical.AtomICoor: [ WARNING ] IcoorAtomID::atom_id(): Cannot get atom_id for POLYMER_LOWER of residue TYR 1.  Returning BOGUS ID instead.\n",
            "core.conformation.Residue: [ WARNING ] missing an atom: 1  H   that depends on a nonexistent polymer connection!\n",
            "core.conformation.Residue: [ WARNING ]  --> generating it using idealized coordinates.\n",
            "protocols.grafting.util: Deleting 9 residues from 962 to 970\n",
            "core.conformation.Conformation: Found disulfide between residues 771 845\n",
            "core.conformation.Conformation: Found disulfide between residues 891 961\n",
            "protocols.grafting.util: insert_point 961\n",
            "protocols.grafting.util: insert_start 962\n",
            "protocols.grafting.util: insert_end 970\n",
            "protocols.grafting.util: insert_length 9\n",
            "protocols.grafting.util: FOLD_TREE  EDGE 1 961 -1  EDGE 961 962 1  EDGE 962 971 -1\n",
            "protocols.grafting.GraftMoverBase: Insertion complete.\n",
            "protocols.grafting.AnchoredGraftMover: Setting default movemap\n",
            "protocols.grafting.CCDEndsGraftMover: Start: 961\n",
            "protocols.grafting.CCDEndsGraftMover: Original End: 971\n",
            "protocols.grafting.CCDEndsGraftMover: End: 971\n",
            "protocols.grafting.CCDEndsGraftMover: Insert Length: 9\n",
            "protocols.grafting.CCDEndsGraftMover: LOOP start: 960  stop: 963  cut: 961  size: 4  skip rate: 0  extended?: False\n",
            "protocols.grafting.CCDEndsGraftMover:\n",
            "protocols.grafting.CCDEndsGraftMover: LOOP start: 969  stop: 972  cut: 970  size: 4  skip rate: 0  extended?: False\n",
            "protocols.grafting.CCDEndsGraftMover:\n",
            "protocols.loops.FoldTreeFromLoopsWrapper: old foldtree FOLD_TREE  EDGE 1 961 -1  EDGE 961 971 1  EDGE 971 980 -1  EDGE 961 962 -2  C    N    EDGE 962 963 -2  C    N    EDGE 963 964 -2  C    N    EDGE 964 965 -2  C    N    EDGE 965 966 -2  C    N    EDGE 966 967 -2  C    N    EDGE 967 968 -2  C    N    EDGE 968 969 -2  C    N    EDGE 969 970 -2  C    N   \n",
            "New foldtree FOLD_TREE  EDGE 1 959 -1  EDGE 959 961 -1  EDGE 959 964 1  EDGE 964 962 -1  EDGE 964 968 -1  EDGE 968 970 -1  EDGE 968 973 2  EDGE 973 971 -1  EDGE 973 980 -1\n",
            "protocols.grafting.util: Loop: 1\n",
            "protocols.grafting.util: Add variant to: 961\n",
            "protocols.grafting.util: Loop: 2\n",
            "protocols.grafting.util: Add variant to: 970\n",
            "protocols.grafting.util: idealized  960\n",
            "protocols.grafting.util: idealized  961\n",
            "protocols.grafting.util: ideal 962\n",
            "protocols.grafting.util: ideal 970\n",
            "protocols.grafting.util: idealized 971\n",
            "protocols.grafting.util: idealized 972\n",
            "protocols.grafting.CCDEndsGraftMover: start 3006.72\n",
            "protocols.grafting.CCDEndsGraftMover: round 1\n",
            "protocols.grafting.CCDEndsGraftMover: Graft Closed early - returning\n",
            "protocols.grafting.CCDEndsGraftMover: 1 2359.49\n",
            "protocols.grafting.CCDEndsGraftMover: finish 2359.49\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 294 rotamers at 35 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.grafting.CCDEndsGraftMover: Graft meets ideal geometry true\n",
            "protocols.grafting.CCDEndsGraftMover: Complete\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L3-9-cis6-1 data: 1 cutoff: 10\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: Adding general dihedral constraints for L3\n",
            "protocols.antibody.design.AntibodyDesignMover: Checking Geometry\n",
            "protocols.antibody.design.AntibodyDesignMover: DATA GRAFT_CLOSURE 1 rabd/my_ab.pdb L3 L3-9-cis6-1 2fbjL_L3_0001\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H1\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H2\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H3\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L1\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L2\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L3\n",
            "protocols.antibody.clusters.CDRClusterSet: Adding cacheable cluster data to pose\n",
            "protocols.antibody.design.util: Setting CDR sequence in NativeAntibodySeq\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: H1-13-1 data: 583 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/H1-13-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 26 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: H2-10-1 data: 328 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/H2-10-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 20 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: Adding general dihedral constraints for H3\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L1-16-1 data: 117 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/L1-16-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 32 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L2-8-1 data: 632 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/L2-8-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 16 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L3-9-cis6-1 data: 1 cutoff: 10\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: Adding general dihedral constraints for L3\n",
            "protocols.antibody.design.AntibodyDesignMover: Beginning MinProtocol\n",
            "protocols.antibody.design.AntibodyDesignMover: Inner round: 1\n",
            "protocols.antibody.design.AntibodyDesignMover: Add min neighbors : L1\n",
            "protocols.antibody.design.AntibodyDesignMover: Add min neighbors : H3\n",
            "basic.io.database: [ WARNING ] Unable to locate database file /sampling/antibodies/antibody_database_rosetta_design.db\n",
            "antibody.database.AntibodyDatabaseManager: The antibody design database is not specified by the user.\n",
            "antibody.database.AntibodyDatabaseManager: Reading from: /usr/local/lib/python3.10/dist-packages/pyrosetta/database//sampling/antibodies/antibody_database_rosetta_design_north_paper.db\n",
            "antibody.database.AntibodyDatabaseManager: Loaded L1-16-1 with 47 datapoints.\n",
            "antibody.database.AntibodyDatabaseManager: Skipped L3-9-cis6-1 with 1 data point\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Data pre-loaded.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: applying prob task op\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Using conservative op for L3\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Using original bb sequence for conservative design.\n",
            "protocols.task_operations.ConservativeDesignOperation: Loading conservative mutational data\n",
            "core.pack.pack_rotamers: built 1710 rotamers at 110 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating PDInteractionGraph\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs start: 17125.5\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs end: 14239.9\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 852\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 853\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 899\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 900\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 966\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 967\n",
            "core.optimization.LineMinimizer: Inaccurate G in step= 7.45058e-09 Deriv= -183.305 Finite Diff= 6.20635e+09\n",
            "core.optimization.LineMinimizer: Inaccurate G in step= 2.58705e-09 Deriv= -56618.4 Finite Diff= 1.78736e+10\n",
            "core.optimization.LineMinimizer: Inaccurate G in step= 3.31881e-07 Deriv= -528.463 Finite Diff= 1.39327e+08\n",
            "core.optimization.Minimizer: Line search failed even after resetting Hessian; aborting at iter#85\n",
            "antibody.design.GeneralAntibodyModeler: CDR min start: 14239.9\n",
            "antibody.design.GeneralAntibodyModeler: CDR min end: 4169.33\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: applying prob task op\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Using conservative op for L3\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Using original bb sequence for conservative design.\n",
            "core.pack.pack_rotamers: built 1635 rotamers at 108 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating PDInteractionGraph\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs start: 4169.33\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs end: 8412.14\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 852\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 853\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 899\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 900\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 966\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 967\n",
            "core.optimization.LineMinimizer: Inaccurate G in step= 9.53674e-09 Deriv= -45661 Finite Diff= 6.89792e+09\n",
            "core.optimization.LineMinimizer: Inaccurate G in step= 3.5704e-08 Deriv= -45661 Finite Diff= 1.84244e+09\n",
            "core.optimization.Minimizer: Line search failed even after resetting Hessian; aborting at iter#3\n",
            "antibody.design.GeneralAntibodyModeler: CDR min start: 4169.33\n",
            "antibody.design.GeneralAntibodyModeler: CDR min end: 4169.33\n",
            "protocols.antibody.design.AntibodyDesignMover: Min Cycle complete\n",
            "protocols.antibody.design.AntibodyDesignMover:\n",
            "protocols.antibody.design.AntibodyDesignMover: SCORE 4 4169.33 0\n",
            "protocols.antibody.design.AntibodyDesignMover: FINAL 4 3831.2 1\n",
            "protocols.antibody.design.AntibodyDesignMover:\n",
            "protocols.antibody.design.AntibodyDesignMover: Last Accepted Score: 3831.2\n",
            "protocols.antibody.design.AntibodyDesignMover: Lowest        Score:3831.2\n",
            "protocols.antibody.design.AntibodyDesignMover:\n",
            "protocols.antibody.design.AntibodyDesignMover: Outer round: 5\n",
            "protocols.antibody.design.AntibodyDesignMover: Grafting CDR from cluster L3-11-cis7-1 fragment 2nxyC_L3_0001\n",
            "core.scoring.ScoreFunctionFactory: SCOREFUNCTION: ref2015\n",
            "core.scoring.ScoreFunctionFactory: SCOREFUNCTION: ref2015\n",
            "protocols.grafting.CCDEndsGraftMover: Start: 961 End: 971 NterO: 3 CterO: 3\n",
            "protocols.grafting.util: Superimposing overhang residues\n",
            "core.chemical.AtomICoor: [ WARNING ] IcoorAtomID::atom_id(): Cannot get atom_id for POLYMER_LOWER of residue TYR 1.  Returning BOGUS ID instead.\n",
            "core.conformation.Residue: [ WARNING ] missing an atom: 1  H   that depends on a nonexistent polymer connection!\n",
            "core.conformation.Residue: [ WARNING ]  --> generating it using idealized coordinates.\n",
            "protocols.grafting.util: Deleting 9 residues from 962 to 970\n",
            "core.conformation.Conformation: Found disulfide between residues 771 845\n",
            "core.conformation.Conformation: Found disulfide between residues 891 961\n",
            "protocols.grafting.util: insert_point 961\n",
            "protocols.grafting.util: insert_start 962\n",
            "protocols.grafting.util: insert_end 972\n",
            "protocols.grafting.util: insert_length 11\n",
            "protocols.grafting.util: FOLD_TREE  EDGE 1 961 -1  EDGE 961 962 1  EDGE 962 971 -1\n",
            "protocols.grafting.GraftMoverBase: Insertion complete.\n",
            "protocols.grafting.AnchoredGraftMover: Setting default movemap\n",
            "protocols.grafting.CCDEndsGraftMover: Start: 961\n",
            "protocols.grafting.CCDEndsGraftMover: Original End: 971\n",
            "protocols.grafting.CCDEndsGraftMover: End: 973\n",
            "protocols.grafting.CCDEndsGraftMover: Insert Length: 11\n",
            "protocols.grafting.CCDEndsGraftMover: LOOP start: 960  stop: 963  cut: 961  size: 4  skip rate: 0  extended?: False\n",
            "protocols.grafting.CCDEndsGraftMover:\n",
            "protocols.grafting.CCDEndsGraftMover: LOOP start: 971  stop: 974  cut: 972  size: 4  skip rate: 0  extended?: False\n",
            "protocols.grafting.CCDEndsGraftMover:\n",
            "protocols.loops.FoldTreeFromLoopsWrapper: old foldtree FOLD_TREE  EDGE 1 961 -1  EDGE 961 973 1  EDGE 973 982 -1  EDGE 961 962 -2  C    N    EDGE 962 963 -2  C    N    EDGE 963 964 -2  C    N    EDGE 964 965 -2  C    N    EDGE 965 966 -2  C    N    EDGE 966 967 -2  C    N    EDGE 967 968 -2  C    N    EDGE 968 969 -2  C    N    EDGE 969 970 -2  C    N    EDGE 970 971 -2  C    N    EDGE 971 972 -2  C    N   \n",
            "New foldtree FOLD_TREE  EDGE 1 959 -1  EDGE 959 961 -1  EDGE 959 964 1  EDGE 964 962 -1  EDGE 964 970 -1  EDGE 970 972 -1  EDGE 970 975 2  EDGE 975 973 -1  EDGE 975 982 -1\n",
            "protocols.grafting.util: Loop: 1\n",
            "protocols.grafting.util: Add variant to: 961\n",
            "protocols.grafting.util: Loop: 2\n",
            "protocols.grafting.util: Add variant to: 972\n",
            "protocols.grafting.util: idealized  960\n",
            "protocols.grafting.util: idealized  961\n",
            "protocols.grafting.util: ideal 962\n",
            "protocols.grafting.util: ideal 972\n",
            "protocols.grafting.util: idealized 973\n",
            "protocols.grafting.util: idealized 974\n",
            "protocols.grafting.CCDEndsGraftMover: start 3027.9\n",
            "protocols.grafting.CCDEndsGraftMover: round 1\n",
            "protocols.grafting.CCDEndsGraftMover: Graft Closed early - returning\n",
            "protocols.grafting.CCDEndsGraftMover: 1 2418.1\n",
            "protocols.grafting.CCDEndsGraftMover: finish 2418.1\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 333 rotamers at 37 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.grafting.CCDEndsGraftMover: Graft meets ideal geometry true\n",
            "protocols.grafting.CCDEndsGraftMover: Complete\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L3-11-cis7-1 data: 2 cutoff: 10\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: Adding general dihedral constraints for L3\n",
            "protocols.antibody.design.AntibodyDesignMover: Checking Geometry\n",
            "protocols.antibody.design.AntibodyDesignMover: DATA GRAFT_CLOSURE 1 rabd/my_ab.pdb L3 L3-11-cis7-1 2nxyC_L3_0001\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H1\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H2\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H3\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L1\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L2\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L3\n",
            "protocols.antibody.clusters.CDRClusterSet: Adding cacheable cluster data to pose\n",
            "protocols.antibody.design.util: Setting CDR sequence in NativeAntibodySeq\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: H1-13-1 data: 583 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/H1-13-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 26 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: H2-10-1 data: 328 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/H2-10-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 20 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: Adding general dihedral constraints for H3\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L1-16-1 data: 117 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/L1-16-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 32 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L2-8-1 data: 632 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/L2-8-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 16 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L3-11-cis7-1 data: 2 cutoff: 10\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: Adding general dihedral constraints for L3\n",
            "protocols.antibody.design.AntibodyDesignMover: Beginning MinProtocol\n",
            "protocols.antibody.design.AntibodyDesignMover: Inner round: 1\n",
            "protocols.antibody.design.AntibodyDesignMover: Add min neighbors : L1\n",
            "protocols.antibody.design.AntibodyDesignMover: Add min neighbors : H3\n",
            "basic.io.database: [ WARNING ] Unable to locate database file /sampling/antibodies/antibody_database_rosetta_design.db\n",
            "antibody.database.AntibodyDatabaseManager: The antibody design database is not specified by the user.\n",
            "antibody.database.AntibodyDatabaseManager: Reading from: /usr/local/lib/python3.10/dist-packages/pyrosetta/database//sampling/antibodies/antibody_database_rosetta_design_north_paper.db\n",
            "antibody.database.AntibodyDatabaseManager: Loaded L1-16-1 with 47 datapoints.\n",
            "antibody.database.AntibodyDatabaseManager: Skipped L3-11-cis7-1 with 1 data point\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Data pre-loaded.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: applying prob task op\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Using conservative op for L3\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Using original bb sequence for conservative design.\n",
            "protocols.task_operations.ConservativeDesignOperation: Loading conservative mutational data\n",
            "core.pack.pack_rotamers: built 2013 rotamers at 117 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating PDInteractionGraph\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs start: 22945.3\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs end: 20201.7\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 852\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 853\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 899\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 900\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 967\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 968\n",
            "core.optimization.LineMinimizer: Inaccurate G in step= 7.45058e-09 Deriv= -75.2358 Finite Diff= 9.89658e+09\n",
            "core.optimization.LineMinimizer: Inaccurate G in step= 5.2984e-07 Deriv= -207.344 Finite Diff= 1.39165e+08\n",
            "core.optimization.Minimizer: Line search failed even after resetting Hessian; aborting at iter#125\n",
            "antibody.design.GeneralAntibodyModeler: CDR min start: 20201.7\n",
            "antibody.design.GeneralAntibodyModeler: CDR min end: 8421.48\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: applying prob task op\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Using conservative op for L3\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Using original bb sequence for conservative design.\n",
            "core.pack.pack_rotamers: built 1868 rotamers at 112 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating PDInteractionGraph\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs start: 8421.48\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs end: 10926.8\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 852\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 853\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 899\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 900\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 967\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 968\n",
            "antibody.design.GeneralAntibodyModeler: CDR min start: 8421.48\n",
            "antibody.design.GeneralAntibodyModeler: CDR min end: 3838.59\n",
            "protocols.antibody.design.AntibodyDesignMover: Min Cycle complete\n",
            "protocols.antibody.design.AntibodyDesignMover:\n",
            "protocols.antibody.design.AntibodyDesignMover: SCORE 5 3838.59 0\n",
            "protocols.antibody.design.AntibodyDesignMover: FINAL 5 3831.2 1\n",
            "protocols.antibody.design.AntibodyDesignMover:\n",
            "protocols.antibody.design.AntibodyDesignMover: Last Accepted Score: 3831.2\n",
            "protocols.antibody.design.AntibodyDesignMover: Lowest        Score:3831.2\n",
            "protocols.antibody.design.AntibodyDesignMover:\n",
            "protocols.antibody.design.AntibodyDesignMover: Outer cycles attempted: 5 Outer cycles completed: 5\n",
            "protocols.antibody.design.AntibodyDesignMover:\n",
            "protocols.antibody.design.AntibodyDesignMover: Native Pose (MC): 27545.2\n",
            "protocols.antibody.design.AntibodyDesignMover: Final Pose (MC): 3831.2\n",
            "protocols.antibody.design.AntibodyDesignMover:\n",
            "protocols.moves.TrialCounter:              unk trials=      5;  accepts= 0.4000;  energy_drop/trial= -4742.80701\n",
            "protocols.antibody.design.AntibodyDesignMover:\n",
            "protocols.antibody.design.AntibodyDesignMover: Native Pose (Total): 13400.9\n",
            "protocols.antibody.design.AntibodyDesignMover: Benchmarked Pose (Total): 27545.2\n",
            "protocols.antibody.design.AntibodyDesignMover: Final Pose (Total): 3831.2\n",
            "protocols.antibody.design.AntibodyDesignMover: \n",
            "------------------------------------------------------------\n",
            " Scores                       Weight   Raw Score Wghtd.Score\n",
            "------------------------------------------------------------\n",
            " fa_atr                       1.000   -5766.983   -5766.983\n",
            " fa_rep                       0.550    7255.103    3990.307\n",
            " fa_sol                       1.000    3563.472    3563.472\n",
            " fa_intra_rep                 0.005    3019.010      15.095\n",
            " fa_intra_sol_xover4          1.000     176.832     176.832\n",
            " lk_ball_wtd                  1.000     -70.022     -70.022\n",
            " fa_elec                      1.000   -1575.802   -1575.802\n",
            " pro_close                    1.250       9.794      12.242\n",
            " hbond_sr_bb                  1.000     -74.066     -74.066\n",
            " hbond_lr_bb                  1.000    -505.159    -505.159\n",
            " hbond_bb_sc                  1.000    -136.937    -136.937\n",
            " hbond_sc                     1.000     -75.464     -75.464\n",
            " dslf_fa13                    1.250      -1.062      -1.328\n",
            " atom_pair_constraint         0.010       0.000       0.000\n",
            " omega                        0.400     702.615     281.046\n",
            " fa_dun                       0.700    5257.610    3680.327\n",
            " p_aa_pp                      0.600    -283.350    -170.010\n",
            " yhh_planarity                0.625       0.962       0.601\n",
            " ref                          1.000     375.098     375.098\n",
            " rama_prepro                  0.450     248.779     111.950\n",
            "---------------------------------------------------\n",
            " Total weighted score:                     3831.199\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H1\n",
            "protocols.antibody.cluster.CDRClusterMatcher: Length: 13 Omega: TTTTTTTTTTTTT\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H2\n",
            "protocols.antibody.cluster.CDRClusterMatcher: Length: 10 Omega: TTTTTTTTTT\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H3\n",
            "protocols.antibody.cluster.CDRClusterMatcher: Length: 13 Omega: TTTTTTTTTTTTT\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L1\n",
            "protocols.antibody.cluster.CDRClusterMatcher: Length: 16 Omega: TTTTTTTTCTTCTTTT\n",
            "protocols.antibody.cluster.CDRClusterMatcher: [ WARNING ]\n",
            "protocols.antibody.cluster.CDRClusterMatcher: [ WARNING ] *** No known cluster of CDR length 16 omega of TTTTTTTTCTTCTTTT found. ***\n",
            "protocols.antibody.cluster.CDRClusterMatcher: [ WARNING ] *** Consider using the command-line option -allow_omega_mismatches_for_north_clusters to find the closest cluster! ***\n",
            "protocols.antibody.cluster.CDRClusterMatcher: [ WARNING ]\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L2\n",
            "protocols.antibody.cluster.CDRClusterMatcher: Length: 8 Omega: TTTTTTTT\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L3\n",
            "protocols.antibody.cluster.CDRClusterMatcher: Length: 9 Omega: TTTTTTTTT\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using explicit constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using interface constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: [ WARNING ] In InterfaceAnalyzerMover, calculator Sasa_1 already exists, this is hopefully correct for your purposes\n",
            "protocols.analysis.InterfaceAnalyzerMover: [ WARNING ] In InterfaceAnalyzerMover, calculator InterfaceNeighborDefinition_1 already exists, this is hopefully correct for your purposes\n",
            "protocols.analysis.InterfaceAnalyzerMover: [ WARNING ] In InterfaceAnalyzerMover, calculator InterfaceDeltaEnergetics_1 already exists, this is hopefully correct for your purposes\n",
            "protocols.analysis.InterfaceAnalyzerMover: [ WARNING ] In InterfaceAnalyzerMover, calculator NumberHBonds_1 already exists, this is hopefully correct for your purposes\n",
            "protocols.analysis.InterfaceAnalyzerMover: [ WARNING ] In InterfaceAnalyzerMover, calculator BuriedUnsatisfiedPolars_1 already exists, this is hopefully correct for your purposes\n",
            "protocols.analysis.InterfaceAnalyzerMover: Making an interface set with fixed chains\n",
            "protocols.analysis.InterfaceAnalyzerMover: Interface set residues total: 117\n",
            "protocols.analysis.InterfaceAnalyzerMover: NOT detecting disulfides in the separated pose.\n",
            "protocols.analysis.InterfaceAnalyzerMover: Calculating dSASA\n",
            "protocols.analysis.InterfaceAnalyzerMover: Calculating per-res dSASA data\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 1600 rotamers at 117 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.analysis.InterfaceAnalyzerMover: included_nres: 980\n",
            "protocols.analysis.InterfaceAnalyzerMover: Making hbond ScoreFunction with bb-bb energy method options\n",
            "core.scoring.ScoreFunctionFactory: SCOREFUNCTION: ref2015\n",
            "protocols.analysis.InterfaceAnalyzerMover: Found Hbond between chains: 9 and 13\n",
            "protocols.analysis.InterfaceAnalyzerMover: Found Hbond between chains: 9 and 13\n",
            "protocols.analysis.InterfaceAnalyzerMover: Found Hbond between chains: 12 and 9\n",
            "protocols.analysis.InterfaceAnalyzerMover: Found Hbond between chains: 9 and 13\n",
            "protocols.analysis.InterfaceAnalyzerMover: Found Hbond between chains: 13 and 10\n",
            "protocols.analysis.InterfaceAnalyzerMover: Computing delta unsat polar residues...\n",
            "protocols.analysis.InterfaceAnalyzerMover: Computing Shape Complementarity Score...\n",
            "protocols.analysis.InterfaceAnalyzerMover: Upstream chain(s) numbers: 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11,\n",
            "protocols.analysis.InterfaceAnalyzerMover: Downstream chain(s) numbers: 12, 13,\n",
            "basic.io.database: Database file opened: sampling/antibodies/cluster_center_dihedrals.txt\n",
            "protocols.antibody.AntibodyNumberingParser: Antibody numbering scheme definitions read successfully\n",
            "protocols.antibody.AntibodyNumberingParser: Antibody CDR definition read successfully\n",
            "antibody.AntibodyInfo: Successfully finished the CDR definition\n",
            "antibody.AntibodyInfo: AC Detecting Regular CDR H3 Stem Type\n",
            "antibody.AntibodyInfo: SRWGGDGFYAMDYW\n",
            "antibody.AntibodyInfo: AC Finished Detecting Regular CDR H3 Stem Type: KINKED\n",
            "antibody.AntibodyInfo: AC Finished Detecting Regular CDR H3 Stem Type: Kink: 1 Extended: 0\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H1\n",
            "protocols.antibody.cluster.CDRClusterMatcher: Length: 13 Omega: TTTTTTTTTTTTT\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H2\n",
            "protocols.antibody.cluster.CDRClusterMatcher: Length: 10 Omega: TTTTTTTTTT\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H3\n",
            "protocols.antibody.cluster.CDRClusterMatcher: Length: 13 Omega: TTTTTTTTTTTTT\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L1\n",
            "protocols.antibody.cluster.CDRClusterMatcher: Length: 11 Omega: TTTTTTTTTTT\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L2\n",
            "protocols.antibody.cluster.CDRClusterMatcher: Length: 8 Omega: TTTTTTTT\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L3\n",
            "protocols.antibody.cluster.CDRClusterMatcher: Length: 9 Omega: TTTTTTCTT\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H1\n",
            "protocols.antibody.cluster.CDRClusterMatcher: Length: 13 Omega: TTTTTTTTTTTTT\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H2\n",
            "protocols.antibody.cluster.CDRClusterMatcher: Length: 10 Omega: TTTTTTTTTT\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H3\n",
            "protocols.antibody.cluster.CDRClusterMatcher: Length: 13 Omega: TTTTTTTTTTTTT\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L1\n",
            "protocols.antibody.cluster.CDRClusterMatcher: Length: 11 Omega: TTTTTTTTTTT\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L2\n",
            "protocols.antibody.cluster.CDRClusterMatcher: Length: 8 Omega: TTTTTTTT\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L3\n",
            "protocols.antibody.cluster.CDRClusterMatcher: Length: 9 Omega: TTTTTTCTT\n",
            "protocols.antibody.clusters.CDRClusterSet: Adding cacheable cluster data to pose\n",
            "core.scoring.ScoreFunctionFactory: SCOREFUNCTION: ref2015\n",
            "core.scoring.ScoreFunctionFactory: SCOREFUNCTION: ref2015\n",
            "core.scoring.ScoreFunctionFactory: SCOREFUNCTION: ref2015\n",
            "basic.io.database: [ WARNING ] Unable to locate database file /sampling/antibodies/antibody_database_rosetta_design.db\n",
            "antibody.database.AntibodyDatabaseManager: The antibody design database is not specified by the user.\n",
            "antibody.database.AntibodyDatabaseManager: Reading from: /usr/local/lib/python3.10/dist-packages/pyrosetta/database//sampling/antibodies/antibody_database_rosetta_design_north_paper.db\n",
            "antibody.database.AntibodyDatabaseManager: Checking loadable structures\n",
            "antibody.database.AntibodyDatabaseManager: Skipping 12 structures from reading\n",
            "core.scoring.ScoreFunctionFactory: SCOREFUNCTION: ref2015\n",
            "core.scoring.ScoreFunctionFactory: SCOREFUNCTION: ref2015\n",
            "antibody.database.AntibodyDatabaseManager: Total L1 in CDRSet: 191\n",
            "antibody.database.AntibodyDatabaseManager: Total L3 in CDRSet: 230\n",
            "antibody.database.AntibodyDatabaseManager: 300 Cached\n",
            "antibody.database.AntibodyDatabaseManager: cdrs loaded from database.\n",
            "protocols.antibody.design.AntibodyDesignMover: ///// Total CDRs in set /////\n",
            "protocols.antibody.design.AntibodyDesignMover: /// L1 191\n",
            "protocols.antibody.design.AntibodyDesignMover: /// L3 230\n",
            "protocols.antibody.design.AntibodyDesignMover: Total possible CDR combinations: 43930\n",
            "core.scoring.ScoreFunction: \n",
            "------------------------------------------------------------\n",
            " Scores                       Weight   Raw Score Wghtd.Score\n",
            "------------------------------------------------------------\n",
            " fa_atr                       1.000   -5695.317   -5695.317\n",
            " fa_rep                       0.550   24719.870   13595.929\n",
            " fa_sol                       1.000    3501.170    3501.170\n",
            " fa_intra_rep                 0.005    2661.048      13.305\n",
            " fa_intra_sol_xover4          1.000     175.028     175.028\n",
            " lk_ball_wtd                  1.000     -57.980     -57.980\n",
            " fa_elec                      1.000   -1588.297   -1588.297\n",
            " pro_close                    1.250      10.290      12.862\n",
            " hbond_sr_bb                  1.000     -74.000     -74.000\n",
            " hbond_lr_bb                  1.000    -511.787    -511.787\n",
            " hbond_bb_sc                  1.000    -130.727    -130.727\n",
            " hbond_sc                     1.000     -78.766     -78.766\n",
            " dslf_fa13                    1.250      -1.572      -1.965\n",
            " atom_pair_constraint         0.010       0.000       0.000\n",
            " omega                        0.400     474.068     189.627\n",
            " fa_dun                       0.700    5380.790    3766.553\n",
            " p_aa_pp                      0.600    -289.096    -173.458\n",
            " yhh_planarity                0.625       0.000       0.000\n",
            " ref                          1.000     372.765     372.765\n",
            " rama_prepro                  0.450     191.068      85.981\n",
            "---------------------------------------------------\n",
            " Total weighted score:                    13400.923\n",
            "protocols.antibody.design.AntibodyDesignMover: Randomizing starting CDRs\n",
            "protocols.antibody.design.AntibodyDesignMover: Grafting CDR from cluster L1-11-1 fragment 1i3gL_L1_0001\n",
            "protocols.grafting.CCDEndsGraftMover: Start: 891 End: 903 NterO: 3 CterO: 3\n",
            "protocols.grafting.util: Superimposing overhang residues\n",
            "core.chemical.AtomICoor: [ WARNING ] IcoorAtomID::atom_id(): Cannot get atom_id for POLYMER_LOWER of residue ILE 1.  Returning BOGUS ID instead.\n",
            "core.conformation.Residue: [ WARNING ] missing an atom: 1  H   that depends on a nonexistent polymer connection!\n",
            "core.conformation.Residue: [ WARNING ]  --> generating it using idealized coordinates.\n",
            "protocols.grafting.util: Deleting 11 residues from 892 to 902\n",
            "core.conformation.Conformation: Found disulfide between residues 771 845\n",
            "core.conformation.Conformation: Found disulfide between residues 891 945\n",
            "protocols.grafting.util: insert_point 891\n",
            "protocols.grafting.util: insert_start 892\n",
            "protocols.grafting.util: insert_end 902\n",
            "protocols.grafting.util: insert_length 11\n",
            "protocols.grafting.util: FOLD_TREE  EDGE 1 891 -1  EDGE 891 892 1  EDGE 892 964 -1\n",
            "protocols.grafting.GraftMoverBase: Insertion complete.\n",
            "protocols.grafting.AnchoredGraftMover: Setting default movemap\n",
            "protocols.grafting.CCDEndsGraftMover: Start: 891\n",
            "protocols.grafting.CCDEndsGraftMover: Original End: 903\n",
            "protocols.grafting.CCDEndsGraftMover: End: 903\n",
            "protocols.grafting.CCDEndsGraftMover: Insert Length: 11\n",
            "protocols.grafting.CCDEndsGraftMover: LOOP start: 890  stop: 893  cut: 891  size: 4  skip rate: 0  extended?: False\n",
            "protocols.grafting.CCDEndsGraftMover:\n",
            "protocols.grafting.CCDEndsGraftMover: LOOP start: 901  stop: 904  cut: 902  size: 4  skip rate: 0  extended?: False\n",
            "protocols.grafting.CCDEndsGraftMover:\n",
            "protocols.loops.FoldTreeFromLoopsWrapper: old foldtree FOLD_TREE  EDGE 1 891 -1  EDGE 891 903 1  EDGE 903 975 -1  EDGE 891 892 -2  C    N    EDGE 892 893 -2  C    N    EDGE 893 894 -2  C    N    EDGE 894 895 -2  C    N    EDGE 895 896 -2  C    N    EDGE 896 897 -2  C    N    EDGE 897 898 -2  C    N    EDGE 898 899 -2  C    N    EDGE 899 900 -2  C    N    EDGE 900 901 -2  C    N    EDGE 901 902 -2  C    N   \n",
            "New foldtree FOLD_TREE  EDGE 1 889 -1  EDGE 889 891 -1  EDGE 889 894 1  EDGE 894 892 -1  EDGE 894 900 -1  EDGE 900 902 -1  EDGE 900 905 2  EDGE 905 903 -1  EDGE 905 975 -1\n",
            "protocols.grafting.util: Loop: 1\n",
            "protocols.grafting.util: Add variant to: 891\n",
            "protocols.grafting.util: Loop: 2\n",
            "protocols.grafting.util: Add variant to: 902\n",
            "protocols.grafting.util: idealized  890\n",
            "protocols.grafting.util: idealized  891\n",
            "protocols.grafting.util: ideal 892\n",
            "protocols.grafting.util: ideal 902\n",
            "protocols.grafting.util: idealized 903\n",
            "protocols.grafting.util: idealized 904\n",
            "protocols.grafting.CCDEndsGraftMover: start 1752.63\n",
            "protocols.grafting.CCDEndsGraftMover: round 1\n",
            "protocols.loops.util: PepBondGeom: 902 - 42 L  Ca-C-N --  99.5 (min): 159.155 : 134.5 (max)\n",
            "protocols.grafting.CCDEndsGraftMover: 1 1360.61\n",
            "protocols.grafting.CCDEndsGraftMover: round 2\n",
            "protocols.loops.util: PepBondGeom: 902 - 42 L  Ca-C-N --  99.5 (min): 159.1 : 134.5 (max)\n",
            "protocols.grafting.CCDEndsGraftMover: round 3\n",
            "protocols.grafting.CCDEndsGraftMover: Graft Closed early - returning\n",
            "protocols.grafting.CCDEndsGraftMover: 3 1244.39\n",
            "protocols.grafting.CCDEndsGraftMover: finish 1244.39\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 531 rotamers at 47 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.grafting.CCDEndsGraftMover: Graft meets ideal geometry true\n",
            "protocols.grafting.CCDEndsGraftMover: Complete\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L1-11-1 data: 188 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/L1-11-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 22 constraints\n",
            "protocols.antibody.design.AntibodyDesignMover: Checking Geometry\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H1\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H2\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H3\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L1\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L2\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L3\n",
            "protocols.antibody.clusters.CDRClusterSet: Adding cacheable cluster data to pose\n",
            "protocols.antibody.design.util: Setting CDR sequence in NativeAntibodySeq\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: H1-13-1 data: 583 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/H1-13-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 26 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: H2-10-1 data: 328 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/H2-10-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 20 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: Adding general dihedral constraints for H3\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L1-11-1 data: 188 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/L1-11-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 22 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L2-8-1 data: 632 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/L2-8-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 16 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L3-9-cis7-1 data: 419 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/L3-9-cis7-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 18 constraints\n",
            "protocols.antibody.design.AntibodyDesignMover: Min Cycle complete\n",
            "protocols.antibody.design.AntibodyDesignMover: Attempts 1 L1\n",
            "protocols.antibody.design.AntibodyDesignMover: Grafting CDR from cluster L3-9-cis7-1 fragment 1i8kA_L3_0001\n",
            "protocols.grafting.CCDEndsGraftMover: Start: 956 End: 966 NterO: 3 CterO: 3\n",
            "protocols.grafting.util: Superimposing overhang residues\n",
            "core.chemical.AtomICoor: [ WARNING ] IcoorAtomID::atom_id(): Cannot get atom_id for POLYMER_LOWER of residue TYR 1.  Returning BOGUS ID instead.\n",
            "core.conformation.Residue: [ WARNING ] missing an atom: 1  H   that depends on a nonexistent polymer connection!\n",
            "core.conformation.Residue: [ WARNING ]  --> generating it using idealized coordinates.\n",
            "protocols.grafting.util: Deleting 9 residues from 957 to 965\n",
            "core.conformation.Conformation: Found disulfide between residues 771 845\n",
            "core.conformation.Conformation: Found disulfide between residues 891 956\n",
            "protocols.grafting.util: insert_point 956\n",
            "protocols.grafting.util: insert_start 957\n",
            "protocols.grafting.util: insert_end 965\n",
            "protocols.grafting.util: insert_length 9\n",
            "protocols.grafting.util: FOLD_TREE  EDGE 1 956 -1  EDGE 956 957 1  EDGE 957 966 -1\n",
            "protocols.grafting.GraftMoverBase: Insertion complete.\n",
            "protocols.grafting.AnchoredGraftMover: Setting default movemap\n",
            "protocols.grafting.CCDEndsGraftMover: Start: 956\n",
            "protocols.grafting.CCDEndsGraftMover: Original End: 966\n",
            "protocols.grafting.CCDEndsGraftMover: End: 966\n",
            "protocols.grafting.CCDEndsGraftMover: Insert Length: 9\n",
            "protocols.grafting.CCDEndsGraftMover: LOOP start: 955  stop: 958  cut: 956  size: 4  skip rate: 0  extended?: False\n",
            "protocols.grafting.CCDEndsGraftMover:\n",
            "protocols.grafting.CCDEndsGraftMover: LOOP start: 964  stop: 967  cut: 965  size: 4  skip rate: 0  extended?: False\n",
            "protocols.grafting.CCDEndsGraftMover:\n",
            "protocols.loops.FoldTreeFromLoopsWrapper: old foldtree FOLD_TREE  EDGE 1 956 -1  EDGE 956 966 1  EDGE 966 975 -1  EDGE 956 957 -2  C    N    EDGE 957 958 -2  C    N    EDGE 958 959 -2  C    N    EDGE 959 960 -2  C    N    EDGE 960 961 -2  C    N    EDGE 961 962 -2  C    N    EDGE 962 963 -2  C    N    EDGE 963 964 -2  C    N    EDGE 964 965 -2  C    N   \n",
            "New foldtree FOLD_TREE  EDGE 1 954 -1  EDGE 954 956 -1  EDGE 954 959 1  EDGE 959 957 -1  EDGE 959 963 -1  EDGE 963 965 -1  EDGE 963 968 2  EDGE 968 966 -1  EDGE 968 975 -1\n",
            "protocols.grafting.util: Loop: 1\n",
            "protocols.grafting.util: Add variant to: 956\n",
            "protocols.grafting.util: Loop: 2\n",
            "protocols.grafting.util: Add variant to: 965\n",
            "protocols.grafting.util: idealized  955\n",
            "protocols.grafting.util: idealized  956\n",
            "protocols.grafting.util: ideal 957\n",
            "protocols.grafting.util: ideal 965\n",
            "protocols.grafting.util: idealized 966\n",
            "protocols.grafting.util: idealized 967\n",
            "protocols.grafting.CCDEndsGraftMover: start 1592.56\n",
            "protocols.grafting.CCDEndsGraftMover: round 1\n",
            "protocols.grafting.CCDEndsGraftMover: Graft Closed early - returning\n",
            "protocols.grafting.CCDEndsGraftMover: 1 1238.28\n",
            "protocols.grafting.CCDEndsGraftMover: finish 1238.28\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 254 rotamers at 39 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.grafting.CCDEndsGraftMover: Graft meets ideal geometry true\n",
            "protocols.grafting.CCDEndsGraftMover: Complete\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L3-9-cis7-1 data: 419 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/L3-9-cis7-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 18 constraints\n",
            "protocols.antibody.design.AntibodyDesignMover: Checking Geometry\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H1\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H2\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H3\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L1\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L2\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L3\n",
            "protocols.antibody.clusters.CDRClusterSet: Adding cacheable cluster data to pose\n",
            "protocols.antibody.design.util: Setting CDR sequence in NativeAntibodySeq\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: H1-13-1 data: 583 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/H1-13-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 26 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: H2-10-1 data: 328 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/H2-10-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 20 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: Adding general dihedral constraints for H3\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L1-11-1 data: 188 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/L1-11-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 22 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L2-8-1 data: 632 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/L2-8-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 16 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L3-9-cis7-1 data: 419 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/L3-9-cis7-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 18 constraints\n",
            "protocols.antibody.design.AntibodyDesignMover: Min Cycle complete\n",
            "protocols.antibody.design.AntibodyDesignMover: Attempts 1 L3\n",
            "protocols.antibody.design.AntibodyDesignMover: Running basic monte carlo algorithm\n",
            "protocols.antibody.design.AntibodyDesignMover: Outer round: 1\n",
            "protocols.antibody.design.AntibodyDesignMover: Grafting CDR from cluster L3-8-cis6-1 fragment 1e6oL_L3_0001\n",
            "protocols.grafting.CCDEndsGraftMover: Start: 956 End: 966 NterO: 3 CterO: 3\n",
            "protocols.grafting.util: Superimposing overhang residues\n",
            "core.chemical.AtomICoor: [ WARNING ] IcoorAtomID::atom_id(): Cannot get atom_id for POLYMER_LOWER of residue TYR 1.  Returning BOGUS ID instead.\n",
            "core.conformation.Residue: [ WARNING ] missing an atom: 1  H   that depends on a nonexistent polymer connection!\n",
            "core.conformation.Residue: [ WARNING ]  --> generating it using idealized coordinates.\n",
            "protocols.grafting.util: Deleting 9 residues from 957 to 965\n",
            "core.conformation.Conformation: Found disulfide between residues 771 845\n",
            "core.conformation.Conformation: Reverting out-of-date disulfide to thiol type at resid 891\n",
            "core.conformation.Conformation: Reverting out-of-date disulfide to thiol type at resid 956\n",
            "protocols.grafting.util: insert_point 956\n",
            "protocols.grafting.util: insert_start 957\n",
            "protocols.grafting.util: insert_end 964\n",
            "protocols.grafting.util: insert_length 8\n",
            "protocols.grafting.util: FOLD_TREE  EDGE 1 956 -1  EDGE 956 957 1  EDGE 957 966 -1\n",
            "protocols.grafting.GraftMoverBase: Insertion complete.\n",
            "protocols.grafting.AnchoredGraftMover: Setting default movemap\n",
            "protocols.grafting.CCDEndsGraftMover: Start: 956\n",
            "protocols.grafting.CCDEndsGraftMover: Original End: 966\n",
            "protocols.grafting.CCDEndsGraftMover: End: 965\n",
            "protocols.grafting.CCDEndsGraftMover: Insert Length: 8\n",
            "protocols.grafting.CCDEndsGraftMover: LOOP start: 955  stop: 958  cut: 956  size: 4  skip rate: 0  extended?: False\n",
            "protocols.grafting.CCDEndsGraftMover:\n",
            "protocols.grafting.CCDEndsGraftMover: LOOP start: 963  stop: 966  cut: 964  size: 4  skip rate: 0  extended?: False\n",
            "protocols.grafting.CCDEndsGraftMover:\n",
            "protocols.loops.FoldTreeFromLoopsWrapper: old foldtree FOLD_TREE  EDGE 1 956 -1  EDGE 956 965 1  EDGE 965 974 -1  EDGE 956 957 -2  C    N    EDGE 957 958 -2  C    N    EDGE 958 959 -2  C    N    EDGE 959 960 -2  C    N    EDGE 960 961 -2  C    N    EDGE 961 962 -2  C    N    EDGE 962 963 -2  C    N    EDGE 963 964 -2  C    N   \n",
            "New foldtree FOLD_TREE  EDGE 1 954 -1  EDGE 954 956 -1  EDGE 954 959 1  EDGE 959 957 -1  EDGE 959 962 -1  EDGE 962 964 -1  EDGE 962 967 2  EDGE 967 965 -1  EDGE 967 974 -1\n",
            "protocols.grafting.util: Loop: 1\n",
            "protocols.grafting.util: Add variant to: 956\n",
            "protocols.grafting.util: Loop: 2\n",
            "protocols.grafting.util: Add variant to: 964\n",
            "protocols.grafting.util: idealized  955\n",
            "protocols.grafting.util: idealized  956\n",
            "protocols.grafting.util: ideal 957\n",
            "protocols.grafting.util: ideal 964\n",
            "protocols.grafting.util: idealized 965\n",
            "protocols.grafting.util: idealized 966\n",
            "protocols.grafting.CCDEndsGraftMover: start 1535.9\n",
            "protocols.grafting.CCDEndsGraftMover: round 1\n",
            "protocols.loops.util: PepBondGeom: 964 - 138 L  Ca-C-N --  99.5 (min): 143.332 : 134.5 (max)\n",
            "protocols.grafting.CCDEndsGraftMover: 1 1368.46\n",
            "protocols.grafting.CCDEndsGraftMover: round 2\n",
            "protocols.grafting.CCDEndsGraftMover: Graft Closed early - returning\n",
            "protocols.grafting.CCDEndsGraftMover: 2 1309.39\n",
            "protocols.grafting.CCDEndsGraftMover: finish 1309.39\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 186 rotamers at 39 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.grafting.CCDEndsGraftMover: Graft meets ideal geometry true\n",
            "protocols.grafting.CCDEndsGraftMover: Complete\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L3-8-cis6-1 data: 4 cutoff: 10\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: Adding general dihedral constraints for L3\n",
            "protocols.antibody.design.AntibodyDesignMover: Checking Geometry\n",
            "protocols.antibody.design.AntibodyDesignMover: DATA GRAFT_CLOSURE 1 rabd/my_ab.pdb L3 L3-8-cis6-1 1e6oL_L3_0001\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H1\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H2\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H3\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L1\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L2\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L3\n",
            "protocols.antibody.clusters.CDRClusterSet: Adding cacheable cluster data to pose\n",
            "protocols.antibody.design.util: Setting CDR sequence in NativeAntibodySeq\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: H1-13-1 data: 583 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/H1-13-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 26 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: H2-10-1 data: 328 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/H2-10-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 20 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: Adding general dihedral constraints for H3\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L1-11-1 data: 188 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/L1-11-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 22 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L2-8-1 data: 632 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/L2-8-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 16 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L3-8-cis6-1 data: 4 cutoff: 10\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: Adding general dihedral constraints for L3\n",
            "protocols.antibody.design.AntibodyDesignMover: Beginning MinProtocol\n",
            "protocols.antibody.design.AntibodyDesignMover: Inner round: 1\n",
            "protocols.antibody.design.AntibodyDesignMover: Add min neighbors : L1\n",
            "protocols.antibody.design.AntibodyDesignMover: Add min neighbors : H3\n",
            "basic.io.database: [ WARNING ] Unable to locate database file /sampling/antibodies/antibody_database_rosetta_design.db\n",
            "antibody.database.AntibodyDatabaseManager: The antibody design database is not specified by the user.\n",
            "antibody.database.AntibodyDatabaseManager: Reading from: /usr/local/lib/python3.10/dist-packages/pyrosetta/database//sampling/antibodies/antibody_database_rosetta_design_north_paper.db\n",
            "antibody.database.AntibodyDatabaseManager: Loaded L1-11-1 with 55 datapoints.\n",
            "antibody.database.AntibodyDatabaseManager: Skipped L3-8-cis6-1 with 3 data point\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Data pre-loaded.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: applying prob task op\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Using conservative op for L3\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Using original bb sequence for conservative design.\n",
            "protocols.task_operations.ConservativeDesignOperation: Loading conservative mutational data\n",
            "core.pack.pack_rotamers: built 1442 rotamers at 92 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating PDInteractionGraph\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs start: 13522\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs end: 9954.28\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 852\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 853\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 897\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 898\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 960\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 961\n",
            "antibody.design.GeneralAntibodyModeler: CDR min start: 9954.28\n",
            "antibody.design.GeneralAntibodyModeler: CDR min end: 5746.66\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: applying prob task op\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Using conservative op for L3\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Using original bb sequence for conservative design.\n",
            "core.pack.pack_rotamers: built 1761 rotamers at 95 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating PDInteractionGraph\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs start: 5746.66\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs end: 3814.7\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 852\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 853\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 897\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 898\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 960\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 961\n",
            "antibody.design.GeneralAntibodyModeler: CDR min start: 3814.7\n",
            "antibody.design.GeneralAntibodyModeler: CDR min end: 2726.07\n",
            "protocols.antibody.design.AntibodyDesignMover: Min Cycle complete\n",
            "protocols.antibody.design.AntibodyDesignMover:\n",
            "protocols.antibody.design.AntibodyDesignMover: SCORE 1 2726.07 1\n",
            "protocols.antibody.design.AntibodyDesignMover: FINAL 1 2726.07 1\n",
            "protocols.antibody.design.AntibodyDesignMover:\n",
            "protocols.antibody.design.AntibodyDesignMover: Last Accepted Score: 2726.07\n",
            "protocols.antibody.design.AntibodyDesignMover: Lowest        Score:2726.07\n",
            "protocols.antibody.design.AntibodyDesignMover:\n",
            "protocols.antibody.design.AntibodyDesignMover: Outer round: 2\n",
            "protocols.antibody.design.AntibodyDesignMover: Grafting CDR from cluster L3-11-1 fragment 3dvgA_L3_0001\n",
            "core.scoring.ScoreFunctionFactory: SCOREFUNCTION: ref2015\n",
            "core.scoring.ScoreFunctionFactory: SCOREFUNCTION: ref2015\n",
            "protocols.grafting.CCDEndsGraftMover: Start: 956 End: 965 NterO: 3 CterO: 3\n",
            "protocols.grafting.util: Superimposing overhang residues\n",
            "core.chemical.AtomICoor: [ WARNING ] IcoorAtomID::atom_id(): Cannot get atom_id for POLYMER_LOWER of residue TYR 1.  Returning BOGUS ID instead.\n",
            "core.conformation.Residue: [ WARNING ] missing an atom: 1  H   that depends on a nonexistent polymer connection!\n",
            "core.conformation.Residue: [ WARNING ]  --> generating it using idealized coordinates.\n",
            "protocols.grafting.util: Deleting 8 residues from 957 to 964\n",
            "core.conformation.Conformation: Found disulfide between residues 771 845\n",
            "protocols.grafting.util: insert_point 956\n",
            "protocols.grafting.util: insert_start 957\n",
            "protocols.grafting.util: insert_end 967\n",
            "protocols.grafting.util: insert_length 11\n",
            "protocols.grafting.util: FOLD_TREE  EDGE 1 956 -1  EDGE 956 957 1  EDGE 957 966 -1\n",
            "protocols.grafting.GraftMoverBase: Insertion complete.\n",
            "protocols.grafting.AnchoredGraftMover: Setting default movemap\n",
            "protocols.grafting.CCDEndsGraftMover: Start: 956\n",
            "protocols.grafting.CCDEndsGraftMover: Original End: 965\n",
            "protocols.grafting.CCDEndsGraftMover: End: 968\n",
            "protocols.grafting.CCDEndsGraftMover: Insert Length: 11\n",
            "protocols.grafting.CCDEndsGraftMover: LOOP start: 955  stop: 958  cut: 956  size: 4  skip rate: 0  extended?: False\n",
            "protocols.grafting.CCDEndsGraftMover:\n",
            "protocols.grafting.CCDEndsGraftMover: LOOP start: 966  stop: 969  cut: 967  size: 4  skip rate: 0  extended?: False\n",
            "protocols.grafting.CCDEndsGraftMover:\n",
            "protocols.loops.FoldTreeFromLoopsWrapper: old foldtree FOLD_TREE  EDGE 1 956 -1  EDGE 956 968 1  EDGE 968 977 -1  EDGE 956 957 -2  C    N    EDGE 957 958 -2  C    N    EDGE 958 959 -2  C    N    EDGE 959 960 -2  C    N    EDGE 960 961 -2  C    N    EDGE 961 962 -2  C    N    EDGE 962 963 -2  C    N    EDGE 963 964 -2  C    N    EDGE 964 965 -2  C    N    EDGE 965 966 -2  C    N    EDGE 966 967 -2  C    N   \n",
            "New foldtree FOLD_TREE  EDGE 1 954 -1  EDGE 954 956 -1  EDGE 954 959 1  EDGE 959 957 -1  EDGE 959 965 -1  EDGE 965 967 -1  EDGE 965 970 2  EDGE 970 968 -1  EDGE 970 977 -1\n",
            "protocols.grafting.util: Loop: 1\n",
            "protocols.grafting.util: Add variant to: 956\n",
            "protocols.grafting.util: Loop: 2\n",
            "protocols.grafting.util: Add variant to: 967\n",
            "protocols.grafting.util: idealized  955\n",
            "protocols.grafting.util: idealized  956\n",
            "protocols.grafting.util: ideal 957\n",
            "protocols.grafting.util: ideal 967\n",
            "protocols.grafting.util: idealized 968\n",
            "protocols.grafting.util: idealized 969\n",
            "protocols.grafting.CCDEndsGraftMover: start 1677.1\n",
            "protocols.grafting.CCDEndsGraftMover: round 1\n",
            "protocols.grafting.CCDEndsGraftMover: Graft Closed early - returning\n",
            "protocols.grafting.CCDEndsGraftMover: 1 1435\n",
            "protocols.grafting.CCDEndsGraftMover: finish 1435\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 322 rotamers at 39 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.grafting.CCDEndsGraftMover: Graft meets ideal geometry true\n",
            "protocols.grafting.CCDEndsGraftMover: Complete\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L3-11-1 data: 34 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/L3-11-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 22 constraints\n",
            "protocols.antibody.design.AntibodyDesignMover: Checking Geometry\n",
            "protocols.antibody.design.AntibodyDesignMover: DATA GRAFT_CLOSURE 1 rabd/my_ab.pdb L3 L3-11-1 3dvgA_L3_0001\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H1\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H2\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H3\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L1\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L2\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L3\n",
            "protocols.antibody.clusters.CDRClusterSet: Adding cacheable cluster data to pose\n",
            "protocols.antibody.design.util: Setting CDR sequence in NativeAntibodySeq\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: H1-13-1 data: 583 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/H1-13-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 26 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: H2-10-1 data: 328 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/H2-10-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 20 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: Adding general dihedral constraints for H3\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L1-11-1 data: 188 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/L1-11-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 22 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L2-8-1 data: 632 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/L2-8-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 16 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L3-11-1 data: 34 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/L3-11-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 22 constraints\n",
            "protocols.antibody.design.AntibodyDesignMover: Beginning MinProtocol\n",
            "protocols.antibody.design.AntibodyDesignMover: Inner round: 1\n",
            "protocols.antibody.design.AntibodyDesignMover: Add min neighbors : L1\n",
            "protocols.antibody.design.AntibodyDesignMover: Add min neighbors : H3\n",
            "basic.io.database: [ WARNING ] Unable to locate database file /sampling/antibodies/antibody_database_rosetta_design.db\n",
            "antibody.database.AntibodyDatabaseManager: The antibody design database is not specified by the user.\n",
            "antibody.database.AntibodyDatabaseManager: Reading from: /usr/local/lib/python3.10/dist-packages/pyrosetta/database//sampling/antibodies/antibody_database_rosetta_design_north_paper.db\n",
            "antibody.database.AntibodyDatabaseManager: Loaded L1-11-1 with 55 datapoints.\n",
            "antibody.database.AntibodyDatabaseManager: Skipped L3-11-1 with 9 data point\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Data pre-loaded.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: applying prob task op\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Using conservative op for L3\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Using original bb sequence for conservative design.\n",
            "protocols.task_operations.ConservativeDesignOperation: Loading conservative mutational data\n",
            "core.pack.pack_rotamers: built 1758 rotamers at 105 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating PDInteractionGraph\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs start: 16728.7\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs end: 10476.6\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 852\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 853\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 897\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 898\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 962\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 963\n",
            "antibody.design.GeneralAntibodyModeler: CDR min start: 10476.6\n",
            "antibody.design.GeneralAntibodyModeler: CDR min end: 5040.94\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: applying prob task op\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Using conservative op for L3\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Using original bb sequence for conservative design.\n",
            "core.pack.pack_rotamers: built 1628 rotamers at 97 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating PDInteractionGraph\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs start: 5040.94\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs end: 4809.35\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 852\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 853\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 897\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 898\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 962\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 963\n",
            "antibody.design.GeneralAntibodyModeler: CDR min start: 4809.35\n",
            "antibody.design.GeneralAntibodyModeler: CDR min end: 2649.69\n",
            "protocols.antibody.design.AntibodyDesignMover: Min Cycle complete\n",
            "protocols.antibody.design.AntibodyDesignMover:\n",
            "protocols.antibody.design.AntibodyDesignMover: SCORE 2 2649.69 1\n",
            "protocols.antibody.design.AntibodyDesignMover: FINAL 2 2649.69 1\n",
            "protocols.antibody.design.AntibodyDesignMover:\n",
            "protocols.antibody.design.AntibodyDesignMover: Last Accepted Score: 2649.69\n",
            "protocols.antibody.design.AntibodyDesignMover: Lowest        Score:2649.69\n",
            "protocols.antibody.design.AntibodyDesignMover:\n",
            "protocols.antibody.design.AntibodyDesignMover: Outer round: 3\n",
            "protocols.antibody.design.AntibodyDesignMover: Grafting CDR from cluster L1-17-1 fragment 1a3rL_L1_0001\n",
            "protocols.grafting.CCDEndsGraftMover: Start: 891 End: 903 NterO: 3 CterO: 3\n",
            "protocols.grafting.util: Superimposing overhang residues\n",
            "core.chemical.AtomICoor: [ WARNING ] IcoorAtomID::atom_id(): Cannot get atom_id for POLYMER_LOWER of residue MET 1.  Returning BOGUS ID instead.\n",
            "core.conformation.Residue: [ WARNING ] missing an atom: 1  H   that depends on a nonexistent polymer connection!\n",
            "core.conformation.Residue: [ WARNING ]  --> generating it using idealized coordinates.\n",
            "protocols.grafting.util: Deleting 11 residues from 892 to 902\n",
            "core.conformation.Conformation: Found disulfide between residues 771 845\n",
            "protocols.grafting.util: insert_point 891\n",
            "protocols.grafting.util: insert_start 892\n",
            "protocols.grafting.util: insert_end 908\n",
            "protocols.grafting.util: insert_length 17\n",
            "protocols.grafting.util: FOLD_TREE  EDGE 1 891 -1  EDGE 891 892 1  EDGE 892 966 -1\n",
            "protocols.grafting.GraftMoverBase: Insertion complete.\n",
            "protocols.grafting.AnchoredGraftMover: Setting default movemap\n",
            "protocols.grafting.CCDEndsGraftMover: Start: 891\n",
            "protocols.grafting.CCDEndsGraftMover: Original End: 903\n",
            "protocols.grafting.CCDEndsGraftMover: End: 909\n",
            "protocols.grafting.CCDEndsGraftMover: Insert Length: 17\n",
            "protocols.grafting.CCDEndsGraftMover: LOOP start: 890  stop: 893  cut: 891  size: 4  skip rate: 0  extended?: False\n",
            "protocols.grafting.CCDEndsGraftMover:\n",
            "protocols.grafting.CCDEndsGraftMover: LOOP start: 907  stop: 910  cut: 908  size: 4  skip rate: 0  extended?: False\n",
            "protocols.grafting.CCDEndsGraftMover:\n",
            "protocols.loops.FoldTreeFromLoopsWrapper: old foldtree FOLD_TREE  EDGE 1 891 -1  EDGE 891 909 1  EDGE 909 983 -1  EDGE 891 892 -2  C    N    EDGE 892 893 -2  C    N    EDGE 893 894 -2  C    N    EDGE 894 895 -2  C    N    EDGE 895 896 -2  C    N    EDGE 896 897 -2  C    N    EDGE 897 898 -2  C    N    EDGE 898 899 -2  C    N    EDGE 899 900 -2  C    N    EDGE 900 901 -2  C    N    EDGE 901 902 -2  C    N    EDGE 902 903 -2  C    N    EDGE 903 904 -2  C    N    EDGE 904 905 -2  C    N    EDGE 905 906 -2  C    N    EDGE 906 907 -2  C    N    EDGE 907 908 -2  C    N   \n",
            "New foldtree FOLD_TREE  EDGE 1 889 -1  EDGE 889 891 -1  EDGE 889 894 1  EDGE 894 892 -1  EDGE 894 906 -1  EDGE 906 908 -1  EDGE 906 911 2  EDGE 911 909 -1  EDGE 911 983 -1\n",
            "protocols.grafting.util: Loop: 1\n",
            "protocols.grafting.util: Add variant to: 891\n",
            "protocols.grafting.util: Loop: 2\n",
            "protocols.grafting.util: Add variant to: 908\n",
            "protocols.grafting.util: idealized  890\n",
            "protocols.grafting.util: idealized  891\n",
            "protocols.grafting.util: ideal 892\n",
            "protocols.grafting.util: ideal 908\n",
            "protocols.grafting.util: idealized 909\n",
            "protocols.grafting.util: idealized 910\n",
            "protocols.grafting.CCDEndsGraftMover: start 1876.67\n",
            "protocols.grafting.CCDEndsGraftMover: round 1\n",
            "protocols.loops.util: PepBondGeom: 908 - 42 L  Ca-C-N --  99.5 (min): 161.86 : 134.5 (max)\n",
            "protocols.grafting.CCDEndsGraftMover: 1 1639.16\n",
            "protocols.grafting.CCDEndsGraftMover: round 2\n",
            "protocols.loops.util: PepBondGeom: 908 - 42 L  Ca-C-N --  99.5 (min): 157.87 : 134.5 (max)\n",
            "protocols.grafting.CCDEndsGraftMover: 2 1636.75\n",
            "protocols.grafting.CCDEndsGraftMover: round 3\n",
            "protocols.loops.util: PepBondGeom: 908 - 42 L  Ca-C-N --  99.5 (min): 152.315 : 134.5 (max)\n",
            "protocols.grafting.CCDEndsGraftMover: round 4\n",
            "protocols.loops.util: PepBondGeom: 42 L  C-N --  1.76828 max: 1.5\n",
            "protocols.grafting.CCDEndsGraftMover: round 5\n",
            "protocols.loops.util: PepBondGeom: 908 - 42 L  Ca-C-N --  99.5 (min): 148.076 : 134.5 (max)\n",
            "protocols.grafting.CCDEndsGraftMover: 5 1637.57\n",
            "protocols.grafting.CCDEndsGraftMover: round 6\n",
            "protocols.grafting.CCDEndsGraftMover: Graft Closed early - returning\n",
            "protocols.grafting.CCDEndsGraftMover: 6 1530.03\n",
            "protocols.grafting.CCDEndsGraftMover: finish 1530.03\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 688 rotamers at 55 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.grafting.CCDEndsGraftMover: Graft meets ideal geometry true\n",
            "protocols.grafting.CCDEndsGraftMover: Complete\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L1-17-1 data: 49 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/L1-17-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 34 constraints\n",
            "protocols.antibody.design.AntibodyDesignMover: Checking Geometry\n",
            "protocols.antibody.design.AntibodyDesignMover: DATA GRAFT_CLOSURE 1 rabd/my_ab.pdb L1 L1-17-1 1a3rL_L1_0001\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H1\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H2\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H3\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L1\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L2\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L3\n",
            "protocols.antibody.clusters.CDRClusterSet: Adding cacheable cluster data to pose\n",
            "protocols.antibody.design.util: Setting CDR sequence in NativeAntibodySeq\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: H1-13-1 data: 583 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/H1-13-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 26 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: H2-10-1 data: 328 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/H2-10-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 20 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: Adding general dihedral constraints for H3\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L1-17-1 data: 49 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/L1-17-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 34 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L2-8-1 data: 632 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/L2-8-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 16 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L3-11-1 data: 34 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/L3-11-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 22 constraints\n",
            "protocols.antibody.design.AntibodyDesignMover: Beginning MinProtocol\n",
            "protocols.antibody.design.AntibodyDesignMover: Inner round: 1\n",
            "protocols.antibody.design.AntibodyDesignMover: Add min neighbors : L2\n",
            "protocols.antibody.design.AntibodyDesignMover: Add min neighbors : L3\n",
            "basic.io.database: [ WARNING ] Unable to locate database file /sampling/antibodies/antibody_database_rosetta_design.db\n",
            "antibody.database.AntibodyDatabaseManager: The antibody design database is not specified by the user.\n",
            "antibody.database.AntibodyDatabaseManager: Reading from: /usr/local/lib/python3.10/dist-packages/pyrosetta/database//sampling/antibodies/antibody_database_rosetta_design_north_paper.db\n",
            "antibody.database.AntibodyDatabaseManager: Loaded L1-17-1 with 16 datapoints.\n",
            "antibody.database.AntibodyDatabaseManager: Skipped L3-11-1 with 9 data point\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Data pre-loaded.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: applying prob task op\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Using conservative op for L3\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Using original bb sequence for conservative design.\n",
            "protocols.task_operations.ConservativeDesignOperation: Loading conservative mutational data\n",
            "core.pack.pack_rotamers: built 1805 rotamers at 97 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating PDInteractionGraph\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs start: 24939.1\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs end: 20679.2\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 900\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 901\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 926\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 927\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 968\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 969\n",
            "antibody.design.GeneralAntibodyModeler: CDR min start: 20679.2\n",
            "antibody.design.GeneralAntibodyModeler: CDR min end: 4620.27\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: applying prob task op\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Using conservative op for L3\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Using original bb sequence for conservative design.\n",
            "core.pack.pack_rotamers: built 2217 rotamers at 100 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating PDInteractionGraph\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs start: 4620.27\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs end: 8469.1\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 900\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 901\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 926\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 927\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 968\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 969\n",
            "antibody.design.GeneralAntibodyModeler: CDR min start: 4620.27\n",
            "antibody.design.GeneralAntibodyModeler: CDR min end: 3626.95\n",
            "protocols.antibody.design.AntibodyDesignMover: Min Cycle complete\n",
            "protocols.antibody.design.AntibodyDesignMover:\n",
            "protocols.antibody.design.AntibodyDesignMover: SCORE 3 3626.95 0\n",
            "protocols.antibody.design.AntibodyDesignMover: FINAL 3 2649.69 1\n",
            "protocols.antibody.design.AntibodyDesignMover:\n",
            "protocols.antibody.design.AntibodyDesignMover: Last Accepted Score: 2649.69\n",
            "protocols.antibody.design.AntibodyDesignMover: Lowest        Score:2649.69\n",
            "protocols.antibody.design.AntibodyDesignMover:\n",
            "protocols.antibody.design.AntibodyDesignMover: Outer round: 4\n",
            "protocols.antibody.design.AntibodyDesignMover: Grafting CDR from cluster L3-9-cis7-1 fragment 2v17L_L3_0001\n",
            "core.scoring.ScoreFunctionFactory: SCOREFUNCTION: ref2015\n",
            "core.scoring.ScoreFunctionFactory: SCOREFUNCTION: ref2015\n",
            "protocols.grafting.CCDEndsGraftMover: Start: 956 End: 968 NterO: 3 CterO: 3\n",
            "protocols.grafting.util: Superimposing overhang residues\n",
            "core.chemical.AtomICoor: [ WARNING ] IcoorAtomID::atom_id(): Cannot get atom_id for POLYMER_LOWER of residue TYR 1.  Returning BOGUS ID instead.\n",
            "core.conformation.Residue: [ WARNING ] missing an atom: 1  H   that depends on a nonexistent polymer connection!\n",
            "core.conformation.Residue: [ WARNING ]  --> generating it using idealized coordinates.\n",
            "protocols.grafting.util: Deleting 11 residues from 957 to 967\n",
            "core.conformation.Conformation: Found disulfide between residues 771 845\n",
            "protocols.grafting.util: insert_point 956\n",
            "protocols.grafting.util: insert_start 957\n",
            "protocols.grafting.util: insert_end 965\n",
            "protocols.grafting.util: insert_length 9\n",
            "protocols.grafting.util: FOLD_TREE  EDGE 1 956 -1  EDGE 956 957 1  EDGE 957 966 -1\n",
            "protocols.grafting.GraftMoverBase: Insertion complete.\n",
            "protocols.grafting.AnchoredGraftMover: Setting default movemap\n",
            "protocols.grafting.CCDEndsGraftMover: Start: 956\n",
            "protocols.grafting.CCDEndsGraftMover: Original End: 968\n",
            "protocols.grafting.CCDEndsGraftMover: End: 966\n",
            "protocols.grafting.CCDEndsGraftMover: Insert Length: 9\n",
            "protocols.grafting.CCDEndsGraftMover: LOOP start: 955  stop: 958  cut: 956  size: 4  skip rate: 0  extended?: False\n",
            "protocols.grafting.CCDEndsGraftMover:\n",
            "protocols.grafting.CCDEndsGraftMover: LOOP start: 964  stop: 967  cut: 965  size: 4  skip rate: 0  extended?: False\n",
            "protocols.grafting.CCDEndsGraftMover:\n",
            "protocols.loops.FoldTreeFromLoopsWrapper: old foldtree FOLD_TREE  EDGE 1 956 -1  EDGE 956 966 1  EDGE 966 975 -1  EDGE 956 957 -2  C    N    EDGE 957 958 -2  C    N    EDGE 958 959 -2  C    N    EDGE 959 960 -2  C    N    EDGE 960 961 -2  C    N    EDGE 961 962 -2  C    N    EDGE 962 963 -2  C    N    EDGE 963 964 -2  C    N    EDGE 964 965 -2  C    N   \n",
            "New foldtree FOLD_TREE  EDGE 1 954 -1  EDGE 954 956 -1  EDGE 954 959 1  EDGE 959 957 -1  EDGE 959 963 -1  EDGE 963 965 -1  EDGE 963 968 2  EDGE 968 966 -1  EDGE 968 975 -1\n",
            "protocols.grafting.util: Loop: 1\n",
            "protocols.grafting.util: Add variant to: 956\n",
            "protocols.grafting.util: Loop: 2\n",
            "protocols.grafting.util: Add variant to: 965\n",
            "protocols.grafting.util: idealized  955\n",
            "protocols.grafting.util: idealized  956\n",
            "protocols.grafting.util: ideal 957\n",
            "protocols.grafting.util: ideal 965\n",
            "protocols.grafting.util: idealized 966\n",
            "protocols.grafting.util: idealized 967\n",
            "protocols.grafting.CCDEndsGraftMover: start 2011.6\n",
            "protocols.grafting.CCDEndsGraftMover: round 1\n",
            "protocols.grafting.CCDEndsGraftMover: Graft Closed early - returning\n",
            "protocols.grafting.CCDEndsGraftMover: 1 1414.37\n",
            "protocols.grafting.CCDEndsGraftMover: finish 1414.37\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 228 rotamers at 36 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.grafting.CCDEndsGraftMover: Graft meets ideal geometry true\n",
            "protocols.grafting.CCDEndsGraftMover: Complete\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L3-9-cis7-1 data: 419 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/L3-9-cis7-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 18 constraints\n",
            "protocols.antibody.design.AntibodyDesignMover: Checking Geometry\n",
            "protocols.antibody.design.AntibodyDesignMover: DATA GRAFT_CLOSURE 1 rabd/my_ab.pdb L3 L3-9-cis7-1 2v17L_L3_0001\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H1\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H2\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H3\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L1\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L2\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L3\n",
            "protocols.antibody.clusters.CDRClusterSet: Adding cacheable cluster data to pose\n",
            "protocols.antibody.design.util: Setting CDR sequence in NativeAntibodySeq\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: H1-13-1 data: 583 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/H1-13-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 26 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: H2-10-1 data: 328 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/H2-10-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 20 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: Adding general dihedral constraints for H3\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L1-11-1 data: 188 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/L1-11-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 22 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L2-8-1 data: 632 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/L2-8-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 16 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L3-9-cis7-1 data: 419 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/L3-9-cis7-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 18 constraints\n",
            "protocols.antibody.design.AntibodyDesignMover: Beginning MinProtocol\n",
            "protocols.antibody.design.AntibodyDesignMover: Inner round: 1\n",
            "protocols.antibody.design.AntibodyDesignMover: Add min neighbors : L1\n",
            "protocols.antibody.design.AntibodyDesignMover: Add min neighbors : H3\n",
            "basic.io.database: [ WARNING ] Unable to locate database file /sampling/antibodies/antibody_database_rosetta_design.db\n",
            "antibody.database.AntibodyDatabaseManager: The antibody design database is not specified by the user.\n",
            "antibody.database.AntibodyDatabaseManager: Reading from: /usr/local/lib/python3.10/dist-packages/pyrosetta/database//sampling/antibodies/antibody_database_rosetta_design_north_paper.db\n",
            "antibody.database.AntibodyDatabaseManager: Loaded L1-11-1 with 55 datapoints.\n",
            "antibody.database.AntibodyDatabaseManager: Loaded L3-9-cis7-1 with 179 datapoints.\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Data pre-loaded.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: applying prob task op\n",
            "core.pack.pack_rotamers: built 1119 rotamers at 96 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating PDInteractionGraph\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs start: 7249.9\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs end: 8759.18\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 852\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 853\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 897\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 898\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 961\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 962\n",
            "antibody.design.GeneralAntibodyModeler: CDR min start: 7249.9\n",
            "antibody.design.GeneralAntibodyModeler: CDR min end: 2454.12\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: applying prob task op\n",
            "core.pack.pack_rotamers: built 1049 rotamers at 96 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating PDInteractionGraph\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs start: 2454.12\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs end: 4522.63\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 852\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 853\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 897\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 898\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 961\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 962\n",
            "antibody.design.GeneralAntibodyModeler: CDR min start: 2454.12\n",
            "antibody.design.GeneralAntibodyModeler: CDR min end: 2454.03\n",
            "protocols.antibody.design.AntibodyDesignMover: Min Cycle complete\n",
            "protocols.antibody.design.AntibodyDesignMover:\n",
            "protocols.antibody.design.AntibodyDesignMover: SCORE 4 2454.03 1\n",
            "protocols.antibody.design.AntibodyDesignMover: FINAL 4 2454.03 1\n",
            "protocols.antibody.design.AntibodyDesignMover:\n",
            "protocols.antibody.design.AntibodyDesignMover: Last Accepted Score: 2454.03\n",
            "protocols.antibody.design.AntibodyDesignMover: Lowest        Score:2454.03\n",
            "protocols.antibody.design.AntibodyDesignMover:\n",
            "protocols.antibody.design.AntibodyDesignMover: Outer round: 5\n",
            "protocols.antibody.design.AntibodyDesignMover: Grafting CDR from cluster L3-8-1 fragment 1c5dA_L3_0001\n",
            "protocols.grafting.CCDEndsGraftMover: Start: 956 End: 966 NterO: 3 CterO: 3\n",
            "protocols.grafting.util: Superimposing overhang residues\n",
            "core.chemical.AtomICoor: [ WARNING ] IcoorAtomID::atom_id(): Cannot get atom_id for POLYMER_LOWER of residue TYR 1.  Returning BOGUS ID instead.\n",
            "core.conformation.Residue: [ WARNING ] missing an atom: 1  H   that depends on a nonexistent polymer connection!\n",
            "core.conformation.Residue: [ WARNING ]  --> generating it using idealized coordinates.\n",
            "protocols.grafting.util: Deleting 9 residues from 957 to 965\n",
            "core.conformation.Conformation: Found disulfide between residues 771 845\n",
            "protocols.grafting.util: insert_point 956\n",
            "protocols.grafting.util: insert_start 957\n",
            "protocols.grafting.util: insert_end 964\n",
            "protocols.grafting.util: insert_length 8\n",
            "protocols.grafting.util: FOLD_TREE  EDGE 1 956 -1  EDGE 956 957 1  EDGE 957 966 -1\n",
            "protocols.grafting.GraftMoverBase: Insertion complete.\n",
            "protocols.grafting.AnchoredGraftMover: Setting default movemap\n",
            "protocols.grafting.CCDEndsGraftMover: Start: 956\n",
            "protocols.grafting.CCDEndsGraftMover: Original End: 966\n",
            "protocols.grafting.CCDEndsGraftMover: End: 965\n",
            "protocols.grafting.CCDEndsGraftMover: Insert Length: 8\n",
            "protocols.grafting.CCDEndsGraftMover: LOOP start: 955  stop: 958  cut: 956  size: 4  skip rate: 0  extended?: False\n",
            "protocols.grafting.CCDEndsGraftMover:\n",
            "protocols.grafting.CCDEndsGraftMover: LOOP start: 963  stop: 966  cut: 964  size: 4  skip rate: 0  extended?: False\n",
            "protocols.grafting.CCDEndsGraftMover:\n",
            "protocols.loops.FoldTreeFromLoopsWrapper: old foldtree FOLD_TREE  EDGE 1 956 -1  EDGE 956 965 1  EDGE 965 974 -1  EDGE 956 957 -2  C    N    EDGE 957 958 -2  C    N    EDGE 958 959 -2  C    N    EDGE 959 960 -2  C    N    EDGE 960 961 -2  C    N    EDGE 961 962 -2  C    N    EDGE 962 963 -2  C    N    EDGE 963 964 -2  C    N   \n",
            "New foldtree FOLD_TREE  EDGE 1 954 -1  EDGE 954 956 -1  EDGE 954 959 1  EDGE 959 957 -1  EDGE 959 962 -1  EDGE 962 964 -1  EDGE 962 967 2  EDGE 967 965 -1  EDGE 967 974 -1\n",
            "protocols.grafting.util: Loop: 1\n",
            "protocols.grafting.util: Add variant to: 956\n",
            "protocols.grafting.util: Loop: 2\n",
            "protocols.grafting.util: Add variant to: 964\n",
            "protocols.grafting.util: idealized  955\n",
            "protocols.grafting.util: idealized  956\n",
            "protocols.grafting.util: ideal 957\n",
            "protocols.grafting.util: ideal 964\n",
            "protocols.grafting.util: idealized 965\n",
            "protocols.grafting.util: idealized 966\n",
            "protocols.grafting.CCDEndsGraftMover: start 1981.44\n",
            "protocols.grafting.CCDEndsGraftMover: round 1\n",
            "protocols.loops.util: PepBondGeom: 964 - 138 L  Ca-C-N --  99.5 (min): 135.329 : 134.5 (max)\n",
            "protocols.grafting.CCDEndsGraftMover: 1 1443.1\n",
            "protocols.grafting.CCDEndsGraftMover: round 2\n",
            "protocols.grafting.CCDEndsGraftMover: Graft Closed early - returning\n",
            "protocols.grafting.CCDEndsGraftMover: 2 1416.61\n",
            "protocols.grafting.CCDEndsGraftMover: finish 1416.61\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 221 rotamers at 38 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.grafting.CCDEndsGraftMover: Graft meets ideal geometry true\n",
            "protocols.grafting.CCDEndsGraftMover: Complete\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L3-8-1 data: 39 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/L3-8-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 16 constraints\n",
            "protocols.antibody.design.AntibodyDesignMover: Checking Geometry\n",
            "protocols.antibody.design.AntibodyDesignMover: DATA GRAFT_CLOSURE 1 rabd/my_ab.pdb L3 L3-8-1 1c5dA_L3_0001\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H1\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H2\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H3\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L1\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L2\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L3\n",
            "protocols.antibody.clusters.CDRClusterSet: Adding cacheable cluster data to pose\n",
            "protocols.antibody.design.util: Setting CDR sequence in NativeAntibodySeq\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: H1-13-1 data: 583 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/H1-13-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 26 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: H2-10-1 data: 328 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/H2-10-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 20 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: Adding general dihedral constraints for H3\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L1-11-1 data: 188 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/L1-11-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 22 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L2-8-1 data: 632 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/L2-8-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 16 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L3-8-1 data: 39 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/L3-8-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 16 constraints\n",
            "protocols.antibody.design.AntibodyDesignMover: Beginning MinProtocol\n",
            "protocols.antibody.design.AntibodyDesignMover: Inner round: 1\n",
            "protocols.antibody.design.AntibodyDesignMover: Add min neighbors : L1\n",
            "protocols.antibody.design.AntibodyDesignMover: Add min neighbors : H3\n",
            "basic.io.database: [ WARNING ] Unable to locate database file /sampling/antibodies/antibody_database_rosetta_design.db\n",
            "antibody.database.AntibodyDatabaseManager: The antibody design database is not specified by the user.\n",
            "antibody.database.AntibodyDatabaseManager: Reading from: /usr/local/lib/python3.10/dist-packages/pyrosetta/database//sampling/antibodies/antibody_database_rosetta_design_north_paper.db\n",
            "antibody.database.AntibodyDatabaseManager: Loaded L1-11-1 with 55 datapoints.\n",
            "antibody.database.AntibodyDatabaseManager: Loaded L3-8-1 with 14 datapoints.\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Data pre-loaded.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: applying prob task op\n",
            "core.pack.pack_rotamers: built 1248 rotamers at 90 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating PDInteractionGraph\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs start: 4069.2\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs end: 6193.29\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 852\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 853\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 897\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 898\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 960\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 961\n",
            "antibody.design.GeneralAntibodyModeler: CDR min start: 4069.2\n",
            "antibody.design.GeneralAntibodyModeler: CDR min end: 2434.08\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: applying prob task op\n",
            "core.pack.pack_rotamers: built 1087 rotamers at 92 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating PDInteractionGraph\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs start: 2434.08\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs end: 3608.62\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 852\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 853\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 897\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 898\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 960\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 961\n",
            "antibody.design.GeneralAntibodyModeler: CDR min start: 2434.08\n",
            "antibody.design.GeneralAntibodyModeler: CDR min end: 2431.31\n",
            "protocols.antibody.design.AntibodyDesignMover: Min Cycle complete\n",
            "protocols.antibody.design.AntibodyDesignMover:\n",
            "protocols.antibody.design.AntibodyDesignMover: SCORE 5 2431.31 1\n",
            "protocols.antibody.design.AntibodyDesignMover: FINAL 5 2431.31 1\n",
            "protocols.antibody.design.AntibodyDesignMover:\n",
            "protocols.antibody.design.AntibodyDesignMover: Last Accepted Score: 2431.31\n",
            "protocols.antibody.design.AntibodyDesignMover: Lowest        Score:2431.31\n",
            "protocols.antibody.design.AntibodyDesignMover:\n",
            "protocols.antibody.design.AntibodyDesignMover: Outer cycles attempted: 5 Outer cycles completed: 5\n",
            "protocols.antibody.design.AntibodyDesignMover:\n",
            "protocols.antibody.design.AntibodyDesignMover: Native Pose (MC): 12088.7\n",
            "protocols.antibody.design.AntibodyDesignMover: Final Pose (MC): 2431.31\n",
            "protocols.antibody.design.AntibodyDesignMover:\n",
            "protocols.moves.TrialCounter:              unk trials=      5;  accepts= 0.8000;  energy_drop/trial= -1931.47745\n",
            "protocols.antibody.design.AntibodyDesignMover:\n",
            "protocols.antibody.design.AntibodyDesignMover: Native Pose (Total): 13400.9\n",
            "protocols.antibody.design.AntibodyDesignMover: Benchmarked Pose (Total): 12088.7\n",
            "protocols.antibody.design.AntibodyDesignMover: Final Pose (Total): 2431.31\n",
            "protocols.antibody.design.AntibodyDesignMover: \n",
            "------------------------------------------------------------\n",
            " Scores                       Weight   Raw Score Wghtd.Score\n",
            "------------------------------------------------------------\n",
            " fa_atr                       1.000   -5608.694   -5608.694\n",
            " fa_rep                       0.550    5155.577    2835.567\n",
            " fa_sol                       1.000    3440.533    3440.533\n",
            " fa_intra_rep                 0.005    2762.807      13.814\n",
            " fa_intra_sol_xover4          1.000     177.515     177.515\n",
            " lk_ball_wtd                  1.000     -68.501     -68.501\n",
            " fa_elec                      1.000   -1596.067   -1596.067\n",
            " pro_close                    1.250       7.823       9.779\n",
            " hbond_sr_bb                  1.000     -74.256     -74.256\n",
            " hbond_lr_bb                  1.000    -510.160    -510.160\n",
            " hbond_bb_sc                  1.000    -139.425    -139.425\n",
            " hbond_sc                     1.000     -75.948     -75.948\n",
            " dslf_fa13                    1.250      -0.697      -0.871\n",
            " atom_pair_constraint         0.010       0.000       0.000\n",
            " omega                        0.400     505.704     202.282\n",
            " fa_dun                       0.700    5032.802    3522.961\n",
            " p_aa_pp                      0.600    -281.258    -168.755\n",
            " yhh_planarity                0.625       0.153       0.095\n",
            " ref                          1.000     379.135     379.135\n",
            " rama_prepro                  0.450     205.121      92.304\n",
            "---------------------------------------------------\n",
            " Total weighted score:                     2431.309\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H1\n",
            "protocols.antibody.cluster.CDRClusterMatcher: Length: 13 Omega: TTTTTTTTTTTTT\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H2\n",
            "protocols.antibody.cluster.CDRClusterMatcher: Length: 10 Omega: TTTTTTTTTT\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H3\n",
            "protocols.antibody.cluster.CDRClusterMatcher: Length: 13 Omega: TTTTTTTTTTTTT\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L1\n",
            "protocols.antibody.cluster.CDRClusterMatcher: Length: 11 Omega: TTTTTTTTTTT\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L2\n",
            "protocols.antibody.cluster.CDRClusterMatcher: Length: 8 Omega: TTTTTTTT\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L3\n",
            "protocols.antibody.cluster.CDRClusterMatcher: Length: 8 Omega: TTTTTTTT\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using explicit constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using interface constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: [ WARNING ] In InterfaceAnalyzerMover, calculator Sasa_1 already exists, this is hopefully correct for your purposes\n",
            "protocols.analysis.InterfaceAnalyzerMover: [ WARNING ] In InterfaceAnalyzerMover, calculator InterfaceNeighborDefinition_1 already exists, this is hopefully correct for your purposes\n",
            "protocols.analysis.InterfaceAnalyzerMover: [ WARNING ] In InterfaceAnalyzerMover, calculator InterfaceDeltaEnergetics_1 already exists, this is hopefully correct for your purposes\n",
            "protocols.analysis.InterfaceAnalyzerMover: [ WARNING ] In InterfaceAnalyzerMover, calculator NumberHBonds_1 already exists, this is hopefully correct for your purposes\n",
            "protocols.analysis.InterfaceAnalyzerMover: [ WARNING ] In InterfaceAnalyzerMover, calculator BuriedUnsatisfiedPolars_1 already exists, this is hopefully correct for your purposes\n",
            "protocols.analysis.InterfaceAnalyzerMover: Making an interface set with fixed chains\n",
            "protocols.analysis.InterfaceAnalyzerMover: Interface set residues total: 93\n",
            "protocols.analysis.InterfaceAnalyzerMover: NOT detecting disulfides in the separated pose.\n",
            "protocols.analysis.InterfaceAnalyzerMover: Calculating dSASA\n",
            "protocols.analysis.InterfaceAnalyzerMover: Calculating per-res dSASA data\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 1145 rotamers at 93 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.analysis.InterfaceAnalyzerMover: included_nres: 974\n",
            "protocols.analysis.InterfaceAnalyzerMover: Making hbond ScoreFunction with bb-bb energy method options\n",
            "core.scoring.ScoreFunctionFactory: SCOREFUNCTION: ref2015\n",
            "protocols.analysis.InterfaceAnalyzerMover: Found Hbond between chains: 13 and 9\n",
            "protocols.analysis.InterfaceAnalyzerMover: Found Hbond between chains: 9 and 13\n",
            "protocols.analysis.InterfaceAnalyzerMover: Found Hbond between chains: 12 and 9\n",
            "protocols.analysis.InterfaceAnalyzerMover: Found Hbond between chains: 9 and 13\n",
            "protocols.analysis.InterfaceAnalyzerMover: Found Hbond between chains: 13 and 10\n",
            "protocols.analysis.InterfaceAnalyzerMover: Computing delta unsat polar residues...\n",
            "protocols.analysis.InterfaceAnalyzerMover: Computing Shape Complementarity Score...\n",
            "protocols.analysis.InterfaceAnalyzerMover: Upstream chain(s) numbers: 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11,\n",
            "protocols.analysis.InterfaceAnalyzerMover: Downstream chain(s) numbers: 12, 13,\n"
          ]
        }
      ],
      "source": [
        "if not os.getenv(\"DEBUG\"):\n",
        "    ### BEGIN SOLUTION\n",
        "    pose = original_pose.clone()\n",
        "    rabd = XmlObjects.static_get_mover('<AntibodyDesignMover name=\"RAbD\" graft_design_cdrs=\"L1,L3\" seq_design_cdrs=\"L1,L3\" random_start=\"1\"/> light_chain=\"kappa\"')\n",
        "    rabd.set_light_chain(\"kappa\")\n",
        "    rabd.apply(pose)\n",
        "\n",
        "    # OR (REUSE code from above)\n",
        "    pose = original_pose.clone()\n",
        "    rabd2.set_seq_design_cdrs(cdrs)\n",
        "    rabd2.set_graft_design_cdrs(cdrs)\n",
        "    rabd2.set_random_start(True)\n",
        "    rabd2.set_light_chain(\"kappa\")\n",
        "    rabd2.apply(pose)\n",
        "\n",
        "    ### END SOLUTION"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "M3w-y4uA2LMz"
      },
      "source": [
        "Would starting from a random CDR help anywhere?  Perhaps if you want an entirely new cluster or length to break a patent or remove some off target effects?  We will use it below to start de novo design with docking."
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "d2BFgL_62LMz"
      },
      "source": [
        "### Tut A4. RAbD Framework Components\n",
        "\n",
        "This tutorial will give you some exprience with an antibody design protocol using the `RosettaAntibdyDesign` components.  We will take the light chain CDRs from a malaria antibody and graft them into our antibody.  In the tutorial we are interested in stabilizing the grafted CDRs in relation to the whole antibody, instead of interface design to an antigen.  \n",
        "\n",
        "We will graft the CDRs in, minimize the structure with CDR dihedral constraints (that use the CDR clusters) to not purturb the CDRs too much, and then design the framework around the CDRs while designing the CDRs and neighbors.  The result should be mutations that better accomodate our new CDRs.  This can be useful for humanizing potential antibodies or framework switching, where we want the binding properties of certain CDRs, but the stability or immunological profile of a different framework.  \n",
        "\n",
        "We are using an XML here for simplicity - all components are available in PyRosetta, but harder to setup.\n",
        "\n",
        "#### 1. Copy the Files\n",
        "\n",
        "        cp ../inputs/rabd/ab_design_components.xml .\n",
        "\t\tcp ../inputs/rabd/malaria_cdrs.pdb .\n",
        "\n",
        "\n",
        "Take a look at the xml.  \n",
        "\n",
        "- We are using the `AntibodyCDRGrafter` to do the grafting of our CDRs.  \n",
        "- We then add constraints using `CDRDihderalConstraintMovers` for each CDR with use the CDR clusters determinedy by RosettaAntibody to keep from perturbing the CDRs too much.  \n",
        "- Finally, we do a round of pack/min/pack using the `RestrictToCDRsAndNeighborsOperation` and the `CDRResidueSelector`.  This task operation controls what we pack and design.  It first limits packing and design to only the CDRs and its neighbors.  By specifying the `design_framework=1` option we allow the neighbor framework residues to design, while the CDRs and antigen neighbors will only repack.  If we wanted to disable antigen repacking, we would pass the _DisableAntibodyRegionOperation_ task operation.  Using this, we can specify any antibody region as `antibody_region`, `cdr_region`, or `antigen_region` and we can disable just design or both packing and design.  \n",
        "\n",
        "\n",
        "These task operations allow us to chisel exactly what we want to design in antibody, sans a residue-specific resfile (though we could combine these with one of them!).  All of these tools are available in-code.  If you've done the design workshop, you will know how to use them here. Checkout `rosetta.protocols.antibody.task_operations` for a list of them.\n",
        "\n",
        "Finally, we use the new SimpleMetric system to obtain our final sequence of the CDRs to compare to our native antibody as well as pymol selections of our CDRs - which you have been introduced to in the previous tutorial.\n",
        "\n",
        "\n",
        "##### PyRosetta Locations\n",
        "\n",
        "`rosetta.protocols.antibody.task_operations`\n",
        "\n",
        "`rosetta.protocols.antibody.constraints`\n",
        "\n",
        "`rosetta.protocols.antibody.residue_selectors`\n",
        "\n",
        "##### Documentation\n",
        "\n",
        "- <https://www.rosettacommons.org/docs/latest/scripting_documentation/RosettaScripts/SimpleMetrics/SimpleMetrics>\n",
        "\n",
        "- More Documentation is available here:\n",
        "\n",
        " - <https://www.rosettacommons.org/docs/latest/scripting_documentation/RosettaScripts/RosettaScripts>\n",
        "\n",
        " - <https://www.rosettacommons.org/docs/latest/scripting_documentation/RosettaScripts/TaskOperations/TaskOperations-RosettaScripts#antibody-and-cdr-specific-operations>\n",
        "\n",
        " - <https://www.rosettacommons.org/docs/latest/scripting_documentation/RosettaScripts/Movers/Movers-RosettaScripts#antibody-modeling-and-design-movers>\n",
        "\n",
        "#### 2. Run the protocol or copy the output (357 seconds).  \n",
        "\n",
        "\n",
        "#### 3. Look at the score file as you have before.  Are the sequences different between what we started with?  How about the interaction energies?"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 57,
      "metadata": {
        "id": "ZLZpaVtP2LMz",
        "outputId": "d0d72384-d8ad-43f5-a021-6e218b412734",
        "colab": {
          "base_uri": "https://localhost:8080/"
        }
      },
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "256"
            ]
          },
          "metadata": {},
          "execution_count": 57
        }
      ],
      "source": [
        "os.system('cp rabd/5jwa.pdb')"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 59,
      "metadata": {
        "id": "KOQLe-WY2LMz",
        "outputId": "00286d1f-1d9c-4138-c17b-2f8f1469e82b",
        "colab": {
          "base_uri": "https://localhost:8080/"
        }
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "PyRosetta-4 2023 [Rosetta PyRosetta4.MinSizeRel.python310.ubuntu 2024.01+release.00b79147e63be743438188f93a3f069ca75106d6 2023-12-25T16:35:48] retrieved from: http://www.pyrosetta.org\n",
            "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n",
            "core.init: Checking for fconfig files in pwd and ./rosetta/flags\n",
            "core.init: Rosetta version: PyRosetta4.MinSizeRel.python310.ubuntu r366 2024.01+release.00b79147e63 00b79147e63be743438188f93a3f069ca75106d6 http://www.pyrosetta.org 2023-12-25T16:35:48\n",
            "core.init: command: PyRosetta -ex1 -ex2aro -database /usr/local/lib/python3.10/dist-packages/pyrosetta/database\n",
            "basic.random.init_random_generator: 'RNG device' seed mode, using '/dev/urandom', seed=-1692218034 seed_offset=0 real_seed=-1692218034\n",
            "basic.random.init_random_generator: RandomGenerator:init: Normal mode, seed=-1692218034 RG_type=mt19937\n",
            "protocols.rosetta_scripts.RosettaScriptsParser: Validating input script...\n",
            "protocols.rosetta_scripts.RosettaScriptsParser: ...done\n",
            "protocols.rosetta_scripts.RosettaScriptsParser: Parsed script:\n",
            "<ROSETTASCRIPTS>\n",
            "\t<RESIDUE_SELECTORS>\n",
            "\t\t<CDR cdrs=\"L1,L2,L3\" name=\"light_cdrs\"/>\n",
            "\t\t<CDR cdrs=\"L1\" name=\"L1\"/>\n",
            "\t\t<CDR cdrs=\"L2\" name=\"L2\"/>\n",
            "\t\t<CDR cdrs=\"L3\" name=\"L3\"/>\n",
            "\t\t<AntibodyRegion name=\"antigen\" region=\"antigen_region\"/>\n",
            "\t\t<AntibodyRegion name=\"framework\" region=\"framework_region\"/>\n",
            "\t\t<AntibodyRegion name=\"cdrs\" region=\"cdr_region\"/>\n",
            "\t</RESIDUE_SELECTORS>\n",
            "\t<TASKOPERATIONS>\n",
            "\t\t<RestrictToCDRsAndNeighbors cdrs=\"L1,L2,L3\" design_cdrs=\"1\" design_framework=\"1\" name=\"restrict_to_cdrs\"/>\n",
            "\t\t<AddCDRProfilesOperation cdrs=\"L1,L2,L3\" fallback_strategy=\"CONSERVATIVE\" name=\"profiles\"/>\n",
            "\t</TASKOPERATIONS>\n",
            "\t<MOVE_MAP_FACTORIES>\n",
            "\t\t<MoveMapFactory bb=\"0\" chi=\"0\" name=\"movemap_cdrs\">\n",
            "\t\t\t<Backbone residue_selector=\"light_cdrs\"/>\n",
            "\t\t\t<Chi residue_selector=\"light_cdrs\"/>\n",
            "\t\t</MoveMapFactory>\n",
            "\t</MOVE_MAP_FACTORIES>\n",
            "\t<SIMPLE_METRICS>\n",
            "\t\t<SasaMetric name=\"sasa\" residue_selector=\"light_cdrs\"/>\n",
            "\t\t<SelectedResiduesPyMOLMetric name=\"cdr_selection\" residue_selector=\"light_cdrs\"/>\n",
            "\t\t<SequenceMetric custom_type=\"L1\" name=\"L1_seq\" residue_selector=\"L1\"/>\n",
            "\t\t<SequenceMetric custom_type=\"L2\" name=\"L2_seq\" residue_selector=\"L2\"/>\n",
            "\t\t<SequenceMetric custom_type=\"L3\" name=\"L3_seq\" residue_selector=\"L3\"/>\n",
            "\t\t<InteractionEnergyMetric name=\"cdr-int\" residue_selector=\"light_cdrs\" residue_selector2=\"antigen\"/>\n",
            "\t</SIMPLE_METRICS>\n",
            "\t<MOVERS>\n",
            "\t\t<CDRDihedralConstraintMover cdr=\"L1\" name=\"dih_mover_L1\" use_cluster_csts=\"1\"/>\n",
            "\t\t<CDRDihedralConstraintMover cdr=\"L2\" name=\"dih_mover_L2\" use_cluster_csts=\"1\"/>\n",
            "\t\t<CDRDihedralConstraintMover cdr=\"L3\" name=\"dih_mover_L3\" use_cluster_csts=\"1\"/>\n",
            "\t\t<AntibodyCDRGrafter cdr_definition=\"North\" cdrs=\"L1,L2,L3\" donor_structure_from_pdb=\"malaria_cdrs.pdb\" input_ab_scheme=\"AHO_Scheme\" name=\"grafter\" optimize_cdr4_if_neighbor=\"1\" optimize_cdrs=\"0\" use_secondary_graft_mover=\"1\"/>\n",
            "\t\t<PackRotamersMover name=\"packrot\" task_operations=\"restrict_to_cdrs,profiles\"/>\n",
            "\t\t<MinMover cartesian=\"1\" movemap_factory=\"movemap_cdrs\" name=\"minmover\" tolerance=\".1\"/>\n",
            "\t\t<RunSimpleMetrics metrics=\"sasa,cdr_selection,L1_seq,L2_seq,L3_seq,cdr-int\" name=\"design_metrics\" prefix=\"design_\"/>\n",
            "\t\t<RunSimpleMetrics metrics=\"sasa,cdr_selection,L1_seq,L2_seq,L3_seq,cdr-int\" name=\"native_metrics\" prefix=\"native_\"/>\n",
            "\t</MOVERS>\n",
            "\t<PROTOCOLS>\n",
            "\t\t<Add mover_name=\"native_metrics\"/>\n",
            "\t\t<Add mover_name=\"grafter\"/>\n",
            "\t\t<Add mover_name=\"dih_mover_L1\"/>\n",
            "\t\t<Add mover_name=\"dih_mover_L2\"/>\n",
            "\t\t<Add mover_name=\"dih_mover_L3\"/>\n",
            "\t\t<Add mover_name=\"packrot\"/>\n",
            "\t\t<Add mover_name=\"minmover\"/>\n",
            "\t\t<Add mover_name=\"packrot\"/>\n",
            "\t\t<Add mover_name=\"minmover\"/>\n",
            "\t\t<Add mover_name=\"design_metrics\"/>\n",
            "\t</PROTOCOLS>\n",
            "</ROSETTASCRIPTS>\n",
            "core.scoring.ScoreFunctionFactory: SCOREFUNCTION: ref2015\n",
            "protocols.antibody.residue_selector.CDRResidueSelector: Setting CDRs from settings\n",
            "protocols.antibody.residue_selector.CDRResidueSelector: Setting CDRs from settings\n",
            "protocols.antibody.residue_selector.CDRResidueSelector: Setting CDRs from settings\n",
            "protocols.antibody.residue_selector.CDRResidueSelector: Setting CDRs from settings\n",
            "protocols.jd2.parser.TaskOperationLoader: Defined TaskOperation named \"restrict_to_cdrs\" of type RestrictToCDRsAndNeighbors\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Setting CDRs from settings\n",
            "protocols.jd2.parser.TaskOperationLoader: Defined TaskOperation named \"profiles\" of type AddCDRProfilesOperation\n",
            "protocols.jd2.parser.MoveMapFactoryLoader: Defined MoveMap named \"movemap_cdrs\"\n",
            "core.scoring.ScoreFunctionFactory: SCOREFUNCTION: ref2015\n",
            "protocols.rosetta_scripts.RosettaScriptsParser: Defined mover named \"dih_mover_L1\" of type CDRDihedralConstraintMover\n",
            "protocols.rosetta_scripts.RosettaScriptsParser: Defined mover named \"dih_mover_L2\" of type CDRDihedralConstraintMover\n",
            "protocols.rosetta_scripts.RosettaScriptsParser: Defined mover named \"dih_mover_L3\" of type CDRDihedralConstraintMover\n",
            "protocols.antibody.AntibodyCDRGrafter: Setting CDRs from settings\n",
            "core.import_pose.import_pose: File 'malaria_cdrs.pdb' automatically determined to be of type PDB\n",
            "core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue CYS:CtermProteinFull 387\n",
            "core.conformation.Conformation: [ WARNING ] missing heavyatom:  OXT on residue ASP:CtermProteinFull 601\n",
            "core.conformation.Conformation: Found disulfide between residues 8 22\n",
            "core.conformation.Conformation: Found disulfide between residues 24 36\n",
            "core.conformation.Conformation: Found disulfide between residues 43 58\n",
            "core.conformation.Conformation: Found disulfide between residues 52 70\n",
            "core.conformation.Conformation: Found disulfide between residues 72 83\n",
            "core.conformation.Conformation: Found disulfide between residues 88 98\n",
            "core.conformation.Conformation: Found disulfide between residues 93 111\n",
            "core.conformation.Conformation: Found disulfide between residues 113 127\n",
            "core.conformation.Conformation: Found disulfide between residues 135 146\n",
            "core.conformation.Conformation: Found disulfide between residues 139 155\n",
            "core.conformation.Conformation: Found disulfide between residues 157 170\n",
            "core.conformation.Conformation: Found disulfide between residues 197 261\n",
            "core.conformation.Conformation: Found disulfide between residues 307 367\n",
            "core.conformation.Conformation: Found disulfide between residues 407 481\n",
            "core.conformation.Conformation: Found disulfide between residues 527 582\n",
            "protocols.rosetta_scripts.RosettaScriptsParser: Defined mover named \"grafter\" of type AntibodyCDRGrafter\n",
            "core.pack.task.xml_util: Object packrot reading the following task_operations: Adding the following task operations\n",
            "restrict_to_cdrs profiles\n",
            "protocols.rosetta_scripts.RosettaScriptsParser: Defined mover named \"packrot\" of type PackRotamersMover\n",
            "protocols.minimization_packing.MinMover: Found set MoveMap factory. Using this to define MoveMap.\n",
            "core.select.movemap.util: Found MoveMapFactory movemap_cdrs\n",
            "protocols.rosetta_scripts.RosettaScriptsParser: Defined mover named \"minmover\" of type MinMover\n",
            "core.simple_metrics.util: Added simple metric SasaMetric.\n",
            "core.simple_metrics.util: Added simple metric SelectedResiduesPyMOLMetric.\n",
            "core.simple_metrics.util: Added simple metric SequenceMetric.\n",
            "core.simple_metrics.util: Added simple metric SequenceMetric.\n",
            "core.simple_metrics.util: Added simple metric SequenceMetric.\n",
            "core.simple_metrics.util: Added simple metric InteractionEnergyMetric.\n",
            "protocols.rosetta_scripts.RosettaScriptsParser: Defined mover named \"design_metrics\" of type RunSimpleMetrics\n",
            "core.simple_metrics.util: Added simple metric SasaMetric.\n",
            "core.simple_metrics.util: Added simple metric SelectedResiduesPyMOLMetric.\n",
            "core.simple_metrics.util: Added simple metric SequenceMetric.\n",
            "core.simple_metrics.util: Added simple metric SequenceMetric.\n",
            "core.simple_metrics.util: Added simple metric SequenceMetric.\n",
            "core.simple_metrics.util: Added simple metric InteractionEnergyMetric.\n",
            "protocols.rosetta_scripts.RosettaScriptsParser: Defined mover named \"native_metrics\" of type RunSimpleMetrics\n",
            "protocols.rosetta_scripts.ParsedProtocol: ParsedProtocol mover with the following settings\n",
            "protocols.rosetta_scripts.ParsedProtocol: Added mover \"native_metrics\"\n",
            "protocols.rosetta_scripts.ParsedProtocol: Added mover \"grafter\"\n",
            "protocols.rosetta_scripts.ParsedProtocol: Added mover \"dih_mover_L1\"\n",
            "protocols.rosetta_scripts.ParsedProtocol: Added mover \"dih_mover_L2\"\n",
            "protocols.rosetta_scripts.ParsedProtocol: Added mover \"dih_mover_L3\"\n",
            "protocols.rosetta_scripts.ParsedProtocol: Added mover \"packrot\"\n",
            "protocols.rosetta_scripts.ParsedProtocol: Added mover \"minmover\"\n",
            "protocols.rosetta_scripts.ParsedProtocol: Added mover \"packrot\"\n",
            "protocols.rosetta_scripts.ParsedProtocol: Added mover \"minmover\"\n",
            "protocols.rosetta_scripts.ParsedProtocol: Added mover \"design_metrics\"\n"
          ]
        }
      ],
      "source": [
        "import pyrosetta\n",
        "from pyrosetta import rosetta\n",
        "\n",
        "if not os.getenv(\"DEBUG\"):\n",
        "    pyrosetta.init()  # Ensure PyRosetta is initialized\n",
        "    pose = original_pose.clone()\n",
        "    parser = rosetta.protocols.rosetta_scripts.RosettaScriptsParser()\n",
        "\n",
        "    # Read the XML file\n",
        "    xml_path = \"rabd/ab_design_components.xml\"\n",
        "\n",
        "    # Parse the XML file and create a Mover\n",
        "    protocol = parser.generate_mover(xml_path)\n",
        "\n",
        "    # Apply the Mover to the pose\n",
        "    #protocol.apply(pose)"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "1pqCYeik2LM0"
      },
      "source": [
        "## Tutorial B:  Optimizing Interface Energy (opt-dG)\n",
        "\n",
        "### Tut B1. Optimizing dG\n",
        "\n",
        "Here, we want to set the protocol to optimize the interface energy during Monte Carlo instead of total energy.  The interface energy is calculated by the InterfaceAnalyzerMover through a specialized MonteCarlo object called `MonteCarloInterface`.  This is useful to improve binding energy and generally results in better interface energies .  Resulting models should still be pruned for high total energy.  This was benchmarked in the paper, and has been used for real-life designs to the HIV epitope (165 seconds for 1 decoy).\n",
        "\n",
        "Use the provided XML or set this up through code.\n",
        "\n",
        "    <AntibodyDesignMover name=\"RAbD\" seq_design_cdrs=\"L1,L3\" graft_design_cdrs=\"L1,L3\" mc_optimize_dG=\"1\" />\n"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 61,
      "metadata": {
        "nbgrader": {
          "grade": true,
          "grade_id": "cell-7664283d2dba2e99",
          "locked": false,
          "points": 0,
          "schema_version": 3,
          "solution": true
        },
        "id": "oqwdFr9Q2LM0",
        "outputId": "58571d22-fdc6-458c-b1fb-907a6fad0989",
        "colab": {
          "base_uri": "https://localhost:8080/"
        }
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "\u001b[1;30;43mStreaming output truncated to the last 5000 lines.\u001b[0m\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L3\n",
            "protocols.antibody.cluster.CDRClusterMatcher: Length: 9 Omega: TTTTTTCTT\n",
            "protocols.antibody.clusters.CDRClusterSet: Adding cacheable cluster data to pose\n",
            "protocols.antibody.design.util: Reading CDRSetOptions from: /sampling/antibodies/design/default_instructions.txt\n",
            "protocols.antibody.design.util: Reading CDRGraftDesignOptions from: /sampling/antibodies/design/default_instructions.txt\n",
            "protocols.antibody.design.util: Reading CDRSeqDesignOptions from: /sampling/antibodies/design/default_instructions.txt\n",
            "core.scoring.ScoreFunctionFactory: SCOREFUNCTION: ref2015\n",
            "core.scoring.ScoreFunctionFactory: SCOREFUNCTION: ref2015\n",
            "core.scoring.ScoreFunctionFactory: SCOREFUNCTION: ref2015\n",
            "basic.io.database: [ WARNING ] Unable to locate database file /sampling/antibodies/antibody_database_rosetta_design.db\n",
            "antibody.database.AntibodyDatabaseManager: The antibody design database is not specified by the user.\n",
            "antibody.database.AntibodyDatabaseManager: Reading from: /usr/local/lib/python3.10/dist-packages/pyrosetta/database//sampling/antibodies/antibody_database_rosetta_design_north_paper.db\n",
            "antibody.database.AntibodyDatabaseManager: Checking loadable structures\n",
            "antibody.database.AntibodyDatabaseManager: Skipping 12 structures from reading\n",
            "core.scoring.ScoreFunctionFactory: SCOREFUNCTION: ref2015\n",
            "core.scoring.ScoreFunctionFactory: SCOREFUNCTION: ref2015\n",
            "antibody.database.AntibodyDatabaseManager: Total L1 in CDRSet: 191\n",
            "antibody.database.AntibodyDatabaseManager: Total L3 in CDRSet: 230\n",
            "antibody.database.AntibodyDatabaseManager: 300 Cached\n",
            "antibody.database.AntibodyDatabaseManager: cdrs loaded from database.\n",
            "protocols.antibody.design.AntibodyDesignMover: ///// Total CDRs in set /////\n",
            "protocols.antibody.design.AntibodyDesignMover: /// L1 191\n",
            "protocols.antibody.design.AntibodyDesignMover: /// L3 230\n",
            "protocols.antibody.design.AntibodyDesignMover: Total possible CDR combinations: 43930\n",
            "core.scoring.ScoreFunction: \n",
            "------------------------------------------------------------\n",
            " Scores                       Weight   Raw Score Wghtd.Score\n",
            "------------------------------------------------------------\n",
            " fa_atr                       1.000   -5695.317   -5695.317\n",
            " fa_rep                       0.550   24719.870   13595.929\n",
            " fa_sol                       1.000    3501.170    3501.170\n",
            " fa_intra_rep                 0.005    2661.048      13.305\n",
            " fa_intra_sol_xover4          1.000     175.028     175.028\n",
            " lk_ball_wtd                  1.000     -57.980     -57.980\n",
            " fa_elec                      1.000   -1588.297   -1588.297\n",
            " pro_close                    1.250      10.290      12.862\n",
            " hbond_sr_bb                  1.000     -74.000     -74.000\n",
            " hbond_lr_bb                  1.000    -511.787    -511.787\n",
            " hbond_bb_sc                  1.000    -130.727    -130.727\n",
            " hbond_sc                     1.000     -78.766     -78.766\n",
            " dslf_fa13                    1.250      -1.572      -1.965\n",
            " atom_pair_constraint         0.010       0.000       0.000\n",
            " omega                        0.400     474.068     189.627\n",
            " fa_dun                       0.700    5380.790    3766.553\n",
            " p_aa_pp                      0.600    -289.096    -173.458\n",
            " yhh_planarity                0.625       0.000       0.000\n",
            " ref                          1.000     372.765     372.765\n",
            " rama_prepro                  0.450     191.068      85.981\n",
            "---------------------------------------------------\n",
            " Total weighted score:                    13400.923\n",
            "protocols.monte_carlo.MonteCarloInterface: Detection of disulfides in the separated pose is set to 0\n",
            "protocols.analysis.InterfaceAnalyzerMover: Repacking separated Pose before running calculations\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using explicit constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using interface constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: [ WARNING ] In InterfaceAnalyzerMover, calculator Sasa_1 already exists, this is hopefully correct for your purposes\n",
            "protocols.analysis.InterfaceAnalyzerMover: [ WARNING ] In InterfaceAnalyzerMover, calculator InterfaceNeighborDefinition_1 already exists, this is hopefully correct for your purposes\n",
            "protocols.analysis.InterfaceAnalyzerMover: [ WARNING ] In InterfaceAnalyzerMover, calculator InterfaceDeltaEnergetics_1 already exists, this is hopefully correct for your purposes\n",
            "protocols.analysis.InterfaceAnalyzerMover: [ WARNING ] In InterfaceAnalyzerMover, calculator NumberHBonds_1 already exists, this is hopefully correct for your purposes\n",
            "protocols.analysis.InterfaceAnalyzerMover: [ WARNING ] In InterfaceAnalyzerMover, calculator BuriedUnsatisfiedPolars_1 already exists, this is hopefully correct for your purposes\n",
            "protocols.analysis.InterfaceAnalyzerMover: Making an interface set with fixed chains\n",
            "protocols.analysis.InterfaceAnalyzerMover: Interface set residues total: 100\n",
            "protocols.analysis.InterfaceAnalyzerMover: NOT detecting disulfides in the separated pose.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 2678 rotamers at 100 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.analysis.InterfaceAnalyzerMover: included_nres: 975\n",
            "protocols.monte_carlo.MonteCarloInterface: Detection of disulfides in the separated pose is set to 0\n",
            "protocols.analysis.InterfaceAnalyzerMover: Repacking separated Pose before running calculations\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using explicit constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using interface constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: [ WARNING ] In InterfaceAnalyzerMover, calculator Sasa_1 already exists, this is hopefully correct for your purposes\n",
            "protocols.analysis.InterfaceAnalyzerMover: [ WARNING ] In InterfaceAnalyzerMover, calculator InterfaceNeighborDefinition_1 already exists, this is hopefully correct for your purposes\n",
            "protocols.analysis.InterfaceAnalyzerMover: [ WARNING ] In InterfaceAnalyzerMover, calculator InterfaceDeltaEnergetics_1 already exists, this is hopefully correct for your purposes\n",
            "protocols.analysis.InterfaceAnalyzerMover: [ WARNING ] In InterfaceAnalyzerMover, calculator NumberHBonds_1 already exists, this is hopefully correct for your purposes\n",
            "protocols.analysis.InterfaceAnalyzerMover: [ WARNING ] In InterfaceAnalyzerMover, calculator BuriedUnsatisfiedPolars_1 already exists, this is hopefully correct for your purposes\n",
            "protocols.analysis.InterfaceAnalyzerMover: Making an interface set with fixed chains\n",
            "protocols.analysis.InterfaceAnalyzerMover: Interface set residues total: 100\n",
            "protocols.analysis.InterfaceAnalyzerMover: NOT detecting disulfides in the separated pose.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 2678 rotamers at 100 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.analysis.InterfaceAnalyzerMover: included_nres: 975\n",
            "protocols.antibody.design.AntibodyDesignMover: Running basic monte carlo algorithm\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using explicit constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using interface constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Making an interface set with fixed chains\n",
            "protocols.analysis.InterfaceAnalyzerMover: Interface set residues total: 100\n",
            "protocols.analysis.InterfaceAnalyzerMover: NOT detecting disulfides in the separated pose.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 2678 rotamers at 100 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.analysis.InterfaceAnalyzerMover: included_nres: 975\n",
            "protocols.antibody.design.AntibodyDesignMover: Outer round: 1\n",
            "protocols.antibody.design.AntibodyDesignMover: Grafting CDR from cluster L1-12-2 fragment 1rhhA_L1_0001\n",
            "protocols.grafting.CCDEndsGraftMover: Start: 891 End: 903 NterO: 3 CterO: 3\n",
            "protocols.grafting.util: Superimposing overhang residues\n",
            "core.chemical.AtomICoor: [ WARNING ] IcoorAtomID::atom_id(): Cannot get atom_id for POLYMER_LOWER of residue LEU 1.  Returning BOGUS ID instead.\n",
            "core.conformation.Residue: [ WARNING ] missing an atom: 1  H   that depends on a nonexistent polymer connection!\n",
            "core.conformation.Residue: [ WARNING ]  --> generating it using idealized coordinates.\n",
            "protocols.grafting.util: Deleting 11 residues from 892 to 902\n",
            "core.conformation.Conformation: Found disulfide between residues 771 845\n",
            "core.conformation.Conformation: Found disulfide between residues 891 945\n",
            "protocols.grafting.util: insert_point 891\n",
            "protocols.grafting.util: insert_start 892\n",
            "protocols.grafting.util: insert_end 903\n",
            "protocols.grafting.util: insert_length 12\n",
            "protocols.grafting.util: FOLD_TREE  EDGE 1 891 -1  EDGE 891 892 1  EDGE 892 964 -1\n",
            "protocols.grafting.GraftMoverBase: Insertion complete.\n",
            "protocols.grafting.AnchoredGraftMover: Setting default movemap\n",
            "protocols.grafting.CCDEndsGraftMover: Start: 891\n",
            "protocols.grafting.CCDEndsGraftMover: Original End: 903\n",
            "protocols.grafting.CCDEndsGraftMover: End: 904\n",
            "protocols.grafting.CCDEndsGraftMover: Insert Length: 12\n",
            "protocols.grafting.CCDEndsGraftMover: LOOP start: 890  stop: 893  cut: 891  size: 4  skip rate: 0  extended?: False\n",
            "protocols.grafting.CCDEndsGraftMover:\n",
            "protocols.grafting.CCDEndsGraftMover: LOOP start: 902  stop: 905  cut: 903  size: 4  skip rate: 0  extended?: False\n",
            "protocols.grafting.CCDEndsGraftMover:\n",
            "protocols.loops.FoldTreeFromLoopsWrapper: old foldtree FOLD_TREE  EDGE 1 891 -1  EDGE 891 904 1  EDGE 904 976 -1  EDGE 891 892 -2  C    N    EDGE 892 893 -2  C    N    EDGE 893 894 -2  C    N    EDGE 894 895 -2  C    N    EDGE 895 896 -2  C    N    EDGE 896 897 -2  C    N    EDGE 897 898 -2  C    N    EDGE 898 899 -2  C    N    EDGE 899 900 -2  C    N    EDGE 900 901 -2  C    N    EDGE 901 902 -2  C    N    EDGE 902 903 -2  C    N   \n",
            "New foldtree FOLD_TREE  EDGE 1 889 -1  EDGE 889 891 -1  EDGE 889 894 1  EDGE 894 892 -1  EDGE 894 901 -1  EDGE 901 903 -1  EDGE 901 906 2  EDGE 906 904 -1  EDGE 906 976 -1\n",
            "protocols.grafting.util: Loop: 1\n",
            "protocols.grafting.util: Add variant to: 891\n",
            "protocols.grafting.util: Loop: 2\n",
            "protocols.grafting.util: Add variant to: 903\n",
            "protocols.grafting.util: idealized  890\n",
            "protocols.grafting.util: idealized  891\n",
            "protocols.grafting.util: ideal 892\n",
            "protocols.grafting.util: ideal 903\n",
            "protocols.grafting.util: idealized 904\n",
            "protocols.grafting.util: idealized 905\n",
            "protocols.grafting.CCDEndsGraftMover: start 1741.59\n",
            "protocols.grafting.CCDEndsGraftMover: round 1\n",
            "protocols.grafting.CCDEndsGraftMover: Graft Closed early - returning\n",
            "protocols.grafting.CCDEndsGraftMover: 1 1287.47\n",
            "protocols.grafting.CCDEndsGraftMover: finish 1287.47\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 1728 rotamers at 51 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.grafting.CCDEndsGraftMover: Graft meets ideal geometry true\n",
            "protocols.grafting.CCDEndsGraftMover: Complete\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L1-12-2 data: 9 cutoff: 10\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: Adding general dihedral constraints for L1\n",
            "protocols.antibody.design.AntibodyDesignMover: Checking Geometry\n",
            "protocols.antibody.design.AntibodyDesignMover: DATA GRAFT_CLOSURE 1 rabd/my_ab.pdb L1 L1-12-2 1rhhA_L1_0001\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H1\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H2\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H3\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L1\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L2\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L3\n",
            "protocols.antibody.clusters.CDRClusterSet: Adding cacheable cluster data to pose\n",
            "protocols.antibody.design.util: Setting CDR sequence in NativeAntibodySeq\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: H1-13-1 data: 583 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/H1-13-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 26 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: H2-10-1 data: 328 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/H2-10-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 20 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: Adding general dihedral constraints for H3\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L1-12-2 data: 9 cutoff: 10\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: Adding general dihedral constraints for L1\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L2-8-1 data: 632 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/L2-8-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 16 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L3-9-cis7-1 data: 419 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/L3-9-cis7-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 18 constraints\n",
            "protocols.antibody.design.AntibodyDesignMover: Beginning MinProtocol\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using explicit constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using interface constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Making an interface set with fixed chains\n",
            "protocols.analysis.InterfaceAnalyzerMover: Interface set residues total: 102\n",
            "protocols.analysis.InterfaceAnalyzerMover: NOT detecting disulfides in the separated pose.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 2962 rotamers at 102 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.analysis.InterfaceAnalyzerMover: included_nres: 976\n",
            "protocols.antibody.design.AntibodyDesignMover: Inner round: 1\n",
            "protocols.antibody.design.AntibodyDesignMover: Add min neighbors : L2\n",
            "protocols.antibody.design.AntibodyDesignMover: Add min neighbors : L3\n",
            "basic.io.database: [ WARNING ] Unable to locate database file /sampling/antibodies/antibody_database_rosetta_design.db\n",
            "antibody.database.AntibodyDatabaseManager: The antibody design database is not specified by the user.\n",
            "antibody.database.AntibodyDatabaseManager: Reading from: /usr/local/lib/python3.10/dist-packages/pyrosetta/database//sampling/antibodies/antibody_database_rosetta_design_north_paper.db\n",
            "antibody.database.AntibodyDatabaseManager: Skipped L1-12-2 with 5 data point\n",
            "antibody.database.AntibodyDatabaseManager: Loaded L3-9-cis7-1 with 179 datapoints.\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Data pre-loaded.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: applying prob task op\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Using conservative op for L1\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Using original bb sequence for conservative design.\n",
            "protocols.task_operations.ConservativeDesignOperation: Loading conservative mutational data\n",
            "core.pack.pack_rotamers: built 5257 rotamers at 80 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating PDInteractionGraph\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs start: 17229.7\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs end: 6124.92\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using explicit constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using interface constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Making an interface set with fixed chains\n",
            "protocols.analysis.InterfaceAnalyzerMover: Interface set residues total: 101\n",
            "protocols.analysis.InterfaceAnalyzerMover: NOT detecting disulfides in the separated pose.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 2736 rotamers at 101 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.analysis.InterfaceAnalyzerMover: included_nres: 976\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 897\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 898\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 921\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 922\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 962\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 963\n",
            "antibody.design.GeneralAntibodyModeler: CDR min start: 6124.92\n",
            "antibody.design.GeneralAntibodyModeler: CDR min end: 4148.39\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using explicit constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using interface constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Making an interface set with fixed chains\n",
            "protocols.analysis.InterfaceAnalyzerMover: Interface set residues total: 104\n",
            "protocols.analysis.InterfaceAnalyzerMover: NOT detecting disulfides in the separated pose.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 2650 rotamers at 104 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.analysis.InterfaceAnalyzerMover: included_nres: 976\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: applying prob task op\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Using conservative op for L1\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Using original bb sequence for conservative design.\n",
            "core.pack.pack_rotamers: built 5555 rotamers at 76 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating PDInteractionGraph\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs start: 4148.39\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs end: 4330.91\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using explicit constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using interface constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Making an interface set with fixed chains\n",
            "protocols.analysis.InterfaceAnalyzerMover: Interface set residues total: 104\n",
            "protocols.analysis.InterfaceAnalyzerMover: NOT detecting disulfides in the separated pose.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 2794 rotamers at 104 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.analysis.InterfaceAnalyzerMover: included_nres: 976\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 897\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 898\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 921\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 922\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 962\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 963\n",
            "antibody.design.GeneralAntibodyModeler: CDR min start: 4330.91\n",
            "antibody.design.GeneralAntibodyModeler: CDR min end: 4103.71\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using explicit constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using interface constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Making an interface set with fixed chains\n",
            "protocols.analysis.InterfaceAnalyzerMover: Interface set residues total: 104\n",
            "protocols.analysis.InterfaceAnalyzerMover: NOT detecting disulfides in the separated pose.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 2800 rotamers at 104 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.analysis.InterfaceAnalyzerMover: included_nres: 976\n",
            "protocols.antibody.design.AntibodyDesignMover: Min Cycle complete\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using explicit constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using interface constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Making an interface set with fixed chains\n",
            "protocols.analysis.InterfaceAnalyzerMover: Interface set residues total: 104\n",
            "protocols.analysis.InterfaceAnalyzerMover: NOT detecting disulfides in the separated pose.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 2800 rotamers at 104 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.analysis.InterfaceAnalyzerMover: included_nres: 976\n",
            "protocols.antibody.design.AntibodyDesignMover:\n",
            "protocols.antibody.design.AntibodyDesignMover: SCORE 1 3498.2 1\n",
            "protocols.antibody.design.AntibodyDesignMover: FINAL 1 3498.2 1\n",
            "protocols.antibody.design.AntibodyDesignMover:\n",
            "protocols.antibody.design.AntibodyDesignMover: Last Accepted Score: 3498.2\n",
            "protocols.antibody.design.AntibodyDesignMover: Lowest        Score:3498.2\n",
            "protocols.antibody.design.AntibodyDesignMover:\n",
            "protocols.antibody.design.AntibodyDesignMover: Outer round: 2\n",
            "protocols.antibody.design.AntibodyDesignMover: Grafting CDR from cluster L3-8-cis6-1 fragment 1e6oL_L3_0001\n",
            "protocols.grafting.CCDEndsGraftMover: Start: 957 End: 967 NterO: 3 CterO: 3\n",
            "protocols.grafting.util: Superimposing overhang residues\n",
            "core.chemical.AtomICoor: [ WARNING ] IcoorAtomID::atom_id(): Cannot get atom_id for POLYMER_LOWER of residue TYR 1.  Returning BOGUS ID instead.\n",
            "core.conformation.Residue: [ WARNING ] missing an atom: 1  H   that depends on a nonexistent polymer connection!\n",
            "core.conformation.Residue: [ WARNING ]  --> generating it using idealized coordinates.\n",
            "protocols.grafting.util: Deleting 9 residues from 958 to 966\n",
            "core.conformation.Conformation: Found disulfide between residues 771 845\n",
            "core.conformation.Conformation: Found disulfide between residues 891 957\n",
            "protocols.grafting.util: insert_point 957\n",
            "protocols.grafting.util: insert_start 958\n",
            "protocols.grafting.util: insert_end 965\n",
            "protocols.grafting.util: insert_length 8\n",
            "protocols.grafting.util: FOLD_TREE  EDGE 1 957 -1  EDGE 957 958 1  EDGE 958 967 -1\n",
            "protocols.grafting.GraftMoverBase: Insertion complete.\n",
            "protocols.grafting.AnchoredGraftMover: Setting default movemap\n",
            "protocols.grafting.CCDEndsGraftMover: Start: 957\n",
            "protocols.grafting.CCDEndsGraftMover: Original End: 967\n",
            "protocols.grafting.CCDEndsGraftMover: End: 966\n",
            "protocols.grafting.CCDEndsGraftMover: Insert Length: 8\n",
            "protocols.grafting.CCDEndsGraftMover: LOOP start: 956  stop: 959  cut: 957  size: 4  skip rate: 0  extended?: False\n",
            "protocols.grafting.CCDEndsGraftMover:\n",
            "protocols.grafting.CCDEndsGraftMover: LOOP start: 964  stop: 967  cut: 965  size: 4  skip rate: 0  extended?: False\n",
            "protocols.grafting.CCDEndsGraftMover:\n",
            "protocols.loops.FoldTreeFromLoopsWrapper: old foldtree FOLD_TREE  EDGE 1 957 -1  EDGE 957 966 1  EDGE 966 975 -1  EDGE 957 958 -2  C    N    EDGE 958 959 -2  C    N    EDGE 959 960 -2  C    N    EDGE 960 961 -2  C    N    EDGE 961 962 -2  C    N    EDGE 962 963 -2  C    N    EDGE 963 964 -2  C    N    EDGE 964 965 -2  C    N   \n",
            "New foldtree FOLD_TREE  EDGE 1 955 -1  EDGE 955 957 -1  EDGE 955 960 1  EDGE 960 958 -1  EDGE 960 963 -1  EDGE 963 965 -1  EDGE 963 968 2  EDGE 968 966 -1  EDGE 968 975 -1\n",
            "protocols.grafting.util: Loop: 1\n",
            "protocols.grafting.util: Add variant to: 957\n",
            "protocols.grafting.util: Loop: 2\n",
            "protocols.grafting.util: Add variant to: 965\n",
            "protocols.grafting.util: idealized  956\n",
            "protocols.grafting.util: idealized  957\n",
            "protocols.grafting.util: ideal 958\n",
            "protocols.grafting.util: ideal 965\n",
            "protocols.grafting.util: idealized 966\n",
            "protocols.grafting.util: idealized 967\n",
            "protocols.grafting.CCDEndsGraftMover: start 1730.91\n",
            "protocols.grafting.CCDEndsGraftMover: round 1\n",
            "protocols.loops.util: PepBondGeom: 965 - 138 L  Ca-C-N --  99.5 (min): 134.93 : 134.5 (max)\n",
            "protocols.grafting.CCDEndsGraftMover: 1 1415.49\n",
            "protocols.grafting.CCDEndsGraftMover: round 2\n",
            "protocols.grafting.CCDEndsGraftMover: Graft Closed early - returning\n",
            "protocols.grafting.CCDEndsGraftMover: 2 1322.53\n",
            "protocols.grafting.CCDEndsGraftMover: finish 1322.53\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 910 rotamers at 34 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.grafting.CCDEndsGraftMover: Graft meets ideal geometry true\n",
            "protocols.grafting.CCDEndsGraftMover: Complete\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L3-8-cis6-1 data: 4 cutoff: 10\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: Adding general dihedral constraints for L3\n",
            "protocols.antibody.design.AntibodyDesignMover: Checking Geometry\n",
            "protocols.antibody.design.AntibodyDesignMover: DATA GRAFT_CLOSURE 1 rabd/my_ab.pdb L3 L3-8-cis6-1 1e6oL_L3_0001\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H1\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H2\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H3\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L1\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L2\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L3\n",
            "protocols.antibody.clusters.CDRClusterSet: Adding cacheable cluster data to pose\n",
            "protocols.antibody.design.util: Setting CDR sequence in NativeAntibodySeq\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: H1-13-1 data: 583 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/H1-13-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 26 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: H2-10-1 data: 328 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/H2-10-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 20 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: Adding general dihedral constraints for H3\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L1-12-2 data: 9 cutoff: 10\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: Adding general dihedral constraints for L1\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L2-8-1 data: 632 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/L2-8-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 16 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L3-8-cis6-1 data: 4 cutoff: 10\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: Adding general dihedral constraints for L3\n",
            "protocols.antibody.design.AntibodyDesignMover: Beginning MinProtocol\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using explicit constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using interface constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Making an interface set with fixed chains\n",
            "protocols.analysis.InterfaceAnalyzerMover: Interface set residues total: 99\n",
            "protocols.analysis.InterfaceAnalyzerMover: NOT detecting disulfides in the separated pose.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 2548 rotamers at 99 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.analysis.InterfaceAnalyzerMover: included_nres: 975\n",
            "protocols.antibody.design.AntibodyDesignMover: Inner round: 1\n",
            "protocols.antibody.design.AntibodyDesignMover: Add min neighbors : L1\n",
            "protocols.antibody.design.AntibodyDesignMover: Add min neighbors : H3\n",
            "basic.io.database: [ WARNING ] Unable to locate database file /sampling/antibodies/antibody_database_rosetta_design.db\n",
            "antibody.database.AntibodyDatabaseManager: The antibody design database is not specified by the user.\n",
            "antibody.database.AntibodyDatabaseManager: Reading from: /usr/local/lib/python3.10/dist-packages/pyrosetta/database//sampling/antibodies/antibody_database_rosetta_design_north_paper.db\n",
            "antibody.database.AntibodyDatabaseManager: Skipped L1-12-2 with 5 data point\n",
            "antibody.database.AntibodyDatabaseManager: Skipped L3-8-cis6-1 with 3 data point\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Data pre-loaded.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: applying prob task op\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Using conservative op for L1\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Using conservative op for L3\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Using original bb sequence for conservative design.\n",
            "protocols.task_operations.ConservativeDesignOperation: Loading conservative mutational data\n",
            "core.pack.pack_rotamers: built 6086 rotamers at 90 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating PDInteractionGraph\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs start: 9791.1\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs end: 6214.3\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using explicit constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using interface constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Making an interface set with fixed chains\n",
            "protocols.analysis.InterfaceAnalyzerMover: Interface set residues total: 99\n",
            "protocols.analysis.InterfaceAnalyzerMover: NOT detecting disulfides in the separated pose.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 2657 rotamers at 99 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.analysis.InterfaceAnalyzerMover: included_nres: 975\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 852\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 853\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 897\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 898\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 961\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 962\n",
            "antibody.design.GeneralAntibodyModeler: CDR min start: 6214.3\n",
            "antibody.design.GeneralAntibodyModeler: CDR min end: 3408.52\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using explicit constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using interface constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Making an interface set with fixed chains\n",
            "protocols.analysis.InterfaceAnalyzerMover: Interface set residues total: 99\n",
            "protocols.analysis.InterfaceAnalyzerMover: NOT detecting disulfides in the separated pose.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 2842 rotamers at 99 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.analysis.InterfaceAnalyzerMover: included_nres: 975\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: applying prob task op\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Using conservative op for L1\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Using conservative op for L3\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Using original bb sequence for conservative design.\n",
            "core.pack.pack_rotamers: built 5657 rotamers at 90 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating PDInteractionGraph\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs start: 3408.52\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs end: 4001.69\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using explicit constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using interface constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Making an interface set with fixed chains\n",
            "protocols.analysis.InterfaceAnalyzerMover: Interface set residues total: 98\n",
            "protocols.analysis.InterfaceAnalyzerMover: NOT detecting disulfides in the separated pose.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 2888 rotamers at 98 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.analysis.InterfaceAnalyzerMover: included_nres: 975\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 852\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 853\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 897\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 898\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 961\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 962\n",
            "antibody.design.GeneralAntibodyModeler: CDR min start: 3408.52\n",
            "antibody.design.GeneralAntibodyModeler: CDR min end: 3407.67\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using explicit constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using interface constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Making an interface set with fixed chains\n",
            "protocols.analysis.InterfaceAnalyzerMover: Interface set residues total: 99\n",
            "protocols.analysis.InterfaceAnalyzerMover: NOT detecting disulfides in the separated pose.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 2844 rotamers at 99 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.analysis.InterfaceAnalyzerMover: included_nres: 975\n",
            "protocols.antibody.design.AntibodyDesignMover: Min Cycle complete\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using explicit constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using interface constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Making an interface set with fixed chains\n",
            "protocols.analysis.InterfaceAnalyzerMover: Interface set residues total: 99\n",
            "protocols.analysis.InterfaceAnalyzerMover: NOT detecting disulfides in the separated pose.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 2844 rotamers at 99 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.analysis.InterfaceAnalyzerMover: included_nres: 975\n",
            "protocols.antibody.design.AntibodyDesignMover:\n",
            "protocols.antibody.design.AntibodyDesignMover: SCORE 2 2815.26 1\n",
            "protocols.antibody.design.AntibodyDesignMover: FINAL 2 2815.26 1\n",
            "protocols.antibody.design.AntibodyDesignMover:\n",
            "protocols.antibody.design.AntibodyDesignMover: Last Accepted Score: 2815.26\n",
            "protocols.antibody.design.AntibodyDesignMover: Lowest        Score:2815.26\n",
            "protocols.antibody.design.AntibodyDesignMover:\n",
            "protocols.antibody.design.AntibodyDesignMover: Outer round: 3\n",
            "protocols.antibody.design.AntibodyDesignMover: Grafting CDR from cluster L1-10-1 fragment 1mimL_L1_0001\n",
            "protocols.grafting.CCDEndsGraftMover: Start: 891 End: 904 NterO: 3 CterO: 3\n",
            "protocols.grafting.util: Superimposing overhang residues\n",
            "core.chemical.AtomICoor: [ WARNING ] IcoorAtomID::atom_id(): Cannot get atom_id for POLYMER_LOWER of residue MET 1.  Returning BOGUS ID instead.\n",
            "core.conformation.Residue: [ WARNING ] missing an atom: 1  H   that depends on a nonexistent polymer connection!\n",
            "core.conformation.Residue: [ WARNING ]  --> generating it using idealized coordinates.\n",
            "protocols.grafting.util: Deleting 12 residues from 892 to 903\n",
            "core.conformation.Conformation: Found disulfide between residues 771 845\n",
            "core.conformation.Conformation: Found disulfide between residues 891 945\n",
            "protocols.grafting.util: insert_point 891\n",
            "protocols.grafting.util: insert_start 892\n",
            "protocols.grafting.util: insert_end 901\n",
            "protocols.grafting.util: insert_length 10\n",
            "protocols.grafting.util: FOLD_TREE  EDGE 1 891 -1  EDGE 891 892 1  EDGE 892 963 -1\n",
            "protocols.grafting.GraftMoverBase: Insertion complete.\n",
            "protocols.grafting.AnchoredGraftMover: Setting default movemap\n",
            "protocols.grafting.CCDEndsGraftMover: Start: 891\n",
            "protocols.grafting.CCDEndsGraftMover: Original End: 904\n",
            "protocols.grafting.CCDEndsGraftMover: End: 902\n",
            "protocols.grafting.CCDEndsGraftMover: Insert Length: 10\n",
            "protocols.grafting.CCDEndsGraftMover: LOOP start: 890  stop: 893  cut: 891  size: 4  skip rate: 0  extended?: False\n",
            "protocols.grafting.CCDEndsGraftMover:\n",
            "protocols.grafting.CCDEndsGraftMover: LOOP start: 900  stop: 903  cut: 901  size: 4  skip rate: 0  extended?: False\n",
            "protocols.grafting.CCDEndsGraftMover:\n",
            "protocols.loops.FoldTreeFromLoopsWrapper: old foldtree FOLD_TREE  EDGE 1 891 -1  EDGE 891 902 1  EDGE 902 973 -1  EDGE 891 892 -2  C    N    EDGE 892 893 -2  C    N    EDGE 893 894 -2  C    N    EDGE 894 895 -2  C    N    EDGE 895 896 -2  C    N    EDGE 896 897 -2  C    N    EDGE 897 898 -2  C    N    EDGE 898 899 -2  C    N    EDGE 899 900 -2  C    N    EDGE 900 901 -2  C    N   \n",
            "New foldtree FOLD_TREE  EDGE 1 889 -1  EDGE 889 891 -1  EDGE 889 894 1  EDGE 894 892 -1  EDGE 894 899 -1  EDGE 899 901 -1  EDGE 899 904 2  EDGE 904 902 -1  EDGE 904 973 -1\n",
            "protocols.grafting.util: Loop: 1\n",
            "protocols.grafting.util: Add variant to: 891\n",
            "protocols.grafting.util: Loop: 2\n",
            "protocols.grafting.util: Add variant to: 901\n",
            "protocols.grafting.util: idealized  890\n",
            "protocols.grafting.util: idealized  891\n",
            "protocols.grafting.util: ideal 892\n",
            "protocols.grafting.util: ideal 901\n",
            "protocols.grafting.util: idealized 902\n",
            "protocols.grafting.util: idealized 903\n",
            "protocols.grafting.CCDEndsGraftMover: start 1588.37\n",
            "protocols.grafting.CCDEndsGraftMover: round 1\n",
            "protocols.loops.util: PepBondGeom: 901 - 42 L  Ca-C-N --  99.5 (min): 136.473 : 134.5 (max)\n",
            "protocols.grafting.CCDEndsGraftMover: 1 1272.61\n",
            "protocols.grafting.CCDEndsGraftMover: round 2\n",
            "protocols.loops.util: PepBondGeom: 901 - 42 L  Ca-C-N --  99.5 (min): 140.026 : 134.5 (max)\n",
            "protocols.grafting.CCDEndsGraftMover: round 3\n",
            "protocols.loops.util: PepBondGeom: 42 L  C-N --  1.54574 max: 1.5\n",
            "protocols.grafting.CCDEndsGraftMover: 3 1272.19\n",
            "protocols.grafting.CCDEndsGraftMover: round 4\n",
            "protocols.loops.util: PepBondGeom: 42 L  C-N --  1.50009 max: 1.5\n",
            "protocols.grafting.CCDEndsGraftMover: round 5\n",
            "protocols.loops.util: PepBondGeom: 901 - 42 L  Ca-C-N --  99.5 (min): 136.581 : 134.5 (max)\n",
            "protocols.grafting.CCDEndsGraftMover: 5 1265.81\n",
            "protocols.grafting.CCDEndsGraftMover: round 6\n",
            "protocols.loops.util: PepBondGeom: 42 L  C-N --  1.54014 max: 1.5\n",
            "protocols.grafting.CCDEndsGraftMover: round 7\n",
            "protocols.loops.util: PepBondGeom: 901 - 42 L  Ca-C-N --  99.5 (min): 137.271 : 134.5 (max)\n",
            "protocols.grafting.CCDEndsGraftMover: round 8\n",
            "protocols.loops.util: PepBondGeom: 901 - 42 L  Ca-C-N --  99.5 (min): 139.78 : 134.5 (max)\n",
            "protocols.grafting.CCDEndsGraftMover: round 9\n",
            "protocols.loops.util: PepBondGeom: 42 L  C-N --  1.52281 max: 1.5\n",
            "protocols.grafting.CCDEndsGraftMover: round 10\n",
            "protocols.grafting.CCDEndsGraftMover: Graft Closed early - returning\n",
            "protocols.grafting.CCDEndsGraftMover: 10 1251.36\n",
            "protocols.grafting.CCDEndsGraftMover: finish 1251.36\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 982 rotamers at 36 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.grafting.CCDEndsGraftMover: Graft meets ideal geometry true\n",
            "protocols.grafting.CCDEndsGraftMover: Complete\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L1-10-1 data: 42 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/L1-10-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 20 constraints\n",
            "protocols.antibody.design.AntibodyDesignMover: Checking Geometry\n",
            "protocols.antibody.design.AntibodyDesignMover: DATA GRAFT_CLOSURE 1 rabd/my_ab.pdb L1 L1-10-1 1mimL_L1_0001\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H1\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H2\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H3\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L1\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L2\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L3\n",
            "protocols.antibody.clusters.CDRClusterSet: Adding cacheable cluster data to pose\n",
            "protocols.antibody.design.util: Setting CDR sequence in NativeAntibodySeq\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: H1-13-1 data: 583 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/H1-13-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 26 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: H2-10-1 data: 328 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/H2-10-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 20 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: Adding general dihedral constraints for H3\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L1-10-1 data: 42 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/L1-10-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 20 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L2-8-1 data: 632 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/L2-8-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 16 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L3-8-cis6-1 data: 4 cutoff: 10\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: Adding general dihedral constraints for L3\n",
            "protocols.antibody.design.AntibodyDesignMover: Beginning MinProtocol\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using explicit constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using interface constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Making an interface set with fixed chains\n",
            "protocols.analysis.InterfaceAnalyzerMover: Interface set residues total: 96\n",
            "protocols.analysis.InterfaceAnalyzerMover: NOT detecting disulfides in the separated pose.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 2707 rotamers at 96 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.analysis.InterfaceAnalyzerMover: included_nres: 973\n",
            "protocols.antibody.design.AntibodyDesignMover: Inner round: 1\n",
            "protocols.antibody.design.AntibodyDesignMover: Add min neighbors : L2\n",
            "protocols.antibody.design.AntibodyDesignMover: Add min neighbors : L3\n",
            "basic.io.database: [ WARNING ] Unable to locate database file /sampling/antibodies/antibody_database_rosetta_design.db\n",
            "antibody.database.AntibodyDatabaseManager: The antibody design database is not specified by the user.\n",
            "antibody.database.AntibodyDatabaseManager: Reading from: /usr/local/lib/python3.10/dist-packages/pyrosetta/database//sampling/antibodies/antibody_database_rosetta_design_north_paper.db\n",
            "antibody.database.AntibodyDatabaseManager: Loaded L1-10-1 with 15 datapoints.\n",
            "antibody.database.AntibodyDatabaseManager: Skipped L3-8-cis6-1 with 3 data point\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Data pre-loaded.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: applying prob task op\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Using conservative op for L3\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Using original bb sequence for conservative design.\n",
            "protocols.task_operations.ConservativeDesignOperation: Loading conservative mutational data\n",
            "core.pack.pack_rotamers: built 3955 rotamers at 74 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating PDInteractionGraph\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs start: 7425\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs end: 4040.87\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using explicit constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using interface constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Making an interface set with fixed chains\n",
            "protocols.analysis.InterfaceAnalyzerMover: Interface set residues total: 94\n",
            "protocols.analysis.InterfaceAnalyzerMover: NOT detecting disulfides in the separated pose.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 2637 rotamers at 94 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.analysis.InterfaceAnalyzerMover: included_nres: 973\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 896\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 897\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 919\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 920\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 959\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 960\n",
            "antibody.design.GeneralAntibodyModeler: CDR min start: 4040.87\n",
            "antibody.design.GeneralAntibodyModeler: CDR min end: 3428.36\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using explicit constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using interface constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Making an interface set with fixed chains\n",
            "protocols.analysis.InterfaceAnalyzerMover: Interface set residues total: 96\n",
            "protocols.analysis.InterfaceAnalyzerMover: NOT detecting disulfides in the separated pose.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 2633 rotamers at 96 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.analysis.InterfaceAnalyzerMover: included_nres: 973\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: applying prob task op\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Using conservative op for L3\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Using original bb sequence for conservative design.\n",
            "core.pack.pack_rotamers: built 4739 rotamers at 74 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating PDInteractionGraph\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs start: 3428.36\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs end: 2805.07\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using explicit constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using interface constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Making an interface set with fixed chains\n",
            "protocols.analysis.InterfaceAnalyzerMover: Interface set residues total: 96\n",
            "protocols.analysis.InterfaceAnalyzerMover: NOT detecting disulfides in the separated pose.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 2734 rotamers at 96 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.analysis.InterfaceAnalyzerMover: included_nres: 973\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 896\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 897\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 919\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 920\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 959\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 960\n",
            "antibody.design.GeneralAntibodyModeler: CDR min start: 2805.07\n",
            "antibody.design.GeneralAntibodyModeler: CDR min end: 2431.96\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using explicit constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using interface constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Making an interface set with fixed chains\n",
            "protocols.analysis.InterfaceAnalyzerMover: Interface set residues total: 95\n",
            "protocols.analysis.InterfaceAnalyzerMover: NOT detecting disulfides in the separated pose.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 2637 rotamers at 95 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.analysis.InterfaceAnalyzerMover: included_nres: 973\n",
            "protocols.antibody.design.AntibodyDesignMover: Min Cycle complete\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using explicit constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using interface constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Making an interface set with fixed chains\n",
            "protocols.analysis.InterfaceAnalyzerMover: Interface set residues total: 95\n",
            "protocols.analysis.InterfaceAnalyzerMover: NOT detecting disulfides in the separated pose.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 2637 rotamers at 95 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.analysis.InterfaceAnalyzerMover: included_nres: 973\n",
            "protocols.antibody.design.AntibodyDesignMover:\n",
            "protocols.antibody.design.AntibodyDesignMover: SCORE 3 1857.01 1\n",
            "protocols.antibody.design.AntibodyDesignMover: FINAL 3 1857.01 1\n",
            "protocols.antibody.design.AntibodyDesignMover:\n",
            "protocols.antibody.design.AntibodyDesignMover: Last Accepted Score: 1857.01\n",
            "protocols.antibody.design.AntibodyDesignMover: Lowest        Score:1857.01\n",
            "protocols.antibody.design.AntibodyDesignMover:\n",
            "protocols.antibody.design.AntibodyDesignMover: Outer round: 4\n",
            "protocols.antibody.design.AntibodyDesignMover: Grafting CDR from cluster L1-10-2 fragment 1otsF_L1_0001\n",
            "protocols.grafting.CCDEndsGraftMover: Start: 891 End: 902 NterO: 3 CterO: 3\n",
            "protocols.grafting.util: Superimposing overhang residues\n",
            "core.chemical.AtomICoor: [ WARNING ] IcoorAtomID::atom_id(): Cannot get atom_id for POLYMER_LOWER of residue MET 1.  Returning BOGUS ID instead.\n",
            "core.conformation.Residue: [ WARNING ] missing an atom: 1  H   that depends on a nonexistent polymer connection!\n",
            "core.conformation.Residue: [ WARNING ]  --> generating it using idealized coordinates.\n",
            "protocols.grafting.util: Deleting 10 residues from 892 to 901\n",
            "core.conformation.Conformation: Found disulfide between residues 771 845\n",
            "core.conformation.Conformation: Found disulfide between residues 891 945\n",
            "protocols.grafting.util: insert_point 891\n",
            "protocols.grafting.util: insert_start 892\n",
            "protocols.grafting.util: insert_end 901\n",
            "protocols.grafting.util: insert_length 10\n",
            "protocols.grafting.util: FOLD_TREE  EDGE 1 891 -1  EDGE 891 892 1  EDGE 892 963 -1\n",
            "protocols.grafting.GraftMoverBase: Insertion complete.\n",
            "protocols.grafting.AnchoredGraftMover: Setting default movemap\n",
            "protocols.grafting.CCDEndsGraftMover: Start: 891\n",
            "protocols.grafting.CCDEndsGraftMover: Original End: 902\n",
            "protocols.grafting.CCDEndsGraftMover: End: 902\n",
            "protocols.grafting.CCDEndsGraftMover: Insert Length: 10\n",
            "protocols.grafting.CCDEndsGraftMover: LOOP start: 890  stop: 893  cut: 891  size: 4  skip rate: 0  extended?: False\n",
            "protocols.grafting.CCDEndsGraftMover:\n",
            "protocols.grafting.CCDEndsGraftMover: LOOP start: 900  stop: 903  cut: 901  size: 4  skip rate: 0  extended?: False\n",
            "protocols.grafting.CCDEndsGraftMover:\n",
            "protocols.loops.FoldTreeFromLoopsWrapper: old foldtree FOLD_TREE  EDGE 1 891 -1  EDGE 891 902 1  EDGE 902 973 -1  EDGE 891 892 -2  C    N    EDGE 892 893 -2  C    N    EDGE 893 894 -2  C    N    EDGE 894 895 -2  C    N    EDGE 895 896 -2  C    N    EDGE 896 897 -2  C    N    EDGE 897 898 -2  C    N    EDGE 898 899 -2  C    N    EDGE 899 900 -2  C    N    EDGE 900 901 -2  C    N   \n",
            "New foldtree FOLD_TREE  EDGE 1 889 -1  EDGE 889 891 -1  EDGE 889 894 1  EDGE 894 892 -1  EDGE 894 899 -1  EDGE 899 901 -1  EDGE 899 904 2  EDGE 904 902 -1  EDGE 904 973 -1\n",
            "protocols.grafting.util: Loop: 1\n",
            "protocols.grafting.util: Add variant to: 891\n",
            "protocols.grafting.util: Loop: 2\n",
            "protocols.grafting.util: Add variant to: 901\n",
            "protocols.grafting.util: idealized  890\n",
            "protocols.grafting.util: idealized  891\n",
            "protocols.grafting.util: ideal 892\n",
            "protocols.grafting.util: ideal 901\n",
            "protocols.grafting.util: idealized 902\n",
            "protocols.grafting.util: idealized 903\n",
            "protocols.grafting.CCDEndsGraftMover: start 1568.82\n",
            "protocols.grafting.CCDEndsGraftMover: round 1\n",
            "protocols.grafting.CCDEndsGraftMover: Graft Closed early - returning\n",
            "protocols.grafting.CCDEndsGraftMover: 1 1250.19\n",
            "protocols.grafting.CCDEndsGraftMover: finish 1250.19\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 1471 rotamers at 43 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.grafting.CCDEndsGraftMover: Graft meets ideal geometry true\n",
            "protocols.grafting.CCDEndsGraftMover: Complete\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L1-10-2 data: 3 cutoff: 10\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: Adding general dihedral constraints for L1\n",
            "protocols.antibody.design.AntibodyDesignMover: Checking Geometry\n",
            "protocols.antibody.design.AntibodyDesignMover: DATA GRAFT_CLOSURE 1 rabd/my_ab.pdb L1 L1-10-2 1otsF_L1_0001\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H1\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H2\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H3\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L1\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L2\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L3\n",
            "protocols.antibody.clusters.CDRClusterSet: Adding cacheable cluster data to pose\n",
            "protocols.antibody.design.util: Setting CDR sequence in NativeAntibodySeq\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: H1-13-1 data: 583 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/H1-13-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 26 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: H2-10-1 data: 328 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/H2-10-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 20 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: Adding general dihedral constraints for H3\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L1-10-2 data: 3 cutoff: 10\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: Adding general dihedral constraints for L1\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L2-8-1 data: 632 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/L2-8-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 16 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L3-8-cis6-1 data: 4 cutoff: 10\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: Adding general dihedral constraints for L3\n",
            "protocols.antibody.design.AntibodyDesignMover: Beginning MinProtocol\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using explicit constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using interface constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Making an interface set with fixed chains\n",
            "protocols.analysis.InterfaceAnalyzerMover: Interface set residues total: 94\n",
            "protocols.analysis.InterfaceAnalyzerMover: NOT detecting disulfides in the separated pose.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 2420 rotamers at 94 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.analysis.InterfaceAnalyzerMover: included_nres: 973\n",
            "protocols.antibody.design.AntibodyDesignMover: Inner round: 1\n",
            "protocols.antibody.design.AntibodyDesignMover: Add min neighbors : L2\n",
            "protocols.antibody.design.AntibodyDesignMover: Add min neighbors : L3\n",
            "basic.io.database: [ WARNING ] Unable to locate database file /sampling/antibodies/antibody_database_rosetta_design.db\n",
            "antibody.database.AntibodyDatabaseManager: The antibody design database is not specified by the user.\n",
            "antibody.database.AntibodyDatabaseManager: Reading from: /usr/local/lib/python3.10/dist-packages/pyrosetta/database//sampling/antibodies/antibody_database_rosetta_design_north_paper.db\n",
            "antibody.database.AntibodyDatabaseManager: Skipped L1-10-2 with 2 data point\n",
            "antibody.database.AntibodyDatabaseManager: Skipped L3-8-cis6-1 with 3 data point\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Data pre-loaded.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: applying prob task op\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Using conservative op for L1\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Using conservative op for L3\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Using original bb sequence for conservative design.\n",
            "protocols.task_operations.ConservativeDesignOperation: Loading conservative mutational data\n",
            "core.pack.pack_rotamers: built 5701 rotamers at 74 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating PDInteractionGraph\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs start: 4410.62\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs end: 3829.35\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using explicit constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using interface constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Making an interface set with fixed chains\n",
            "protocols.analysis.InterfaceAnalyzerMover: Interface set residues total: 93\n",
            "protocols.analysis.InterfaceAnalyzerMover: NOT detecting disulfides in the separated pose.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 2286 rotamers at 93 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.analysis.InterfaceAnalyzerMover: included_nres: 973\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 896\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 897\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 919\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 920\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 959\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 960\n",
            "antibody.design.GeneralAntibodyModeler: CDR min start: 3829.35\n",
            "antibody.design.GeneralAntibodyModeler: CDR min end: 3440.67\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using explicit constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using interface constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Making an interface set with fixed chains\n",
            "protocols.analysis.InterfaceAnalyzerMover: Interface set residues total: 94\n",
            "protocols.analysis.InterfaceAnalyzerMover: NOT detecting disulfides in the separated pose.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 2307 rotamers at 94 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.analysis.InterfaceAnalyzerMover: included_nres: 973\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: applying prob task op\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Using conservative op for L1\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Using conservative op for L3\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Using original bb sequence for conservative design.\n",
            "core.pack.pack_rotamers: built 5365 rotamers at 72 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating PDInteractionGraph\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs start: 3440.67\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs end: 3491.94\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using explicit constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using interface constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Making an interface set with fixed chains\n",
            "protocols.analysis.InterfaceAnalyzerMover: Interface set residues total: 96\n",
            "protocols.analysis.InterfaceAnalyzerMover: NOT detecting disulfides in the separated pose.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 2344 rotamers at 96 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.analysis.InterfaceAnalyzerMover: included_nres: 973\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 896\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 897\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 919\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 920\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 959\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 960\n",
            "antibody.design.GeneralAntibodyModeler: CDR min start: 3440.67\n",
            "antibody.design.GeneralAntibodyModeler: CDR min end: 3436.01\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using explicit constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using interface constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Making an interface set with fixed chains\n",
            "protocols.analysis.InterfaceAnalyzerMover: Interface set residues total: 94\n",
            "protocols.analysis.InterfaceAnalyzerMover: NOT detecting disulfides in the separated pose.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 2307 rotamers at 94 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.analysis.InterfaceAnalyzerMover: included_nres: 973\n",
            "protocols.antibody.design.AntibodyDesignMover: Min Cycle complete\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using explicit constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using interface constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Making an interface set with fixed chains\n",
            "protocols.analysis.InterfaceAnalyzerMover: Interface set residues total: 94\n",
            "protocols.analysis.InterfaceAnalyzerMover: NOT detecting disulfides in the separated pose.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 2307 rotamers at 94 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.analysis.InterfaceAnalyzerMover: included_nres: 973\n",
            "protocols.antibody.design.AntibodyDesignMover:\n",
            "protocols.antibody.design.AntibodyDesignMover: SCORE 4 2838.4 0\n",
            "protocols.antibody.design.AntibodyDesignMover: FINAL 4 1857.01 1\n",
            "protocols.antibody.design.AntibodyDesignMover:\n",
            "protocols.antibody.design.AntibodyDesignMover: Last Accepted Score: 1857.01\n",
            "protocols.antibody.design.AntibodyDesignMover: Lowest        Score:1857.01\n",
            "protocols.antibody.design.AntibodyDesignMover:\n",
            "protocols.antibody.design.AntibodyDesignMover: Outer round: 5\n",
            "protocols.antibody.design.AntibodyDesignMover: Grafting CDR from cluster L3-9-1 fragment 3c08L_L3_0001\n",
            "core.scoring.ScoreFunctionFactory: SCOREFUNCTION: ref2015\n",
            "core.scoring.ScoreFunctionFactory: SCOREFUNCTION: ref2015\n",
            "protocols.grafting.CCDEndsGraftMover: Start: 955 End: 964 NterO: 3 CterO: 3\n",
            "protocols.grafting.util: Superimposing overhang residues\n",
            "core.chemical.AtomICoor: [ WARNING ] IcoorAtomID::atom_id(): Cannot get atom_id for POLYMER_LOWER of residue TYR 1.  Returning BOGUS ID instead.\n",
            "core.conformation.Residue: [ WARNING ] missing an atom: 1  H   that depends on a nonexistent polymer connection!\n",
            "core.conformation.Residue: [ WARNING ]  --> generating it using idealized coordinates.\n",
            "protocols.grafting.util: Deleting 8 residues from 956 to 963\n",
            "core.conformation.Conformation: Found disulfide between residues 771 845\n",
            "core.conformation.Conformation: Found disulfide between residues 891 955\n",
            "protocols.grafting.util: insert_point 955\n",
            "protocols.grafting.util: insert_start 956\n",
            "protocols.grafting.util: insert_end 964\n",
            "protocols.grafting.util: insert_length 9\n",
            "protocols.grafting.util: FOLD_TREE  EDGE 1 955 -1  EDGE 955 956 1  EDGE 956 965 -1\n",
            "protocols.grafting.GraftMoverBase: Insertion complete.\n",
            "protocols.grafting.AnchoredGraftMover: Setting default movemap\n",
            "protocols.grafting.CCDEndsGraftMover: Start: 955\n",
            "protocols.grafting.CCDEndsGraftMover: Original End: 964\n",
            "protocols.grafting.CCDEndsGraftMover: End: 965\n",
            "protocols.grafting.CCDEndsGraftMover: Insert Length: 9\n",
            "protocols.grafting.CCDEndsGraftMover: LOOP start: 954  stop: 957  cut: 955  size: 4  skip rate: 0  extended?: False\n",
            "protocols.grafting.CCDEndsGraftMover:\n",
            "protocols.grafting.CCDEndsGraftMover: LOOP start: 963  stop: 966  cut: 964  size: 4  skip rate: 0  extended?: False\n",
            "protocols.grafting.CCDEndsGraftMover:\n",
            "protocols.loops.FoldTreeFromLoopsWrapper: old foldtree FOLD_TREE  EDGE 1 955 -1  EDGE 955 965 1  EDGE 965 974 -1  EDGE 955 956 -2  C    N    EDGE 956 957 -2  C    N    EDGE 957 958 -2  C    N    EDGE 958 959 -2  C    N    EDGE 959 960 -2  C    N    EDGE 960 961 -2  C    N    EDGE 961 962 -2  C    N    EDGE 962 963 -2  C    N    EDGE 963 964 -2  C    N   \n",
            "New foldtree FOLD_TREE  EDGE 1 953 -1  EDGE 953 955 -1  EDGE 953 958 1  EDGE 958 956 -1  EDGE 958 962 -1  EDGE 962 964 -1  EDGE 962 967 2  EDGE 967 965 -1  EDGE 967 974 -1\n",
            "protocols.grafting.util: Loop: 1\n",
            "protocols.grafting.util: Add variant to: 955\n",
            "protocols.grafting.util: Loop: 2\n",
            "protocols.grafting.util: Add variant to: 964\n",
            "protocols.grafting.util: idealized  954\n",
            "protocols.grafting.util: idealized  955\n",
            "protocols.grafting.util: ideal 956\n",
            "protocols.grafting.util: ideal 964\n",
            "protocols.grafting.util: idealized 965\n",
            "protocols.grafting.util: idealized 966\n",
            "protocols.grafting.CCDEndsGraftMover: start 2168.65\n",
            "protocols.grafting.CCDEndsGraftMover: round 1\n",
            "protocols.loops.util: PepBondGeom: 138 L  C-N --  1.51363 max: 1.5\n",
            "protocols.grafting.CCDEndsGraftMover: 1 1755.41\n",
            "protocols.grafting.CCDEndsGraftMover: round 2\n",
            "protocols.grafting.CCDEndsGraftMover: Graft Closed early - returning\n",
            "protocols.grafting.CCDEndsGraftMover: 2 1736.71\n",
            "protocols.grafting.CCDEndsGraftMover: finish 1736.71\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 977 rotamers at 36 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.grafting.CCDEndsGraftMover: Graft meets ideal geometry true\n",
            "protocols.grafting.CCDEndsGraftMover: Complete\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L3-9-1 data: 36 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/L3-9-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 18 constraints\n",
            "protocols.antibody.design.AntibodyDesignMover: Checking Geometry\n",
            "protocols.antibody.design.AntibodyDesignMover: DATA GRAFT_CLOSURE 1 rabd/my_ab.pdb L3 L3-9-1 3c08L_L3_0001\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H1\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H2\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H3\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L1\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L2\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L3\n",
            "protocols.antibody.clusters.CDRClusterSet: Adding cacheable cluster data to pose\n",
            "protocols.antibody.design.util: Setting CDR sequence in NativeAntibodySeq\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: H1-13-1 data: 583 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/H1-13-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 26 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: H2-10-1 data: 328 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/H2-10-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 20 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: Adding general dihedral constraints for H3\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L1-10-1 data: 42 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/L1-10-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 20 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L2-8-1 data: 632 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/L2-8-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 16 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L3-9-1 data: 36 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/L3-9-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 18 constraints\n",
            "protocols.antibody.design.AntibodyDesignMover: Beginning MinProtocol\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using explicit constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using interface constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Making an interface set with fixed chains\n",
            "protocols.analysis.InterfaceAnalyzerMover: Interface set residues total: 101\n",
            "protocols.analysis.InterfaceAnalyzerMover: NOT detecting disulfides in the separated pose.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 2917 rotamers at 101 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.analysis.InterfaceAnalyzerMover: included_nres: 974\n",
            "protocols.antibody.design.AntibodyDesignMover: Inner round: 1\n",
            "protocols.antibody.design.AntibodyDesignMover: Add min neighbors : L1\n",
            "protocols.antibody.design.AntibodyDesignMover: Add min neighbors : H3\n",
            "basic.io.database: [ WARNING ] Unable to locate database file /sampling/antibodies/antibody_database_rosetta_design.db\n",
            "antibody.database.AntibodyDatabaseManager: The antibody design database is not specified by the user.\n",
            "antibody.database.AntibodyDatabaseManager: Reading from: /usr/local/lib/python3.10/dist-packages/pyrosetta/database//sampling/antibodies/antibody_database_rosetta_design_north_paper.db\n",
            "antibody.database.AntibodyDatabaseManager: Loaded L1-10-1 with 15 datapoints.\n",
            "antibody.database.AntibodyDatabaseManager: Loaded L3-9-1 with 14 datapoints.\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Data pre-loaded.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: applying prob task op\n",
            "core.pack.pack_rotamers: built 3187 rotamers at 92 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating PDInteractionGraph\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs start: 12775.5\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs end: 7718.08\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using explicit constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using interface constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Making an interface set with fixed chains\n",
            "protocols.analysis.InterfaceAnalyzerMover: Interface set residues total: 101\n",
            "protocols.analysis.InterfaceAnalyzerMover: NOT detecting disulfides in the separated pose.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 2892 rotamers at 101 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.analysis.InterfaceAnalyzerMover: included_nres: 974\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 852\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 853\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 896\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 897\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 960\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 961\n",
            "antibody.design.GeneralAntibodyModeler: CDR min start: 7718.08\n",
            "antibody.design.GeneralAntibodyModeler: CDR min end: 2536.88\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using explicit constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using interface constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Making an interface set with fixed chains\n",
            "protocols.analysis.InterfaceAnalyzerMover: Interface set residues total: 100\n",
            "protocols.analysis.InterfaceAnalyzerMover: NOT detecting disulfides in the separated pose.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 2994 rotamers at 100 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.analysis.InterfaceAnalyzerMover: included_nres: 974\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: applying prob task op\n",
            "core.pack.pack_rotamers: built 3592 rotamers at 89 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating PDInteractionGraph\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs start: 2536.88\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs end: 2820.28\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using explicit constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using interface constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Making an interface set with fixed chains\n",
            "protocols.analysis.InterfaceAnalyzerMover: Interface set residues total: 99\n",
            "protocols.analysis.InterfaceAnalyzerMover: NOT detecting disulfides in the separated pose.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 2932 rotamers at 99 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.analysis.InterfaceAnalyzerMover: included_nres: 974\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 852\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 853\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 896\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 897\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 960\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 961\n",
            "antibody.design.GeneralAntibodyModeler: CDR min start: 2536.88\n",
            "antibody.design.GeneralAntibodyModeler: CDR min end: 2532.47\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using explicit constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using interface constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Making an interface set with fixed chains\n",
            "protocols.analysis.InterfaceAnalyzerMover: Interface set residues total: 100\n",
            "protocols.analysis.InterfaceAnalyzerMover: NOT detecting disulfides in the separated pose.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 2988 rotamers at 100 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.analysis.InterfaceAnalyzerMover: included_nres: 974\n",
            "protocols.antibody.design.AntibodyDesignMover: Min Cycle complete\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using explicit constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using interface constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Making an interface set with fixed chains\n",
            "protocols.analysis.InterfaceAnalyzerMover: Interface set residues total: 100\n",
            "protocols.analysis.InterfaceAnalyzerMover: NOT detecting disulfides in the separated pose.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 2988 rotamers at 100 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.analysis.InterfaceAnalyzerMover: included_nres: 974\n",
            "protocols.antibody.design.AntibodyDesignMover:\n",
            "protocols.antibody.design.AntibodyDesignMover: SCORE 5 1902.01 0\n",
            "protocols.antibody.design.AntibodyDesignMover: FINAL 5 1857.01 1\n",
            "protocols.antibody.design.AntibodyDesignMover:\n",
            "protocols.antibody.design.AntibodyDesignMover: Last Accepted Score: 1857.01\n",
            "protocols.antibody.design.AntibodyDesignMover: Lowest        Score:1857.01\n",
            "protocols.antibody.design.AntibodyDesignMover:\n",
            "protocols.antibody.design.AntibodyDesignMover: Outer round: 6\n",
            "protocols.antibody.design.AntibodyDesignMover: Grafting CDR from cluster L1-10-2 fragment 1ay1L_L1_0001\n",
            "protocols.grafting.CCDEndsGraftMover: Start: 891 End: 902 NterO: 3 CterO: 3\n",
            "protocols.grafting.util: Superimposing overhang residues\n",
            "core.chemical.AtomICoor: [ WARNING ] IcoorAtomID::atom_id(): Cannot get atom_id for POLYMER_LOWER of residue MET 1.  Returning BOGUS ID instead.\n",
            "core.conformation.Residue: [ WARNING ] missing an atom: 1  H   that depends on a nonexistent polymer connection!\n",
            "core.conformation.Residue: [ WARNING ]  --> generating it using idealized coordinates.\n",
            "protocols.grafting.util: Deleting 10 residues from 892 to 901\n",
            "core.conformation.Conformation: Found disulfide between residues 771 845\n",
            "core.conformation.Conformation: Found disulfide between residues 891 945\n",
            "protocols.grafting.util: insert_point 891\n",
            "protocols.grafting.util: insert_start 892\n",
            "protocols.grafting.util: insert_end 901\n",
            "protocols.grafting.util: insert_length 10\n",
            "protocols.grafting.util: FOLD_TREE  EDGE 1 891 -1  EDGE 891 892 1  EDGE 892 963 -1\n",
            "protocols.grafting.GraftMoverBase: Insertion complete.\n",
            "protocols.grafting.AnchoredGraftMover: Setting default movemap\n",
            "protocols.grafting.CCDEndsGraftMover: Start: 891\n",
            "protocols.grafting.CCDEndsGraftMover: Original End: 902\n",
            "protocols.grafting.CCDEndsGraftMover: End: 902\n",
            "protocols.grafting.CCDEndsGraftMover: Insert Length: 10\n",
            "protocols.grafting.CCDEndsGraftMover: LOOP start: 890  stop: 893  cut: 891  size: 4  skip rate: 0  extended?: False\n",
            "protocols.grafting.CCDEndsGraftMover:\n",
            "protocols.grafting.CCDEndsGraftMover: LOOP start: 900  stop: 903  cut: 901  size: 4  skip rate: 0  extended?: False\n",
            "protocols.grafting.CCDEndsGraftMover:\n",
            "protocols.loops.FoldTreeFromLoopsWrapper: old foldtree FOLD_TREE  EDGE 1 891 -1  EDGE 891 902 1  EDGE 902 973 -1  EDGE 891 892 -2  C    N    EDGE 892 893 -2  C    N    EDGE 893 894 -2  C    N    EDGE 894 895 -2  C    N    EDGE 895 896 -2  C    N    EDGE 896 897 -2  C    N    EDGE 897 898 -2  C    N    EDGE 898 899 -2  C    N    EDGE 899 900 -2  C    N    EDGE 900 901 -2  C    N   \n",
            "New foldtree FOLD_TREE  EDGE 1 889 -1  EDGE 889 891 -1  EDGE 889 894 1  EDGE 894 892 -1  EDGE 894 899 -1  EDGE 899 901 -1  EDGE 899 904 2  EDGE 904 902 -1  EDGE 904 973 -1\n",
            "protocols.grafting.util: Loop: 1\n",
            "protocols.grafting.util: Add variant to: 891\n",
            "protocols.grafting.util: Loop: 2\n",
            "protocols.grafting.util: Add variant to: 901\n",
            "protocols.grafting.util: idealized  890\n",
            "protocols.grafting.util: idealized  891\n",
            "protocols.grafting.util: ideal 892\n",
            "protocols.grafting.util: ideal 901\n",
            "protocols.grafting.util: idealized 902\n",
            "protocols.grafting.util: idealized 903\n",
            "protocols.grafting.CCDEndsGraftMover: start 1617.03\n",
            "protocols.grafting.CCDEndsGraftMover: round 1\n",
            "protocols.loops.util: PepBondGeom: 901 - 42 L  Ca-C-N --  99.5 (min): 156.072 : 134.5 (max)\n",
            "protocols.grafting.CCDEndsGraftMover: 1 1380.3\n",
            "protocols.grafting.CCDEndsGraftMover: round 2\n",
            "protocols.loops.util: PepBondGeom: 42 L  C-N --  1.60027 max: 1.5\n",
            "protocols.grafting.CCDEndsGraftMover: 2 1329.07\n",
            "protocols.grafting.CCDEndsGraftMover: round 3\n",
            "protocols.grafting.CCDEndsGraftMover: Graft Closed early - returning\n",
            "protocols.grafting.CCDEndsGraftMover: 3 1261\n",
            "protocols.grafting.CCDEndsGraftMover: finish 1261\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 1471 rotamers at 43 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.grafting.CCDEndsGraftMover: Graft meets ideal geometry true\n",
            "protocols.grafting.CCDEndsGraftMover: Complete\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L1-10-2 data: 3 cutoff: 10\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: Adding general dihedral constraints for L1\n",
            "protocols.antibody.design.AntibodyDesignMover: Checking Geometry\n",
            "protocols.antibody.design.AntibodyDesignMover: DATA GRAFT_CLOSURE 1 rabd/my_ab.pdb L1 L1-10-2 1ay1L_L1_0001\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H1\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H2\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H3\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L1\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L2\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L3\n",
            "protocols.antibody.clusters.CDRClusterSet: Adding cacheable cluster data to pose\n",
            "protocols.antibody.design.util: Setting CDR sequence in NativeAntibodySeq\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: H1-13-1 data: 583 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/H1-13-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 26 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: H2-10-1 data: 328 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/H2-10-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 20 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: Adding general dihedral constraints for H3\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L1-10-2 data: 3 cutoff: 10\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: Adding general dihedral constraints for L1\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L2-8-1 data: 632 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/L2-8-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 16 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L3-8-cis6-1 data: 4 cutoff: 10\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: Adding general dihedral constraints for L3\n",
            "protocols.antibody.design.AntibodyDesignMover: Beginning MinProtocol\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using explicit constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using interface constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Making an interface set with fixed chains\n",
            "protocols.analysis.InterfaceAnalyzerMover: Interface set residues total: 94\n",
            "protocols.analysis.InterfaceAnalyzerMover: NOT detecting disulfides in the separated pose.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 2554 rotamers at 94 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.analysis.InterfaceAnalyzerMover: included_nres: 973\n",
            "protocols.antibody.design.AntibodyDesignMover: Inner round: 1\n",
            "protocols.antibody.design.AntibodyDesignMover: Add min neighbors : L2\n",
            "protocols.antibody.design.AntibodyDesignMover: Add min neighbors : L3\n",
            "basic.io.database: [ WARNING ] Unable to locate database file /sampling/antibodies/antibody_database_rosetta_design.db\n",
            "antibody.database.AntibodyDatabaseManager: The antibody design database is not specified by the user.\n",
            "antibody.database.AntibodyDatabaseManager: Reading from: /usr/local/lib/python3.10/dist-packages/pyrosetta/database//sampling/antibodies/antibody_database_rosetta_design_north_paper.db\n",
            "antibody.database.AntibodyDatabaseManager: Skipped L1-10-2 with 2 data point\n",
            "antibody.database.AntibodyDatabaseManager: Skipped L3-8-cis6-1 with 3 data point\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Data pre-loaded.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: applying prob task op\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Using conservative op for L1\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Using conservative op for L3\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Using original bb sequence for conservative design.\n",
            "protocols.task_operations.ConservativeDesignOperation: Loading conservative mutational data\n",
            "core.pack.pack_rotamers: built 6176 rotamers at 76 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating PDInteractionGraph\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs start: 4502.15\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs end: 3813.42\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using explicit constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using interface constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Making an interface set with fixed chains\n",
            "protocols.analysis.InterfaceAnalyzerMover: Interface set residues total: 92\n",
            "protocols.analysis.InterfaceAnalyzerMover: NOT detecting disulfides in the separated pose.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 2408 rotamers at 92 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.analysis.InterfaceAnalyzerMover: included_nres: 973\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 896\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 897\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 919\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 920\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 959\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 960\n",
            "antibody.design.GeneralAntibodyModeler: CDR min start: 3813.42\n",
            "antibody.design.GeneralAntibodyModeler: CDR min end: 3408.47\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using explicit constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using interface constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Making an interface set with fixed chains\n",
            "protocols.analysis.InterfaceAnalyzerMover: Interface set residues total: 92\n",
            "protocols.analysis.InterfaceAnalyzerMover: NOT detecting disulfides in the separated pose.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 2532 rotamers at 92 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.analysis.InterfaceAnalyzerMover: included_nres: 973\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: applying prob task op\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Using conservative op for L1\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Using conservative op for L3\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Using original bb sequence for conservative design.\n",
            "core.pack.pack_rotamers: built 5986 rotamers at 71 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating PDInteractionGraph\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs start: 3408.47\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs end: 3474.41\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using explicit constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using interface constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Making an interface set with fixed chains\n",
            "protocols.analysis.InterfaceAnalyzerMover: Interface set residues total: 94\n",
            "protocols.analysis.InterfaceAnalyzerMover: NOT detecting disulfides in the separated pose.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 2357 rotamers at 94 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.analysis.InterfaceAnalyzerMover: included_nres: 973\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 896\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 897\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 919\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 920\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 959\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 960\n",
            "antibody.design.GeneralAntibodyModeler: CDR min start: 3408.47\n",
            "antibody.design.GeneralAntibodyModeler: CDR min end: 3406.9\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using explicit constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using interface constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Making an interface set with fixed chains\n",
            "protocols.analysis.InterfaceAnalyzerMover: Interface set residues total: 93\n",
            "protocols.analysis.InterfaceAnalyzerMover: NOT detecting disulfides in the separated pose.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 2536 rotamers at 93 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.analysis.InterfaceAnalyzerMover: included_nres: 973\n",
            "protocols.antibody.design.AntibodyDesignMover: Min Cycle complete\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using explicit constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using interface constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Making an interface set with fixed chains\n",
            "protocols.analysis.InterfaceAnalyzerMover: Interface set residues total: 93\n",
            "protocols.analysis.InterfaceAnalyzerMover: NOT detecting disulfides in the separated pose.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 2536 rotamers at 93 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.analysis.InterfaceAnalyzerMover: included_nres: 973\n",
            "protocols.antibody.design.AntibodyDesignMover:\n",
            "protocols.antibody.design.AntibodyDesignMover: SCORE 6 2791.99 0\n",
            "protocols.antibody.design.AntibodyDesignMover: FINAL 6 1857.01 1\n",
            "protocols.antibody.design.AntibodyDesignMover:\n",
            "protocols.antibody.design.AntibodyDesignMover: Last Accepted Score: 1857.01\n",
            "protocols.antibody.design.AntibodyDesignMover: Lowest        Score:1857.01\n",
            "protocols.antibody.design.AntibodyDesignMover:\n",
            "protocols.antibody.design.AntibodyDesignMover: Outer round: 7\n",
            "protocols.antibody.design.AntibodyDesignMover: Grafting CDR from cluster L3-10-cis7,8-1 fragment 1jguL_L3_0001\n",
            "protocols.grafting.CCDEndsGraftMover: Start: 955 End: 964 NterO: 3 CterO: 3\n",
            "protocols.grafting.util: Superimposing overhang residues\n",
            "core.chemical.AtomICoor: [ WARNING ] IcoorAtomID::atom_id(): Cannot get atom_id for POLYMER_LOWER of residue TYR 1.  Returning BOGUS ID instead.\n",
            "core.conformation.Residue: [ WARNING ] missing an atom: 1  H   that depends on a nonexistent polymer connection!\n",
            "core.conformation.Residue: [ WARNING ]  --> generating it using idealized coordinates.\n",
            "protocols.grafting.util: Deleting 8 residues from 956 to 963\n",
            "core.conformation.Conformation: Found disulfide between residues 771 845\n",
            "core.conformation.Conformation: Found disulfide between residues 891 955\n",
            "protocols.grafting.util: insert_point 955\n",
            "protocols.grafting.util: insert_start 956\n",
            "protocols.grafting.util: insert_end 965\n",
            "protocols.grafting.util: insert_length 10\n",
            "protocols.grafting.util: FOLD_TREE  EDGE 1 955 -1  EDGE 955 956 1  EDGE 956 965 -1\n",
            "protocols.grafting.GraftMoverBase: Insertion complete.\n",
            "protocols.grafting.AnchoredGraftMover: Setting default movemap\n",
            "protocols.grafting.CCDEndsGraftMover: Start: 955\n",
            "protocols.grafting.CCDEndsGraftMover: Original End: 964\n",
            "protocols.grafting.CCDEndsGraftMover: End: 966\n",
            "protocols.grafting.CCDEndsGraftMover: Insert Length: 10\n",
            "protocols.grafting.CCDEndsGraftMover: LOOP start: 954  stop: 957  cut: 955  size: 4  skip rate: 0  extended?: False\n",
            "protocols.grafting.CCDEndsGraftMover:\n",
            "protocols.grafting.CCDEndsGraftMover: LOOP start: 964  stop: 967  cut: 965  size: 4  skip rate: 0  extended?: False\n",
            "protocols.grafting.CCDEndsGraftMover:\n",
            "protocols.loops.FoldTreeFromLoopsWrapper: old foldtree FOLD_TREE  EDGE 1 955 -1  EDGE 955 966 1  EDGE 966 975 -1  EDGE 955 956 -2  C    N    EDGE 956 957 -2  C    N    EDGE 957 958 -2  C    N    EDGE 958 959 -2  C    N    EDGE 959 960 -2  C    N    EDGE 960 961 -2  C    N    EDGE 961 962 -2  C    N    EDGE 962 963 -2  C    N    EDGE 963 964 -2  C    N    EDGE 964 965 -2  C    N   \n",
            "New foldtree FOLD_TREE  EDGE 1 953 -1  EDGE 953 955 -1  EDGE 953 958 1  EDGE 958 956 -1  EDGE 958 963 -1  EDGE 963 965 -1  EDGE 963 968 2  EDGE 968 966 -1  EDGE 968 975 -1\n",
            "protocols.grafting.util: Loop: 1\n",
            "protocols.grafting.util: Add variant to: 955\n",
            "protocols.grafting.util: Loop: 2\n",
            "protocols.grafting.util: Add variant to: 965\n",
            "protocols.grafting.util: idealized  954\n",
            "protocols.grafting.util: idealized  955\n",
            "protocols.grafting.util: ideal 956\n",
            "protocols.grafting.util: ideal 965\n",
            "protocols.grafting.util: idealized 966\n",
            "protocols.grafting.util: idealized 967\n",
            "protocols.grafting.CCDEndsGraftMover: start 1639.24\n",
            "protocols.grafting.CCDEndsGraftMover: round 1\n",
            "protocols.grafting.CCDEndsGraftMover: Graft Closed early - returning\n",
            "protocols.grafting.CCDEndsGraftMover: 1 1205.06\n",
            "protocols.grafting.CCDEndsGraftMover: finish 1205.06\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 1086 rotamers at 38 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.grafting.CCDEndsGraftMover: Graft meets ideal geometry true\n",
            "protocols.grafting.CCDEndsGraftMover: Complete\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L3-10-cis7,8-1 data: 5 cutoff: 10\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: Adding general dihedral constraints for L3\n",
            "protocols.antibody.design.AntibodyDesignMover: Checking Geometry\n",
            "protocols.antibody.design.AntibodyDesignMover: DATA GRAFT_CLOSURE 1 rabd/my_ab.pdb L3 L3-10-cis7,8-1 1jguL_L3_0001\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H1\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H2\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H3\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L1\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L2\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L3\n",
            "protocols.antibody.clusters.CDRClusterSet: Adding cacheable cluster data to pose\n",
            "protocols.antibody.design.util: Setting CDR sequence in NativeAntibodySeq\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: H1-13-1 data: 583 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/H1-13-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 26 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: H2-10-1 data: 328 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/H2-10-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 20 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: Adding general dihedral constraints for H3\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L1-10-1 data: 42 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/L1-10-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 20 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L2-8-1 data: 632 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/L2-8-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 16 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L3-10-cis7,8-1 data: 5 cutoff: 10\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: Adding general dihedral constraints for L3\n",
            "protocols.antibody.design.AntibodyDesignMover: Beginning MinProtocol\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using explicit constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using interface constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Making an interface set with fixed chains\n",
            "protocols.analysis.InterfaceAnalyzerMover: Interface set residues total: 99\n",
            "protocols.analysis.InterfaceAnalyzerMover: NOT detecting disulfides in the separated pose.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 2917 rotamers at 99 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.analysis.InterfaceAnalyzerMover: included_nres: 975\n",
            "protocols.antibody.design.AntibodyDesignMover: Inner round: 1\n",
            "protocols.antibody.design.AntibodyDesignMover: Add min neighbors : L1\n",
            "protocols.antibody.design.AntibodyDesignMover: Add min neighbors : H3\n",
            "basic.io.database: [ WARNING ] Unable to locate database file /sampling/antibodies/antibody_database_rosetta_design.db\n",
            "antibody.database.AntibodyDatabaseManager: The antibody design database is not specified by the user.\n",
            "antibody.database.AntibodyDatabaseManager: Reading from: /usr/local/lib/python3.10/dist-packages/pyrosetta/database//sampling/antibodies/antibody_database_rosetta_design_north_paper.db\n",
            "antibody.database.AntibodyDatabaseManager: Loaded L1-10-1 with 15 datapoints.\n",
            "antibody.database.AntibodyDatabaseManager: Skipped L3-10-cis7,8-1 with 1 data point\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Data pre-loaded.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: applying prob task op\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Using conservative op for L3\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Using original bb sequence for conservative design.\n",
            "protocols.task_operations.ConservativeDesignOperation: Loading conservative mutational data\n",
            "core.pack.pack_rotamers: built 4686 rotamers at 92 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating PDInteractionGraph\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs start: 7578.74\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs end: 5101.3\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using explicit constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using interface constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Making an interface set with fixed chains\n",
            "protocols.analysis.InterfaceAnalyzerMover: Interface set residues total: 99\n",
            "protocols.analysis.InterfaceAnalyzerMover: NOT detecting disulfides in the separated pose.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 2699 rotamers at 99 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.analysis.InterfaceAnalyzerMover: included_nres: 975\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 852\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 853\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 896\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 897\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 960\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 961\n",
            "antibody.design.GeneralAntibodyModeler: CDR min start: 5101.3\n",
            "antibody.design.GeneralAntibodyModeler: CDR min end: 3427.01\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using explicit constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using interface constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Making an interface set with fixed chains\n",
            "protocols.analysis.InterfaceAnalyzerMover: Interface set residues total: 100\n",
            "protocols.analysis.InterfaceAnalyzerMover: NOT detecting disulfides in the separated pose.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 2909 rotamers at 100 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.analysis.InterfaceAnalyzerMover: included_nres: 975\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: applying prob task op\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Using conservative op for L3\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Using original bb sequence for conservative design.\n",
            "core.pack.pack_rotamers: built 4425 rotamers at 92 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating PDInteractionGraph\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs start: 3427.01\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs end: 4079.58\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using explicit constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using interface constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Making an interface set with fixed chains\n",
            "protocols.analysis.InterfaceAnalyzerMover: Interface set residues total: 100\n",
            "protocols.analysis.InterfaceAnalyzerMover: NOT detecting disulfides in the separated pose.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 2938 rotamers at 100 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.analysis.InterfaceAnalyzerMover: included_nres: 975\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 852\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 853\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 896\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 897\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 960\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 961\n",
            "antibody.design.GeneralAntibodyModeler: CDR min start: 3427.01\n",
            "antibody.design.GeneralAntibodyModeler: CDR min end: 3425.95\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using explicit constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using interface constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Making an interface set with fixed chains\n",
            "protocols.analysis.InterfaceAnalyzerMover: Interface set residues total: 100\n",
            "protocols.analysis.InterfaceAnalyzerMover: NOT detecting disulfides in the separated pose.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 2902 rotamers at 100 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.analysis.InterfaceAnalyzerMover: included_nres: 975\n",
            "protocols.antibody.design.AntibodyDesignMover: Min Cycle complete\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using explicit constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using interface constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Making an interface set with fixed chains\n",
            "protocols.analysis.InterfaceAnalyzerMover: Interface set residues total: 100\n",
            "protocols.analysis.InterfaceAnalyzerMover: NOT detecting disulfides in the separated pose.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 2909 rotamers at 100 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.analysis.InterfaceAnalyzerMover: included_nres: 975\n",
            "protocols.antibody.design.AntibodyDesignMover:\n",
            "protocols.antibody.design.AntibodyDesignMover: SCORE 7 2864.74 0\n",
            "protocols.antibody.design.AntibodyDesignMover: FINAL 7 1857.01 1\n",
            "protocols.antibody.design.AntibodyDesignMover:\n",
            "protocols.antibody.design.AntibodyDesignMover: Last Accepted Score: 1857.01\n",
            "protocols.antibody.design.AntibodyDesignMover: Lowest        Score:1857.01\n",
            "protocols.antibody.design.AntibodyDesignMover:\n",
            "protocols.antibody.design.AntibodyDesignMover: Outer round: 8\n",
            "protocols.antibody.design.AntibodyDesignMover: Grafting CDR from cluster L3-8-cis6-1 fragment 1tzhA_L3_0001\n",
            "core.scoring.ScoreFunctionFactory: SCOREFUNCTION: ref2015\n",
            "core.scoring.ScoreFunctionFactory: SCOREFUNCTION: ref2015\n",
            "protocols.grafting.CCDEndsGraftMover: Start: 955 End: 964 NterO: 3 CterO: 3\n",
            "protocols.grafting.util: Superimposing overhang residues\n",
            "core.chemical.AtomICoor: [ WARNING ] IcoorAtomID::atom_id(): Cannot get atom_id for POLYMER_LOWER of residue TYR 1.  Returning BOGUS ID instead.\n",
            "core.conformation.Residue: [ WARNING ] missing an atom: 1  H   that depends on a nonexistent polymer connection!\n",
            "core.conformation.Residue: [ WARNING ]  --> generating it using idealized coordinates.\n",
            "protocols.grafting.util: Deleting 8 residues from 956 to 963\n",
            "core.conformation.Conformation: Found disulfide between residues 771 845\n",
            "core.conformation.Conformation: Found disulfide between residues 891 955\n",
            "protocols.grafting.util: insert_point 955\n",
            "protocols.grafting.util: insert_start 956\n",
            "protocols.grafting.util: insert_end 963\n",
            "protocols.grafting.util: insert_length 8\n",
            "protocols.grafting.util: FOLD_TREE  EDGE 1 955 -1  EDGE 955 956 1  EDGE 956 965 -1\n",
            "protocols.grafting.GraftMoverBase: Insertion complete.\n",
            "protocols.grafting.AnchoredGraftMover: Setting default movemap\n",
            "protocols.grafting.CCDEndsGraftMover: Start: 955\n",
            "protocols.grafting.CCDEndsGraftMover: Original End: 964\n",
            "protocols.grafting.CCDEndsGraftMover: End: 964\n",
            "protocols.grafting.CCDEndsGraftMover: Insert Length: 8\n",
            "protocols.grafting.CCDEndsGraftMover: LOOP start: 954  stop: 957  cut: 955  size: 4  skip rate: 0  extended?: False\n",
            "protocols.grafting.CCDEndsGraftMover:\n",
            "protocols.grafting.CCDEndsGraftMover: LOOP start: 962  stop: 965  cut: 963  size: 4  skip rate: 0  extended?: False\n",
            "protocols.grafting.CCDEndsGraftMover:\n",
            "protocols.loops.FoldTreeFromLoopsWrapper: old foldtree FOLD_TREE  EDGE 1 955 -1  EDGE 955 964 1  EDGE 964 973 -1  EDGE 955 956 -2  C    N    EDGE 956 957 -2  C    N    EDGE 957 958 -2  C    N    EDGE 958 959 -2  C    N    EDGE 959 960 -2  C    N    EDGE 960 961 -2  C    N    EDGE 961 962 -2  C    N    EDGE 962 963 -2  C    N   \n",
            "New foldtree FOLD_TREE  EDGE 1 953 -1  EDGE 953 955 -1  EDGE 953 958 1  EDGE 958 956 -1  EDGE 958 961 -1  EDGE 961 963 -1  EDGE 961 966 2  EDGE 966 964 -1  EDGE 966 973 -1\n",
            "protocols.grafting.util: Loop: 1\n",
            "protocols.grafting.util: Add variant to: 955\n",
            "protocols.grafting.util: Loop: 2\n",
            "protocols.grafting.util: Add variant to: 963\n",
            "protocols.grafting.util: idealized  954\n",
            "protocols.grafting.util: idealized  955\n",
            "protocols.grafting.util: ideal 956\n",
            "protocols.grafting.util: ideal 963\n",
            "protocols.grafting.util: idealized 964\n",
            "protocols.grafting.util: idealized 965\n",
            "protocols.grafting.CCDEndsGraftMover: start 1609.64\n",
            "protocols.grafting.CCDEndsGraftMover: round 1\n",
            "protocols.grafting.CCDEndsGraftMover: Graft Closed early - returning\n",
            "protocols.grafting.CCDEndsGraftMover: 1 1261.85\n",
            "protocols.grafting.CCDEndsGraftMover: finish 1261.85\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 977 rotamers at 36 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.grafting.CCDEndsGraftMover: Graft meets ideal geometry true\n",
            "protocols.grafting.CCDEndsGraftMover: Complete\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L3-8-cis6-1 data: 4 cutoff: 10\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: Adding general dihedral constraints for L3\n",
            "protocols.antibody.design.AntibodyDesignMover: Checking Geometry\n",
            "protocols.antibody.design.AntibodyDesignMover: DATA GRAFT_CLOSURE 1 rabd/my_ab.pdb L3 L3-8-cis6-1 1tzhA_L3_0001\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H1\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H2\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H3\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L1\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L2\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L3\n",
            "protocols.antibody.clusters.CDRClusterSet: Adding cacheable cluster data to pose\n",
            "protocols.antibody.design.util: Setting CDR sequence in NativeAntibodySeq\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: H1-13-1 data: 583 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/H1-13-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 26 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: H2-10-1 data: 328 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/H2-10-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 20 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: Adding general dihedral constraints for H3\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L1-10-1 data: 42 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/L1-10-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 20 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L2-8-1 data: 632 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/L2-8-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 16 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L3-8-cis6-1 data: 4 cutoff: 10\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: Adding general dihedral constraints for L3\n",
            "protocols.antibody.design.AntibodyDesignMover: Beginning MinProtocol\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using explicit constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using interface constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Making an interface set with fixed chains\n",
            "protocols.analysis.InterfaceAnalyzerMover: Interface set residues total: 98\n",
            "protocols.analysis.InterfaceAnalyzerMover: NOT detecting disulfides in the separated pose.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 2732 rotamers at 98 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.analysis.InterfaceAnalyzerMover: included_nres: 973\n",
            "protocols.antibody.design.AntibodyDesignMover: Inner round: 1\n",
            "protocols.antibody.design.AntibodyDesignMover: Add min neighbors : L1\n",
            "protocols.antibody.design.AntibodyDesignMover: Add min neighbors : H3\n",
            "basic.io.database: [ WARNING ] Unable to locate database file /sampling/antibodies/antibody_database_rosetta_design.db\n",
            "antibody.database.AntibodyDatabaseManager: The antibody design database is not specified by the user.\n",
            "antibody.database.AntibodyDatabaseManager: Reading from: /usr/local/lib/python3.10/dist-packages/pyrosetta/database//sampling/antibodies/antibody_database_rosetta_design_north_paper.db\n",
            "antibody.database.AntibodyDatabaseManager: Loaded L1-10-1 with 15 datapoints.\n",
            "antibody.database.AntibodyDatabaseManager: Skipped L3-8-cis6-1 with 3 data point\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Data pre-loaded.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: applying prob task op\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Using conservative op for L3\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Using original bb sequence for conservative design.\n",
            "protocols.task_operations.ConservativeDesignOperation: Loading conservative mutational data\n",
            "core.pack.pack_rotamers: built 4062 rotamers at 88 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating PDInteractionGraph\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs start: 3477.78\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs end: 3532.05\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using explicit constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using interface constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Making an interface set with fixed chains\n",
            "protocols.analysis.InterfaceAnalyzerMover: Interface set residues total: 95\n",
            "protocols.analysis.InterfaceAnalyzerMover: NOT detecting disulfides in the separated pose.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 2460 rotamers at 95 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.analysis.InterfaceAnalyzerMover: included_nres: 973\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 852\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 853\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 896\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 897\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 959\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 960\n",
            "antibody.design.GeneralAntibodyModeler: CDR min start: 3532.05\n",
            "antibody.design.GeneralAntibodyModeler: CDR min end: 2436.43\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using explicit constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using interface constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Making an interface set with fixed chains\n",
            "protocols.analysis.InterfaceAnalyzerMover: Interface set residues total: 96\n",
            "protocols.analysis.InterfaceAnalyzerMover: NOT detecting disulfides in the separated pose.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 2576 rotamers at 96 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.analysis.InterfaceAnalyzerMover: included_nres: 973\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: applying prob task op\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Using conservative op for L3\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Using original bb sequence for conservative design.\n",
            "core.pack.pack_rotamers: built 4019 rotamers at 88 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating PDInteractionGraph\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs start: 2436.43\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs end: 2674.44\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using explicit constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using interface constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Making an interface set with fixed chains\n",
            "protocols.analysis.InterfaceAnalyzerMover: Interface set residues total: 95\n",
            "protocols.analysis.InterfaceAnalyzerMover: NOT detecting disulfides in the separated pose.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 2559 rotamers at 95 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.analysis.InterfaceAnalyzerMover: included_nres: 973\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 852\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 853\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 896\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 897\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 959\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 960\n",
            "antibody.design.GeneralAntibodyModeler: CDR min start: 2436.43\n",
            "antibody.design.GeneralAntibodyModeler: CDR min end: 2432.95\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using explicit constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using interface constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Making an interface set with fixed chains\n",
            "protocols.analysis.InterfaceAnalyzerMover: Interface set residues total: 96\n",
            "protocols.analysis.InterfaceAnalyzerMover: NOT detecting disulfides in the separated pose.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 2579 rotamers at 96 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.analysis.InterfaceAnalyzerMover: included_nres: 973\n",
            "protocols.antibody.design.AntibodyDesignMover: Min Cycle complete\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using explicit constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using interface constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Making an interface set with fixed chains\n",
            "protocols.analysis.InterfaceAnalyzerMover: Interface set residues total: 96\n",
            "protocols.analysis.InterfaceAnalyzerMover: NOT detecting disulfides in the separated pose.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 2579 rotamers at 96 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.analysis.InterfaceAnalyzerMover: included_nres: 973\n",
            "protocols.antibody.design.AntibodyDesignMover:\n",
            "protocols.antibody.design.AntibodyDesignMover: SCORE 8 1841.12 1\n",
            "protocols.antibody.design.AntibodyDesignMover: FINAL 8 1841.12 1\n",
            "protocols.antibody.design.AntibodyDesignMover:\n",
            "protocols.antibody.design.AntibodyDesignMover: Last Accepted Score: 1841.12\n",
            "protocols.antibody.design.AntibodyDesignMover: Lowest        Score:1841.12\n",
            "protocols.antibody.design.AntibodyDesignMover:\n",
            "protocols.antibody.design.AntibodyDesignMover: Outer round: 9\n",
            "protocols.antibody.design.AntibodyDesignMover: Grafting CDR from cluster L3-8-cis6-1 fragment 1e6oL_L3_0001\n",
            "protocols.grafting.CCDEndsGraftMover: Start: 955 End: 964 NterO: 3 CterO: 3\n",
            "protocols.grafting.util: Superimposing overhang residues\n",
            "core.chemical.AtomICoor: [ WARNING ] IcoorAtomID::atom_id(): Cannot get atom_id for POLYMER_LOWER of residue TYR 1.  Returning BOGUS ID instead.\n",
            "core.conformation.Residue: [ WARNING ] missing an atom: 1  H   that depends on a nonexistent polymer connection!\n",
            "core.conformation.Residue: [ WARNING ]  --> generating it using idealized coordinates.\n",
            "protocols.grafting.util: Deleting 8 residues from 956 to 963\n",
            "core.conformation.Conformation: Found disulfide between residues 771 845\n",
            "core.conformation.Conformation: Found disulfide between residues 891 955\n",
            "protocols.grafting.util: insert_point 955\n",
            "protocols.grafting.util: insert_start 956\n",
            "protocols.grafting.util: insert_end 963\n",
            "protocols.grafting.util: insert_length 8\n",
            "protocols.grafting.util: FOLD_TREE  EDGE 1 955 -1  EDGE 955 956 1  EDGE 956 965 -1\n",
            "protocols.grafting.GraftMoverBase: Insertion complete.\n",
            "protocols.grafting.AnchoredGraftMover: Setting default movemap\n",
            "protocols.grafting.CCDEndsGraftMover: Start: 955\n",
            "protocols.grafting.CCDEndsGraftMover: Original End: 964\n",
            "protocols.grafting.CCDEndsGraftMover: End: 964\n",
            "protocols.grafting.CCDEndsGraftMover: Insert Length: 8\n",
            "protocols.grafting.CCDEndsGraftMover: LOOP start: 954  stop: 957  cut: 955  size: 4  skip rate: 0  extended?: False\n",
            "protocols.grafting.CCDEndsGraftMover:\n",
            "protocols.grafting.CCDEndsGraftMover: LOOP start: 962  stop: 965  cut: 963  size: 4  skip rate: 0  extended?: False\n",
            "protocols.grafting.CCDEndsGraftMover:\n",
            "protocols.loops.FoldTreeFromLoopsWrapper: old foldtree FOLD_TREE  EDGE 1 955 -1  EDGE 955 964 1  EDGE 964 973 -1  EDGE 955 956 -2  C    N    EDGE 956 957 -2  C    N    EDGE 957 958 -2  C    N    EDGE 958 959 -2  C    N    EDGE 959 960 -2  C    N    EDGE 960 961 -2  C    N    EDGE 961 962 -2  C    N    EDGE 962 963 -2  C    N   \n",
            "New foldtree FOLD_TREE  EDGE 1 953 -1  EDGE 953 955 -1  EDGE 953 958 1  EDGE 958 956 -1  EDGE 958 961 -1  EDGE 961 963 -1  EDGE 961 966 2  EDGE 966 964 -1  EDGE 966 973 -1\n",
            "protocols.grafting.util: Loop: 1\n",
            "protocols.grafting.util: Add variant to: 955\n",
            "protocols.grafting.util: Loop: 2\n",
            "protocols.grafting.util: Add variant to: 963\n",
            "protocols.grafting.util: idealized  954\n",
            "protocols.grafting.util: idealized  955\n",
            "protocols.grafting.util: ideal 956\n",
            "protocols.grafting.util: ideal 963\n",
            "protocols.grafting.util: idealized 964\n",
            "protocols.grafting.util: idealized 965\n",
            "protocols.grafting.CCDEndsGraftMover: start 1462.64\n",
            "protocols.grafting.CCDEndsGraftMover: round 1\n",
            "protocols.grafting.CCDEndsGraftMover: Graft Closed early - returning\n",
            "protocols.grafting.CCDEndsGraftMover: 1 1260.94\n",
            "protocols.grafting.CCDEndsGraftMover: finish 1260.94\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 895 rotamers at 34 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.grafting.CCDEndsGraftMover: Graft meets ideal geometry true\n",
            "protocols.grafting.CCDEndsGraftMover: Complete\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L3-8-cis6-1 data: 4 cutoff: 10\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: Adding general dihedral constraints for L3\n",
            "protocols.antibody.design.AntibodyDesignMover: Checking Geometry\n",
            "protocols.antibody.design.AntibodyDesignMover: DATA GRAFT_CLOSURE 1 rabd/my_ab.pdb L3 L3-8-cis6-1 1e6oL_L3_0001\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H1\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H2\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H3\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L1\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L2\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L3\n",
            "protocols.antibody.clusters.CDRClusterSet: Adding cacheable cluster data to pose\n",
            "protocols.antibody.design.util: Setting CDR sequence in NativeAntibodySeq\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: H1-13-1 data: 583 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/H1-13-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 26 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: H2-10-1 data: 328 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/H2-10-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 20 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: Adding general dihedral constraints for H3\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L1-10-1 data: 42 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/L1-10-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 20 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L2-8-1 data: 632 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/L2-8-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 16 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L3-8-cis6-1 data: 4 cutoff: 10\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: Adding general dihedral constraints for L3\n",
            "protocols.antibody.design.AntibodyDesignMover: Beginning MinProtocol\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using explicit constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using interface constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Making an interface set with fixed chains\n",
            "protocols.analysis.InterfaceAnalyzerMover: Interface set residues total: 98\n",
            "protocols.analysis.InterfaceAnalyzerMover: NOT detecting disulfides in the separated pose.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 2635 rotamers at 98 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.analysis.InterfaceAnalyzerMover: included_nres: 973\n",
            "protocols.antibody.design.AntibodyDesignMover: Inner round: 1\n",
            "protocols.antibody.design.AntibodyDesignMover: Add min neighbors : L1\n",
            "protocols.antibody.design.AntibodyDesignMover: Add min neighbors : H3\n",
            "basic.io.database: [ WARNING ] Unable to locate database file /sampling/antibodies/antibody_database_rosetta_design.db\n",
            "antibody.database.AntibodyDatabaseManager: The antibody design database is not specified by the user.\n",
            "antibody.database.AntibodyDatabaseManager: Reading from: /usr/local/lib/python3.10/dist-packages/pyrosetta/database//sampling/antibodies/antibody_database_rosetta_design_north_paper.db\n",
            "antibody.database.AntibodyDatabaseManager: Loaded L1-10-1 with 15 datapoints.\n",
            "antibody.database.AntibodyDatabaseManager: Skipped L3-8-cis6-1 with 3 data point\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Data pre-loaded.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: applying prob task op\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Using conservative op for L3\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Using original bb sequence for conservative design.\n",
            "protocols.task_operations.ConservativeDesignOperation: Loading conservative mutational data\n",
            "core.pack.pack_rotamers: built 4232 rotamers at 89 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating PDInteractionGraph\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs start: 8183.82\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs end: 3518.58\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using explicit constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using interface constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Making an interface set with fixed chains\n",
            "protocols.analysis.InterfaceAnalyzerMover: Interface set residues total: 96\n",
            "protocols.analysis.InterfaceAnalyzerMover: NOT detecting disulfides in the separated pose.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 2702 rotamers at 96 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.analysis.InterfaceAnalyzerMover: included_nres: 973\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 852\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 853\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 896\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 897\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 959\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 960\n",
            "antibody.design.GeneralAntibodyModeler: CDR min start: 3518.58\n",
            "antibody.design.GeneralAntibodyModeler: CDR min end: 2433.49\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using explicit constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using interface constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Making an interface set with fixed chains\n",
            "protocols.analysis.InterfaceAnalyzerMover: Interface set residues total: 95\n",
            "protocols.analysis.InterfaceAnalyzerMover: NOT detecting disulfides in the separated pose.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 2606 rotamers at 95 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.analysis.InterfaceAnalyzerMover: included_nres: 973\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: applying prob task op\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Using conservative op for L3\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Using original bb sequence for conservative design.\n",
            "core.pack.pack_rotamers: built 4187 rotamers at 88 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating PDInteractionGraph\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs start: 2433.49\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs end: 2594.53\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using explicit constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using interface constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Making an interface set with fixed chains\n",
            "protocols.analysis.InterfaceAnalyzerMover: Interface set residues total: 95\n",
            "protocols.analysis.InterfaceAnalyzerMover: NOT detecting disulfides in the separated pose.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 2579 rotamers at 95 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.analysis.InterfaceAnalyzerMover: included_nres: 973\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 852\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 853\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 896\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 897\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 959\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 960\n",
            "antibody.design.GeneralAntibodyModeler: CDR min start: 2433.49\n",
            "antibody.design.GeneralAntibodyModeler: CDR min end: 2431.5\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using explicit constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using interface constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Making an interface set with fixed chains\n",
            "protocols.analysis.InterfaceAnalyzerMover: Interface set residues total: 95\n",
            "protocols.analysis.InterfaceAnalyzerMover: NOT detecting disulfides in the separated pose.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 2607 rotamers at 95 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.analysis.InterfaceAnalyzerMover: included_nres: 973\n",
            "protocols.antibody.design.AntibodyDesignMover: Min Cycle complete\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using explicit constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using interface constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Making an interface set with fixed chains\n",
            "protocols.analysis.InterfaceAnalyzerMover: Interface set residues total: 95\n",
            "protocols.analysis.InterfaceAnalyzerMover: NOT detecting disulfides in the separated pose.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 2607 rotamers at 95 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.analysis.InterfaceAnalyzerMover: included_nres: 973\n",
            "protocols.antibody.design.AntibodyDesignMover:\n",
            "protocols.antibody.design.AntibodyDesignMover: SCORE 9 1809.59 1\n",
            "protocols.antibody.design.AntibodyDesignMover: FINAL 9 1809.59 1\n",
            "protocols.antibody.design.AntibodyDesignMover:\n",
            "protocols.antibody.design.AntibodyDesignMover: Last Accepted Score: 1809.59\n",
            "protocols.antibody.design.AntibodyDesignMover: Lowest        Score:1809.59\n",
            "protocols.antibody.design.AntibodyDesignMover:\n",
            "protocols.antibody.design.AntibodyDesignMover: Outer round: 10\n",
            "protocols.antibody.design.AntibodyDesignMover: Grafting CDR from cluster L1-15-1 fragment 1iqwL_L1_0001\n",
            "protocols.grafting.CCDEndsGraftMover: Start: 891 End: 902 NterO: 3 CterO: 3\n",
            "protocols.grafting.util: Superimposing overhang residues\n",
            "core.chemical.AtomICoor: [ WARNING ] IcoorAtomID::atom_id(): Cannot get atom_id for POLYMER_LOWER of residue ILE 1.  Returning BOGUS ID instead.\n",
            "core.conformation.Residue: [ WARNING ] missing an atom: 1  H   that depends on a nonexistent polymer connection!\n",
            "core.conformation.Residue: [ WARNING ]  --> generating it using idealized coordinates.\n",
            "protocols.grafting.util: Deleting 10 residues from 892 to 901\n",
            "core.conformation.Conformation: Found disulfide between residues 771 845\n",
            "core.conformation.Conformation: Found disulfide between residues 891 945\n",
            "protocols.grafting.util: insert_point 891\n",
            "protocols.grafting.util: insert_start 892\n",
            "protocols.grafting.util: insert_end 906\n",
            "protocols.grafting.util: insert_length 15\n",
            "protocols.grafting.util: FOLD_TREE  EDGE 1 891 -1  EDGE 891 892 1  EDGE 892 963 -1\n",
            "protocols.grafting.GraftMoverBase: Insertion complete.\n",
            "protocols.grafting.AnchoredGraftMover: Setting default movemap\n",
            "protocols.grafting.CCDEndsGraftMover: Start: 891\n",
            "protocols.grafting.CCDEndsGraftMover: Original End: 902\n",
            "protocols.grafting.CCDEndsGraftMover: End: 907\n",
            "protocols.grafting.CCDEndsGraftMover: Insert Length: 15\n",
            "protocols.grafting.CCDEndsGraftMover: LOOP start: 890  stop: 893  cut: 891  size: 4  skip rate: 0  extended?: False\n",
            "protocols.grafting.CCDEndsGraftMover:\n",
            "protocols.grafting.CCDEndsGraftMover: LOOP start: 905  stop: 908  cut: 906  size: 4  skip rate: 0  extended?: False\n",
            "protocols.grafting.CCDEndsGraftMover:\n",
            "protocols.loops.FoldTreeFromLoopsWrapper: old foldtree FOLD_TREE  EDGE 1 891 -1  EDGE 891 907 1  EDGE 907 978 -1  EDGE 891 892 -2  C    N    EDGE 892 893 -2  C    N    EDGE 893 894 -2  C    N    EDGE 894 895 -2  C    N    EDGE 895 896 -2  C    N    EDGE 896 897 -2  C    N    EDGE 897 898 -2  C    N    EDGE 898 899 -2  C    N    EDGE 899 900 -2  C    N    EDGE 900 901 -2  C    N    EDGE 901 902 -2  C    N    EDGE 902 903 -2  C    N    EDGE 903 904 -2  C    N    EDGE 904 905 -2  C    N    EDGE 905 906 -2  C    N   \n",
            "New foldtree FOLD_TREE  EDGE 1 889 -1  EDGE 889 891 -1  EDGE 889 894 1  EDGE 894 892 -1  EDGE 894 904 -1  EDGE 904 906 -1  EDGE 904 909 2  EDGE 909 907 -1  EDGE 909 978 -1\n",
            "protocols.grafting.util: Loop: 1\n",
            "protocols.grafting.util: Add variant to: 891\n",
            "protocols.grafting.util: Loop: 2\n",
            "protocols.grafting.util: Add variant to: 906\n",
            "protocols.grafting.util: idealized  890\n",
            "protocols.grafting.util: idealized  891\n",
            "protocols.grafting.util: ideal 892\n",
            "protocols.grafting.util: ideal 906\n",
            "protocols.grafting.util: idealized 907\n",
            "protocols.grafting.util: idealized 908\n",
            "protocols.grafting.CCDEndsGraftMover: start 1797.01\n",
            "protocols.grafting.CCDEndsGraftMover: round 1\n",
            "protocols.loops.util: PepBondGeom: 906 - 42 L  Ca-C-N --  99.5 (min): 156.337 : 134.5 (max)\n",
            "protocols.grafting.CCDEndsGraftMover: 1 1535.53\n",
            "protocols.grafting.CCDEndsGraftMover: round 2\n",
            "protocols.grafting.CCDEndsGraftMover: Graft Closed early - returning\n",
            "protocols.grafting.CCDEndsGraftMover: 2 1493.47\n",
            "protocols.grafting.CCDEndsGraftMover: finish 1493.47\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 2364 rotamers at 53 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.grafting.CCDEndsGraftMover: Graft meets ideal geometry true\n",
            "protocols.grafting.CCDEndsGraftMover: Complete\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L1-15-1 data: 31 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/L1-15-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 30 constraints\n",
            "protocols.antibody.design.AntibodyDesignMover: Checking Geometry\n",
            "protocols.antibody.design.AntibodyDesignMover: DATA GRAFT_CLOSURE 1 rabd/my_ab.pdb L1 L1-15-1 1iqwL_L1_0001\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H1\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H2\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H3\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L1\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L2\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L3\n",
            "protocols.antibody.clusters.CDRClusterSet: Adding cacheable cluster data to pose\n",
            "protocols.antibody.design.util: Setting CDR sequence in NativeAntibodySeq\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: H1-13-1 data: 583 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/H1-13-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 26 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: H2-10-1 data: 328 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/H2-10-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 20 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: Adding general dihedral constraints for H3\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L1-15-1 data: 31 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/L1-15-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 30 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L2-8-1 data: 632 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/L2-8-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 16 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L3-8-cis6-1 data: 4 cutoff: 10\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: Adding general dihedral constraints for L3\n",
            "protocols.antibody.design.AntibodyDesignMover: Beginning MinProtocol\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using explicit constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using interface constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Making an interface set with fixed chains\n",
            "protocols.analysis.InterfaceAnalyzerMover: Interface set residues total: 111\n",
            "protocols.analysis.InterfaceAnalyzerMover: NOT detecting disulfides in the separated pose.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 3039 rotamers at 111 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.analysis.InterfaceAnalyzerMover: included_nres: 978\n",
            "protocols.antibody.design.AntibodyDesignMover: Inner round: 1\n",
            "protocols.antibody.design.AntibodyDesignMover: Add min neighbors : L2\n",
            "protocols.antibody.design.AntibodyDesignMover: Add min neighbors : L3\n",
            "basic.io.database: [ WARNING ] Unable to locate database file /sampling/antibodies/antibody_database_rosetta_design.db\n",
            "antibody.database.AntibodyDatabaseManager: The antibody design database is not specified by the user.\n",
            "antibody.database.AntibodyDatabaseManager: Reading from: /usr/local/lib/python3.10/dist-packages/pyrosetta/database//sampling/antibodies/antibody_database_rosetta_design_north_paper.db\n",
            "antibody.database.AntibodyDatabaseManager: Skipped L1-15-1 with 9 data point\n",
            "antibody.database.AntibodyDatabaseManager: Skipped L3-8-cis6-1 with 3 data point\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Data pre-loaded.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: applying prob task op\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Using conservative op for L1\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Using conservative op for L3\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Using original bb sequence for conservative design.\n",
            "protocols.task_operations.ConservativeDesignOperation: Loading conservative mutational data\n",
            "core.pack.pack_rotamers: built 6046 rotamers at 83 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating PDInteractionGraph\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs start: 19946.7\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs end: 17333.6\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using explicit constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using interface constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Making an interface set with fixed chains\n",
            "protocols.analysis.InterfaceAnalyzerMover: Interface set residues total: 111\n",
            "protocols.analysis.InterfaceAnalyzerMover: NOT detecting disulfides in the separated pose.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 3142 rotamers at 111 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.analysis.InterfaceAnalyzerMover: included_nres: 978\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 899\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 900\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 924\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 925\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 964\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 965\n",
            "antibody.design.GeneralAntibodyModeler: CDR min start: 17333.6\n",
            "antibody.design.GeneralAntibodyModeler: CDR min end: 9006.48\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using explicit constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using interface constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Making an interface set with fixed chains\n",
            "protocols.analysis.InterfaceAnalyzerMover: Interface set residues total: 105\n",
            "protocols.analysis.InterfaceAnalyzerMover: NOT detecting disulfides in the separated pose.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 3027 rotamers at 105 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.analysis.InterfaceAnalyzerMover: included_nres: 978\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: applying prob task op\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Using conservative op for L1\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Using conservative op for L3\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Using original bb sequence for conservative design.\n",
            "core.pack.pack_rotamers: built 6144 rotamers at 80 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating PDInteractionGraph\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs start: 9006.48\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs end: 5233.79\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using explicit constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using interface constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Making an interface set with fixed chains\n",
            "protocols.analysis.InterfaceAnalyzerMover: Interface set residues total: 103\n",
            "protocols.analysis.InterfaceAnalyzerMover: NOT detecting disulfides in the separated pose.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 2784 rotamers at 103 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.analysis.InterfaceAnalyzerMover: included_nres: 978\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 899\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 900\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 924\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 925\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 964\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 965\n",
            "antibody.design.GeneralAntibodyModeler: CDR min start: 5233.79\n",
            "antibody.design.GeneralAntibodyModeler: CDR min end: 2485.33\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using explicit constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using interface constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Making an interface set with fixed chains\n",
            "protocols.analysis.InterfaceAnalyzerMover: Interface set residues total: 100\n",
            "protocols.analysis.InterfaceAnalyzerMover: NOT detecting disulfides in the separated pose.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 2614 rotamers at 100 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.analysis.InterfaceAnalyzerMover: included_nres: 978\n",
            "protocols.antibody.design.AntibodyDesignMover: Min Cycle complete\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using explicit constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using interface constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Making an interface set with fixed chains\n",
            "protocols.analysis.InterfaceAnalyzerMover: Interface set residues total: 100\n",
            "protocols.analysis.InterfaceAnalyzerMover: NOT detecting disulfides in the separated pose.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 2614 rotamers at 100 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.analysis.InterfaceAnalyzerMover: included_nres: 978\n",
            "protocols.antibody.design.AntibodyDesignMover:\n",
            "protocols.antibody.design.AntibodyDesignMover: SCORE 10 1839.3 0\n",
            "protocols.antibody.design.AntibodyDesignMover: FINAL 10 1809.59 1\n",
            "protocols.antibody.design.AntibodyDesignMover:\n",
            "protocols.antibody.design.AntibodyDesignMover: Last Accepted Score: 1809.59\n",
            "protocols.antibody.design.AntibodyDesignMover: Lowest        Score:1809.59\n",
            "protocols.antibody.design.AntibodyDesignMover:\n",
            "protocols.antibody.design.AntibodyDesignMover: Outer round: 11\n",
            "protocols.antibody.design.AntibodyDesignMover: Grafting CDR from cluster L1-11-1 fragment 2cmrL_L1_0001\n",
            "protocols.grafting.CCDEndsGraftMover: Start: 891 End: 902 NterO: 3 CterO: 3\n",
            "protocols.grafting.util: Superimposing overhang residues\n",
            "core.chemical.AtomICoor: [ WARNING ] IcoorAtomID::atom_id(): Cannot get atom_id for POLYMER_LOWER of residue ILE 1.  Returning BOGUS ID instead.\n",
            "core.conformation.Residue: [ WARNING ] missing an atom: 1  H   that depends on a nonexistent polymer connection!\n",
            "core.conformation.Residue: [ WARNING ]  --> generating it using idealized coordinates.\n",
            "protocols.grafting.util: Deleting 10 residues from 892 to 901\n",
            "core.conformation.Conformation: Found disulfide between residues 771 845\n",
            "core.conformation.Conformation: Found disulfide between residues 891 945\n",
            "protocols.grafting.util: insert_point 891\n",
            "protocols.grafting.util: insert_start 892\n",
            "protocols.grafting.util: insert_end 902\n",
            "protocols.grafting.util: insert_length 11\n",
            "protocols.grafting.util: FOLD_TREE  EDGE 1 891 -1  EDGE 891 892 1  EDGE 892 963 -1\n",
            "protocols.grafting.GraftMoverBase: Insertion complete.\n",
            "protocols.grafting.AnchoredGraftMover: Setting default movemap\n",
            "protocols.grafting.CCDEndsGraftMover: Start: 891\n",
            "protocols.grafting.CCDEndsGraftMover: Original End: 902\n",
            "protocols.grafting.CCDEndsGraftMover: End: 903\n",
            "protocols.grafting.CCDEndsGraftMover: Insert Length: 11\n",
            "protocols.grafting.CCDEndsGraftMover: LOOP start: 890  stop: 893  cut: 891  size: 4  skip rate: 0  extended?: False\n",
            "protocols.grafting.CCDEndsGraftMover:\n",
            "protocols.grafting.CCDEndsGraftMover: LOOP start: 901  stop: 904  cut: 902  size: 4  skip rate: 0  extended?: False\n",
            "protocols.grafting.CCDEndsGraftMover:\n",
            "protocols.loops.FoldTreeFromLoopsWrapper: old foldtree FOLD_TREE  EDGE 1 891 -1  EDGE 891 903 1  EDGE 903 974 -1  EDGE 891 892 -2  C    N    EDGE 892 893 -2  C    N    EDGE 893 894 -2  C    N    EDGE 894 895 -2  C    N    EDGE 895 896 -2  C    N    EDGE 896 897 -2  C    N    EDGE 897 898 -2  C    N    EDGE 898 899 -2  C    N    EDGE 899 900 -2  C    N    EDGE 900 901 -2  C    N    EDGE 901 902 -2  C    N   \n",
            "New foldtree FOLD_TREE  EDGE 1 889 -1  EDGE 889 891 -1  EDGE 889 894 1  EDGE 894 892 -1  EDGE 894 900 -1  EDGE 900 902 -1  EDGE 900 905 2  EDGE 905 903 -1  EDGE 905 974 -1\n",
            "protocols.grafting.util: Loop: 1\n",
            "protocols.grafting.util: Add variant to: 891\n",
            "protocols.grafting.util: Loop: 2\n",
            "protocols.grafting.util: Add variant to: 902\n",
            "protocols.grafting.util: idealized  890\n",
            "protocols.grafting.util: idealized  891\n",
            "protocols.grafting.util: ideal 892\n",
            "protocols.grafting.util: ideal 902\n",
            "protocols.grafting.util: idealized 903\n",
            "protocols.grafting.util: idealized 904\n",
            "protocols.grafting.CCDEndsGraftMover: start 1736.98\n",
            "protocols.grafting.CCDEndsGraftMover: round 1\n",
            "protocols.loops.util: PepBondGeom: 902 - 42 L  Ca-C-N --  99.5 (min): 155.291 : 134.5 (max)\n",
            "protocols.grafting.CCDEndsGraftMover: 1 1402.22\n",
            "protocols.grafting.CCDEndsGraftMover: round 2\n",
            "protocols.loops.util: PepBondGeom: 902 - 42 L  Ca-C-N --  99.5 (min): 161.097 : 134.5 (max)\n",
            "protocols.grafting.CCDEndsGraftMover: 2 1381.64\n",
            "protocols.grafting.CCDEndsGraftMover: round 3\n",
            "protocols.loops.util: PepBondGeom: 902 - 42 L  Ca-C-N --  99.5 (min): 154.44 : 134.5 (max)\n",
            "protocols.grafting.CCDEndsGraftMover: round 4\n",
            "protocols.loops.util: PepBondGeom: 902 - 42 L  Ca-C-N --  99.5 (min): 161.679 : 134.5 (max)\n",
            "protocols.grafting.CCDEndsGraftMover: round 5\n",
            "protocols.loops.util: PepBondGeom: 902 - 42 L  Ca-C-N --  99.5 (min): 160.569 : 134.5 (max)\n",
            "protocols.grafting.CCDEndsGraftMover: 5 1378.52\n",
            "protocols.grafting.CCDEndsGraftMover: round 6\n",
            "protocols.loops.util: PepBondGeom: 902 - 42 L  Ca-C-N --  99.5 (min): 160.933 : 134.5 (max)\n",
            "protocols.grafting.CCDEndsGraftMover: 6 1373.52\n",
            "protocols.grafting.CCDEndsGraftMover: round 7\n",
            "protocols.loops.util: PepBondGeom: 902 - 42 L  Ca-C-N --  99.5 (min): 162.086 : 134.5 (max)\n",
            "protocols.grafting.CCDEndsGraftMover: round 8\n",
            "protocols.loops.util: PepBondGeom: 902 - 42 L  Ca-C-N --  99.5 (min): 160.368 : 134.5 (max)\n",
            "protocols.grafting.CCDEndsGraftMover: round 9\n",
            "protocols.loops.util: PepBondGeom: 902 - 42 L  Ca-C-N --  99.5 (min): 158.739 : 134.5 (max)\n",
            "protocols.grafting.CCDEndsGraftMover: round 10\n",
            "protocols.loops.util: PepBondGeom: 902 - 42 L  Ca-C-N --  99.5 (min): 163.175 : 134.5 (max)\n",
            "protocols.grafting.CCDEndsGraftMover: round 11\n",
            "protocols.loops.util: PepBondGeom: 902 - 42 L  Ca-C-N --  99.5 (min): 153.843 : 134.5 (max)\n",
            "protocols.grafting.CCDEndsGraftMover: round 12\n",
            "protocols.loops.util: PepBondGeom: 902 - 42 L  Ca-C-N --  99.5 (min): 155.367 : 134.5 (max)\n",
            "protocols.grafting.CCDEndsGraftMover: round 13\n",
            "protocols.loops.util: PepBondGeom: 902 - 42 L  Ca-C-N --  99.5 (min): 155.568 : 134.5 (max)\n",
            "protocols.grafting.CCDEndsGraftMover: round 14\n",
            "protocols.loops.util: PepBondGeom: 902 - 42 L  Ca-C-N --  99.5 (min): 153.721 : 134.5 (max)\n",
            "protocols.grafting.CCDEndsGraftMover: round 15\n",
            "protocols.loops.util: PepBondGeom: 902 - 42 L  Ca-C-N --  99.5 (min): 157.012 : 134.5 (max)\n",
            "protocols.grafting.CCDEndsGraftMover: round 16\n",
            "protocols.loops.util: PepBondGeom: 902 - 42 L  Ca-C-N --  99.5 (min): 157.392 : 134.5 (max)\n",
            "protocols.grafting.CCDEndsGraftMover: round 17\n",
            "protocols.loops.util: PepBondGeom: 902 - 42 L  Ca-C-N --  99.5 (min): 161.754 : 134.5 (max)\n",
            "protocols.grafting.CCDEndsGraftMover: round 18\n",
            "protocols.loops.util: PepBondGeom: 902 - 42 L  Ca-C-N --  99.5 (min): 154.368 : 134.5 (max)\n",
            "protocols.grafting.CCDEndsGraftMover: round 19\n",
            "protocols.grafting.CCDEndsGraftMover: Graft Closed early - returning\n",
            "protocols.grafting.CCDEndsGraftMover: 19 1366.94\n",
            "protocols.grafting.CCDEndsGraftMover: finish 1366.94\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 2178 rotamers at 48 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.grafting.CCDEndsGraftMover: Graft meets ideal geometry true\n",
            "protocols.grafting.CCDEndsGraftMover: Complete\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L1-11-1 data: 188 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/L1-11-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 22 constraints\n",
            "protocols.antibody.design.AntibodyDesignMover: Checking Geometry\n",
            "protocols.antibody.design.AntibodyDesignMover: DATA GRAFT_CLOSURE 1 rabd/my_ab.pdb L1 L1-11-1 2cmrL_L1_0001\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H1\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H2\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H3\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L1\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L2\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L3\n",
            "protocols.antibody.clusters.CDRClusterSet: Adding cacheable cluster data to pose\n",
            "protocols.antibody.design.util: Setting CDR sequence in NativeAntibodySeq\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: H1-13-1 data: 583 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/H1-13-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 26 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: H2-10-1 data: 328 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/H2-10-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 20 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: Adding general dihedral constraints for H3\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L1-11-1 data: 188 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/L1-11-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 22 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L2-8-1 data: 632 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/L2-8-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 16 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L3-8-cis6-1 data: 4 cutoff: 10\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: Adding general dihedral constraints for L3\n",
            "protocols.antibody.design.AntibodyDesignMover: Beginning MinProtocol\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using explicit constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using interface constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Making an interface set with fixed chains\n",
            "protocols.analysis.InterfaceAnalyzerMover: Interface set residues total: 94\n",
            "protocols.analysis.InterfaceAnalyzerMover: NOT detecting disulfides in the separated pose.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 2413 rotamers at 94 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.analysis.InterfaceAnalyzerMover: included_nres: 974\n",
            "protocols.antibody.design.AntibodyDesignMover: Inner round: 1\n",
            "protocols.antibody.design.AntibodyDesignMover: Add min neighbors : L2\n",
            "protocols.antibody.design.AntibodyDesignMover: Add min neighbors : L3\n",
            "basic.io.database: [ WARNING ] Unable to locate database file /sampling/antibodies/antibody_database_rosetta_design.db\n",
            "antibody.database.AntibodyDatabaseManager: The antibody design database is not specified by the user.\n",
            "antibody.database.AntibodyDatabaseManager: Reading from: /usr/local/lib/python3.10/dist-packages/pyrosetta/database//sampling/antibodies/antibody_database_rosetta_design_north_paper.db\n",
            "antibody.database.AntibodyDatabaseManager: Loaded L1-11-1 with 55 datapoints.\n",
            "antibody.database.AntibodyDatabaseManager: Skipped L3-8-cis6-1 with 3 data point\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Data pre-loaded.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: applying prob task op\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Using conservative op for L3\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Using original bb sequence for conservative design.\n",
            "protocols.task_operations.ConservativeDesignOperation: Loading conservative mutational data\n",
            "core.pack.pack_rotamers: built 3331 rotamers at 73 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating PDInteractionGraph\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs start: 7613.14\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs end: 5623.24\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using explicit constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using interface constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Making an interface set with fixed chains\n",
            "protocols.analysis.InterfaceAnalyzerMover: Interface set residues total: 96\n",
            "protocols.analysis.InterfaceAnalyzerMover: NOT detecting disulfides in the separated pose.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 2478 rotamers at 96 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.analysis.InterfaceAnalyzerMover: included_nres: 974\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 897\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 898\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 920\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 921\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 960\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 961\n",
            "antibody.design.GeneralAntibodyModeler: CDR min start: 5623.24\n",
            "antibody.design.GeneralAntibodyModeler: CDR min end: 2493.97\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using explicit constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using interface constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Making an interface set with fixed chains\n",
            "protocols.analysis.InterfaceAnalyzerMover: Interface set residues total: 95\n",
            "protocols.analysis.InterfaceAnalyzerMover: NOT detecting disulfides in the separated pose.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 2677 rotamers at 95 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.analysis.InterfaceAnalyzerMover: included_nres: 974\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: applying prob task op\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Using conservative op for L3\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Using original bb sequence for conservative design.\n",
            "core.pack.pack_rotamers: built 4140 rotamers at 73 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating PDInteractionGraph\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs start: 2493.97\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs end: 2540.36\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using explicit constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using interface constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Making an interface set with fixed chains\n",
            "protocols.analysis.InterfaceAnalyzerMover: Interface set residues total: 95\n",
            "protocols.analysis.InterfaceAnalyzerMover: NOT detecting disulfides in the separated pose.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 2652 rotamers at 95 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.analysis.InterfaceAnalyzerMover: included_nres: 974\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 897\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 898\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 920\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 921\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 960\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 961\n",
            "antibody.design.GeneralAntibodyModeler: CDR min start: 2493.97\n",
            "antibody.design.GeneralAntibodyModeler: CDR min end: 2493.9\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using explicit constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using interface constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Making an interface set with fixed chains\n",
            "protocols.analysis.InterfaceAnalyzerMover: Interface set residues total: 95\n",
            "protocols.analysis.InterfaceAnalyzerMover: NOT detecting disulfides in the separated pose.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 2670 rotamers at 95 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.analysis.InterfaceAnalyzerMover: included_nres: 974\n",
            "protocols.antibody.design.AntibodyDesignMover: Min Cycle complete\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using explicit constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using interface constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Making an interface set with fixed chains\n",
            "protocols.analysis.InterfaceAnalyzerMover: Interface set residues total: 95\n",
            "protocols.analysis.InterfaceAnalyzerMover: NOT detecting disulfides in the separated pose.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 2677 rotamers at 95 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.analysis.InterfaceAnalyzerMover: included_nres: 974\n",
            "protocols.antibody.design.AntibodyDesignMover:\n",
            "protocols.antibody.design.AntibodyDesignMover: SCORE 11 1813.24 0\n",
            "protocols.antibody.design.AntibodyDesignMover: FINAL 11 1809.59 1\n",
            "protocols.antibody.design.AntibodyDesignMover:\n",
            "protocols.antibody.design.AntibodyDesignMover: Last Accepted Score: 1809.59\n",
            "protocols.antibody.design.AntibodyDesignMover: Lowest        Score:1809.59\n",
            "protocols.antibody.design.AntibodyDesignMover:\n",
            "protocols.antibody.design.AntibodyDesignMover: Outer round: 12\n",
            "protocols.antibody.design.AntibodyDesignMover: Grafting CDR from cluster L1-15-1 fragment 2hrpL_L1_0001\n",
            "protocols.grafting.CCDEndsGraftMover: Start: 891 End: 902 NterO: 3 CterO: 3\n",
            "protocols.grafting.util: Superimposing overhang residues\n",
            "core.chemical.AtomICoor: [ WARNING ] IcoorAtomID::atom_id(): Cannot get atom_id for POLYMER_LOWER of residue ILE 1.  Returning BOGUS ID instead.\n",
            "core.conformation.Residue: [ WARNING ] missing an atom: 1  H   that depends on a nonexistent polymer connection!\n",
            "core.conformation.Residue: [ WARNING ]  --> generating it using idealized coordinates.\n",
            "protocols.grafting.util: Deleting 10 residues from 892 to 901\n",
            "core.conformation.Conformation: Found disulfide between residues 771 845\n",
            "core.conformation.Conformation: Found disulfide between residues 891 945\n",
            "protocols.grafting.util: insert_point 891\n",
            "protocols.grafting.util: insert_start 892\n",
            "protocols.grafting.util: insert_end 906\n",
            "protocols.grafting.util: insert_length 15\n",
            "protocols.grafting.util: FOLD_TREE  EDGE 1 891 -1  EDGE 891 892 1  EDGE 892 963 -1\n",
            "protocols.grafting.GraftMoverBase: Insertion complete.\n",
            "protocols.grafting.AnchoredGraftMover: Setting default movemap\n",
            "protocols.grafting.CCDEndsGraftMover: Start: 891\n",
            "protocols.grafting.CCDEndsGraftMover: Original End: 902\n",
            "protocols.grafting.CCDEndsGraftMover: End: 907\n",
            "protocols.grafting.CCDEndsGraftMover: Insert Length: 15\n",
            "protocols.grafting.CCDEndsGraftMover: LOOP start: 890  stop: 893  cut: 891  size: 4  skip rate: 0  extended?: False\n",
            "protocols.grafting.CCDEndsGraftMover:\n",
            "protocols.grafting.CCDEndsGraftMover: LOOP start: 905  stop: 908  cut: 906  size: 4  skip rate: 0  extended?: False\n",
            "protocols.grafting.CCDEndsGraftMover:\n",
            "protocols.loops.FoldTreeFromLoopsWrapper: old foldtree FOLD_TREE  EDGE 1 891 -1  EDGE 891 907 1  EDGE 907 978 -1  EDGE 891 892 -2  C    N    EDGE 892 893 -2  C    N    EDGE 893 894 -2  C    N    EDGE 894 895 -2  C    N    EDGE 895 896 -2  C    N    EDGE 896 897 -2  C    N    EDGE 897 898 -2  C    N    EDGE 898 899 -2  C    N    EDGE 899 900 -2  C    N    EDGE 900 901 -2  C    N    EDGE 901 902 -2  C    N    EDGE 902 903 -2  C    N    EDGE 903 904 -2  C    N    EDGE 904 905 -2  C    N    EDGE 905 906 -2  C    N   \n",
            "New foldtree FOLD_TREE  EDGE 1 889 -1  EDGE 889 891 -1  EDGE 889 894 1  EDGE 894 892 -1  EDGE 894 904 -1  EDGE 904 906 -1  EDGE 904 909 2  EDGE 909 907 -1  EDGE 909 978 -1\n",
            "protocols.grafting.util: Loop: 1\n",
            "protocols.grafting.util: Add variant to: 891\n",
            "protocols.grafting.util: Loop: 2\n",
            "protocols.grafting.util: Add variant to: 906\n",
            "protocols.grafting.util: idealized  890\n",
            "protocols.grafting.util: idealized  891\n",
            "protocols.grafting.util: ideal 892\n",
            "protocols.grafting.util: ideal 906\n",
            "protocols.grafting.util: idealized 907\n",
            "protocols.grafting.util: idealized 908\n",
            "protocols.grafting.CCDEndsGraftMover: start 1814\n",
            "protocols.grafting.CCDEndsGraftMover: round 1\n",
            "protocols.grafting.CCDEndsGraftMover: Graft Closed early - returning\n",
            "protocols.grafting.CCDEndsGraftMover: 1 1427.52\n",
            "protocols.grafting.CCDEndsGraftMover: finish 1427.52\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 2364 rotamers at 53 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.grafting.CCDEndsGraftMover: Graft meets ideal geometry true\n",
            "protocols.grafting.CCDEndsGraftMover: Complete\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L1-15-1 data: 31 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/L1-15-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 30 constraints\n",
            "protocols.antibody.design.AntibodyDesignMover: Checking Geometry\n",
            "protocols.antibody.design.AntibodyDesignMover: DATA GRAFT_CLOSURE 1 rabd/my_ab.pdb L1 L1-15-1 2hrpL_L1_0001\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H1\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H2\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H3\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L1\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L2\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L3\n",
            "protocols.antibody.clusters.CDRClusterSet: Adding cacheable cluster data to pose\n",
            "protocols.antibody.design.util: Setting CDR sequence in NativeAntibodySeq\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: H1-13-1 data: 583 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/H1-13-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 26 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: H2-10-1 data: 328 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/H2-10-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 20 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: Adding general dihedral constraints for H3\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L1-15-1 data: 31 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/L1-15-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 30 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L2-8-1 data: 632 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/L2-8-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 16 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L3-8-cis6-1 data: 4 cutoff: 10\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: Adding general dihedral constraints for L3\n",
            "protocols.antibody.design.AntibodyDesignMover: Beginning MinProtocol\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using explicit constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using interface constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Making an interface set with fixed chains\n",
            "protocols.analysis.InterfaceAnalyzerMover: Interface set residues total: 110\n",
            "protocols.analysis.InterfaceAnalyzerMover: NOT detecting disulfides in the separated pose.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 2994 rotamers at 110 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.analysis.InterfaceAnalyzerMover: included_nres: 978\n",
            "protocols.antibody.design.AntibodyDesignMover: Inner round: 1\n",
            "protocols.antibody.design.AntibodyDesignMover: Add min neighbors : L2\n",
            "protocols.antibody.design.AntibodyDesignMover: Add min neighbors : L3\n",
            "basic.io.database: [ WARNING ] Unable to locate database file /sampling/antibodies/antibody_database_rosetta_design.db\n",
            "antibody.database.AntibodyDatabaseManager: The antibody design database is not specified by the user.\n",
            "antibody.database.AntibodyDatabaseManager: Reading from: /usr/local/lib/python3.10/dist-packages/pyrosetta/database//sampling/antibodies/antibody_database_rosetta_design_north_paper.db\n",
            "antibody.database.AntibodyDatabaseManager: Skipped L1-15-1 with 9 data point\n",
            "antibody.database.AntibodyDatabaseManager: Skipped L3-8-cis6-1 with 3 data point\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Data pre-loaded.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: applying prob task op\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Using conservative op for L1\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Using conservative op for L3\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Using original bb sequence for conservative design.\n",
            "protocols.task_operations.ConservativeDesignOperation: Loading conservative mutational data\n",
            "core.pack.pack_rotamers: built 5977 rotamers at 84 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating PDInteractionGraph\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs start: 26454.3\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs end: 14232.2\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using explicit constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using interface constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Making an interface set with fixed chains\n",
            "protocols.analysis.InterfaceAnalyzerMover: Interface set residues total: 109\n",
            "protocols.analysis.InterfaceAnalyzerMover: NOT detecting disulfides in the separated pose.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 2993 rotamers at 109 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.analysis.InterfaceAnalyzerMover: included_nres: 978\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 899\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 900\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 924\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 925\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 964\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 965\n",
            "antibody.design.GeneralAntibodyModeler: CDR min start: 14232.2\n",
            "antibody.design.GeneralAntibodyModeler: CDR min end: 3656.7\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using explicit constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using interface constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Making an interface set with fixed chains\n",
            "protocols.analysis.InterfaceAnalyzerMover: Interface set residues total: 98\n",
            "protocols.analysis.InterfaceAnalyzerMover: NOT detecting disulfides in the separated pose.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 3088 rotamers at 98 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.analysis.InterfaceAnalyzerMover: included_nres: 978\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: applying prob task op\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Using conservative op for L1\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Using conservative op for L3\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Using original bb sequence for conservative design.\n",
            "core.pack.pack_rotamers: built 6466 rotamers at 82 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating PDInteractionGraph\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs start: 3656.7\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs end: 5499.2\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using explicit constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using interface constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Making an interface set with fixed chains\n",
            "protocols.analysis.InterfaceAnalyzerMover: Interface set residues total: 98\n",
            "protocols.analysis.InterfaceAnalyzerMover: NOT detecting disulfides in the separated pose.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 2647 rotamers at 98 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.analysis.InterfaceAnalyzerMover: included_nres: 978\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 899\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 900\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 924\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 925\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 964\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 965\n",
            "antibody.design.GeneralAntibodyModeler: CDR min start: 3656.7\n",
            "antibody.design.GeneralAntibodyModeler: CDR min end: 3615.86\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using explicit constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using interface constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Making an interface set with fixed chains\n",
            "protocols.analysis.InterfaceAnalyzerMover: Interface set residues total: 98\n",
            "protocols.analysis.InterfaceAnalyzerMover: NOT detecting disulfides in the separated pose.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 3086 rotamers at 98 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.analysis.InterfaceAnalyzerMover: included_nres: 978\n",
            "protocols.antibody.design.AntibodyDesignMover: Min Cycle complete\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using explicit constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using interface constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Making an interface set with fixed chains\n",
            "protocols.analysis.InterfaceAnalyzerMover: Interface set residues total: 98\n",
            "protocols.analysis.InterfaceAnalyzerMover: NOT detecting disulfides in the separated pose.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 3086 rotamers at 98 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.analysis.InterfaceAnalyzerMover: included_nres: 978\n",
            "protocols.antibody.design.AntibodyDesignMover:\n",
            "protocols.antibody.design.AntibodyDesignMover: SCORE 12 2869.33 0\n",
            "protocols.antibody.design.AntibodyDesignMover: FINAL 12 1809.59 1\n",
            "protocols.antibody.design.AntibodyDesignMover:\n",
            "protocols.antibody.design.AntibodyDesignMover: Last Accepted Score: 1809.59\n",
            "protocols.antibody.design.AntibodyDesignMover: Lowest        Score:1809.59\n",
            "protocols.antibody.design.AntibodyDesignMover:\n",
            "protocols.antibody.design.AntibodyDesignMover: Outer round: 13\n",
            "protocols.antibody.design.AntibodyDesignMover: Grafting CDR from cluster L3-10-cis8-1 fragment 1i7zC_L3_0001\n",
            "protocols.grafting.CCDEndsGraftMover: Start: 955 End: 964 NterO: 3 CterO: 3\n",
            "protocols.grafting.util: Superimposing overhang residues\n",
            "core.chemical.AtomICoor: [ WARNING ] IcoorAtomID::atom_id(): Cannot get atom_id for POLYMER_LOWER of residue TYR 1.  Returning BOGUS ID instead.\n",
            "core.conformation.Residue: [ WARNING ] missing an atom: 1  H   that depends on a nonexistent polymer connection!\n",
            "core.conformation.Residue: [ WARNING ]  --> generating it using idealized coordinates.\n",
            "protocols.grafting.util: Deleting 8 residues from 956 to 963\n",
            "core.conformation.Conformation: Found disulfide between residues 771 845\n",
            "core.conformation.Conformation: Found disulfide between residues 891 955\n",
            "protocols.grafting.util: insert_point 955\n",
            "protocols.grafting.util: insert_start 956\n",
            "protocols.grafting.util: insert_end 965\n",
            "protocols.grafting.util: insert_length 10\n",
            "protocols.grafting.util: FOLD_TREE  EDGE 1 955 -1  EDGE 955 956 1  EDGE 956 965 -1\n",
            "protocols.grafting.GraftMoverBase: Insertion complete.\n",
            "protocols.grafting.AnchoredGraftMover: Setting default movemap\n",
            "protocols.grafting.CCDEndsGraftMover: Start: 955\n",
            "protocols.grafting.CCDEndsGraftMover: Original End: 964\n",
            "protocols.grafting.CCDEndsGraftMover: End: 966\n",
            "protocols.grafting.CCDEndsGraftMover: Insert Length: 10\n",
            "protocols.grafting.CCDEndsGraftMover: LOOP start: 954  stop: 957  cut: 955  size: 4  skip rate: 0  extended?: False\n",
            "protocols.grafting.CCDEndsGraftMover:\n",
            "protocols.grafting.CCDEndsGraftMover: LOOP start: 964  stop: 967  cut: 965  size: 4  skip rate: 0  extended?: False\n",
            "protocols.grafting.CCDEndsGraftMover:\n",
            "protocols.loops.FoldTreeFromLoopsWrapper: old foldtree FOLD_TREE  EDGE 1 955 -1  EDGE 955 966 1  EDGE 966 975 -1  EDGE 955 956 -2  C    N    EDGE 956 957 -2  C    N    EDGE 957 958 -2  C    N    EDGE 958 959 -2  C    N    EDGE 959 960 -2  C    N    EDGE 960 961 -2  C    N    EDGE 961 962 -2  C    N    EDGE 962 963 -2  C    N    EDGE 963 964 -2  C    N    EDGE 964 965 -2  C    N   \n",
            "New foldtree FOLD_TREE  EDGE 1 953 -1  EDGE 953 955 -1  EDGE 953 958 1  EDGE 958 956 -1  EDGE 958 963 -1  EDGE 963 965 -1  EDGE 963 968 2  EDGE 968 966 -1  EDGE 968 975 -1\n",
            "protocols.grafting.util: Loop: 1\n",
            "protocols.grafting.util: Add variant to: 955\n",
            "protocols.grafting.util: Loop: 2\n",
            "protocols.grafting.util: Add variant to: 965\n",
            "protocols.grafting.util: idealized  954\n",
            "protocols.grafting.util: idealized  955\n",
            "protocols.grafting.util: ideal 956\n",
            "protocols.grafting.util: ideal 965\n",
            "protocols.grafting.util: idealized 966\n",
            "protocols.grafting.util: idealized 967\n",
            "protocols.grafting.CCDEndsGraftMover: start 1543.44\n",
            "protocols.grafting.CCDEndsGraftMover: round 1\n",
            "protocols.grafting.CCDEndsGraftMover: Graft Closed early - returning\n",
            "protocols.grafting.CCDEndsGraftMover: 1 1310.85\n",
            "protocols.grafting.CCDEndsGraftMover: finish 1310.85\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 1242 rotamers at 38 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.grafting.CCDEndsGraftMover: Graft meets ideal geometry true\n",
            "protocols.grafting.CCDEndsGraftMover: Complete\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L3-10-cis8-1 data: 4 cutoff: 10\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: Adding general dihedral constraints for L3\n",
            "protocols.antibody.design.AntibodyDesignMover: Checking Geometry\n",
            "protocols.antibody.design.AntibodyDesignMover: DATA GRAFT_CLOSURE 1 rabd/my_ab.pdb L3 L3-10-cis8-1 1i7zC_L3_0001\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H1\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H2\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H3\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L1\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L2\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L3\n",
            "protocols.antibody.clusters.CDRClusterSet: Adding cacheable cluster data to pose\n",
            "protocols.antibody.design.util: Setting CDR sequence in NativeAntibodySeq\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: H1-13-1 data: 583 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/H1-13-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 26 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: H2-10-1 data: 328 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/H2-10-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 20 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: Adding general dihedral constraints for H3\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L1-10-1 data: 42 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/L1-10-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 20 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L2-8-1 data: 632 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/L2-8-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 16 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L3-10-cis8-1 data: 4 cutoff: 10\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: Adding general dihedral constraints for L3\n",
            "protocols.antibody.design.AntibodyDesignMover: Beginning MinProtocol\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using explicit constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using interface constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Making an interface set with fixed chains\n",
            "protocols.analysis.InterfaceAnalyzerMover: Interface set residues total: 97\n",
            "protocols.analysis.InterfaceAnalyzerMover: NOT detecting disulfides in the separated pose.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 2444 rotamers at 97 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.analysis.InterfaceAnalyzerMover: included_nres: 975\n",
            "protocols.antibody.design.AntibodyDesignMover: Inner round: 1\n",
            "protocols.antibody.design.AntibodyDesignMover: Add min neighbors : L1\n",
            "protocols.antibody.design.AntibodyDesignMover: Add min neighbors : H3\n",
            "basic.io.database: [ WARNING ] Unable to locate database file /sampling/antibodies/antibody_database_rosetta_design.db\n",
            "antibody.database.AntibodyDatabaseManager: The antibody design database is not specified by the user.\n",
            "antibody.database.AntibodyDatabaseManager: Reading from: /usr/local/lib/python3.10/dist-packages/pyrosetta/database//sampling/antibodies/antibody_database_rosetta_design_north_paper.db\n",
            "antibody.database.AntibodyDatabaseManager: Loaded L1-10-1 with 15 datapoints.\n",
            "antibody.database.AntibodyDatabaseManager: Skipped L3-10-cis8-1 with 1 data point\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Data pre-loaded.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: applying prob task op\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Using conservative op for L3\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Using original bb sequence for conservative design.\n",
            "protocols.task_operations.ConservativeDesignOperation: Loading conservative mutational data\n",
            "core.pack.pack_rotamers: built 3630 rotamers at 93 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating PDInteractionGraph\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs start: 14486.8\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs end: 8786.72\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using explicit constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using interface constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Making an interface set with fixed chains\n",
            "protocols.analysis.InterfaceAnalyzerMover: Interface set residues total: 97\n",
            "protocols.analysis.InterfaceAnalyzerMover: NOT detecting disulfides in the separated pose.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 2413 rotamers at 97 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.analysis.InterfaceAnalyzerMover: included_nres: 975\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 852\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 853\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 896\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 897\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 960\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 961\n",
            "core.optimization.LineMinimizer: Inaccurate G in step= 7.45058e-09 Deriv= -806.245 Finite Diff= 1.1054e+10\n",
            "core.optimization.LineMinimizer: Inaccurate G in step= 6.29404e-09 Deriv= -174.75 Finite Diff= 1.26296e+10\n",
            "core.optimization.LineMinimizer: Inaccurate G in step= 1.77945e-07 Deriv= -1838.26 Finite Diff= 4.46715e+08\n",
            "core.optimization.Minimizer: Line search failed even after resetting Hessian; aborting at iter#81\n",
            "antibody.design.GeneralAntibodyModeler: CDR min start: 8786.72\n",
            "antibody.design.GeneralAntibodyModeler: CDR min end: 2309.91\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using explicit constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using interface constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Making an interface set with fixed chains\n",
            "protocols.analysis.InterfaceAnalyzerMover: Interface set residues total: 95\n",
            "protocols.analysis.InterfaceAnalyzerMover: NOT detecting disulfides in the separated pose.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 2593 rotamers at 95 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.analysis.InterfaceAnalyzerMover: included_nres: 975\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: applying prob task op\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Using conservative op for L3\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Using original bb sequence for conservative design.\n",
            "core.pack.pack_rotamers: built 4192 rotamers at 90 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating PDInteractionGraph\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs start: 2309.91\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs end: 3203.71\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using explicit constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using interface constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Making an interface set with fixed chains\n",
            "protocols.analysis.InterfaceAnalyzerMover: Interface set residues total: 95\n",
            "protocols.analysis.InterfaceAnalyzerMover: NOT detecting disulfides in the separated pose.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 2623 rotamers at 95 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.analysis.InterfaceAnalyzerMover: included_nres: 975\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 852\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 853\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 896\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 897\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 960\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 961\n",
            "core.optimization.LineMinimizer: Inaccurate G in step= 9.53674e-09 Deriv= -3366.53 Finite Diff= 8.33524e+09\n",
            "core.optimization.LineMinimizer: Inaccurate G in step= 1.31492e-07 Deriv= -3366.53 Finite Diff= 6.04529e+08\n",
            "core.optimization.Minimizer: Line search failed even after resetting Hessian; aborting at iter#3\n",
            "antibody.design.GeneralAntibodyModeler: CDR min start: 2309.91\n",
            "antibody.design.GeneralAntibodyModeler: CDR min end: 2309.91\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using explicit constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using interface constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Making an interface set with fixed chains\n",
            "protocols.analysis.InterfaceAnalyzerMover: Interface set residues total: 95\n",
            "protocols.analysis.InterfaceAnalyzerMover: NOT detecting disulfides in the separated pose.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 2593 rotamers at 95 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.analysis.InterfaceAnalyzerMover: included_nres: 975\n",
            "protocols.antibody.design.AntibodyDesignMover: Min Cycle complete\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using explicit constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using interface constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Making an interface set with fixed chains\n",
            "protocols.analysis.InterfaceAnalyzerMover: Interface set residues total: 95\n",
            "protocols.analysis.InterfaceAnalyzerMover: NOT detecting disulfides in the separated pose.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 2593 rotamers at 95 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.analysis.InterfaceAnalyzerMover: included_nres: 975\n",
            "protocols.antibody.design.AntibodyDesignMover:\n",
            "protocols.antibody.design.AntibodyDesignMover: SCORE 13 1440.25 1\n",
            "protocols.antibody.design.AntibodyDesignMover: FINAL 13 1440.25 1\n",
            "protocols.antibody.design.AntibodyDesignMover:\n",
            "protocols.antibody.design.AntibodyDesignMover: Last Accepted Score: 1440.25\n",
            "protocols.antibody.design.AntibodyDesignMover: Lowest        Score:1440.25\n",
            "protocols.antibody.design.AntibodyDesignMover:\n",
            "protocols.antibody.design.AntibodyDesignMover: Outer round: 14\n",
            "protocols.antibody.design.AntibodyDesignMover: Grafting CDR from cluster L1-11-2 fragment 1a0qL_L1_0001\n",
            "protocols.grafting.CCDEndsGraftMover: Start: 891 End: 902 NterO: 3 CterO: 3\n",
            "protocols.grafting.util: Superimposing overhang residues\n",
            "core.chemical.AtomICoor: [ WARNING ] IcoorAtomID::atom_id(): Cannot get atom_id for POLYMER_LOWER of residue ILE 1.  Returning BOGUS ID instead.\n",
            "core.conformation.Residue: [ WARNING ] missing an atom: 1  H   that depends on a nonexistent polymer connection!\n",
            "core.conformation.Residue: [ WARNING ]  --> generating it using idealized coordinates.\n",
            "protocols.grafting.util: Deleting 10 residues from 892 to 901\n",
            "core.conformation.Conformation: Found disulfide between residues 771 845\n",
            "core.conformation.Conformation: Found disulfide between residues 891 945\n",
            "protocols.grafting.util: insert_point 891\n",
            "protocols.grafting.util: insert_start 892\n",
            "protocols.grafting.util: insert_end 902\n",
            "protocols.grafting.util: insert_length 11\n",
            "protocols.grafting.util: FOLD_TREE  EDGE 1 891 -1  EDGE 891 892 1  EDGE 892 965 -1\n",
            "protocols.grafting.GraftMoverBase: Insertion complete.\n",
            "protocols.grafting.AnchoredGraftMover: Setting default movemap\n",
            "protocols.grafting.CCDEndsGraftMover: Start: 891\n",
            "protocols.grafting.CCDEndsGraftMover: Original End: 902\n",
            "protocols.grafting.CCDEndsGraftMover: End: 903\n",
            "protocols.grafting.CCDEndsGraftMover: Insert Length: 11\n",
            "protocols.grafting.CCDEndsGraftMover: LOOP start: 890  stop: 893  cut: 891  size: 4  skip rate: 0  extended?: False\n",
            "protocols.grafting.CCDEndsGraftMover:\n",
            "protocols.grafting.CCDEndsGraftMover: LOOP start: 901  stop: 904  cut: 902  size: 4  skip rate: 0  extended?: False\n",
            "protocols.grafting.CCDEndsGraftMover:\n",
            "protocols.loops.FoldTreeFromLoopsWrapper: old foldtree FOLD_TREE  EDGE 1 891 -1  EDGE 891 903 1  EDGE 903 976 -1  EDGE 891 892 -2  C    N    EDGE 892 893 -2  C    N    EDGE 893 894 -2  C    N    EDGE 894 895 -2  C    N    EDGE 895 896 -2  C    N    EDGE 896 897 -2  C    N    EDGE 897 898 -2  C    N    EDGE 898 899 -2  C    N    EDGE 899 900 -2  C    N    EDGE 900 901 -2  C    N    EDGE 901 902 -2  C    N   \n",
            "New foldtree FOLD_TREE  EDGE 1 889 -1  EDGE 889 891 -1  EDGE 889 894 1  EDGE 894 892 -1  EDGE 894 900 -1  EDGE 900 902 -1  EDGE 900 905 2  EDGE 905 903 -1  EDGE 905 976 -1\n",
            "protocols.grafting.util: Loop: 1\n",
            "protocols.grafting.util: Add variant to: 891\n",
            "protocols.grafting.util: Loop: 2\n",
            "protocols.grafting.util: Add variant to: 902\n",
            "protocols.grafting.util: idealized  890\n",
            "protocols.grafting.util: idealized  891\n",
            "protocols.grafting.util: ideal 892\n",
            "protocols.grafting.util: ideal 902\n",
            "protocols.grafting.util: idealized 903\n",
            "protocols.grafting.util: idealized 904\n",
            "protocols.grafting.CCDEndsGraftMover: start 1840.56\n",
            "protocols.grafting.CCDEndsGraftMover: round 1\n",
            "protocols.loops.util: PepBondGeom: 42 L  C-N --  1.55956 max: 1.5\n",
            "protocols.grafting.CCDEndsGraftMover: 1 1460.35\n",
            "protocols.grafting.CCDEndsGraftMover: round 2\n",
            "protocols.loops.util: PepBondGeom: 42 L  C-N --  1.52163 max: 1.5\n",
            "protocols.grafting.CCDEndsGraftMover: 2 1417.11\n",
            "protocols.grafting.CCDEndsGraftMover: round 3\n",
            "protocols.grafting.CCDEndsGraftMover: Graft Closed early - returning\n",
            "protocols.grafting.CCDEndsGraftMover: 3 1401.9\n",
            "protocols.grafting.CCDEndsGraftMover: finish 1401.9\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 1909 rotamers at 48 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.grafting.CCDEndsGraftMover: Graft meets ideal geometry true\n",
            "protocols.grafting.CCDEndsGraftMover: Complete\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L1-11-2 data: 91 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/L1-11-2.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 22 constraints\n",
            "protocols.antibody.design.AntibodyDesignMover: Checking Geometry\n",
            "protocols.antibody.design.AntibodyDesignMover: DATA GRAFT_CLOSURE 1 rabd/my_ab.pdb L1 L1-11-2 1a0qL_L1_0001\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H1\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H2\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H3\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L1\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L2\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L3\n",
            "protocols.antibody.clusters.CDRClusterSet: Adding cacheable cluster data to pose\n",
            "protocols.antibody.design.util: Setting CDR sequence in NativeAntibodySeq\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: H1-13-1 data: 583 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/H1-13-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 26 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: H2-10-1 data: 328 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/H2-10-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 20 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: Adding general dihedral constraints for H3\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L1-11-2 data: 91 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/L1-11-2.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 22 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L2-8-1 data: 632 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/L2-8-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 16 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L3-10-cis8-1 data: 4 cutoff: 10\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: Adding general dihedral constraints for L3\n",
            "protocols.antibody.design.AntibodyDesignMover: Beginning MinProtocol\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using explicit constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using interface constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Making an interface set with fixed chains\n",
            "protocols.analysis.InterfaceAnalyzerMover: Interface set residues total: 98\n",
            "protocols.analysis.InterfaceAnalyzerMover: NOT detecting disulfides in the separated pose.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 2369 rotamers at 98 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.analysis.InterfaceAnalyzerMover: included_nres: 976\n",
            "protocols.antibody.design.AntibodyDesignMover: Inner round: 1\n",
            "protocols.antibody.design.AntibodyDesignMover: Add min neighbors : L2\n",
            "protocols.antibody.design.AntibodyDesignMover: Add min neighbors : L3\n",
            "basic.io.database: [ WARNING ] Unable to locate database file /sampling/antibodies/antibody_database_rosetta_design.db\n",
            "antibody.database.AntibodyDatabaseManager: The antibody design database is not specified by the user.\n",
            "antibody.database.AntibodyDatabaseManager: Reading from: /usr/local/lib/python3.10/dist-packages/pyrosetta/database//sampling/antibodies/antibody_database_rosetta_design_north_paper.db\n",
            "antibody.database.AntibodyDatabaseManager: Loaded L1-11-2 with 35 datapoints.\n",
            "antibody.database.AntibodyDatabaseManager: Skipped L3-10-cis8-1 with 1 data point\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Data pre-loaded.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: applying prob task op\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Using conservative op for L3\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Using original bb sequence for conservative design.\n",
            "protocols.task_operations.ConservativeDesignOperation: Loading conservative mutational data\n",
            "core.pack.pack_rotamers: built 3608 rotamers at 80 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating PDInteractionGraph\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs start: 8214.64\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs end: 4798.73\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using explicit constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using interface constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Making an interface set with fixed chains\n",
            "protocols.analysis.InterfaceAnalyzerMover: Interface set residues total: 97\n",
            "protocols.analysis.InterfaceAnalyzerMover: NOT detecting disulfides in the separated pose.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 2230 rotamers at 97 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.analysis.InterfaceAnalyzerMover: included_nres: 976\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 897\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 898\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 920\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 921\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 961\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 962\n",
            "antibody.design.GeneralAntibodyModeler: CDR min start: 4798.73\n",
            "antibody.design.GeneralAntibodyModeler: CDR min end: 2411.44\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using explicit constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using interface constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Making an interface set with fixed chains\n",
            "protocols.analysis.InterfaceAnalyzerMover: Interface set residues total: 95\n",
            "protocols.analysis.InterfaceAnalyzerMover: NOT detecting disulfides in the separated pose.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 2232 rotamers at 95 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.analysis.InterfaceAnalyzerMover: included_nres: 976\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: applying prob task op\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Using conservative op for L3\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Using original bb sequence for conservative design.\n",
            "core.pack.pack_rotamers: built 3749 rotamers at 76 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating PDInteractionGraph\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs start: 2411.44\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs end: 3215.3\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using explicit constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using interface constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Making an interface set with fixed chains\n",
            "protocols.analysis.InterfaceAnalyzerMover: Interface set residues total: 96\n",
            "protocols.analysis.InterfaceAnalyzerMover: NOT detecting disulfides in the separated pose.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 2356 rotamers at 96 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.analysis.InterfaceAnalyzerMover: included_nres: 976\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 897\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 898\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 920\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 921\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 961\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 962\n",
            "antibody.design.GeneralAntibodyModeler: CDR min start: 2411.44\n",
            "antibody.design.GeneralAntibodyModeler: CDR min end: 2410.07\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using explicit constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using interface constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Making an interface set with fixed chains\n",
            "protocols.analysis.InterfaceAnalyzerMover: Interface set residues total: 95\n",
            "protocols.analysis.InterfaceAnalyzerMover: NOT detecting disulfides in the separated pose.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 2240 rotamers at 95 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.analysis.InterfaceAnalyzerMover: included_nres: 976\n",
            "protocols.antibody.design.AntibodyDesignMover: Min Cycle complete\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using explicit constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using interface constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Making an interface set with fixed chains\n",
            "protocols.analysis.InterfaceAnalyzerMover: Interface set residues total: 95\n",
            "protocols.analysis.InterfaceAnalyzerMover: NOT detecting disulfides in the separated pose.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 2232 rotamers at 95 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.analysis.InterfaceAnalyzerMover: included_nres: 976\n",
            "protocols.antibody.design.AntibodyDesignMover:\n",
            "protocols.antibody.design.AntibodyDesignMover: SCORE 14 1607.59 0\n",
            "protocols.antibody.design.AntibodyDesignMover: FINAL 14 1440.25 1\n",
            "protocols.antibody.design.AntibodyDesignMover:\n",
            "protocols.antibody.design.AntibodyDesignMover: Last Accepted Score: 1440.25\n",
            "protocols.antibody.design.AntibodyDesignMover: Lowest        Score:1440.25\n",
            "protocols.antibody.design.AntibodyDesignMover:\n",
            "protocols.antibody.design.AntibodyDesignMover: Outer round: 15\n",
            "protocols.antibody.design.AntibodyDesignMover: Grafting CDR from cluster L3-7-1 fragment 1dfbL_L3_0001\n",
            "protocols.grafting.CCDEndsGraftMover: Start: 955 End: 966 NterO: 3 CterO: 3\n",
            "protocols.grafting.util: Superimposing overhang residues\n",
            "core.chemical.AtomICoor: [ WARNING ] IcoorAtomID::atom_id(): Cannot get atom_id for POLYMER_LOWER of residue TYR 1.  Returning BOGUS ID instead.\n",
            "core.conformation.Residue: [ WARNING ] missing an atom: 1  H   that depends on a nonexistent polymer connection!\n",
            "core.conformation.Residue: [ WARNING ]  --> generating it using idealized coordinates.\n",
            "protocols.grafting.util: Deleting 10 residues from 956 to 965\n",
            "core.conformation.Conformation: Found disulfide between residues 771 845\n",
            "core.conformation.Conformation: Found disulfide between residues 891 955\n",
            "protocols.grafting.util: insert_point 955\n",
            "protocols.grafting.util: insert_start 956\n",
            "protocols.grafting.util: insert_end 962\n",
            "protocols.grafting.util: insert_length 7\n",
            "protocols.grafting.util: FOLD_TREE  EDGE 1 955 -1  EDGE 955 956 1  EDGE 956 965 -1\n",
            "protocols.grafting.GraftMoverBase: Insertion complete.\n",
            "protocols.grafting.AnchoredGraftMover: Setting default movemap\n",
            "protocols.grafting.CCDEndsGraftMover: Start: 955\n",
            "protocols.grafting.CCDEndsGraftMover: Original End: 966\n",
            "protocols.grafting.CCDEndsGraftMover: End: 963\n",
            "protocols.grafting.CCDEndsGraftMover: Insert Length: 7\n",
            "protocols.grafting.CCDEndsGraftMover: LOOP start: 954  stop: 957  cut: 955  size: 4  skip rate: 0  extended?: False\n",
            "protocols.grafting.CCDEndsGraftMover:\n",
            "protocols.grafting.CCDEndsGraftMover: LOOP start: 961  stop: 964  cut: 962  size: 4  skip rate: 0  extended?: False\n",
            "protocols.grafting.CCDEndsGraftMover:\n",
            "protocols.loops.FoldTreeFromLoopsWrapper: old foldtree FOLD_TREE  EDGE 1 955 -1  EDGE 955 963 1  EDGE 963 972 -1  EDGE 955 956 -2  C    N    EDGE 956 957 -2  C    N    EDGE 957 958 -2  C    N    EDGE 958 959 -2  C    N    EDGE 959 960 -2  C    N    EDGE 960 961 -2  C    N    EDGE 961 962 -2  C    N   \n",
            "New foldtree FOLD_TREE  EDGE 1 953 -1  EDGE 953 955 -1  EDGE 953 958 1  EDGE 958 956 -1  EDGE 958 960 -1  EDGE 960 962 -1  EDGE 960 965 2  EDGE 965 963 -1  EDGE 965 972 -1\n",
            "protocols.grafting.util: Loop: 1\n",
            "protocols.grafting.util: Add variant to: 955\n",
            "protocols.grafting.util: Loop: 2\n",
            "protocols.grafting.util: Add variant to: 962\n",
            "protocols.grafting.util: idealized  954\n",
            "protocols.grafting.util: idealized  955\n",
            "protocols.grafting.util: ideal 956\n",
            "protocols.grafting.util: ideal 962\n",
            "protocols.grafting.util: idealized 963\n",
            "protocols.grafting.util: idealized 964\n",
            "protocols.grafting.CCDEndsGraftMover: start 1584.35\n",
            "protocols.grafting.CCDEndsGraftMover: round 1\n",
            "protocols.loops.util: PepBondGeom: 962 - 138 L  Ca-C-N --  99.5 (min): 139.664 : 134.5 (max)\n",
            "protocols.grafting.CCDEndsGraftMover: 1 1390.39\n",
            "protocols.grafting.CCDEndsGraftMover: round 2\n",
            "protocols.loops.util: PepBondGeom: 106 L  C-N --  1.52913 max: 1.5\n",
            "protocols.grafting.CCDEndsGraftMover: 2 1337.72\n",
            "protocols.grafting.CCDEndsGraftMover: round 3\n",
            "protocols.grafting.CCDEndsGraftMover: Graft Closed early - returning\n",
            "protocols.grafting.CCDEndsGraftMover: 3 1320.57\n",
            "protocols.grafting.CCDEndsGraftMover: finish 1320.57\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 846 rotamers at 38 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.grafting.CCDEndsGraftMover: Graft meets ideal geometry true\n",
            "protocols.grafting.CCDEndsGraftMover: Complete\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L3-7-1 data: 2 cutoff: 10\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: Adding general dihedral constraints for L3\n",
            "protocols.antibody.design.AntibodyDesignMover: Checking Geometry\n",
            "protocols.antibody.design.AntibodyDesignMover: DATA GRAFT_CLOSURE 1 rabd/my_ab.pdb L3 L3-7-1 1dfbL_L3_0001\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H1\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H2\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H3\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L1\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L2\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L3\n",
            "protocols.antibody.clusters.CDRClusterSet: Adding cacheable cluster data to pose\n",
            "protocols.antibody.design.util: Setting CDR sequence in NativeAntibodySeq\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: H1-13-1 data: 583 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/H1-13-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 26 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: H2-10-1 data: 328 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/H2-10-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 20 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: Adding general dihedral constraints for H3\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L1-10-1 data: 42 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/L1-10-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 20 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L2-8-1 data: 632 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/L2-8-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 16 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L3-7-1 data: 2 cutoff: 10\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: Adding general dihedral constraints for L3\n",
            "protocols.antibody.design.AntibodyDesignMover: Beginning MinProtocol\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using explicit constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using interface constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Making an interface set with fixed chains\n",
            "protocols.analysis.InterfaceAnalyzerMover: Interface set residues total: 91\n",
            "protocols.analysis.InterfaceAnalyzerMover: NOT detecting disulfides in the separated pose.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 2461 rotamers at 91 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.analysis.InterfaceAnalyzerMover: included_nres: 972\n",
            "protocols.antibody.design.AntibodyDesignMover: Inner round: 1\n",
            "protocols.antibody.design.AntibodyDesignMover: Add min neighbors : L1\n",
            "protocols.antibody.design.AntibodyDesignMover: Add min neighbors : H3\n",
            "basic.io.database: [ WARNING ] Unable to locate database file /sampling/antibodies/antibody_database_rosetta_design.db\n",
            "antibody.database.AntibodyDatabaseManager: The antibody design database is not specified by the user.\n",
            "antibody.database.AntibodyDatabaseManager: Reading from: /usr/local/lib/python3.10/dist-packages/pyrosetta/database//sampling/antibodies/antibody_database_rosetta_design_north_paper.db\n",
            "antibody.database.AntibodyDatabaseManager: Loaded L1-10-1 with 15 datapoints.\n",
            "antibody.database.AntibodyDatabaseManager: Skipped L3-7-1 with 2 data point\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Data pre-loaded.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: applying prob task op\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Using conservative op for L3\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Using original bb sequence for conservative design.\n",
            "protocols.task_operations.ConservativeDesignOperation: Loading conservative mutational data\n",
            "core.pack.pack_rotamers: built 5267 rotamers at 85 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating PDInteractionGraph\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs start: 6606.06\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs end: 2876.19\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using explicit constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using interface constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Making an interface set with fixed chains\n",
            "protocols.analysis.InterfaceAnalyzerMover: Interface set residues total: 91\n",
            "protocols.analysis.InterfaceAnalyzerMover: NOT detecting disulfides in the separated pose.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 2368 rotamers at 91 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.analysis.InterfaceAnalyzerMover: included_nres: 972\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 852\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 853\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 896\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 897\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 959\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 960\n",
            "antibody.design.GeneralAntibodyModeler: CDR min start: 2876.19\n",
            "antibody.design.GeneralAntibodyModeler: CDR min end: 1971.51\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using explicit constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using interface constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Making an interface set with fixed chains\n",
            "protocols.analysis.InterfaceAnalyzerMover: Interface set residues total: 92\n",
            "protocols.analysis.InterfaceAnalyzerMover: NOT detecting disulfides in the separated pose.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 2656 rotamers at 92 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.analysis.InterfaceAnalyzerMover: included_nres: 972\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: applying prob task op\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Using conservative op for L3\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Using original bb sequence for conservative design.\n",
            "core.pack.pack_rotamers: built 4594 rotamers at 82 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating PDInteractionGraph\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs start: 1971.51\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs end: 2195.94\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using explicit constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using interface constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Making an interface set with fixed chains\n",
            "protocols.analysis.InterfaceAnalyzerMover: Interface set residues total: 92\n",
            "protocols.analysis.InterfaceAnalyzerMover: NOT detecting disulfides in the separated pose.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 2516 rotamers at 92 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.analysis.InterfaceAnalyzerMover: included_nres: 972\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 852\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 853\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 896\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 897\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 959\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 960\n",
            "antibody.design.GeneralAntibodyModeler: CDR min start: 1971.51\n",
            "antibody.design.GeneralAntibodyModeler: CDR min end: 1970.75\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using explicit constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using interface constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Making an interface set with fixed chains\n",
            "protocols.analysis.InterfaceAnalyzerMover: Interface set residues total: 92\n",
            "protocols.analysis.InterfaceAnalyzerMover: NOT detecting disulfides in the separated pose.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 2647 rotamers at 92 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.analysis.InterfaceAnalyzerMover: included_nres: 972\n",
            "protocols.antibody.design.AntibodyDesignMover: Min Cycle complete\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using explicit constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using interface constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Making an interface set with fixed chains\n",
            "protocols.analysis.InterfaceAnalyzerMover: Interface set residues total: 92\n",
            "protocols.analysis.InterfaceAnalyzerMover: NOT detecting disulfides in the separated pose.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 2656 rotamers at 92 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.analysis.InterfaceAnalyzerMover: included_nres: 972\n",
            "protocols.antibody.design.AntibodyDesignMover:\n",
            "protocols.antibody.design.AntibodyDesignMover: SCORE 15 1333.26 1\n",
            "protocols.antibody.design.AntibodyDesignMover: FINAL 15 1333.26 1\n",
            "protocols.antibody.design.AntibodyDesignMover:\n",
            "protocols.antibody.design.AntibodyDesignMover: Last Accepted Score: 1333.26\n",
            "protocols.antibody.design.AntibodyDesignMover: Lowest        Score:1333.26\n",
            "protocols.antibody.design.AntibodyDesignMover:\n",
            "protocols.antibody.design.AntibodyDesignMover: Outer round: 16\n",
            "protocols.antibody.design.AntibodyDesignMover: Grafting CDR from cluster L1-16-1 fragment 1i9jL_L1_0001\n",
            "protocols.grafting.CCDEndsGraftMover: Start: 891 End: 902 NterO: 3 CterO: 3\n",
            "protocols.grafting.util: Superimposing overhang residues\n",
            "core.chemical.AtomICoor: [ WARNING ] IcoorAtomID::atom_id(): Cannot get atom_id for POLYMER_LOWER of residue ILE 1.  Returning BOGUS ID instead.\n",
            "core.conformation.Residue: [ WARNING ] missing an atom: 1  H   that depends on a nonexistent polymer connection!\n",
            "core.conformation.Residue: [ WARNING ]  --> generating it using idealized coordinates.\n",
            "protocols.grafting.util: Deleting 10 residues from 892 to 901\n",
            "core.conformation.Conformation: Found disulfide between residues 771 845\n",
            "core.conformation.Conformation: Found disulfide between residues 891 945\n",
            "protocols.grafting.util: insert_point 891\n",
            "protocols.grafting.util: insert_start 892\n",
            "protocols.grafting.util: insert_end 907\n",
            "protocols.grafting.util: insert_length 16\n",
            "protocols.grafting.util: FOLD_TREE  EDGE 1 891 -1  EDGE 891 892 1  EDGE 892 962 -1\n",
            "protocols.grafting.GraftMoverBase: Insertion complete.\n",
            "protocols.grafting.AnchoredGraftMover: Setting default movemap\n",
            "protocols.grafting.CCDEndsGraftMover: Start: 891\n",
            "protocols.grafting.CCDEndsGraftMover: Original End: 902\n",
            "protocols.grafting.CCDEndsGraftMover: End: 908\n",
            "protocols.grafting.CCDEndsGraftMover: Insert Length: 16\n",
            "protocols.grafting.CCDEndsGraftMover: LOOP start: 890  stop: 893  cut: 891  size: 4  skip rate: 0  extended?: False\n",
            "protocols.grafting.CCDEndsGraftMover:\n",
            "protocols.grafting.CCDEndsGraftMover: LOOP start: 906  stop: 909  cut: 907  size: 4  skip rate: 0  extended?: False\n",
            "protocols.grafting.CCDEndsGraftMover:\n",
            "protocols.loops.FoldTreeFromLoopsWrapper: old foldtree FOLD_TREE  EDGE 1 891 -1  EDGE 891 908 1  EDGE 908 978 -1  EDGE 891 892 -2  C    N    EDGE 892 893 -2  C    N    EDGE 893 894 -2  C    N    EDGE 894 895 -2  C    N    EDGE 895 896 -2  C    N    EDGE 896 897 -2  C    N    EDGE 897 898 -2  C    N    EDGE 898 899 -2  C    N    EDGE 899 900 -2  C    N    EDGE 900 901 -2  C    N    EDGE 901 902 -2  C    N    EDGE 902 903 -2  C    N    EDGE 903 904 -2  C    N    EDGE 904 905 -2  C    N    EDGE 905 906 -2  C    N    EDGE 906 907 -2  C    N   \n",
            "New foldtree FOLD_TREE  EDGE 1 889 -1  EDGE 889 891 -1  EDGE 889 894 1  EDGE 894 892 -1  EDGE 894 905 -1  EDGE 905 907 -1  EDGE 905 910 2  EDGE 910 908 -1  EDGE 910 978 -1\n",
            "protocols.grafting.util: Loop: 1\n",
            "protocols.grafting.util: Add variant to: 891\n",
            "protocols.grafting.util: Loop: 2\n",
            "protocols.grafting.util: Add variant to: 907\n",
            "protocols.grafting.util: idealized  890\n",
            "protocols.grafting.util: idealized  891\n",
            "protocols.grafting.util: ideal 892\n",
            "protocols.grafting.util: ideal 907\n",
            "protocols.grafting.util: idealized 908\n",
            "protocols.grafting.util: idealized 909\n",
            "protocols.grafting.CCDEndsGraftMover: start 1808.04\n",
            "protocols.grafting.CCDEndsGraftMover: round 1\n",
            "protocols.grafting.CCDEndsGraftMover: Graft Closed early - returning\n",
            "protocols.grafting.CCDEndsGraftMover: 1 1444.61\n",
            "protocols.grafting.CCDEndsGraftMover: finish 1444.61\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 2006 rotamers at 52 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.grafting.CCDEndsGraftMover: Graft meets ideal geometry true\n",
            "protocols.grafting.CCDEndsGraftMover: Complete\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L1-16-1 data: 117 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/L1-16-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 32 constraints\n",
            "protocols.antibody.design.AntibodyDesignMover: Checking Geometry\n",
            "protocols.antibody.design.AntibodyDesignMover: DATA GRAFT_CLOSURE 1 rabd/my_ab.pdb L1 L1-16-1 1i9jL_L1_0001\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H1\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H2\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H3\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L1\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L2\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L3\n",
            "protocols.antibody.clusters.CDRClusterSet: Adding cacheable cluster data to pose\n",
            "protocols.antibody.design.util: Setting CDR sequence in NativeAntibodySeq\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: H1-13-1 data: 583 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/H1-13-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 26 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: H2-10-1 data: 328 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/H2-10-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 20 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: Adding general dihedral constraints for H3\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L1-16-1 data: 117 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/L1-16-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 32 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L2-8-1 data: 632 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/L2-8-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 16 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L3-7-1 data: 2 cutoff: 10\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: Adding general dihedral constraints for L3\n",
            "protocols.antibody.design.AntibodyDesignMover: Beginning MinProtocol\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using explicit constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using interface constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Making an interface set with fixed chains\n",
            "protocols.analysis.InterfaceAnalyzerMover: Interface set residues total: 115\n",
            "protocols.analysis.InterfaceAnalyzerMover: NOT detecting disulfides in the separated pose.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 3283 rotamers at 115 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.analysis.InterfaceAnalyzerMover: included_nres: 978\n",
            "protocols.antibody.design.AntibodyDesignMover: Inner round: 1\n",
            "protocols.antibody.design.AntibodyDesignMover: Add min neighbors : L2\n",
            "protocols.antibody.design.AntibodyDesignMover: Add min neighbors : L3\n",
            "basic.io.database: [ WARNING ] Unable to locate database file /sampling/antibodies/antibody_database_rosetta_design.db\n",
            "antibody.database.AntibodyDatabaseManager: The antibody design database is not specified by the user.\n",
            "antibody.database.AntibodyDatabaseManager: Reading from: /usr/local/lib/python3.10/dist-packages/pyrosetta/database//sampling/antibodies/antibody_database_rosetta_design_north_paper.db\n",
            "antibody.database.AntibodyDatabaseManager: Loaded L1-16-1 with 47 datapoints.\n",
            "antibody.database.AntibodyDatabaseManager: Skipped L3-7-1 with 2 data point\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Data pre-loaded.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: applying prob task op\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Using conservative op for L3\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Using original bb sequence for conservative design.\n",
            "protocols.task_operations.ConservativeDesignOperation: Loading conservative mutational data\n",
            "core.pack.pack_rotamers: built 4012 rotamers at 84 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating PDInteractionGraph\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs start: 25307.8\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs end: 15930.9\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using explicit constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using interface constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Making an interface set with fixed chains\n",
            "protocols.analysis.InterfaceAnalyzerMover: Interface set residues total: 115\n",
            "protocols.analysis.InterfaceAnalyzerMover: NOT detecting disulfides in the separated pose.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 3186 rotamers at 115 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.analysis.InterfaceAnalyzerMover: included_nres: 978\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 899\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 900\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 925\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 926\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 965\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 966\n",
            "antibody.design.GeneralAntibodyModeler: CDR min start: 15930.9\n",
            "antibody.design.GeneralAntibodyModeler: CDR min end: 4795.75\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using explicit constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using interface constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Making an interface set with fixed chains\n",
            "protocols.analysis.InterfaceAnalyzerMover: Interface set residues total: 119\n",
            "protocols.analysis.InterfaceAnalyzerMover: NOT detecting disulfides in the separated pose.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 3540 rotamers at 119 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.analysis.InterfaceAnalyzerMover: included_nres: 978\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: applying prob task op\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Using conservative op for L3\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Using original bb sequence for conservative design.\n",
            "core.pack.pack_rotamers: built 4927 rotamers at 88 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating PDInteractionGraph\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs start: 4795.75\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs end: 8009.42\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using explicit constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using interface constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Making an interface set with fixed chains\n",
            "protocols.analysis.InterfaceAnalyzerMover: Interface set residues total: 119\n",
            "protocols.analysis.InterfaceAnalyzerMover: NOT detecting disulfides in the separated pose.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 3479 rotamers at 119 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.analysis.InterfaceAnalyzerMover: included_nres: 978\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 899\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 900\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 925\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 926\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 965\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 966\n",
            "antibody.design.GeneralAntibodyModeler: CDR min start: 4795.75\n",
            "antibody.design.GeneralAntibodyModeler: CDR min end: 4769.8\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using explicit constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using interface constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Making an interface set with fixed chains\n",
            "protocols.analysis.InterfaceAnalyzerMover: Interface set residues total: 119\n",
            "protocols.analysis.InterfaceAnalyzerMover: NOT detecting disulfides in the separated pose.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 3538 rotamers at 119 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.analysis.InterfaceAnalyzerMover: included_nres: 978\n",
            "protocols.antibody.design.AntibodyDesignMover: Min Cycle complete\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using explicit constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using interface constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Making an interface set with fixed chains\n",
            "protocols.analysis.InterfaceAnalyzerMover: Interface set residues total: 119\n",
            "protocols.analysis.InterfaceAnalyzerMover: NOT detecting disulfides in the separated pose.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 3538 rotamers at 119 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.analysis.InterfaceAnalyzerMover: included_nres: 978\n",
            "protocols.antibody.design.AntibodyDesignMover:\n",
            "protocols.antibody.design.AntibodyDesignMover: SCORE 16 4081.13 0\n",
            "protocols.antibody.design.AntibodyDesignMover: FINAL 16 1333.26 1\n",
            "protocols.antibody.design.AntibodyDesignMover:\n",
            "protocols.antibody.design.AntibodyDesignMover: Last Accepted Score: 1333.26\n",
            "protocols.antibody.design.AntibodyDesignMover: Lowest        Score:1333.26\n",
            "protocols.antibody.design.AntibodyDesignMover:\n",
            "protocols.antibody.design.AntibodyDesignMover: Outer round: 17\n",
            "protocols.antibody.design.AntibodyDesignMover: Grafting CDR from cluster L1-15-2 fragment 1i7zA_L1_0001\n",
            "protocols.grafting.CCDEndsGraftMover: Start: 891 End: 902 NterO: 3 CterO: 3\n",
            "protocols.grafting.util: Superimposing overhang residues\n",
            "core.chemical.AtomICoor: [ WARNING ] IcoorAtomID::atom_id(): Cannot get atom_id for POLYMER_LOWER of residue ILE 1.  Returning BOGUS ID instead.\n",
            "core.conformation.Residue: [ WARNING ] missing an atom: 1  H   that depends on a nonexistent polymer connection!\n",
            "core.conformation.Residue: [ WARNING ]  --> generating it using idealized coordinates.\n",
            "protocols.grafting.util: Deleting 10 residues from 892 to 901\n",
            "core.conformation.Conformation: Found disulfide between residues 771 845\n",
            "core.conformation.Conformation: Found disulfide between residues 891 945\n",
            "protocols.grafting.util: insert_point 891\n",
            "protocols.grafting.util: insert_start 892\n",
            "protocols.grafting.util: insert_end 906\n",
            "protocols.grafting.util: insert_length 15\n",
            "protocols.grafting.util: FOLD_TREE  EDGE 1 891 -1  EDGE 891 892 1  EDGE 892 962 -1\n",
            "protocols.grafting.GraftMoverBase: Insertion complete.\n",
            "protocols.grafting.AnchoredGraftMover: Setting default movemap\n",
            "protocols.grafting.CCDEndsGraftMover: Start: 891\n",
            "protocols.grafting.CCDEndsGraftMover: Original End: 902\n",
            "protocols.grafting.CCDEndsGraftMover: End: 907\n",
            "protocols.grafting.CCDEndsGraftMover: Insert Length: 15\n",
            "protocols.grafting.CCDEndsGraftMover: LOOP start: 890  stop: 893  cut: 891  size: 4  skip rate: 0  extended?: False\n",
            "protocols.grafting.CCDEndsGraftMover:\n",
            "protocols.grafting.CCDEndsGraftMover: LOOP start: 905  stop: 908  cut: 906  size: 4  skip rate: 0  extended?: False\n",
            "protocols.grafting.CCDEndsGraftMover:\n",
            "protocols.loops.FoldTreeFromLoopsWrapper: old foldtree FOLD_TREE  EDGE 1 891 -1  EDGE 891 907 1  EDGE 907 977 -1  EDGE 891 892 -2  C    N    EDGE 892 893 -2  C    N    EDGE 893 894 -2  C    N    EDGE 894 895 -2  C    N    EDGE 895 896 -2  C    N    EDGE 896 897 -2  C    N    EDGE 897 898 -2  C    N    EDGE 898 899 -2  C    N    EDGE 899 900 -2  C    N    EDGE 900 901 -2  C    N    EDGE 901 902 -2  C    N    EDGE 902 903 -2  C    N    EDGE 903 904 -2  C    N    EDGE 904 905 -2  C    N    EDGE 905 906 -2  C    N   \n",
            "New foldtree FOLD_TREE  EDGE 1 889 -1  EDGE 889 891 -1  EDGE 889 894 1  EDGE 894 892 -1  EDGE 894 904 -1  EDGE 904 906 -1  EDGE 904 909 2  EDGE 909 907 -1  EDGE 909 977 -1\n",
            "protocols.grafting.util: Loop: 1\n",
            "protocols.grafting.util: Add variant to: 891\n",
            "protocols.grafting.util: Loop: 2\n",
            "protocols.grafting.util: Add variant to: 906\n",
            "protocols.grafting.util: idealized  890\n",
            "protocols.grafting.util: idealized  891\n",
            "protocols.grafting.util: ideal 892\n",
            "protocols.grafting.util: ideal 906\n",
            "protocols.grafting.util: idealized 907\n",
            "protocols.grafting.util: idealized 908\n",
            "protocols.grafting.CCDEndsGraftMover: start 1706.01\n",
            "protocols.grafting.CCDEndsGraftMover: round 1\n",
            "protocols.grafting.CCDEndsGraftMover: Graft Closed early - returning\n",
            "protocols.grafting.CCDEndsGraftMover: 1 1387.9\n",
            "protocols.grafting.CCDEndsGraftMover: finish 1387.9\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 2005 rotamers at 51 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.grafting.CCDEndsGraftMover: Graft meets ideal geometry true\n",
            "protocols.grafting.CCDEndsGraftMover: Complete\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L1-15-2 data: 3 cutoff: 10\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: Adding general dihedral constraints for L1\n",
            "protocols.antibody.design.AntibodyDesignMover: Checking Geometry\n",
            "protocols.antibody.design.AntibodyDesignMover: DATA GRAFT_CLOSURE 1 rabd/my_ab.pdb L1 L1-15-2 1i7zA_L1_0001\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H1\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H2\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H3\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L1\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L2\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L3\n",
            "protocols.antibody.clusters.CDRClusterSet: Adding cacheable cluster data to pose\n",
            "protocols.antibody.design.util: Setting CDR sequence in NativeAntibodySeq\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: H1-13-1 data: 583 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/H1-13-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 26 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: H2-10-1 data: 328 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/H2-10-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 20 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: Adding general dihedral constraints for H3\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L1-15-2 data: 3 cutoff: 10\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: Adding general dihedral constraints for L1\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L2-8-1 data: 632 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/L2-8-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 16 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L3-7-1 data: 2 cutoff: 10\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: Adding general dihedral constraints for L3\n",
            "protocols.antibody.design.AntibodyDesignMover: Beginning MinProtocol\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using explicit constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using interface constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Making an interface set with fixed chains\n",
            "protocols.analysis.InterfaceAnalyzerMover: Interface set residues total: 106\n",
            "protocols.analysis.InterfaceAnalyzerMover: NOT detecting disulfides in the separated pose.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 3157 rotamers at 106 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.analysis.InterfaceAnalyzerMover: included_nres: 977\n",
            "protocols.antibody.design.AntibodyDesignMover: Inner round: 1\n",
            "protocols.antibody.design.AntibodyDesignMover: Add min neighbors : L2\n",
            "protocols.antibody.design.AntibodyDesignMover: Add min neighbors : L3\n",
            "basic.io.database: [ WARNING ] Unable to locate database file /sampling/antibodies/antibody_database_rosetta_design.db\n",
            "antibody.database.AntibodyDatabaseManager: The antibody design database is not specified by the user.\n",
            "antibody.database.AntibodyDatabaseManager: Reading from: /usr/local/lib/python3.10/dist-packages/pyrosetta/database//sampling/antibodies/antibody_database_rosetta_design_north_paper.db\n",
            "antibody.database.AntibodyDatabaseManager: Skipped L1-15-2 with 2 data point\n",
            "antibody.database.AntibodyDatabaseManager: Skipped L3-7-1 with 2 data point\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Data pre-loaded.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: applying prob task op\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Using conservative op for L1\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Using conservative op for L3\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Using original bb sequence for conservative design.\n",
            "protocols.task_operations.ConservativeDesignOperation: Loading conservative mutational data\n",
            "core.pack.pack_rotamers: built 6786 rotamers at 81 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating PDInteractionGraph\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs start: 17639.1\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs end: 10012.4\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using explicit constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using interface constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Making an interface set with fixed chains\n",
            "protocols.analysis.InterfaceAnalyzerMover: Interface set residues total: 107\n",
            "protocols.analysis.InterfaceAnalyzerMover: NOT detecting disulfides in the separated pose.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 2994 rotamers at 107 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.analysis.InterfaceAnalyzerMover: included_nres: 977\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 899\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 900\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 924\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 925\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 964\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 965\n",
            "antibody.design.GeneralAntibodyModeler: CDR min start: 10012.4\n",
            "antibody.design.GeneralAntibodyModeler: CDR min end: 2992.96\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using explicit constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using interface constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Making an interface set with fixed chains\n",
            "protocols.analysis.InterfaceAnalyzerMover: Interface set residues total: 105\n",
            "protocols.analysis.InterfaceAnalyzerMover: NOT detecting disulfides in the separated pose.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 2916 rotamers at 105 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.analysis.InterfaceAnalyzerMover: included_nres: 977\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: applying prob task op\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Using conservative op for L1\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Using conservative op for L3\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Using original bb sequence for conservative design.\n",
            "core.pack.pack_rotamers: built 6870 rotamers at 81 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating PDInteractionGraph\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs start: 2992.96\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs end: 3791.45\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using explicit constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using interface constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Making an interface set with fixed chains\n",
            "protocols.analysis.InterfaceAnalyzerMover: Interface set residues total: 105\n",
            "protocols.analysis.InterfaceAnalyzerMover: NOT detecting disulfides in the separated pose.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 2875 rotamers at 105 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.analysis.InterfaceAnalyzerMover: included_nres: 977\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 899\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 900\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 924\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 925\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 964\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 965\n",
            "antibody.design.GeneralAntibodyModeler: CDR min start: 2992.96\n",
            "antibody.design.GeneralAntibodyModeler: CDR min end: 2937.2\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using explicit constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using interface constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Making an interface set with fixed chains\n",
            "protocols.analysis.InterfaceAnalyzerMover: Interface set residues total: 104\n",
            "protocols.analysis.InterfaceAnalyzerMover: NOT detecting disulfides in the separated pose.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 2978 rotamers at 104 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.analysis.InterfaceAnalyzerMover: included_nres: 977\n",
            "protocols.antibody.design.AntibodyDesignMover: Min Cycle complete\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using explicit constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using interface constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Making an interface set with fixed chains\n",
            "protocols.analysis.InterfaceAnalyzerMover: Interface set residues total: 104\n",
            "protocols.analysis.InterfaceAnalyzerMover: NOT detecting disulfides in the separated pose.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 2978 rotamers at 104 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.analysis.InterfaceAnalyzerMover: included_nres: 977\n",
            "protocols.antibody.design.AntibodyDesignMover:\n",
            "protocols.antibody.design.AntibodyDesignMover: SCORE 17 2206.99 0\n",
            "protocols.antibody.design.AntibodyDesignMover: FINAL 17 1333.26 1\n",
            "protocols.antibody.design.AntibodyDesignMover:\n",
            "protocols.antibody.design.AntibodyDesignMover: Last Accepted Score: 1333.26\n",
            "protocols.antibody.design.AntibodyDesignMover: Lowest        Score:1333.26\n",
            "protocols.antibody.design.AntibodyDesignMover:\n",
            "protocols.antibody.design.AntibodyDesignMover: Outer round: 18\n",
            "protocols.antibody.design.AntibodyDesignMover: Grafting CDR from cluster L1-17-1 fragment 1efqA_L1_0001\n",
            "protocols.grafting.CCDEndsGraftMover: Start: 891 End: 902 NterO: 3 CterO: 3\n",
            "protocols.grafting.util: Superimposing overhang residues\n",
            "core.chemical.AtomICoor: [ WARNING ] IcoorAtomID::atom_id(): Cannot get atom_id for POLYMER_LOWER of residue ILE 1.  Returning BOGUS ID instead.\n",
            "core.conformation.Residue: [ WARNING ] missing an atom: 1  H   that depends on a nonexistent polymer connection!\n",
            "core.conformation.Residue: [ WARNING ]  --> generating it using idealized coordinates.\n",
            "protocols.grafting.util: Deleting 10 residues from 892 to 901\n",
            "core.conformation.Conformation: Found disulfide between residues 771 845\n",
            "core.conformation.Conformation: Found disulfide between residues 891 945\n",
            "protocols.grafting.util: insert_point 891\n",
            "protocols.grafting.util: insert_start 892\n",
            "protocols.grafting.util: insert_end 908\n",
            "protocols.grafting.util: insert_length 17\n",
            "protocols.grafting.util: FOLD_TREE  EDGE 1 891 -1  EDGE 891 892 1  EDGE 892 962 -1\n",
            "protocols.grafting.GraftMoverBase: Insertion complete.\n",
            "protocols.grafting.AnchoredGraftMover: Setting default movemap\n",
            "protocols.grafting.CCDEndsGraftMover: Start: 891\n",
            "protocols.grafting.CCDEndsGraftMover: Original End: 902\n",
            "protocols.grafting.CCDEndsGraftMover: End: 909\n",
            "protocols.grafting.CCDEndsGraftMover: Insert Length: 17\n",
            "protocols.grafting.CCDEndsGraftMover: LOOP start: 890  stop: 893  cut: 891  size: 4  skip rate: 0  extended?: False\n",
            "protocols.grafting.CCDEndsGraftMover:\n",
            "protocols.grafting.CCDEndsGraftMover: LOOP start: 907  stop: 910  cut: 908  size: 4  skip rate: 0  extended?: False\n",
            "protocols.grafting.CCDEndsGraftMover:\n",
            "protocols.loops.FoldTreeFromLoopsWrapper: old foldtree FOLD_TREE  EDGE 1 891 -1  EDGE 891 909 1  EDGE 909 979 -1  EDGE 891 892 -2  C    N    EDGE 892 893 -2  C    N    EDGE 893 894 -2  C    N    EDGE 894 895 -2  C    N    EDGE 895 896 -2  C    N    EDGE 896 897 -2  C    N    EDGE 897 898 -2  C    N    EDGE 898 899 -2  C    N    EDGE 899 900 -2  C    N    EDGE 900 901 -2  C    N    EDGE 901 902 -2  C    N    EDGE 902 903 -2  C    N    EDGE 903 904 -2  C    N    EDGE 904 905 -2  C    N    EDGE 905 906 -2  C    N    EDGE 906 907 -2  C    N    EDGE 907 908 -2  C    N   \n",
            "New foldtree FOLD_TREE  EDGE 1 889 -1  EDGE 889 891 -1  EDGE 889 894 1  EDGE 894 892 -1  EDGE 894 906 -1  EDGE 906 908 -1  EDGE 906 911 2  EDGE 911 909 -1  EDGE 911 979 -1\n",
            "protocols.grafting.util: Loop: 1\n",
            "protocols.grafting.util: Add variant to: 891\n",
            "protocols.grafting.util: Loop: 2\n",
            "protocols.grafting.util: Add variant to: 908\n",
            "protocols.grafting.util: idealized  890\n",
            "protocols.grafting.util: idealized  891\n",
            "protocols.grafting.util: ideal 892\n",
            "protocols.grafting.util: ideal 908\n",
            "protocols.grafting.util: idealized 909\n",
            "protocols.grafting.util: idealized 910\n",
            "protocols.grafting.CCDEndsGraftMover: start 1843.86\n",
            "protocols.grafting.CCDEndsGraftMover: round 1\n",
            "protocols.grafting.CCDEndsGraftMover: Graft Closed early - returning\n",
            "protocols.grafting.CCDEndsGraftMover: 1 1487.45\n",
            "protocols.grafting.CCDEndsGraftMover: finish 1487.45\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 2058 rotamers at 55 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.grafting.CCDEndsGraftMover: Graft meets ideal geometry true\n",
            "protocols.grafting.CCDEndsGraftMover: Complete\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L1-17-1 data: 49 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/L1-17-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 34 constraints\n",
            "protocols.antibody.design.AntibodyDesignMover: Checking Geometry\n",
            "protocols.antibody.design.AntibodyDesignMover: DATA GRAFT_CLOSURE 1 rabd/my_ab.pdb L1 L1-17-1 1efqA_L1_0001\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H1\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H2\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H3\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L1\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L2\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L3\n",
            "protocols.antibody.clusters.CDRClusterSet: Adding cacheable cluster data to pose\n",
            "protocols.antibody.design.util: Setting CDR sequence in NativeAntibodySeq\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: H1-13-1 data: 583 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/H1-13-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 26 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: H2-10-1 data: 328 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/H2-10-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 20 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: Adding general dihedral constraints for H3\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L1-17-1 data: 49 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/L1-17-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 34 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L2-8-1 data: 632 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/L2-8-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 16 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L3-7-1 data: 2 cutoff: 10\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: Adding general dihedral constraints for L3\n",
            "protocols.antibody.design.AntibodyDesignMover: Beginning MinProtocol\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using explicit constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using interface constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Making an interface set with fixed chains\n",
            "protocols.analysis.InterfaceAnalyzerMover: Interface set residues total: 121\n",
            "protocols.analysis.InterfaceAnalyzerMover: NOT detecting disulfides in the separated pose.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 3707 rotamers at 121 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.analysis.InterfaceAnalyzerMover: included_nres: 979\n",
            "protocols.antibody.design.AntibodyDesignMover: Inner round: 1\n",
            "protocols.antibody.design.AntibodyDesignMover: Add min neighbors : L2\n",
            "protocols.antibody.design.AntibodyDesignMover: Add min neighbors : L3\n",
            "basic.io.database: [ WARNING ] Unable to locate database file /sampling/antibodies/antibody_database_rosetta_design.db\n",
            "antibody.database.AntibodyDatabaseManager: The antibody design database is not specified by the user.\n",
            "antibody.database.AntibodyDatabaseManager: Reading from: /usr/local/lib/python3.10/dist-packages/pyrosetta/database//sampling/antibodies/antibody_database_rosetta_design_north_paper.db\n",
            "antibody.database.AntibodyDatabaseManager: Loaded L1-17-1 with 16 datapoints.\n",
            "antibody.database.AntibodyDatabaseManager: Skipped L3-7-1 with 2 data point\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Data pre-loaded.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: applying prob task op\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Using conservative op for L3\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Using original bb sequence for conservative design.\n",
            "protocols.task_operations.ConservativeDesignOperation: Loading conservative mutational data\n",
            "core.pack.pack_rotamers: built 4229 rotamers at 88 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating PDInteractionGraph\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs start: 27989.8\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs end: 20909.2\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using explicit constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using interface constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Making an interface set with fixed chains\n",
            "protocols.analysis.InterfaceAnalyzerMover: Interface set residues total: 121\n",
            "protocols.analysis.InterfaceAnalyzerMover: NOT detecting disulfides in the separated pose.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 3622 rotamers at 121 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.analysis.InterfaceAnalyzerMover: included_nres: 979\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 900\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 901\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 926\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 927\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 966\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 967\n",
            "core.optimization.LineMinimizer: Inaccurate G in step= 3.72529e-09 Deriv= -19.5108 Finite Diff= 4.4257e+09\n",
            "antibody.design.GeneralAntibodyModeler: CDR min start: 20909.2\n",
            "antibody.design.GeneralAntibodyModeler: CDR min end: 5896.87\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using explicit constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using interface constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Making an interface set with fixed chains\n",
            "protocols.analysis.InterfaceAnalyzerMover: Interface set residues total: 126\n",
            "protocols.analysis.InterfaceAnalyzerMover: NOT detecting disulfides in the separated pose.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 3959 rotamers at 126 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.analysis.InterfaceAnalyzerMover: included_nres: 979\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: applying prob task op\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Using conservative op for L3\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Using original bb sequence for conservative design.\n",
            "core.pack.pack_rotamers: built 4565 rotamers at 92 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating PDInteractionGraph\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs start: 5896.87\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs end: 7647.24\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using explicit constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using interface constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Making an interface set with fixed chains\n",
            "protocols.analysis.InterfaceAnalyzerMover: Interface set residues total: 126\n",
            "protocols.analysis.InterfaceAnalyzerMover: NOT detecting disulfides in the separated pose.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 3889 rotamers at 126 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.analysis.InterfaceAnalyzerMover: included_nres: 979\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 900\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 901\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 926\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 927\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 966\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 967\n",
            "antibody.design.GeneralAntibodyModeler: CDR min start: 5896.87\n",
            "antibody.design.GeneralAntibodyModeler: CDR min end: 5893.39\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using explicit constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using interface constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Making an interface set with fixed chains\n",
            "protocols.analysis.InterfaceAnalyzerMover: Interface set residues total: 127\n",
            "protocols.analysis.InterfaceAnalyzerMover: NOT detecting disulfides in the separated pose.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 3977 rotamers at 127 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.analysis.InterfaceAnalyzerMover: included_nres: 979\n",
            "protocols.antibody.design.AntibodyDesignMover: Min Cycle complete\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using explicit constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using interface constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Making an interface set with fixed chains\n",
            "protocols.analysis.InterfaceAnalyzerMover: Interface set residues total: 126\n",
            "protocols.analysis.InterfaceAnalyzerMover: NOT detecting disulfides in the separated pose.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 3959 rotamers at 126 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.analysis.InterfaceAnalyzerMover: included_nres: 979\n",
            "protocols.antibody.design.AntibodyDesignMover:\n",
            "protocols.antibody.design.AntibodyDesignMover: SCORE 18 5083.31 0\n",
            "protocols.antibody.design.AntibodyDesignMover: FINAL 18 1333.26 1\n",
            "protocols.antibody.design.AntibodyDesignMover:\n",
            "protocols.antibody.design.AntibodyDesignMover: Last Accepted Score: 1333.26\n",
            "protocols.antibody.design.AntibodyDesignMover: Lowest        Score:1333.26\n",
            "protocols.antibody.design.AntibodyDesignMover:\n",
            "protocols.antibody.design.AntibodyDesignMover: Outer round: 19\n",
            "protocols.antibody.design.AntibodyDesignMover: Grafting CDR from cluster L1-10-2 fragment 1ay1L_L1_0001\n",
            "protocols.grafting.CCDEndsGraftMover: Start: 891 End: 902 NterO: 3 CterO: 3\n",
            "protocols.grafting.util: Superimposing overhang residues\n",
            "core.chemical.AtomICoor: [ WARNING ] IcoorAtomID::atom_id(): Cannot get atom_id for POLYMER_LOWER of residue MET 1.  Returning BOGUS ID instead.\n",
            "core.conformation.Residue: [ WARNING ] missing an atom: 1  H   that depends on a nonexistent polymer connection!\n",
            "core.conformation.Residue: [ WARNING ]  --> generating it using idealized coordinates.\n",
            "protocols.grafting.util: Deleting 10 residues from 892 to 901\n",
            "core.conformation.Conformation: Found disulfide between residues 771 845\n",
            "core.conformation.Conformation: Found disulfide between residues 891 945\n",
            "protocols.grafting.util: insert_point 891\n",
            "protocols.grafting.util: insert_start 892\n",
            "protocols.grafting.util: insert_end 901\n",
            "protocols.grafting.util: insert_length 10\n",
            "protocols.grafting.util: FOLD_TREE  EDGE 1 891 -1  EDGE 891 892 1  EDGE 892 962 -1\n",
            "protocols.grafting.GraftMoverBase: Insertion complete.\n",
            "protocols.grafting.AnchoredGraftMover: Setting default movemap\n",
            "protocols.grafting.CCDEndsGraftMover: Start: 891\n",
            "protocols.grafting.CCDEndsGraftMover: Original End: 902\n",
            "protocols.grafting.CCDEndsGraftMover: End: 902\n",
            "protocols.grafting.CCDEndsGraftMover: Insert Length: 10\n",
            "protocols.grafting.CCDEndsGraftMover: LOOP start: 890  stop: 893  cut: 891  size: 4  skip rate: 0  extended?: False\n",
            "protocols.grafting.CCDEndsGraftMover:\n",
            "protocols.grafting.CCDEndsGraftMover: LOOP start: 900  stop: 903  cut: 901  size: 4  skip rate: 0  extended?: False\n",
            "protocols.grafting.CCDEndsGraftMover:\n",
            "protocols.loops.FoldTreeFromLoopsWrapper: old foldtree FOLD_TREE  EDGE 1 891 -1  EDGE 891 902 1  EDGE 902 972 -1  EDGE 891 892 -2  C    N    EDGE 892 893 -2  C    N    EDGE 893 894 -2  C    N    EDGE 894 895 -2  C    N    EDGE 895 896 -2  C    N    EDGE 896 897 -2  C    N    EDGE 897 898 -2  C    N    EDGE 898 899 -2  C    N    EDGE 899 900 -2  C    N    EDGE 900 901 -2  C    N   \n",
            "New foldtree FOLD_TREE  EDGE 1 889 -1  EDGE 889 891 -1  EDGE 889 894 1  EDGE 894 892 -1  EDGE 894 899 -1  EDGE 899 901 -1  EDGE 899 904 2  EDGE 904 902 -1  EDGE 904 972 -1\n",
            "protocols.grafting.util: Loop: 1\n",
            "protocols.grafting.util: Add variant to: 891\n",
            "protocols.grafting.util: Loop: 2\n",
            "protocols.grafting.util: Add variant to: 901\n",
            "protocols.grafting.util: idealized  890\n",
            "protocols.grafting.util: idealized  891\n",
            "protocols.grafting.util: ideal 892\n",
            "protocols.grafting.util: ideal 901\n",
            "protocols.grafting.util: idealized 902\n",
            "protocols.grafting.util: idealized 903\n",
            "protocols.grafting.CCDEndsGraftMover: start 1650.77\n",
            "protocols.grafting.CCDEndsGraftMover: round 1\n",
            "protocols.loops.util: PepBondGeom: 42 L  C-N --  1.61374 max: 1.5\n",
            "protocols.grafting.CCDEndsGraftMover: 1 1383.84\n",
            "protocols.grafting.CCDEndsGraftMover: round 2\n",
            "protocols.grafting.CCDEndsGraftMover: Graft Closed early - returning\n",
            "protocols.grafting.CCDEndsGraftMover: 2 1274.55\n",
            "protocols.grafting.CCDEndsGraftMover: finish 1274.55\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 1560 rotamers at 42 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.grafting.CCDEndsGraftMover: Graft meets ideal geometry true\n",
            "protocols.grafting.CCDEndsGraftMover: Complete\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L1-10-2 data: 3 cutoff: 10\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: Adding general dihedral constraints for L1\n",
            "protocols.antibody.design.AntibodyDesignMover: Checking Geometry\n",
            "protocols.antibody.design.AntibodyDesignMover: DATA GRAFT_CLOSURE 1 rabd/my_ab.pdb L1 L1-10-2 1ay1L_L1_0001\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H1\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H2\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H3\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L1\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L2\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L3\n",
            "protocols.antibody.clusters.CDRClusterSet: Adding cacheable cluster data to pose\n",
            "protocols.antibody.design.util: Setting CDR sequence in NativeAntibodySeq\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: H1-13-1 data: 583 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/H1-13-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 26 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: H2-10-1 data: 328 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/H2-10-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 20 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: Adding general dihedral constraints for H3\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L1-10-2 data: 3 cutoff: 10\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: Adding general dihedral constraints for L1\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L2-8-1 data: 632 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/L2-8-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 16 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L3-7-1 data: 2 cutoff: 10\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: Adding general dihedral constraints for L3\n",
            "protocols.antibody.design.AntibodyDesignMover: Beginning MinProtocol\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using explicit constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using interface constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Making an interface set with fixed chains\n",
            "protocols.analysis.InterfaceAnalyzerMover: Interface set residues total: 92\n",
            "protocols.analysis.InterfaceAnalyzerMover: NOT detecting disulfides in the separated pose.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 2449 rotamers at 92 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.analysis.InterfaceAnalyzerMover: included_nres: 972\n",
            "protocols.antibody.design.AntibodyDesignMover: Inner round: 1\n",
            "protocols.antibody.design.AntibodyDesignMover: Add min neighbors : L2\n",
            "protocols.antibody.design.AntibodyDesignMover: Add min neighbors : L3\n",
            "basic.io.database: [ WARNING ] Unable to locate database file /sampling/antibodies/antibody_database_rosetta_design.db\n",
            "antibody.database.AntibodyDatabaseManager: The antibody design database is not specified by the user.\n",
            "antibody.database.AntibodyDatabaseManager: Reading from: /usr/local/lib/python3.10/dist-packages/pyrosetta/database//sampling/antibodies/antibody_database_rosetta_design_north_paper.db\n",
            "antibody.database.AntibodyDatabaseManager: Skipped L1-10-2 with 2 data point\n",
            "antibody.database.AntibodyDatabaseManager: Skipped L3-7-1 with 2 data point\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Data pre-loaded.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: applying prob task op\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Using conservative op for L1\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Using conservative op for L3\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Using original bb sequence for conservative design.\n",
            "protocols.task_operations.ConservativeDesignOperation: Loading conservative mutational data\n",
            "core.pack.pack_rotamers: built 5932 rotamers at 70 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating PDInteractionGraph\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs start: 8311.91\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs end: 2444.48\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using explicit constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using interface constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Making an interface set with fixed chains\n",
            "protocols.analysis.InterfaceAnalyzerMover: Interface set residues total: 89\n",
            "protocols.analysis.InterfaceAnalyzerMover: NOT detecting disulfides in the separated pose.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 2526 rotamers at 89 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.analysis.InterfaceAnalyzerMover: included_nres: 972\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 896\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 897\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 919\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 920\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 959\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 960\n",
            "antibody.design.GeneralAntibodyModeler: CDR min start: 2444.48\n",
            "antibody.design.GeneralAntibodyModeler: CDR min end: 1956.88\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using explicit constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using interface constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Making an interface set with fixed chains\n",
            "protocols.analysis.InterfaceAnalyzerMover: Interface set residues total: 89\n",
            "protocols.analysis.InterfaceAnalyzerMover: NOT detecting disulfides in the separated pose.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 2501 rotamers at 89 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.analysis.InterfaceAnalyzerMover: included_nres: 972\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: applying prob task op\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Using conservative op for L1\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Using conservative op for L3\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Using original bb sequence for conservative design.\n",
            "core.pack.pack_rotamers: built 6593 rotamers at 68 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating PDInteractionGraph\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs start: 1956.88\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs end: 2138.65\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using explicit constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using interface constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Making an interface set with fixed chains\n",
            "protocols.analysis.InterfaceAnalyzerMover: Interface set residues total: 89\n",
            "protocols.analysis.InterfaceAnalyzerMover: NOT detecting disulfides in the separated pose.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 2317 rotamers at 89 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.analysis.InterfaceAnalyzerMover: included_nres: 972\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 896\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 897\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 919\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 920\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 959\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 960\n",
            "antibody.design.GeneralAntibodyModeler: CDR min start: 1956.88\n",
            "antibody.design.GeneralAntibodyModeler: CDR min end: 1956.22\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using explicit constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using interface constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Making an interface set with fixed chains\n",
            "protocols.analysis.InterfaceAnalyzerMover: Interface set residues total: 89\n",
            "protocols.analysis.InterfaceAnalyzerMover: NOT detecting disulfides in the separated pose.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 2501 rotamers at 89 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.analysis.InterfaceAnalyzerMover: included_nres: 972\n",
            "protocols.antibody.design.AntibodyDesignMover: Min Cycle complete\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using explicit constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using interface constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Making an interface set with fixed chains\n",
            "protocols.analysis.InterfaceAnalyzerMover: Interface set residues total: 89\n",
            "protocols.analysis.InterfaceAnalyzerMover: NOT detecting disulfides in the separated pose.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 2501 rotamers at 89 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.analysis.InterfaceAnalyzerMover: included_nres: 972\n",
            "protocols.antibody.design.AntibodyDesignMover:\n",
            "protocols.antibody.design.AntibodyDesignMover: SCORE 19 1288.01 1\n",
            "protocols.antibody.design.AntibodyDesignMover: FINAL 19 1288.01 1\n",
            "protocols.antibody.design.AntibodyDesignMover:\n",
            "protocols.antibody.design.AntibodyDesignMover: Last Accepted Score: 1288.01\n",
            "protocols.antibody.design.AntibodyDesignMover: Lowest        Score:1288.01\n",
            "protocols.antibody.design.AntibodyDesignMover:\n",
            "protocols.antibody.design.AntibodyDesignMover: Outer round: 20\n",
            "protocols.antibody.design.AntibodyDesignMover: Grafting CDR from cluster L1-17-1 fragment 1h3pL_L1_0001\n",
            "protocols.grafting.CCDEndsGraftMover: Start: 891 End: 902 NterO: 3 CterO: 3\n",
            "protocols.grafting.util: Superimposing overhang residues\n",
            "core.chemical.AtomICoor: [ WARNING ] IcoorAtomID::atom_id(): Cannot get atom_id for POLYMER_LOWER of residue MET 1.  Returning BOGUS ID instead.\n",
            "core.conformation.Residue: [ WARNING ] missing an atom: 1  H   that depends on a nonexistent polymer connection!\n",
            "core.conformation.Residue: [ WARNING ]  --> generating it using idealized coordinates.\n",
            "protocols.grafting.util: Deleting 10 residues from 892 to 901\n",
            "core.conformation.Conformation: Found disulfide between residues 771 845\n",
            "core.conformation.Conformation: Found disulfide between residues 891 945\n",
            "protocols.grafting.util: insert_point 891\n",
            "protocols.grafting.util: insert_start 892\n",
            "protocols.grafting.util: insert_end 908\n",
            "protocols.grafting.util: insert_length 17\n",
            "protocols.grafting.util: FOLD_TREE  EDGE 1 891 -1  EDGE 891 892 1  EDGE 892 962 -1\n",
            "protocols.grafting.GraftMoverBase: Insertion complete.\n",
            "protocols.grafting.AnchoredGraftMover: Setting default movemap\n",
            "protocols.grafting.CCDEndsGraftMover: Start: 891\n",
            "protocols.grafting.CCDEndsGraftMover: Original End: 902\n",
            "protocols.grafting.CCDEndsGraftMover: End: 909\n",
            "protocols.grafting.CCDEndsGraftMover: Insert Length: 17\n",
            "protocols.grafting.CCDEndsGraftMover: LOOP start: 890  stop: 893  cut: 891  size: 4  skip rate: 0  extended?: False\n",
            "protocols.grafting.CCDEndsGraftMover:\n",
            "protocols.grafting.CCDEndsGraftMover: LOOP start: 907  stop: 910  cut: 908  size: 4  skip rate: 0  extended?: False\n",
            "protocols.grafting.CCDEndsGraftMover:\n",
            "protocols.loops.FoldTreeFromLoopsWrapper: old foldtree FOLD_TREE  EDGE 1 891 -1  EDGE 891 909 1  EDGE 909 979 -1  EDGE 891 892 -2  C    N    EDGE 892 893 -2  C    N    EDGE 893 894 -2  C    N    EDGE 894 895 -2  C    N    EDGE 895 896 -2  C    N    EDGE 896 897 -2  C    N    EDGE 897 898 -2  C    N    EDGE 898 899 -2  C    N    EDGE 899 900 -2  C    N    EDGE 900 901 -2  C    N    EDGE 901 902 -2  C    N    EDGE 902 903 -2  C    N    EDGE 903 904 -2  C    N    EDGE 904 905 -2  C    N    EDGE 905 906 -2  C    N    EDGE 906 907 -2  C    N    EDGE 907 908 -2  C    N   \n",
            "New foldtree FOLD_TREE  EDGE 1 889 -1  EDGE 889 891 -1  EDGE 889 894 1  EDGE 894 892 -1  EDGE 894 906 -1  EDGE 906 908 -1  EDGE 906 911 2  EDGE 911 909 -1  EDGE 911 979 -1\n",
            "protocols.grafting.util: Loop: 1\n",
            "protocols.grafting.util: Add variant to: 891\n",
            "protocols.grafting.util: Loop: 2\n",
            "protocols.grafting.util: Add variant to: 908\n",
            "protocols.grafting.util: idealized  890\n",
            "protocols.grafting.util: idealized  891\n",
            "protocols.grafting.util: ideal 892\n",
            "protocols.grafting.util: ideal 908\n",
            "protocols.grafting.util: idealized 909\n",
            "protocols.grafting.util: idealized 910\n",
            "protocols.grafting.CCDEndsGraftMover: start 1740.74\n",
            "protocols.grafting.CCDEndsGraftMover: round 1\n",
            "protocols.grafting.CCDEndsGraftMover: Graft Closed early - returning\n",
            "protocols.grafting.CCDEndsGraftMover: 1 1479.97\n",
            "protocols.grafting.CCDEndsGraftMover: finish 1479.97\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 2086 rotamers at 56 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.grafting.CCDEndsGraftMover: Graft meets ideal geometry true\n",
            "protocols.grafting.CCDEndsGraftMover: Complete\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L1-17-1 data: 49 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/L1-17-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 34 constraints\n",
            "protocols.antibody.design.AntibodyDesignMover: Checking Geometry\n",
            "protocols.antibody.design.AntibodyDesignMover: DATA GRAFT_CLOSURE 1 rabd/my_ab.pdb L1 L1-17-1 1h3pL_L1_0001\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H1\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H2\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H3\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L1\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L2\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L3\n",
            "protocols.antibody.clusters.CDRClusterSet: Adding cacheable cluster data to pose\n",
            "protocols.antibody.design.util: Setting CDR sequence in NativeAntibodySeq\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: H1-13-1 data: 583 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/H1-13-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 26 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: H2-10-1 data: 328 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/H2-10-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 20 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: Adding general dihedral constraints for H3\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L1-17-1 data: 49 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/L1-17-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 34 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L2-8-1 data: 632 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/L2-8-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 16 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L3-7-1 data: 2 cutoff: 10\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: Adding general dihedral constraints for L3\n",
            "protocols.antibody.design.AntibodyDesignMover: Beginning MinProtocol\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using explicit constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using interface constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Making an interface set with fixed chains\n",
            "protocols.analysis.InterfaceAnalyzerMover: Interface set residues total: 117\n",
            "protocols.analysis.InterfaceAnalyzerMover: NOT detecting disulfides in the separated pose.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 3591 rotamers at 117 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.analysis.InterfaceAnalyzerMover: included_nres: 979\n",
            "protocols.antibody.design.AntibodyDesignMover: Inner round: 1\n",
            "protocols.antibody.design.AntibodyDesignMover: Add min neighbors : L2\n",
            "protocols.antibody.design.AntibodyDesignMover: Add min neighbors : L3\n",
            "basic.io.database: [ WARNING ] Unable to locate database file /sampling/antibodies/antibody_database_rosetta_design.db\n",
            "antibody.database.AntibodyDatabaseManager: The antibody design database is not specified by the user.\n",
            "antibody.database.AntibodyDatabaseManager: Reading from: /usr/local/lib/python3.10/dist-packages/pyrosetta/database//sampling/antibodies/antibody_database_rosetta_design_north_paper.db\n",
            "antibody.database.AntibodyDatabaseManager: Loaded L1-17-1 with 16 datapoints.\n",
            "antibody.database.AntibodyDatabaseManager: Skipped L3-7-1 with 2 data point\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Data pre-loaded.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: applying prob task op\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Using conservative op for L3\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Using original bb sequence for conservative design.\n",
            "protocols.task_operations.ConservativeDesignOperation: Loading conservative mutational data\n",
            "core.pack.pack_rotamers: built 4368 rotamers at 88 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating PDInteractionGraph\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs start: 26157.2\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs end: 21608.7\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using explicit constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using interface constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Making an interface set with fixed chains\n",
            "protocols.analysis.InterfaceAnalyzerMover: Interface set residues total: 117\n",
            "protocols.analysis.InterfaceAnalyzerMover: NOT detecting disulfides in the separated pose.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 3865 rotamers at 117 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.analysis.InterfaceAnalyzerMover: included_nres: 979\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 900\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 901\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 926\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 927\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 966\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 967\n",
            "antibody.design.GeneralAntibodyModeler: CDR min start: 21608.7\n",
            "antibody.design.GeneralAntibodyModeler: CDR min end: 4875.22\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using explicit constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using interface constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Making an interface set with fixed chains\n",
            "protocols.analysis.InterfaceAnalyzerMover: Interface set residues total: 100\n",
            "protocols.analysis.InterfaceAnalyzerMover: NOT detecting disulfides in the separated pose.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 2886 rotamers at 100 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.analysis.InterfaceAnalyzerMover: included_nres: 979\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: applying prob task op\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Using conservative op for L3\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Using original bb sequence for conservative design.\n",
            "core.pack.pack_rotamers: built 5088 rotamers at 81 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating PDInteractionGraph\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs start: 4875.22\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs end: 2671.65\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using explicit constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using interface constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Making an interface set with fixed chains\n",
            "protocols.analysis.InterfaceAnalyzerMover: Interface set residues total: 100\n",
            "protocols.analysis.InterfaceAnalyzerMover: NOT detecting disulfides in the separated pose.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 2759 rotamers at 100 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.analysis.InterfaceAnalyzerMover: included_nres: 979\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 900\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 901\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 926\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 927\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 966\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 967\n",
            "antibody.design.GeneralAntibodyModeler: CDR min start: 2671.65\n",
            "antibody.design.GeneralAntibodyModeler: CDR min end: 2096.31\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using explicit constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using interface constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Making an interface set with fixed chains\n",
            "protocols.analysis.InterfaceAnalyzerMover: Interface set residues total: 97\n",
            "protocols.analysis.InterfaceAnalyzerMover: NOT detecting disulfides in the separated pose.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 2939 rotamers at 97 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.analysis.InterfaceAnalyzerMover: included_nres: 979\n",
            "protocols.antibody.design.AntibodyDesignMover: Min Cycle complete\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using explicit constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using interface constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Making an interface set with fixed chains\n",
            "protocols.analysis.InterfaceAnalyzerMover: Interface set residues total: 97\n",
            "protocols.analysis.InterfaceAnalyzerMover: NOT detecting disulfides in the separated pose.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 2939 rotamers at 97 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.analysis.InterfaceAnalyzerMover: included_nres: 979\n",
            "protocols.antibody.design.AntibodyDesignMover:\n",
            "protocols.antibody.design.AntibodyDesignMover: SCORE 20 1320.58 0\n",
            "protocols.antibody.design.AntibodyDesignMover: FINAL 20 1288.01 1\n",
            "protocols.antibody.design.AntibodyDesignMover:\n",
            "protocols.antibody.design.AntibodyDesignMover: Last Accepted Score: 1288.01\n",
            "protocols.antibody.design.AntibodyDesignMover: Lowest        Score:1288.01\n",
            "protocols.antibody.design.AntibodyDesignMover:\n",
            "protocols.antibody.design.AntibodyDesignMover: Outer round: 21\n",
            "protocols.antibody.design.AntibodyDesignMover: Grafting CDR from cluster L3-7-1 fragment 1dfbL_L3_0001\n",
            "protocols.grafting.CCDEndsGraftMover: Start: 955 End: 963 NterO: 3 CterO: 3\n",
            "protocols.grafting.util: Superimposing overhang residues\n",
            "core.chemical.AtomICoor: [ WARNING ] IcoorAtomID::atom_id(): Cannot get atom_id for POLYMER_LOWER of residue TYR 1.  Returning BOGUS ID instead.\n",
            "core.conformation.Residue: [ WARNING ] missing an atom: 1  H   that depends on a nonexistent polymer connection!\n",
            "core.conformation.Residue: [ WARNING ]  --> generating it using idealized coordinates.\n",
            "protocols.grafting.util: Deleting 7 residues from 956 to 962\n",
            "core.conformation.Conformation: Found disulfide between residues 771 845\n",
            "core.conformation.Conformation: Found disulfide between residues 891 955\n",
            "protocols.grafting.util: insert_point 955\n",
            "protocols.grafting.util: insert_start 956\n",
            "protocols.grafting.util: insert_end 962\n",
            "protocols.grafting.util: insert_length 7\n",
            "protocols.grafting.util: FOLD_TREE  EDGE 1 955 -1  EDGE 955 956 1  EDGE 956 965 -1\n",
            "protocols.grafting.GraftMoverBase: Insertion complete.\n",
            "protocols.grafting.AnchoredGraftMover: Setting default movemap\n",
            "protocols.grafting.CCDEndsGraftMover: Start: 955\n",
            "protocols.grafting.CCDEndsGraftMover: Original End: 963\n",
            "protocols.grafting.CCDEndsGraftMover: End: 963\n",
            "protocols.grafting.CCDEndsGraftMover: Insert Length: 7\n",
            "protocols.grafting.CCDEndsGraftMover: LOOP start: 954  stop: 957  cut: 955  size: 4  skip rate: 0  extended?: False\n",
            "protocols.grafting.CCDEndsGraftMover:\n",
            "protocols.grafting.CCDEndsGraftMover: LOOP start: 961  stop: 964  cut: 962  size: 4  skip rate: 0  extended?: False\n",
            "protocols.grafting.CCDEndsGraftMover:\n",
            "protocols.loops.FoldTreeFromLoopsWrapper: old foldtree FOLD_TREE  EDGE 1 955 -1  EDGE 955 963 1  EDGE 963 972 -1  EDGE 955 956 -2  C    N    EDGE 956 957 -2  C    N    EDGE 957 958 -2  C    N    EDGE 958 959 -2  C    N    EDGE 959 960 -2  C    N    EDGE 960 961 -2  C    N    EDGE 961 962 -2  C    N   \n",
            "New foldtree FOLD_TREE  EDGE 1 953 -1  EDGE 953 955 -1  EDGE 953 958 1  EDGE 958 956 -1  EDGE 958 960 -1  EDGE 960 962 -1  EDGE 960 965 2  EDGE 965 963 -1  EDGE 965 972 -1\n",
            "protocols.grafting.util: Loop: 1\n",
            "protocols.grafting.util: Add variant to: 955\n",
            "protocols.grafting.util: Loop: 2\n",
            "protocols.grafting.util: Add variant to: 962\n",
            "protocols.grafting.util: idealized  954\n",
            "protocols.grafting.util: idealized  955\n",
            "protocols.grafting.util: ideal 956\n",
            "protocols.grafting.util: ideal 962\n",
            "protocols.grafting.util: idealized 963\n",
            "protocols.grafting.util: idealized 964\n",
            "protocols.grafting.CCDEndsGraftMover: start 1574.55\n",
            "protocols.grafting.CCDEndsGraftMover: round 1\n",
            "protocols.loops.util: PepBondGeom: 138 L  C-N --  1.50622 max: 1.5\n",
            "protocols.grafting.CCDEndsGraftMover: 1 1280.47\n",
            "protocols.grafting.CCDEndsGraftMover: round 2\n",
            "protocols.grafting.CCDEndsGraftMover: Graft Closed early - returning\n",
            "protocols.grafting.CCDEndsGraftMover: 2 1252.2\n",
            "protocols.grafting.CCDEndsGraftMover: finish 1252.2\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 810 rotamers at 31 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.grafting.CCDEndsGraftMover: Graft meets ideal geometry true\n",
            "protocols.grafting.CCDEndsGraftMover: Complete\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L3-7-1 data: 2 cutoff: 10\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: Adding general dihedral constraints for L3\n",
            "protocols.antibody.design.AntibodyDesignMover: Checking Geometry\n",
            "protocols.antibody.design.AntibodyDesignMover: DATA GRAFT_CLOSURE 1 rabd/my_ab.pdb L3 L3-7-1 1dfbL_L3_0001\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H1\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H2\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H3\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L1\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L2\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L3\n",
            "protocols.antibody.clusters.CDRClusterSet: Adding cacheable cluster data to pose\n",
            "protocols.antibody.design.util: Setting CDR sequence in NativeAntibodySeq\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: H1-13-1 data: 583 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/H1-13-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 26 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: H2-10-1 data: 328 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/H2-10-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 20 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: Adding general dihedral constraints for H3\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L1-10-2 data: 3 cutoff: 10\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: Adding general dihedral constraints for L1\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L2-8-1 data: 632 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/L2-8-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 16 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L3-7-1 data: 2 cutoff: 10\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: Adding general dihedral constraints for L3\n",
            "protocols.antibody.design.AntibodyDesignMover: Beginning MinProtocol\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using explicit constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using interface constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Making an interface set with fixed chains\n",
            "protocols.analysis.InterfaceAnalyzerMover: Interface set residues total: 89\n",
            "protocols.analysis.InterfaceAnalyzerMover: NOT detecting disulfides in the separated pose.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 2570 rotamers at 89 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.analysis.InterfaceAnalyzerMover: included_nres: 972\n",
            "protocols.antibody.design.AntibodyDesignMover: Inner round: 1\n",
            "protocols.antibody.design.AntibodyDesignMover: Add min neighbors : L1\n",
            "protocols.antibody.design.AntibodyDesignMover: Add min neighbors : H3\n",
            "basic.io.database: [ WARNING ] Unable to locate database file /sampling/antibodies/antibody_database_rosetta_design.db\n",
            "antibody.database.AntibodyDatabaseManager: The antibody design database is not specified by the user.\n",
            "antibody.database.AntibodyDatabaseManager: Reading from: /usr/local/lib/python3.10/dist-packages/pyrosetta/database//sampling/antibodies/antibody_database_rosetta_design_north_paper.db\n",
            "antibody.database.AntibodyDatabaseManager: Skipped L1-10-2 with 2 data point\n",
            "antibody.database.AntibodyDatabaseManager: Skipped L3-7-1 with 2 data point\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Data pre-loaded.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: applying prob task op\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Using conservative op for L1\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Using conservative op for L3\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Using original bb sequence for conservative design.\n",
            "protocols.task_operations.ConservativeDesignOperation: Loading conservative mutational data\n",
            "core.pack.pack_rotamers: built 7432 rotamers at 83 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating PDInteractionGraph\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs start: 5707.67\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs end: 2530.11\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using explicit constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using interface constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Making an interface set with fixed chains\n",
            "protocols.analysis.InterfaceAnalyzerMover: Interface set residues total: 88\n",
            "protocols.analysis.InterfaceAnalyzerMover: NOT detecting disulfides in the separated pose.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 2341 rotamers at 88 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.analysis.InterfaceAnalyzerMover: included_nres: 972\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 852\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 853\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 896\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 897\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 959\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 960\n",
            "antibody.design.GeneralAntibodyModeler: CDR min start: 2530.11\n",
            "antibody.design.GeneralAntibodyModeler: CDR min end: 1953.94\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using explicit constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using interface constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Making an interface set with fixed chains\n",
            "protocols.analysis.InterfaceAnalyzerMover: Interface set residues total: 88\n",
            "protocols.analysis.InterfaceAnalyzerMover: NOT detecting disulfides in the separated pose.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 2487 rotamers at 88 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.analysis.InterfaceAnalyzerMover: included_nres: 972\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: applying prob task op\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Using conservative op for L1\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Using conservative op for L3\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Using original bb sequence for conservative design.\n",
            "core.pack.pack_rotamers: built 6727 rotamers at 83 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating PDInteractionGraph\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs start: 1953.94\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs end: 2246.92\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using explicit constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using interface constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Making an interface set with fixed chains\n",
            "protocols.analysis.InterfaceAnalyzerMover: Interface set residues total: 88\n",
            "protocols.analysis.InterfaceAnalyzerMover: NOT detecting disulfides in the separated pose.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 2349 rotamers at 88 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.analysis.InterfaceAnalyzerMover: included_nres: 972\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 852\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 853\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 896\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 897\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 959\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 960\n",
            "antibody.design.GeneralAntibodyModeler: CDR min start: 1953.94\n",
            "antibody.design.GeneralAntibodyModeler: CDR min end: 1953.13\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using explicit constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using interface constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Making an interface set with fixed chains\n",
            "protocols.analysis.InterfaceAnalyzerMover: Interface set residues total: 88\n",
            "protocols.analysis.InterfaceAnalyzerMover: NOT detecting disulfides in the separated pose.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 2484 rotamers at 88 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.analysis.InterfaceAnalyzerMover: included_nres: 972\n",
            "protocols.antibody.design.AntibodyDesignMover: Min Cycle complete\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using explicit constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using interface constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Making an interface set with fixed chains\n",
            "protocols.analysis.InterfaceAnalyzerMover: Interface set residues total: 88\n",
            "protocols.analysis.InterfaceAnalyzerMover: NOT detecting disulfides in the separated pose.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 2487 rotamers at 88 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.analysis.InterfaceAnalyzerMover: included_nres: 972\n",
            "protocols.antibody.design.AntibodyDesignMover:\n",
            "protocols.antibody.design.AntibodyDesignMover: SCORE 21 1305.32 0\n",
            "protocols.antibody.design.AntibodyDesignMover: FINAL 21 1288.01 1\n",
            "protocols.antibody.design.AntibodyDesignMover:\n",
            "protocols.antibody.design.AntibodyDesignMover: Last Accepted Score: 1288.01\n",
            "protocols.antibody.design.AntibodyDesignMover: Lowest        Score:1288.01\n",
            "protocols.antibody.design.AntibodyDesignMover:\n",
            "protocols.antibody.design.AntibodyDesignMover: Outer round: 22\n",
            "protocols.antibody.design.AntibodyDesignMover: Grafting CDR from cluster L1-16-1 fragment 1t2qL_L1_0001\n",
            "protocols.grafting.CCDEndsGraftMover: Start: 891 End: 902 NterO: 3 CterO: 3\n",
            "protocols.grafting.util: Superimposing overhang residues\n",
            "core.chemical.AtomICoor: [ WARNING ] IcoorAtomID::atom_id(): Cannot get atom_id for POLYMER_LOWER of residue ILE 1.  Returning BOGUS ID instead.\n",
            "core.conformation.Residue: [ WARNING ] missing an atom: 1  H   that depends on a nonexistent polymer connection!\n",
            "core.conformation.Residue: [ WARNING ]  --> generating it using idealized coordinates.\n",
            "protocols.grafting.util: Deleting 10 residues from 892 to 901\n",
            "core.conformation.Conformation: Found disulfide between residues 771 845\n",
            "core.conformation.Conformation: Found disulfide between residues 891 945\n",
            "protocols.grafting.util: insert_point 891\n",
            "protocols.grafting.util: insert_start 892\n",
            "protocols.grafting.util: insert_end 907\n",
            "protocols.grafting.util: insert_length 16\n",
            "protocols.grafting.util: FOLD_TREE  EDGE 1 891 -1  EDGE 891 892 1  EDGE 892 962 -1\n",
            "protocols.grafting.GraftMoverBase: Insertion complete.\n",
            "protocols.grafting.AnchoredGraftMover: Setting default movemap\n",
            "protocols.grafting.CCDEndsGraftMover: Start: 891\n",
            "protocols.grafting.CCDEndsGraftMover: Original End: 902\n",
            "protocols.grafting.CCDEndsGraftMover: End: 908\n",
            "protocols.grafting.CCDEndsGraftMover: Insert Length: 16\n",
            "protocols.grafting.CCDEndsGraftMover: LOOP start: 890  stop: 893  cut: 891  size: 4  skip rate: 0  extended?: False\n",
            "protocols.grafting.CCDEndsGraftMover:\n",
            "protocols.grafting.CCDEndsGraftMover: LOOP start: 906  stop: 909  cut: 907  size: 4  skip rate: 0  extended?: False\n",
            "protocols.grafting.CCDEndsGraftMover:\n",
            "protocols.loops.FoldTreeFromLoopsWrapper: old foldtree FOLD_TREE  EDGE 1 891 -1  EDGE 891 908 1  EDGE 908 978 -1  EDGE 891 892 -2  C    N    EDGE 892 893 -2  C    N    EDGE 893 894 -2  C    N    EDGE 894 895 -2  C    N    EDGE 895 896 -2  C    N    EDGE 896 897 -2  C    N    EDGE 897 898 -2  C    N    EDGE 898 899 -2  C    N    EDGE 899 900 -2  C    N    EDGE 900 901 -2  C    N    EDGE 901 902 -2  C    N    EDGE 902 903 -2  C    N    EDGE 903 904 -2  C    N    EDGE 904 905 -2  C    N    EDGE 905 906 -2  C    N    EDGE 906 907 -2  C    N   \n",
            "New foldtree FOLD_TREE  EDGE 1 889 -1  EDGE 889 891 -1  EDGE 889 894 1  EDGE 894 892 -1  EDGE 894 905 -1  EDGE 905 907 -1  EDGE 905 910 2  EDGE 910 908 -1  EDGE 910 978 -1\n",
            "protocols.grafting.util: Loop: 1\n",
            "protocols.grafting.util: Add variant to: 891\n",
            "protocols.grafting.util: Loop: 2\n",
            "protocols.grafting.util: Add variant to: 907\n",
            "protocols.grafting.util: idealized  890\n",
            "protocols.grafting.util: idealized  891\n",
            "protocols.grafting.util: ideal 892\n",
            "protocols.grafting.util: ideal 907\n",
            "protocols.grafting.util: idealized 908\n",
            "protocols.grafting.util: idealized 909\n",
            "protocols.grafting.CCDEndsGraftMover: start 1665.01\n",
            "protocols.grafting.CCDEndsGraftMover: round 1\n",
            "protocols.loops.util: PepBondGeom: 907 - 42 L  Ca-C-N --  99.5 (min): 158.004 : 134.5 (max)\n",
            "protocols.grafting.CCDEndsGraftMover: 1 1453.33\n",
            "protocols.grafting.CCDEndsGraftMover: round 2\n",
            "protocols.loops.util: PepBondGeom: 907 - 42 L  Ca-C-N --  99.5 (min): 163.511 : 134.5 (max)\n",
            "protocols.grafting.CCDEndsGraftMover: 2 1447.48\n",
            "protocols.grafting.CCDEndsGraftMover: round 3\n",
            "protocols.loops.util: PepBondGeom: 907 - 42 L  Ca-C-N --  99.5 (min): 155.858 : 134.5 (max)\n",
            "protocols.grafting.CCDEndsGraftMover: round 4\n",
            "protocols.grafting.CCDEndsGraftMover: Graft Closed early - returning\n",
            "protocols.grafting.CCDEndsGraftMover: 4 1349.88\n",
            "protocols.grafting.CCDEndsGraftMover: finish 1349.88\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 2033 rotamers at 53 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.grafting.CCDEndsGraftMover: Graft meets ideal geometry true\n",
            "protocols.grafting.CCDEndsGraftMover: Complete\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L1-16-1 data: 117 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/L1-16-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 32 constraints\n",
            "protocols.antibody.design.AntibodyDesignMover: Checking Geometry\n",
            "protocols.antibody.design.AntibodyDesignMover: DATA GRAFT_CLOSURE 1 rabd/my_ab.pdb L1 L1-16-1 1t2qL_L1_0001\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H1\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H2\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H3\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L1\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L2\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L3\n",
            "protocols.antibody.clusters.CDRClusterSet: Adding cacheable cluster data to pose\n",
            "protocols.antibody.design.util: Setting CDR sequence in NativeAntibodySeq\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: H1-13-1 data: 583 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/H1-13-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 26 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: H2-10-1 data: 328 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/H2-10-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 20 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: Adding general dihedral constraints for H3\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L1-16-1 data: 117 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/L1-16-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 32 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L2-8-1 data: 632 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/L2-8-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 16 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L3-7-1 data: 2 cutoff: 10\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: Adding general dihedral constraints for L3\n",
            "protocols.antibody.design.AntibodyDesignMover: Beginning MinProtocol\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using explicit constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using interface constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Making an interface set with fixed chains\n",
            "protocols.analysis.InterfaceAnalyzerMover: Interface set residues total: 115\n",
            "protocols.analysis.InterfaceAnalyzerMover: NOT detecting disulfides in the separated pose.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 3119 rotamers at 115 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.analysis.InterfaceAnalyzerMover: included_nres: 978\n",
            "protocols.antibody.design.AntibodyDesignMover: Inner round: 1\n",
            "protocols.antibody.design.AntibodyDesignMover: Add min neighbors : L2\n",
            "protocols.antibody.design.AntibodyDesignMover: Add min neighbors : L3\n",
            "basic.io.database: [ WARNING ] Unable to locate database file /sampling/antibodies/antibody_database_rosetta_design.db\n",
            "antibody.database.AntibodyDatabaseManager: The antibody design database is not specified by the user.\n",
            "antibody.database.AntibodyDatabaseManager: Reading from: /usr/local/lib/python3.10/dist-packages/pyrosetta/database//sampling/antibodies/antibody_database_rosetta_design_north_paper.db\n",
            "antibody.database.AntibodyDatabaseManager: Loaded L1-16-1 with 47 datapoints.\n",
            "antibody.database.AntibodyDatabaseManager: Skipped L3-7-1 with 2 data point\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Data pre-loaded.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: applying prob task op\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Using conservative op for L3\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Using original bb sequence for conservative design.\n",
            "protocols.task_operations.ConservativeDesignOperation: Loading conservative mutational data\n",
            "core.pack.pack_rotamers: built 3895 rotamers at 86 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating PDInteractionGraph\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs start: 22978.7\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs end: 14367.4\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using explicit constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using interface constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Making an interface set with fixed chains\n",
            "protocols.analysis.InterfaceAnalyzerMover: Interface set residues total: 115\n",
            "protocols.analysis.InterfaceAnalyzerMover: NOT detecting disulfides in the separated pose.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 3283 rotamers at 115 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.analysis.InterfaceAnalyzerMover: included_nres: 978\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 899\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 900\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 925\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 926\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 965\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 966\n",
            "antibody.design.GeneralAntibodyModeler: CDR min start: 14367.4\n",
            "antibody.design.GeneralAntibodyModeler: CDR min end: 2929.27\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using explicit constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using interface constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Making an interface set with fixed chains\n",
            "protocols.analysis.InterfaceAnalyzerMover: Interface set residues total: 102\n",
            "protocols.analysis.InterfaceAnalyzerMover: NOT detecting disulfides in the separated pose.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 3132 rotamers at 102 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.analysis.InterfaceAnalyzerMover: included_nres: 978\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: applying prob task op\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Using conservative op for L3\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Using original bb sequence for conservative design.\n",
            "core.pack.pack_rotamers: built 4167 rotamers at 85 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating PDInteractionGraph\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs start: 2929.27\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs end: 2858.58\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using explicit constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using interface constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Making an interface set with fixed chains\n",
            "protocols.analysis.InterfaceAnalyzerMover: Interface set residues total: 102\n",
            "protocols.analysis.InterfaceAnalyzerMover: NOT detecting disulfides in the separated pose.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 3083 rotamers at 102 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.analysis.InterfaceAnalyzerMover: included_nres: 978\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 899\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 900\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 925\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 926\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 965\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 966\n",
            "antibody.design.GeneralAntibodyModeler: CDR min start: 2858.58\n",
            "antibody.design.GeneralAntibodyModeler: CDR min end: 2083.83\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using explicit constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using interface constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Making an interface set with fixed chains\n",
            "protocols.analysis.InterfaceAnalyzerMover: Interface set residues total: 100\n",
            "protocols.analysis.InterfaceAnalyzerMover: NOT detecting disulfides in the separated pose.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 2859 rotamers at 100 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.analysis.InterfaceAnalyzerMover: included_nres: 978\n",
            "protocols.antibody.design.AntibodyDesignMover: Min Cycle complete\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using explicit constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using interface constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Making an interface set with fixed chains\n",
            "protocols.analysis.InterfaceAnalyzerMover: Interface set residues total: 100\n",
            "protocols.analysis.InterfaceAnalyzerMover: NOT detecting disulfides in the separated pose.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 2859 rotamers at 100 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.analysis.InterfaceAnalyzerMover: included_nres: 978\n",
            "protocols.antibody.design.AntibodyDesignMover:\n",
            "protocols.antibody.design.AntibodyDesignMover: SCORE 22 1345.94 0\n",
            "protocols.antibody.design.AntibodyDesignMover: FINAL 22 1288.01 1\n",
            "protocols.antibody.design.AntibodyDesignMover:\n",
            "protocols.antibody.design.AntibodyDesignMover: Last Accepted Score: 1288.01\n",
            "protocols.antibody.design.AntibodyDesignMover: Lowest        Score:1288.01\n",
            "protocols.antibody.design.AntibodyDesignMover:\n",
            "protocols.antibody.design.AntibodyDesignMover: Outer round: 23\n",
            "protocols.antibody.design.AntibodyDesignMover: Grafting CDR from cluster L1-15-2 fragment 3dggA_L1_0001\n",
            "protocols.grafting.CCDEndsGraftMover: Start: 891 End: 902 NterO: 3 CterO: 3\n",
            "protocols.grafting.util: Superimposing overhang residues\n",
            "core.chemical.AtomICoor: [ WARNING ] IcoorAtomID::atom_id(): Cannot get atom_id for POLYMER_LOWER of residue ILE 1.  Returning BOGUS ID instead.\n",
            "core.conformation.Residue: [ WARNING ] missing an atom: 1  H   that depends on a nonexistent polymer connection!\n",
            "core.conformation.Residue: [ WARNING ]  --> generating it using idealized coordinates.\n",
            "protocols.grafting.util: Deleting 10 residues from 892 to 901\n",
            "core.conformation.Conformation: Found disulfide between residues 771 845\n",
            "core.conformation.Conformation: Found disulfide between residues 891 945\n",
            "protocols.grafting.util: insert_point 891\n",
            "protocols.grafting.util: insert_start 892\n",
            "protocols.grafting.util: insert_end 906\n",
            "protocols.grafting.util: insert_length 15\n",
            "protocols.grafting.util: FOLD_TREE  EDGE 1 891 -1  EDGE 891 892 1  EDGE 892 962 -1\n",
            "protocols.grafting.GraftMoverBase: Insertion complete.\n",
            "protocols.grafting.AnchoredGraftMover: Setting default movemap\n",
            "protocols.grafting.CCDEndsGraftMover: Start: 891\n",
            "protocols.grafting.CCDEndsGraftMover: Original End: 902\n",
            "protocols.grafting.CCDEndsGraftMover: End: 907\n",
            "protocols.grafting.CCDEndsGraftMover: Insert Length: 15\n",
            "protocols.grafting.CCDEndsGraftMover: LOOP start: 890  stop: 893  cut: 891  size: 4  skip rate: 0  extended?: False\n",
            "protocols.grafting.CCDEndsGraftMover:\n",
            "protocols.grafting.CCDEndsGraftMover: LOOP start: 905  stop: 908  cut: 906  size: 4  skip rate: 0  extended?: False\n",
            "protocols.grafting.CCDEndsGraftMover:\n",
            "protocols.loops.FoldTreeFromLoopsWrapper: old foldtree FOLD_TREE  EDGE 1 891 -1  EDGE 891 907 1  EDGE 907 977 -1  EDGE 891 892 -2  C    N    EDGE 892 893 -2  C    N    EDGE 893 894 -2  C    N    EDGE 894 895 -2  C    N    EDGE 895 896 -2  C    N    EDGE 896 897 -2  C    N    EDGE 897 898 -2  C    N    EDGE 898 899 -2  C    N    EDGE 899 900 -2  C    N    EDGE 900 901 -2  C    N    EDGE 901 902 -2  C    N    EDGE 902 903 -2  C    N    EDGE 903 904 -2  C    N    EDGE 904 905 -2  C    N    EDGE 905 906 -2  C    N   \n",
            "New foldtree FOLD_TREE  EDGE 1 889 -1  EDGE 889 891 -1  EDGE 889 894 1  EDGE 894 892 -1  EDGE 894 904 -1  EDGE 904 906 -1  EDGE 904 909 2  EDGE 909 907 -1  EDGE 909 977 -1\n",
            "protocols.grafting.util: Loop: 1\n",
            "protocols.grafting.util: Add variant to: 891\n",
            "protocols.grafting.util: Loop: 2\n",
            "protocols.grafting.util: Add variant to: 906\n",
            "protocols.grafting.util: idealized  890\n",
            "protocols.grafting.util: idealized  891\n",
            "protocols.grafting.util: ideal 892\n",
            "protocols.grafting.util: ideal 906\n",
            "protocols.grafting.util: idealized 907\n",
            "protocols.grafting.util: idealized 908\n",
            "protocols.grafting.CCDEndsGraftMover: start 1752.78\n",
            "protocols.grafting.CCDEndsGraftMover: round 1\n",
            "protocols.loops.util: PepBondGeom: 906 - 42 L  Ca-C-N --  99.5 (min): 162.32 : 134.5 (max)\n",
            "protocols.grafting.CCDEndsGraftMover: 1 1479.41\n",
            "protocols.grafting.CCDEndsGraftMover: round 2\n",
            "protocols.loops.util: PepBondGeom: 42 L  C-N --  1.50576 max: 1.5\n",
            "protocols.grafting.CCDEndsGraftMover: 2 1454.7\n",
            "protocols.grafting.CCDEndsGraftMover: round 3\n",
            "protocols.loops.util: PepBondGeom: 42 L  C-N --  1.64564 max: 1.5\n",
            "protocols.grafting.CCDEndsGraftMover: round 4\n",
            "protocols.loops.util: PepBondGeom: 42 L  C-N --  1.62532 max: 1.5\n",
            "protocols.grafting.CCDEndsGraftMover: round 5\n",
            "protocols.loops.util: PepBondGeom: 42 L  C-N --  1.56443 max: 1.5\n",
            "protocols.grafting.CCDEndsGraftMover: round 6\n",
            "protocols.loops.util: PepBondGeom: 42 L  C-N --  1.67229 max: 1.5\n",
            "protocols.grafting.CCDEndsGraftMover: 6 1454.51\n",
            "protocols.grafting.CCDEndsGraftMover: round 7\n",
            "protocols.loops.util: PepBondGeom: 42 L  C-N --  1.61424 max: 1.5\n",
            "protocols.grafting.CCDEndsGraftMover: round 8\n",
            "protocols.loops.util: PepBondGeom: 42 L  C-N --  1.67653 max: 1.5\n",
            "protocols.grafting.CCDEndsGraftMover: round 9\n",
            "protocols.loops.util: PepBondGeom: 42 L  C-N --  1.65135 max: 1.5\n",
            "protocols.grafting.CCDEndsGraftMover: round 10\n",
            "protocols.loops.util: PepBondGeom: 42 L  C-N --  1.652 max: 1.5\n",
            "protocols.grafting.CCDEndsGraftMover: round 11\n",
            "protocols.loops.util: PepBondGeom: 42 L  C-N --  1.65589 max: 1.5\n",
            "protocols.grafting.CCDEndsGraftMover: round 12\n",
            "protocols.loops.util: PepBondGeom: 42 L  C-N --  1.65244 max: 1.5\n",
            "protocols.grafting.CCDEndsGraftMover: round 13\n",
            "protocols.loops.util: PepBondGeom: 42 L  C-N --  1.69174 max: 1.5\n",
            "protocols.grafting.CCDEndsGraftMover: 13 1452.05\n",
            "protocols.grafting.CCDEndsGraftMover: round 14\n",
            "protocols.loops.util: PepBondGeom: 42 L  C-N --  1.65543 max: 1.5\n",
            "protocols.grafting.CCDEndsGraftMover: 14 1451.16\n",
            "protocols.grafting.CCDEndsGraftMover: round 15\n",
            "protocols.loops.util: PepBondGeom: 42 L  C-N --  1.69937 max: 1.5\n",
            "protocols.grafting.CCDEndsGraftMover: 15 1442.16\n",
            "protocols.grafting.CCDEndsGraftMover: round 16\n",
            "protocols.loops.util: PepBondGeom: 42 L  C-N --  1.61925 max: 1.5\n",
            "protocols.grafting.CCDEndsGraftMover: 16 1443.12\n",
            "protocols.grafting.CCDEndsGraftMover: round 17\n",
            "protocols.loops.util: PepBondGeom: 42 L  C-N --  1.55639 max: 1.5\n",
            "protocols.grafting.CCDEndsGraftMover: round 18\n",
            "protocols.loops.util: PepBondGeom: 42 L  C-N --  1.56564 max: 1.5\n",
            "protocols.grafting.CCDEndsGraftMover: round 19\n",
            "protocols.loops.util: PepBondGeom: 42 L  C-N --  1.57516 max: 1.5\n",
            "protocols.grafting.CCDEndsGraftMover: round 20\n",
            "protocols.loops.util: PepBondGeom: 42 L  C-N --  1.59094 max: 1.5\n",
            "protocols.grafting.CCDEndsGraftMover: round 21\n",
            "protocols.loops.util: PepBondGeom: 42 L  C-N --  1.52776 max: 1.5\n",
            "protocols.grafting.CCDEndsGraftMover: round 22\n",
            "protocols.loops.util: PepBondGeom: 42 L  C-N --  1.61603 max: 1.5\n",
            "protocols.grafting.CCDEndsGraftMover: round 23\n",
            "protocols.loops.util: PepBondGeom: 42 L  C-N --  1.5844 max: 1.5\n",
            "protocols.grafting.CCDEndsGraftMover: round 24\n",
            "protocols.loops.util: PepBondGeom: 42 L  C-N --  1.57538 max: 1.5\n",
            "protocols.grafting.CCDEndsGraftMover: round 25\n",
            "protocols.loops.util: PepBondGeom: 42 L  C-N --  1.51887 max: 1.5\n",
            "protocols.grafting.CCDEndsGraftMover: round 26\n",
            "protocols.loops.util: PepBondGeom: 42 L  C-N --  1.53418 max: 1.5\n",
            "protocols.grafting.CCDEndsGraftMover: round 27\n",
            "protocols.loops.util: PepBondGeom: 42 L  C-N --  1.57398 max: 1.5\n",
            "protocols.grafting.CCDEndsGraftMover: round 28\n",
            "protocols.loops.util: PepBondGeom: 42 L  C-N --  1.54677 max: 1.5\n",
            "protocols.grafting.CCDEndsGraftMover: round 29\n",
            "protocols.loops.util: PepBondGeom: 42 L  C-N --  1.51855 max: 1.5\n",
            "protocols.grafting.CCDEndsGraftMover: round 30\n",
            "protocols.loops.util: PepBondGeom: 42 L  C-N --  1.57665 max: 1.5\n",
            "protocols.grafting.CCDEndsGraftMover: round 31\n",
            "protocols.loops.util: PepBondGeom: 42 L  C-N --  1.5986 max: 1.5\n",
            "protocols.grafting.CCDEndsGraftMover: round 32\n",
            "protocols.loops.util: PepBondGeom: 42 L  C-N --  1.59894 max: 1.5\n",
            "protocols.grafting.CCDEndsGraftMover: round 33\n",
            "protocols.loops.util: PepBondGeom: 42 L  C-N --  1.58095 max: 1.5\n",
            "protocols.grafting.CCDEndsGraftMover: 33 1444.62\n",
            "protocols.grafting.CCDEndsGraftMover: round 34\n",
            "protocols.loops.util: PepBondGeom: 42 L  C-N --  1.58507 max: 1.5\n",
            "protocols.grafting.CCDEndsGraftMover: 34 1445.21\n",
            "protocols.grafting.CCDEndsGraftMover: round 35\n",
            "protocols.loops.util: PepBondGeom: 42 L  C-N --  1.58325 max: 1.5\n",
            "protocols.grafting.CCDEndsGraftMover: 35 1444\n",
            "protocols.grafting.CCDEndsGraftMover: round 36\n",
            "protocols.loops.util: PepBondGeom: 42 L  C-N --  1.53294 max: 1.5\n",
            "protocols.grafting.CCDEndsGraftMover: round 37\n",
            "protocols.grafting.CCDEndsGraftMover: Graft Closed early - returning\n",
            "protocols.grafting.CCDEndsGraftMover: 37 1455.03\n",
            "protocols.grafting.CCDEndsGraftMover: finish 1455.03\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 2032 rotamers at 52 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.grafting.CCDEndsGraftMover: Graft meets ideal geometry true\n",
            "protocols.grafting.CCDEndsGraftMover: Complete\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L1-15-2 data: 3 cutoff: 10\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: Adding general dihedral constraints for L1\n",
            "protocols.antibody.design.AntibodyDesignMover: Checking Geometry\n",
            "protocols.antibody.design.AntibodyDesignMover: DATA GRAFT_CLOSURE 1 rabd/my_ab.pdb L1 L1-15-2 3dggA_L1_0001\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H1\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H2\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H3\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L1\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L2\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L3\n",
            "protocols.antibody.clusters.CDRClusterSet: Adding cacheable cluster data to pose\n",
            "protocols.antibody.design.util: Setting CDR sequence in NativeAntibodySeq\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: H1-13-1 data: 583 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/H1-13-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 26 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: H2-10-1 data: 328 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/H2-10-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 20 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: Adding general dihedral constraints for H3\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L1-15-2 data: 3 cutoff: 10\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: Adding general dihedral constraints for L1\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L2-8-1 data: 632 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/L2-8-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 16 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L3-7-1 data: 2 cutoff: 10\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: Adding general dihedral constraints for L3\n",
            "protocols.antibody.design.AntibodyDesignMover: Beginning MinProtocol\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using explicit constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using interface constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Making an interface set with fixed chains\n",
            "protocols.analysis.InterfaceAnalyzerMover: Interface set residues total: 108\n",
            "protocols.analysis.InterfaceAnalyzerMover: NOT detecting disulfides in the separated pose.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 3233 rotamers at 108 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.analysis.InterfaceAnalyzerMover: included_nres: 977\n",
            "protocols.antibody.design.AntibodyDesignMover: Inner round: 1\n",
            "protocols.antibody.design.AntibodyDesignMover: Add min neighbors : L2\n",
            "protocols.antibody.design.AntibodyDesignMover: Add min neighbors : L3\n",
            "basic.io.database: [ WARNING ] Unable to locate database file /sampling/antibodies/antibody_database_rosetta_design.db\n",
            "antibody.database.AntibodyDatabaseManager: The antibody design database is not specified by the user.\n",
            "antibody.database.AntibodyDatabaseManager: Reading from: /usr/local/lib/python3.10/dist-packages/pyrosetta/database//sampling/antibodies/antibody_database_rosetta_design_north_paper.db\n",
            "antibody.database.AntibodyDatabaseManager: Skipped L1-15-2 with 2 data point\n",
            "antibody.database.AntibodyDatabaseManager: Skipped L3-7-1 with 2 data point\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Data pre-loaded.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: applying prob task op\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Using conservative op for L1\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Using conservative op for L3\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Using original bb sequence for conservative design.\n",
            "protocols.task_operations.ConservativeDesignOperation: Loading conservative mutational data\n",
            "core.pack.pack_rotamers: built 5907 rotamers at 84 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating PDInteractionGraph\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs start: 20527.9\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs end: 9127.86\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using explicit constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using interface constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Making an interface set with fixed chains\n",
            "protocols.analysis.InterfaceAnalyzerMover: Interface set residues total: 108\n",
            "protocols.analysis.InterfaceAnalyzerMover: NOT detecting disulfides in the separated pose.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 3020 rotamers at 108 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.analysis.InterfaceAnalyzerMover: included_nres: 977\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 899\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 900\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 924\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 925\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 964\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 965\n",
            "antibody.design.GeneralAntibodyModeler: CDR min start: 9127.86\n",
            "antibody.design.GeneralAntibodyModeler: CDR min end: 3422.22\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using explicit constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using interface constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Making an interface set with fixed chains\n",
            "protocols.analysis.InterfaceAnalyzerMover: Interface set residues total: 105\n",
            "protocols.analysis.InterfaceAnalyzerMover: NOT detecting disulfides in the separated pose.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 3005 rotamers at 105 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.analysis.InterfaceAnalyzerMover: included_nres: 977\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: applying prob task op\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Using conservative op for L1\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Using conservative op for L3\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Using original bb sequence for conservative design.\n",
            "core.pack.pack_rotamers: built 6897 rotamers at 82 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating PDInteractionGraph\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs start: 3422.22\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs end: 4986.22\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using explicit constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using interface constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Making an interface set with fixed chains\n",
            "protocols.analysis.InterfaceAnalyzerMover: Interface set residues total: 104\n",
            "protocols.analysis.InterfaceAnalyzerMover: NOT detecting disulfides in the separated pose.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 3108 rotamers at 104 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.analysis.InterfaceAnalyzerMover: included_nres: 977\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 899\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 900\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 924\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 925\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 964\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 965\n",
            "antibody.design.GeneralAntibodyModeler: CDR min start: 4986.22\n",
            "antibody.design.GeneralAntibodyModeler: CDR min end: 2159.11\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using explicit constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using interface constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Making an interface set with fixed chains\n",
            "protocols.analysis.InterfaceAnalyzerMover: Interface set residues total: 100\n",
            "protocols.analysis.InterfaceAnalyzerMover: NOT detecting disulfides in the separated pose.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 2837 rotamers at 100 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.analysis.InterfaceAnalyzerMover: included_nres: 977\n",
            "protocols.antibody.design.AntibodyDesignMover: Min Cycle complete\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using explicit constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using interface constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Making an interface set with fixed chains\n",
            "protocols.analysis.InterfaceAnalyzerMover: Interface set residues total: 100\n",
            "protocols.analysis.InterfaceAnalyzerMover: NOT detecting disulfides in the separated pose.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 2837 rotamers at 100 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.analysis.InterfaceAnalyzerMover: included_nres: 977\n",
            "protocols.antibody.design.AntibodyDesignMover:\n",
            "protocols.antibody.design.AntibodyDesignMover: SCORE 23 1325.87 0\n",
            "protocols.antibody.design.AntibodyDesignMover: FINAL 23 1288.01 1\n",
            "protocols.antibody.design.AntibodyDesignMover:\n",
            "protocols.antibody.design.AntibodyDesignMover: Last Accepted Score: 1288.01\n",
            "protocols.antibody.design.AntibodyDesignMover: Lowest        Score:1288.01\n",
            "protocols.antibody.design.AntibodyDesignMover:\n",
            "protocols.antibody.design.AntibodyDesignMover: Outer round: 24\n",
            "protocols.antibody.design.AntibodyDesignMover: Grafting CDR from cluster L1-11-2 fragment 2bdnL_L1_0001\n",
            "protocols.grafting.CCDEndsGraftMover: Start: 891 End: 902 NterO: 3 CterO: 3\n",
            "protocols.grafting.util: Superimposing overhang residues\n",
            "core.chemical.AtomICoor: [ WARNING ] IcoorAtomID::atom_id(): Cannot get atom_id for POLYMER_LOWER of residue ILE 1.  Returning BOGUS ID instead.\n",
            "core.conformation.Residue: [ WARNING ] missing an atom: 1  H   that depends on a nonexistent polymer connection!\n",
            "core.conformation.Residue: [ WARNING ]  --> generating it using idealized coordinates.\n",
            "protocols.grafting.util: Deleting 10 residues from 892 to 901\n",
            "core.conformation.Conformation: Found disulfide between residues 771 845\n",
            "core.conformation.Conformation: Found disulfide between residues 891 945\n",
            "protocols.grafting.util: insert_point 891\n",
            "protocols.grafting.util: insert_start 892\n",
            "protocols.grafting.util: insert_end 902\n",
            "protocols.grafting.util: insert_length 11\n",
            "protocols.grafting.util: FOLD_TREE  EDGE 1 891 -1  EDGE 891 892 1  EDGE 892 962 -1\n",
            "protocols.grafting.GraftMoverBase: Insertion complete.\n",
            "protocols.grafting.AnchoredGraftMover: Setting default movemap\n",
            "protocols.grafting.CCDEndsGraftMover: Start: 891\n",
            "protocols.grafting.CCDEndsGraftMover: Original End: 902\n",
            "protocols.grafting.CCDEndsGraftMover: End: 903\n",
            "protocols.grafting.CCDEndsGraftMover: Insert Length: 11\n",
            "protocols.grafting.CCDEndsGraftMover: LOOP start: 890  stop: 893  cut: 891  size: 4  skip rate: 0  extended?: False\n",
            "protocols.grafting.CCDEndsGraftMover:\n",
            "protocols.grafting.CCDEndsGraftMover: LOOP start: 901  stop: 904  cut: 902  size: 4  skip rate: 0  extended?: False\n",
            "protocols.grafting.CCDEndsGraftMover:\n",
            "protocols.loops.FoldTreeFromLoopsWrapper: old foldtree FOLD_TREE  EDGE 1 891 -1  EDGE 891 903 1  EDGE 903 973 -1  EDGE 891 892 -2  C    N    EDGE 892 893 -2  C    N    EDGE 893 894 -2  C    N    EDGE 894 895 -2  C    N    EDGE 895 896 -2  C    N    EDGE 896 897 -2  C    N    EDGE 897 898 -2  C    N    EDGE 898 899 -2  C    N    EDGE 899 900 -2  C    N    EDGE 900 901 -2  C    N    EDGE 901 902 -2  C    N   \n",
            "New foldtree FOLD_TREE  EDGE 1 889 -1  EDGE 889 891 -1  EDGE 889 894 1  EDGE 894 892 -1  EDGE 894 900 -1  EDGE 900 902 -1  EDGE 900 905 2  EDGE 905 903 -1  EDGE 905 973 -1\n",
            "protocols.grafting.util: Loop: 1\n",
            "protocols.grafting.util: Add variant to: 891\n",
            "protocols.grafting.util: Loop: 2\n",
            "protocols.grafting.util: Add variant to: 902\n",
            "protocols.grafting.util: idealized  890\n",
            "protocols.grafting.util: idealized  891\n",
            "protocols.grafting.util: ideal 892\n",
            "protocols.grafting.util: ideal 902\n",
            "protocols.grafting.util: idealized 903\n",
            "protocols.grafting.util: idealized 904\n",
            "protocols.grafting.CCDEndsGraftMover: start 1593.6\n",
            "protocols.grafting.CCDEndsGraftMover: round 1\n",
            "protocols.loops.util: PepBondGeom: 902 - 42 L  Ca-C-N --  99.5 (min): 158.33 : 134.5 (max)\n",
            "protocols.grafting.CCDEndsGraftMover: 1 1353.07\n",
            "protocols.grafting.CCDEndsGraftMover: round 2\n",
            "protocols.loops.util: PepBondGeom: 42 L  C-N --  1.56223 max: 1.5\n",
            "protocols.grafting.CCDEndsGraftMover: 2 1344.98\n",
            "protocols.grafting.CCDEndsGraftMover: round 3\n",
            "protocols.loops.util: PepBondGeom: 42 L  C-N --  1.61042 max: 1.5\n",
            "protocols.grafting.CCDEndsGraftMover: round 4\n",
            "protocols.loops.util: PepBondGeom: 42 L  C-N --  1.56282 max: 1.5\n",
            "protocols.grafting.CCDEndsGraftMover: round 5\n",
            "protocols.loops.util: PepBondGeom: 42 L  C-N --  1.5417 max: 1.5\n",
            "protocols.grafting.CCDEndsGraftMover: round 6\n",
            "protocols.loops.util: PepBondGeom: 42 L  C-N --  1.60199 max: 1.5\n",
            "protocols.grafting.CCDEndsGraftMover: round 7\n",
            "protocols.loops.util: PepBondGeom: 902 - 42 L  Ca-C-N --  99.5 (min): 134.524 : 134.5 (max)\n",
            "protocols.grafting.CCDEndsGraftMover: round 8\n",
            "protocols.loops.util: PepBondGeom: 42 L  C-N --  1.6932 max: 1.5\n",
            "protocols.grafting.CCDEndsGraftMover: round 9\n",
            "protocols.loops.util: PepBondGeom: 42 L  C-N --  1.53864 max: 1.5\n",
            "protocols.grafting.CCDEndsGraftMover: round 10\n",
            "protocols.loops.util: PepBondGeom: 42 L  C-N --  1.51391 max: 1.5\n",
            "protocols.grafting.CCDEndsGraftMover: round 11\n",
            "protocols.loops.util: PepBondGeom: 42 L  C-N --  1.6901 max: 1.5\n",
            "protocols.grafting.CCDEndsGraftMover: round 12\n",
            "protocols.loops.util: PepBondGeom: 42 L  C-N --  1.53442 max: 1.5\n",
            "protocols.grafting.CCDEndsGraftMover: 12 1326.44\n",
            "protocols.grafting.CCDEndsGraftMover: round 13\n",
            "protocols.loops.util: PepBondGeom: 42 L  C-N --  1.53428 max: 1.5\n",
            "protocols.grafting.CCDEndsGraftMover: round 14\n",
            "protocols.loops.util: PepBondGeom: 42 L  C-N --  1.57285 max: 1.5\n",
            "protocols.grafting.CCDEndsGraftMover: 14 1321.67\n",
            "protocols.grafting.CCDEndsGraftMover: round 15\n",
            "protocols.loops.util: PepBondGeom: 42 L  C-N --  1.63719 max: 1.5\n",
            "protocols.grafting.CCDEndsGraftMover: round 16\n",
            "protocols.loops.util: PepBondGeom: 42 L  C-N --  1.61668 max: 1.5\n",
            "protocols.grafting.CCDEndsGraftMover: 16 1315.17\n",
            "protocols.grafting.CCDEndsGraftMover: round 17\n",
            "protocols.loops.util: PepBondGeom: 42 L  C-N --  1.60127 max: 1.5\n",
            "protocols.grafting.CCDEndsGraftMover: round 18\n",
            "protocols.loops.util: PepBondGeom: 42 L  C-N --  1.59645 max: 1.5\n",
            "protocols.grafting.CCDEndsGraftMover: round 19\n",
            "protocols.loops.util: PepBondGeom: 42 L  C-N --  1.60945 max: 1.5\n",
            "protocols.grafting.CCDEndsGraftMover: round 20\n",
            "protocols.loops.util: PepBondGeom: 42 L  C-N --  1.53345 max: 1.5\n",
            "protocols.grafting.CCDEndsGraftMover: round 21\n",
            "protocols.loops.util: PepBondGeom: 42 L  C-N --  1.58214 max: 1.5\n",
            "protocols.grafting.CCDEndsGraftMover: round 22\n",
            "protocols.loops.util: PepBondGeom: 42 L  C-N --  1.57179 max: 1.5\n",
            "protocols.grafting.CCDEndsGraftMover: round 23\n",
            "protocols.loops.util: PepBondGeom: 42 L  C-N --  1.56978 max: 1.5\n",
            "protocols.grafting.CCDEndsGraftMover: round 24\n",
            "protocols.loops.util: PepBondGeom: 42 L  C-N --  1.58324 max: 1.5\n",
            "protocols.grafting.CCDEndsGraftMover: round 25\n",
            "protocols.loops.util: PepBondGeom: 42 L  C-N --  1.57322 max: 1.5\n",
            "protocols.grafting.CCDEndsGraftMover: round 26\n",
            "protocols.loops.util: PepBondGeom: 42 L  C-N --  1.60615 max: 1.5\n",
            "protocols.grafting.CCDEndsGraftMover: round 27\n",
            "protocols.loops.util: PepBondGeom: 42 L  C-N --  1.57654 max: 1.5\n",
            "protocols.grafting.CCDEndsGraftMover: round 28\n",
            "protocols.loops.util: PepBondGeom: 42 L  C-N --  1.59504 max: 1.5\n",
            "protocols.grafting.CCDEndsGraftMover: round 29\n",
            "protocols.loops.util: PepBondGeom: 42 L  C-N --  1.61735 max: 1.5\n",
            "protocols.grafting.CCDEndsGraftMover: round 30\n",
            "protocols.loops.util: PepBondGeom: 42 L  C-N --  1.58166 max: 1.5\n",
            "protocols.grafting.CCDEndsGraftMover: round 31\n",
            "protocols.loops.util: PepBondGeom: 42 L  C-N --  1.55731 max: 1.5\n",
            "protocols.grafting.CCDEndsGraftMover: round 32\n",
            "protocols.loops.util: PepBondGeom: 42 L  C-N --  1.57252 max: 1.5\n",
            "protocols.grafting.CCDEndsGraftMover: round 33\n",
            "protocols.loops.util: PepBondGeom: 42 L  C-N --  1.58878 max: 1.5\n",
            "protocols.grafting.CCDEndsGraftMover: 33 1315.74\n",
            "protocols.grafting.CCDEndsGraftMover: round 34\n",
            "protocols.loops.util: PepBondGeom: 42 L  C-N --  1.52049 max: 1.5\n",
            "protocols.grafting.CCDEndsGraftMover: round 35\n",
            "protocols.loops.util: PepBondGeom: 42 L  C-N --  1.53447 max: 1.5\n",
            "protocols.grafting.CCDEndsGraftMover: round 36\n",
            "protocols.loops.util: PepBondGeom: 42 L  C-N --  1.54785 max: 1.5\n",
            "protocols.grafting.CCDEndsGraftMover: round 37\n",
            "protocols.loops.util: PepBondGeom: 42 L  C-N --  1.56746 max: 1.5\n",
            "protocols.grafting.CCDEndsGraftMover: round 38\n",
            "protocols.loops.util: PepBondGeom: 42 L  C-N --  1.52782 max: 1.5\n",
            "protocols.grafting.CCDEndsGraftMover: round 39\n",
            "protocols.loops.util: PepBondGeom: 42 L  C-N --  1.52178 max: 1.5\n",
            "protocols.grafting.CCDEndsGraftMover: round 40\n",
            "protocols.loops.util: PepBondGeom: 42 L  C-N --  1.54814 max: 1.5\n",
            "protocols.grafting.CCDEndsGraftMover: round 41\n",
            "protocols.loops.util: PepBondGeom: 42 L  C-N --  1.53755 max: 1.5\n",
            "protocols.grafting.CCDEndsGraftMover: round 42\n",
            "protocols.loops.util: PepBondGeom: 42 L  C-N --  1.58014 max: 1.5\n",
            "protocols.grafting.CCDEndsGraftMover: round 43\n",
            "protocols.loops.util: PepBondGeom: 42 L  C-N --  1.5334 max: 1.5\n",
            "protocols.grafting.CCDEndsGraftMover: 43 1316.29\n",
            "protocols.grafting.CCDEndsGraftMover: round 44\n",
            "protocols.loops.util: PepBondGeom: 42 L  C-N --  1.53677 max: 1.5\n",
            "protocols.grafting.CCDEndsGraftMover: round 45\n",
            "protocols.loops.util: PepBondGeom: 42 L  C-N --  1.52903 max: 1.5\n",
            "protocols.grafting.CCDEndsGraftMover: round 46\n",
            "protocols.loops.util: PepBondGeom: 42 L  C-N --  1.56305 max: 1.5\n",
            "protocols.grafting.CCDEndsGraftMover: round 47\n",
            "protocols.loops.util: PepBondGeom: 42 L  C-N --  1.52687 max: 1.5\n",
            "protocols.grafting.CCDEndsGraftMover: round 48\n",
            "protocols.loops.util: PepBondGeom: 42 L  C-N --  1.51354 max: 1.5\n",
            "protocols.grafting.CCDEndsGraftMover: round 49\n",
            "protocols.loops.util: PepBondGeom: 42 L  C-N --  1.53549 max: 1.5\n",
            "protocols.grafting.CCDEndsGraftMover: round 50\n",
            "protocols.loops.util: PepBondGeom: 42 L  C-N --  1.55166 max: 1.5\n",
            "protocols.grafting.CCDEndsGraftMover: 50 1312.32\n",
            "protocols.grafting.CCDEndsGraftMover: finish 1312.32\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 1845 rotamers at 47 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.loops.util: PepBondGeom: 42 L  C-N --  1.55166 max: 1.5\n",
            "protocols.grafting.CCDEndsGraftMover: Graft meets ideal geometry true\n",
            "protocols.grafting.CCDEndsGraftMover: Complete\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L1-11-2 data: 91 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/L1-11-2.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 22 constraints\n",
            "protocols.antibody.design.AntibodyDesignMover: Checking Geometry\n",
            "protocols.loops.util: PepBondGeom: 42 L  C-N --  1.55166 max: 1.5\n",
            "protocols.antibody.design.AntibodyDesignMover: Graft not closed. Adapting graft closure\n",
            "protocols.grafting.util: Deleting 10 residues from 892 to 901\n",
            "core.conformation.Conformation: Found disulfide between residues 771 845\n",
            "core.conformation.Conformation: Found disulfide between residues 891 945\n",
            "protocols.grafting.util: insert_point 891\n",
            "protocols.grafting.util: insert_start 892\n",
            "protocols.grafting.util: insert_end 902\n",
            "protocols.grafting.util: insert_length 11\n",
            "protocols.grafting.util: FOLD_TREE  EDGE 1 891 -1  EDGE 891 892 1  EDGE 892 962 -1\n",
            "protocols.grafting.GraftMoverBase: Insertion complete.\n",
            "protocols.grafting.AnchoredGraftMover: Setting default movemap\n",
            "protocols.grafting.util: FOLD_TREE  EDGE 1 889 -1  EDGE 889 903 -1  EDGE 889 905 1  EDGE 905 904 -1  EDGE 905 973 -1\n",
            "protocols.grafting.AnchoredGraftMover: Loops: LOOP  begin  end  cut  skip_rate  extended\n",
            "protocols.grafting.AnchoredGraftMover: LOOP start: 890  stop: 904  cut: 903  size: 15  skip rate: 0  extended?: False\n",
            "protocols.grafting.AnchoredGraftMover:\n",
            "protocols.grafting.AnchoredGraftMover:\n",
            "protocols.grafting.util: Loop: 1\n",
            "protocols.grafting.util: Add variant to: 903\n",
            "protocols.grafting.util: idealized  890\n",
            "protocols.grafting.util: idealized  891\n",
            "protocols.grafting.util: ideal 892\n",
            "protocols.grafting.util: ideal 902\n",
            "protocols.grafting.util: idealized 903\n",
            "protocols.grafting.util: idealized 904\n",
            "protocols.grafting.AnchoredGraftMover: start 8716.67\n",
            "protocols.grafting.AnchoredGraftMover: round 1\n",
            "protocols.grafting.AnchoredGraftMover: Graft Closed early - returning\n",
            "protocols.grafting.AnchoredGraftMover: 1 1449.6\n",
            "protocols.grafting.AnchoredGraftMover: finish 1449.6\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 1845 rotamers at 47 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.grafting.AnchoredGraftMover: Graft meets ideal geometry true\n",
            "protocols.grafting.AnchoredGraftMover: Complete\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L1-11-2 data: 91 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/L1-11-2.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 22 constraints\n",
            "protocols.antibody.design.AntibodyDesignMover: Checking Geometry\n",
            "protocols.antibody.design.AntibodyDesignMover: DATA GRAFT_CLOSURE 1 rabd/my_ab.pdb L1 L1-11-2 2bdnL_L1_0001\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H1\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H2\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H3\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L1\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L2\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L3\n",
            "protocols.antibody.clusters.CDRClusterSet: Adding cacheable cluster data to pose\n",
            "protocols.antibody.design.util: Setting CDR sequence in NativeAntibodySeq\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: H1-13-1 data: 583 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/H1-13-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 26 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: H2-10-1 data: 328 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/H2-10-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 20 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: Adding general dihedral constraints for H3\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L1-11-2 data: 91 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/L1-11-2.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 22 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L2-8-1 data: 632 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/L2-8-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 16 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L3-7-1 data: 2 cutoff: 10\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: Adding general dihedral constraints for L3\n",
            "protocols.antibody.design.AntibodyDesignMover: Beginning MinProtocol\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using explicit constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using interface constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Making an interface set with fixed chains\n",
            "protocols.analysis.InterfaceAnalyzerMover: Interface set residues total: 95\n",
            "protocols.analysis.InterfaceAnalyzerMover: NOT detecting disulfides in the separated pose.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 2542 rotamers at 95 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.analysis.InterfaceAnalyzerMover: included_nres: 973\n",
            "protocols.antibody.design.AntibodyDesignMover: Inner round: 1\n",
            "protocols.antibody.design.AntibodyDesignMover: Add min neighbors : L2\n",
            "protocols.antibody.design.AntibodyDesignMover: Add min neighbors : L3\n",
            "basic.io.database: [ WARNING ] Unable to locate database file /sampling/antibodies/antibody_database_rosetta_design.db\n",
            "antibody.database.AntibodyDatabaseManager: The antibody design database is not specified by the user.\n",
            "antibody.database.AntibodyDatabaseManager: Reading from: /usr/local/lib/python3.10/dist-packages/pyrosetta/database//sampling/antibodies/antibody_database_rosetta_design_north_paper.db\n",
            "antibody.database.AntibodyDatabaseManager: Loaded L1-11-2 with 35 datapoints.\n",
            "antibody.database.AntibodyDatabaseManager: Skipped L3-7-1 with 2 data point\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Data pre-loaded.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: applying prob task op\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Using conservative op for L3\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Using original bb sequence for conservative design.\n",
            "protocols.task_operations.ConservativeDesignOperation: Loading conservative mutational data\n",
            "core.pack.pack_rotamers: built 3278 rotamers at 71 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating PDInteractionGraph\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs start: 8250.6\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs end: 4466.17\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using explicit constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using interface constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Making an interface set with fixed chains\n",
            "protocols.analysis.InterfaceAnalyzerMover: Interface set residues total: 95\n",
            "protocols.analysis.InterfaceAnalyzerMover: NOT detecting disulfides in the separated pose.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 2579 rotamers at 95 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.analysis.InterfaceAnalyzerMover: included_nres: 973\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 897\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 898\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 920\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 921\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 960\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 961\n",
            "antibody.design.GeneralAntibodyModeler: CDR min start: 4466.17\n",
            "antibody.design.GeneralAntibodyModeler: CDR min end: 2147.12\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using explicit constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using interface constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Making an interface set with fixed chains\n",
            "protocols.analysis.InterfaceAnalyzerMover: Interface set residues total: 91\n",
            "protocols.analysis.InterfaceAnalyzerMover: NOT detecting disulfides in the separated pose.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 2362 rotamers at 91 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.analysis.InterfaceAnalyzerMover: included_nres: 973\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: applying prob task op\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Using conservative op for L3\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Using original bb sequence for conservative design.\n",
            "core.pack.pack_rotamers: built 4008 rotamers at 71 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating PDInteractionGraph\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs start: 2147.12\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs end: 2178.63\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using explicit constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using interface constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Making an interface set with fixed chains\n",
            "protocols.analysis.InterfaceAnalyzerMover: Interface set residues total: 91\n",
            "protocols.analysis.InterfaceAnalyzerMover: NOT detecting disulfides in the separated pose.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 2422 rotamers at 91 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.analysis.InterfaceAnalyzerMover: included_nres: 973\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 897\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 898\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 920\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 921\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 960\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 961\n",
            "antibody.design.GeneralAntibodyModeler: CDR min start: 2178.63\n",
            "antibody.design.GeneralAntibodyModeler: CDR min end: 2092.15\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using explicit constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using interface constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Making an interface set with fixed chains\n",
            "protocols.analysis.InterfaceAnalyzerMover: Interface set residues total: 92\n",
            "protocols.analysis.InterfaceAnalyzerMover: NOT detecting disulfides in the separated pose.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 2577 rotamers at 92 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.analysis.InterfaceAnalyzerMover: included_nres: 973\n",
            "protocols.antibody.design.AntibodyDesignMover: Min Cycle complete\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using explicit constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using interface constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Making an interface set with fixed chains\n",
            "protocols.analysis.InterfaceAnalyzerMover: Interface set residues total: 92\n",
            "protocols.analysis.InterfaceAnalyzerMover: NOT detecting disulfides in the separated pose.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 2577 rotamers at 92 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.analysis.InterfaceAnalyzerMover: included_nres: 973\n",
            "protocols.antibody.design.AntibodyDesignMover:\n",
            "protocols.antibody.design.AntibodyDesignMover: SCORE 24 1335.85 0\n",
            "protocols.antibody.design.AntibodyDesignMover: FINAL 24 1288.01 1\n",
            "protocols.antibody.design.AntibodyDesignMover:\n",
            "protocols.antibody.design.AntibodyDesignMover: Last Accepted Score: 1288.01\n",
            "protocols.antibody.design.AntibodyDesignMover: Lowest        Score:1288.01\n",
            "protocols.antibody.design.AntibodyDesignMover:\n",
            "protocols.antibody.design.AntibodyDesignMover: Outer round: 25\n",
            "protocols.antibody.design.AntibodyDesignMover: Grafting CDR from cluster L1-10-2 fragment 1otsF_L1_0001\n",
            "protocols.grafting.CCDEndsGraftMover: Start: 891 End: 902 NterO: 3 CterO: 3\n",
            "protocols.grafting.util: Superimposing overhang residues\n",
            "core.chemical.AtomICoor: [ WARNING ] IcoorAtomID::atom_id(): Cannot get atom_id for POLYMER_LOWER of residue MET 1.  Returning BOGUS ID instead.\n",
            "core.conformation.Residue: [ WARNING ] missing an atom: 1  H   that depends on a nonexistent polymer connection!\n",
            "core.conformation.Residue: [ WARNING ]  --> generating it using idealized coordinates.\n",
            "protocols.grafting.util: Deleting 10 residues from 892 to 901\n",
            "core.conformation.Conformation: Found disulfide between residues 771 845\n",
            "core.conformation.Conformation: Found disulfide between residues 891 945\n",
            "protocols.grafting.util: insert_point 891\n",
            "protocols.grafting.util: insert_start 892\n",
            "protocols.grafting.util: insert_end 901\n",
            "protocols.grafting.util: insert_length 10\n",
            "protocols.grafting.util: FOLD_TREE  EDGE 1 891 -1  EDGE 891 892 1  EDGE 892 962 -1\n",
            "protocols.grafting.GraftMoverBase: Insertion complete.\n",
            "protocols.grafting.AnchoredGraftMover: Setting default movemap\n",
            "protocols.grafting.CCDEndsGraftMover: Start: 891\n",
            "protocols.grafting.CCDEndsGraftMover: Original End: 902\n",
            "protocols.grafting.CCDEndsGraftMover: End: 902\n",
            "protocols.grafting.CCDEndsGraftMover: Insert Length: 10\n",
            "protocols.grafting.CCDEndsGraftMover: LOOP start: 890  stop: 893  cut: 891  size: 4  skip rate: 0  extended?: False\n",
            "protocols.grafting.CCDEndsGraftMover:\n",
            "protocols.grafting.CCDEndsGraftMover: LOOP start: 900  stop: 903  cut: 901  size: 4  skip rate: 0  extended?: False\n",
            "protocols.grafting.CCDEndsGraftMover:\n",
            "protocols.loops.FoldTreeFromLoopsWrapper: old foldtree FOLD_TREE  EDGE 1 891 -1  EDGE 891 902 1  EDGE 902 972 -1  EDGE 891 892 -2  C    N    EDGE 892 893 -2  C    N    EDGE 893 894 -2  C    N    EDGE 894 895 -2  C    N    EDGE 895 896 -2  C    N    EDGE 896 897 -2  C    N    EDGE 897 898 -2  C    N    EDGE 898 899 -2  C    N    EDGE 899 900 -2  C    N    EDGE 900 901 -2  C    N   \n",
            "New foldtree FOLD_TREE  EDGE 1 889 -1  EDGE 889 891 -1  EDGE 889 894 1  EDGE 894 892 -1  EDGE 894 899 -1  EDGE 899 901 -1  EDGE 899 904 2  EDGE 904 902 -1  EDGE 904 972 -1\n",
            "protocols.grafting.util: Loop: 1\n",
            "protocols.grafting.util: Add variant to: 891\n",
            "protocols.grafting.util: Loop: 2\n",
            "protocols.grafting.util: Add variant to: 901\n",
            "protocols.grafting.util: idealized  890\n",
            "protocols.grafting.util: idealized  891\n",
            "protocols.grafting.util: ideal 892\n",
            "protocols.grafting.util: ideal 901\n",
            "protocols.grafting.util: idealized 902\n",
            "protocols.grafting.util: idealized 903\n",
            "protocols.grafting.CCDEndsGraftMover: start 1534.57\n",
            "protocols.grafting.CCDEndsGraftMover: round 1\n",
            "protocols.grafting.CCDEndsGraftMover: Graft Closed early - returning\n",
            "protocols.grafting.CCDEndsGraftMover: 1 1257.68\n",
            "protocols.grafting.CCDEndsGraftMover: finish 1257.68\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 1563 rotamers at 43 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.grafting.CCDEndsGraftMover: Graft meets ideal geometry true\n",
            "protocols.grafting.CCDEndsGraftMover: Complete\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L1-10-2 data: 3 cutoff: 10\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: Adding general dihedral constraints for L1\n",
            "protocols.antibody.design.AntibodyDesignMover: Checking Geometry\n",
            "protocols.antibody.design.AntibodyDesignMover: DATA GRAFT_CLOSURE 1 rabd/my_ab.pdb L1 L1-10-2 1otsF_L1_0001\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H1\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H2\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H3\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L1\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L2\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L3\n",
            "protocols.antibody.clusters.CDRClusterSet: Adding cacheable cluster data to pose\n",
            "protocols.antibody.design.util: Setting CDR sequence in NativeAntibodySeq\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: H1-13-1 data: 583 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/H1-13-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 26 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: H2-10-1 data: 328 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/H2-10-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 20 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: Adding general dihedral constraints for H3\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L1-10-2 data: 3 cutoff: 10\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: Adding general dihedral constraints for L1\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L2-8-1 data: 632 cutoff: 10\n",
            "core.scoring.constraints.ConstraintsIO: read constraints from /usr/local/lib/python3.10/dist-packages/pyrosetta/database/sampling/antibodies/cluster_based_constraints/CircularHarmonic/outliers_false_liberal_use_means/L2-8-1.txt\n",
            "core.scoring.constraints.ConstraintsIO: Read in 16 constraints\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: L3-7-1 data: 2 cutoff: 10\n",
            "protocols.antibody.constraints.CDRDihedralConstraintMover: Adding general dihedral constraints for L3\n",
            "protocols.antibody.design.AntibodyDesignMover: Beginning MinProtocol\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using explicit constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using interface constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Making an interface set with fixed chains\n",
            "protocols.analysis.InterfaceAnalyzerMover: Interface set residues total: 92\n",
            "protocols.analysis.InterfaceAnalyzerMover: NOT detecting disulfides in the separated pose.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 2387 rotamers at 92 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.analysis.InterfaceAnalyzerMover: included_nres: 972\n",
            "protocols.antibody.design.AntibodyDesignMover: Inner round: 1\n",
            "protocols.antibody.design.AntibodyDesignMover: Add min neighbors : L2\n",
            "protocols.antibody.design.AntibodyDesignMover: Add min neighbors : L3\n",
            "basic.io.database: [ WARNING ] Unable to locate database file /sampling/antibodies/antibody_database_rosetta_design.db\n",
            "antibody.database.AntibodyDatabaseManager: The antibody design database is not specified by the user.\n",
            "antibody.database.AntibodyDatabaseManager: Reading from: /usr/local/lib/python3.10/dist-packages/pyrosetta/database//sampling/antibodies/antibody_database_rosetta_design_north_paper.db\n",
            "antibody.database.AntibodyDatabaseManager: Skipped L1-10-2 with 2 data point\n",
            "antibody.database.AntibodyDatabaseManager: Skipped L3-7-1 with 2 data point\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Data pre-loaded.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: applying prob task op\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Using conservative op for L1\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Using conservative op for L3\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Using original bb sequence for conservative design.\n",
            "protocols.task_operations.ConservativeDesignOperation: Loading conservative mutational data\n",
            "core.pack.pack_rotamers: built 5954 rotamers at 69 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating PDInteractionGraph\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs start: 6193.04\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs end: 2774.51\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using explicit constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using interface constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Making an interface set with fixed chains\n",
            "protocols.analysis.InterfaceAnalyzerMover: Interface set residues total: 91\n",
            "protocols.analysis.InterfaceAnalyzerMover: NOT detecting disulfides in the separated pose.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 2339 rotamers at 91 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.analysis.InterfaceAnalyzerMover: included_nres: 972\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 896\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 897\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 919\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 920\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 959\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 960\n",
            "antibody.design.GeneralAntibodyModeler: CDR min start: 2774.51\n",
            "antibody.design.GeneralAntibodyModeler: CDR min end: 2018.86\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using explicit constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using interface constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Making an interface set with fixed chains\n",
            "protocols.analysis.InterfaceAnalyzerMover: Interface set residues total: 89\n",
            "protocols.analysis.InterfaceAnalyzerMover: NOT detecting disulfides in the separated pose.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 2399 rotamers at 89 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.analysis.InterfaceAnalyzerMover: included_nres: 972\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: applying prob task op\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Using conservative op for L1\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Using conservative op for L3\n",
            "protocols.antibody.task_operations.AddCDRProfilesOperation: Using original bb sequence for conservative design.\n",
            "core.pack.pack_rotamers: built 6836 rotamers at 69 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating PDInteractionGraph\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs start: 2018.86\n",
            "antibody.design.GeneralAntibodyModeler: repacking cdrs end: 2291.44\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using explicit constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using interface constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Making an interface set with fixed chains\n",
            "protocols.analysis.InterfaceAnalyzerMover: Interface set residues total: 89\n",
            "protocols.analysis.InterfaceAnalyzerMover: NOT detecting disulfides in the separated pose.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 2359 rotamers at 89 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.analysis.InterfaceAnalyzerMover: included_nres: 972\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 896\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 897\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 919\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 920\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 959\n",
            "protocols.loops.loops_main: Added cutpoint variant to residue 960\n",
            "antibody.design.GeneralAntibodyModeler: CDR min start: 2018.86\n",
            "antibody.design.GeneralAntibodyModeler: CDR min end: 2018.22\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using explicit constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using interface constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Making an interface set with fixed chains\n",
            "protocols.analysis.InterfaceAnalyzerMover: Interface set residues total: 89\n",
            "protocols.analysis.InterfaceAnalyzerMover: NOT detecting disulfides in the separated pose.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 2400 rotamers at 89 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.analysis.InterfaceAnalyzerMover: included_nres: 972\n",
            "protocols.antibody.design.AntibodyDesignMover: Min Cycle complete\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using explicit constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using interface constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Making an interface set with fixed chains\n",
            "protocols.analysis.InterfaceAnalyzerMover: Interface set residues total: 89\n",
            "protocols.analysis.InterfaceAnalyzerMover: NOT detecting disulfides in the separated pose.\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 2399 rotamers at 89 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.analysis.InterfaceAnalyzerMover: included_nres: 972\n",
            "protocols.antibody.design.AntibodyDesignMover:\n",
            "protocols.antibody.design.AntibodyDesignMover: SCORE 25 1295.03 0\n",
            "protocols.antibody.design.AntibodyDesignMover: FINAL 25 1288.01 1\n",
            "protocols.antibody.design.AntibodyDesignMover:\n",
            "protocols.antibody.design.AntibodyDesignMover: Last Accepted Score: 1288.01\n",
            "protocols.antibody.design.AntibodyDesignMover: Lowest        Score:1288.01\n",
            "protocols.antibody.design.AntibodyDesignMover:\n",
            "protocols.antibody.design.AntibodyDesignMover: Outer cycles attempted: 25 Outer cycles completed: 25\n",
            "protocols.antibody.design.AntibodyDesignMover:\n",
            "protocols.antibody.design.AntibodyDesignMover: Native Pose (MC): 12617.3\n",
            "protocols.antibody.design.AntibodyDesignMover: Final Pose (MC): 1288.01\n",
            "protocols.antibody.design.AntibodyDesignMover:\n",
            "protocols.moves.TrialCounter:              unk trials=     25;  accepts= 0.3200;  energy_drop/trial= -453.17259\n",
            "protocols.antibody.design.AntibodyDesignMover:\n",
            "protocols.antibody.design.AntibodyDesignMover: Native Pose (Total): 13400.9\n",
            "protocols.antibody.design.AntibodyDesignMover: Final Pose (Total): 1956.22\n",
            "protocols.antibody.design.AntibodyDesignMover: \n",
            "------------------------------------------------------------\n",
            " Scores                       Weight   Raw Score Wghtd.Score\n",
            "------------------------------------------------------------\n",
            " fa_atr                       1.000   -5548.607   -5548.607\n",
            " fa_rep                       0.550    4367.284    2402.006\n",
            " fa_sol                       1.000    3396.838    3396.838\n",
            " fa_intra_rep                 0.005    2814.476      14.072\n",
            " fa_intra_sol_xover4          1.000     175.134     175.134\n",
            " lk_ball_wtd                  1.000     -67.745     -67.745\n",
            " fa_elec                      1.000   -1577.392   -1577.392\n",
            " pro_close                    1.250       7.395       9.244\n",
            " hbond_sr_bb                  1.000     -73.182     -73.182\n",
            " hbond_lr_bb                  1.000    -505.473    -505.473\n",
            " hbond_bb_sc                  1.000    -144.432    -144.432\n",
            " hbond_sc                     1.000     -72.838     -72.838\n",
            " dslf_fa13                    1.250      -0.876      -1.095\n",
            " atom_pair_constraint         0.010       0.000       0.000\n",
            " omega                        0.400     470.264     188.106\n",
            " fa_dun                       0.700    4944.281    3460.997\n",
            " p_aa_pp                      0.600    -279.463    -167.678\n",
            " yhh_planarity                0.625       0.051       0.032\n",
            " ref                          1.000     374.883     374.883\n",
            " rama_prepro                  0.450     207.452      93.353\n",
            "---------------------------------------------------\n",
            " Total weighted score:                     1956.224\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H1\n",
            "protocols.antibody.cluster.CDRClusterMatcher: Length: 13 Omega: TTTTTTTTTTTTT\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H2\n",
            "protocols.antibody.cluster.CDRClusterMatcher: Length: 10 Omega: TTTTTTTTTT\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for H3\n",
            "protocols.antibody.cluster.CDRClusterMatcher: Length: 13 Omega: TTTTTTTTTTTTT\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L1\n",
            "protocols.antibody.cluster.CDRClusterMatcher: Length: 10 Omega: TTTTTTTTTT\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L2\n",
            "protocols.antibody.cluster.CDRClusterMatcher: Length: 8 Omega: TTTTTTTT\n",
            "antibody.AntibodyInfo: Setting up CDR Cluster for L3\n",
            "protocols.antibody.cluster.CDRClusterMatcher: Length: 7 Omega: TTTTTTT\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using explicit constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: Using interface constructor\n",
            "protocols.analysis.InterfaceAnalyzerMover: [ WARNING ] In InterfaceAnalyzerMover, calculator Sasa_1 already exists, this is hopefully correct for your purposes\n",
            "protocols.analysis.InterfaceAnalyzerMover: [ WARNING ] In InterfaceAnalyzerMover, calculator InterfaceNeighborDefinition_1 already exists, this is hopefully correct for your purposes\n",
            "protocols.analysis.InterfaceAnalyzerMover: [ WARNING ] In InterfaceAnalyzerMover, calculator InterfaceDeltaEnergetics_1 already exists, this is hopefully correct for your purposes\n",
            "protocols.analysis.InterfaceAnalyzerMover: [ WARNING ] In InterfaceAnalyzerMover, calculator NumberHBonds_1 already exists, this is hopefully correct for your purposes\n",
            "protocols.analysis.InterfaceAnalyzerMover: [ WARNING ] In InterfaceAnalyzerMover, calculator BuriedUnsatisfiedPolars_1 already exists, this is hopefully correct for your purposes\n",
            "protocols.analysis.InterfaceAnalyzerMover: Making an interface set with fixed chains\n",
            "protocols.analysis.InterfaceAnalyzerMover: Interface set residues total: 89\n",
            "protocols.analysis.InterfaceAnalyzerMover: NOT detecting disulfides in the separated pose.\n",
            "protocols.analysis.InterfaceAnalyzerMover: Calculating dSASA\n",
            "protocols.analysis.InterfaceAnalyzerMover: Calculating per-res dSASA data\n",
            "core.pack.task: Packer task: initialize from command line()\n",
            "core.pack.pack_rotamers: built 2501 rotamers at 89 positions.\n",
            "core.pack.interaction_graph.interaction_graph_factory: Instantiating DensePDInteractionGraph\n",
            "protocols.analysis.InterfaceAnalyzerMover: included_nres: 972\n",
            "protocols.analysis.InterfaceAnalyzerMover: Making hbond ScoreFunction with bb-bb energy method options\n",
            "core.scoring.ScoreFunctionFactory: SCOREFUNCTION: ref2015\n",
            "protocols.analysis.InterfaceAnalyzerMover: Found Hbond between chains: 9 and 13\n",
            "protocols.analysis.InterfaceAnalyzerMover: Found Hbond between chains: 13 and 9\n",
            "protocols.analysis.InterfaceAnalyzerMover: Found Hbond between chains: 9 and 12\n",
            "protocols.analysis.InterfaceAnalyzerMover: Found Hbond between chains: 12 and 9\n",
            "protocols.analysis.InterfaceAnalyzerMover: Computing delta unsat polar residues...\n",
            "protocols.analysis.InterfaceAnalyzerMover: Computing Shape Complementarity Score...\n",
            "protocols.analysis.InterfaceAnalyzerMover: Upstream chain(s) numbers: 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11,\n",
            "protocols.analysis.InterfaceAnalyzerMover: Downstream chain(s) numbers: 12, 13,\n"
          ]
        }
      ],
      "source": [
        "if not os.getenv(\"DEBUG\"):\n",
        "    ### BEGIN SOLUTION\n",
        "\n",
        "    pose = original_pose.clone()\n",
        "    rabd = XmlObjects.static_get_mover('<AntibodyDesignMover name=\"RAbD\" graft_design_cdrs=\"L1,L3\" seq_design_cdrs=\"L1,L3\" mc_optimize_dG=\"1\" light_chain=\"kappa\"/>')\n",
        "    rabd.apply(pose)\n",
        "\n",
        "    ### END SOLUTION"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "qJhIgXTX2LM0"
      },
      "source": [
        "Load the scorefile with nstruct=5 from `expected_outputs/rabd/tutB1_score.sc`\n",
        "\n",
        "Compare this data to tutorial A2.  Are the interface energies better?  Has the Shape Complementarity improved (sc score) improved?"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "DseDWc7E2LM5"
      },
      "source": [
        "### Tut B2. Optimizing Interface Energy and Total Score (opt-dG and opt-E)\n",
        "\n",
        "Here, we want to set the protocol to optimize the interface energy during Monte Carlo, but we want to add some total energy to the weight.  Because the overall numbers of total energy will dominate the overall numbers, we only add a small weight for total energy.  This has not been fully benchmarked, but if your models have very bad total energy when using opt-dG - consider using it.  (178 sec for 1 nstruct)\n",
        "\n",
        "\n",
        "    <AntibodyDesignMover name=\"RAbD\" seq_design_cdrs=\"L1,L3\" graft_design_cdrs=\"L1,L3\" mc_optimize_dG=\"1\" mc_total_weight=\".01 mc_interface_weight=\".99 light_chain=\"kappa\"/>\n",
        "\n"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "nbgrader": {
          "grade": true,
          "grade_id": "cell-119248d958256181",
          "locked": false,
          "points": 0,
          "schema_version": 3,
          "solution": true
        },
        "id": "0_wYRnXB2LM5"
      },
      "outputs": [],
      "source": [
        "if not os.getenv(\"DEBUG\"):\n",
        "    ### BEGIN SOLUTION\n",
        "    pose = original_pose.clone()\n",
        "    rabd = XmlObjects.static_get_mover('<AntibodyDesignMover name=\"RAbD\" seq_design_cdrs=\"L1,L3\" graft_design_cdrs=\"L1,L3\" mc_optimize_dG=\"1\" mc_total_weight=\".01 mc_interface_weight=\".99 light_chain=\"kappa\"/>')\n",
        "    rabd.apply(pose)\n",
        "    ### END SOLUTION"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "8yR_usHD2LM5"
      },
      "source": [
        "Use the scorefile from an nstruct=5 run to compare total energies (`total_score`) of this run vs the one right before it located at `expected_outputs/rabd/tutB2_score.sc`.  Are the total scores better?"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "tk9vKUyH2LM5"
      },
      "source": [
        "## Tutorial C: Towards DeNovo Design: Integrated Dock/Design\n",
        "\n",
        "This tutorial takes a long time to run as docking is fairly slow - even with the optimizations that are part of RAbD.  PLEASE USE THE PREGENERATED OUTPUT.  The top 10 designs from each tutorial and associated scorefiles of a 1000 nstruct cluster run are in the output directory.  Note that we are starting these tutorials with a pre-relaxed structure in order to get more reliable rosetta energies.  Since we are running a large nstruct, we will escape the local energy well that this leads us into.\n",
        "\n",
        "### Tut C1. RosettaDocking\n",
        "\n",
        "In this example, we use integrated RosettaDock (with sequence design during the high-res step) to sample the antibody-antigen orientation, but we don't care where the antibody binds to the antigen.  Just that it binds. IE - No Constraints. The RAbD protocol always has at least Paratope SiteConstraints enabled to make sure any docking is contained to the paratope (like most good docking programs).\n",
        "\n",
        "This takes too long to run, so PLEASE USE THE OUTPUT GENERATED FOR YOU. We will use opt-dG here and for these tutorials, we will be sequence-designing all cdrs to begin to create a better interface.  Note that sequence design happens whereever packing occurs - Including during high-resolution docking.\n",
        "\n",
        "\t<AntibodyDesignMover name=\"RAbD\" mc_optimize_dG=\"1\" do_dock=\"1\" seq_design_cdrs=\"L1,L2,L3,H1,H2,H3\" graft_design_cdrs=\"L1,L2,L3,H1,H2\" light_chain=\"kappa\"/>\n",
        "\n",
        "**Use pymol to load the files, and load tutC1_score from the expected_outputs directory as a pandas dataframe.**\n",
        "\n",
        "\t\tpymol my_ab.pdb expected_outputs/rabd/top10_C1/*\n",
        "\t\t@color_cdrs.pml\n",
        "\t\tcenter full_epitope\n",
        "\n",
        "Where is the antibody in the resulting designs?  Are the interfaces restricted to the Paratope?  Has the epitope moved relative to the starting interface?\n",
        "\n"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "jWHFZWoO2LM5"
      },
      "source": [
        "### Tut C2. Auto Epitope Constraints\n",
        "\n",
        "Allow Dock-Design, incorporating auto-generated SiteConstraints to keep the antibody around the starting interface residues.  These residues are determined by being within 6A to the CDR residues (This interface distance can be customized).  Again, these results are provided for you.\n",
        "\n",
        "\n",
        "\t<AntibodyDesignMover name=\"RAbD\" mc_optimize_dG=\"1\" do_dock=\"1\" use_epitope_csts=\"1\"\n",
        "\t   seq_design_cdrs=\"L1,L2,L3,H1,H2,H3\" graft_design_cdrs=\"L1,L2,L3,H1,H2\" light_chain=\"kappa\"/>\n",
        "\n",
        "**Use pymol to load the files and checkout the scores in `expected_outputs/rabd/tutC2_score.sc` as before.**\n",
        "\n",
        "\n",
        "\tpymol my_ab.pdb expected_outputs/rabd/top10_C2/*\n",
        "\t@color_cdrs.pml\n",
        "\tcenter full_epitope\n",
        "\n",
        "How do these compare with with the previous tutorial?  Are the antibodies closer to the starting interface?  Are the scores better?\n",
        "\n"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "EKoUs5DY2LM5"
      },
      "source": [
        "### Tut C3. Specific Residue Epitope Constraints\n",
        "\n",
        "Allow Dock-Design, as above, but specify the Epitope Residues and Paratope CDRs to guide the dock/design to have these interact.\n",
        "\n",
        "For now, we are more focused on the light chain. We could do this as a two-stage process, where we first optimize positioning and CDRs of the light chain and then the heavy chain or simply add heavy chain CDRs to the paratope CDRs option.\n",
        "\n",
        "\n",
        "\t<AntibodyDesignMover name=\"RAbD\" mc_optimize_dG=\"1\" do_dock=\"1\" use_epitope_csts=\"1\"\n",
        "\t\tepitope_residues=\"38J,52J,34K,37K\" paratope_cdrs=\"L1,L3\"\n",
        "\t\tseq_design_cdrs=\"L1,L2,L3,H1,H2,H3\" graft_design_cdrs=\"L1,L2,L3,H1,H2\" light_chain=\"kappa\"/>\n",
        "\n",
        "**Again, load these into Pymol and take a look at the scorefile in a dataframe.**  \n",
        "\n",
        "\tpymol my_ab.pdb expected_outputs/rabd/top10_C3/*\n",
        "\t@color_cdrs.pml\n",
        "\tcenter full_epitope\n",
        "\n",
        "Now that we have specified where we want the interface to be and are additionally designing more CDRS, how do the enegies compare?  Are we starting to get a decent interface with the lowest energy structure?\n",
        "\n",
        "How do these compare with the previous runs?  "
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "TleL2LXZ2LM6"
      },
      "source": [
        "## Tutorial D: Advanced Settings and CDR Instruction File Customization\n",
        "\n",
        "Once again, all output files are in `expected_outputs`.  Please use these if you want - as many of these take around 10 minutes to run.\n",
        "\n",
        "### Tut D1.  CDR Instruction File\n",
        "\n",
        "More complicated design runs can be created by using the Antibody Design Instruction file.  This file allows complete customization of the design run. See below for a review of the syntax of the file and possible customization.  An instruction file is provided where we use conservative design on L1 and graft in L1, H2, and H1 CDRs at a longer length to attempt to create a larger interface area.  More info on instruction file syntax can be found at the end of this tutorial. (150 seconds on my laptop for nstruc 1)\n",
        "\n",
        "\tcp ../inputs/my_instruction_file.txt .\n",
        "\tcp ../inputs/default_cdr_instructions.txt .\n",
        "\n",
        "\n",
        "Take a look at the default CDR instructions.  These are loaded by default into Rosetta.  There is syntax examples at the end of the file. Run the XML or attempt to use it in-code.\n",
        "\n",
        "\t<AntibodyDesignMover name=\"RAbD\" instruction_file=\"my_instruction_file.txt\"\n",
        "\t   seq_design_cdrs=\"L1,L3,H1,H2,H3\" graft_design_cdrs=\"L1,H2,H1\" random_start=\"1\" light_chain=\"kappa\"/>\n",
        "\n"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "nbgrader": {
          "grade": true,
          "grade_id": "cell-41c91bb1703d613c",
          "locked": false,
          "points": 0,
          "schema_version": 3,
          "solution": true
        },
        "id": "GqGzptts2LM6"
      },
      "outputs": [],
      "source": [
        "if not os.getenv(\"DEBUG\"):\n",
        "    ### BEGIN SOLUTION\n",
        "\n",
        "    pose = original_pose.clone()\n",
        "    rabd = XmlObjects.static_get_mover('<AntibodyDesignMover name=\"RAbD\" instruction_file=\"my_instruction_file.txt\" seq_design_cdrs=\"L1,L3,H1,H2,H3\" graft_design_cdrs=\"L1,H2,H1\" random_start=\"1\" light_chain=\"kappa\"/>')\n",
        "    rabd.apply(pose)\n",
        "\n",
        "    ### END SOLUTION"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "vDyMZbKK2LM6"
      },
      "source": [
        "### Tut D2.  Dissallow AAs and the Resfile\n",
        "\n",
        "Here, we will disallow ANY sequence design into Proline residues and Cysteine residues, while giving a resfile to further LIMIT design and packing as specific positions. These can be given as 3 or 1 letter codes and mixed codes such as PRO and C are accepted. Note that the resfile does NOT turn any residues ON, it is simply used to optionally LIMIT design residue types and design and packing positions.\n",
        "    \n",
        "Resfile syntax can be found here: [https://www.rosettacommons.org/docs/wiki/rosetta_basics/file_types/resfiles] Note that specifying a resfile and dissalowing aa are only currently available as cmd-line options that are read by RAbD.\n",
        "\n",
        "Runtime is less than a minute for nstruct 1.\n",
        "\n",
        "\tcp ../inputs/rabd/my_resfile.resfile .\n",
        "\n",
        "Take a look at the resfile.  Can you describe what it is we are doing with it?\n",
        "Unfortunately, at the moment, resfile setting is only available as a cmd-line option that needs to be set in the `init()` function as `-resfile my_resfile.resfile`\n",
        "\n",
        "\t<AntibodyDesignMover name=\"RAbD\" seq_design_cdrs=\"L1,L3,H1,H2,H3\" light_chain=\"kappa\"/>\n",
        "\n",
        "\n"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "C1_wuo_K2LM6"
      },
      "source": [
        "### Tut D3. Mintype\n",
        "\n",
        "Here, we will change the mintype to relax.  This mintype enables Flexible-Backbone design as we have seen in previous workshops.  Our default is to use min/pack cycles, but relax typically works better.  However, it also takes considerably more time!  This tutorial takes about 339 seconds for one struct!\n",
        "\n",
        "\n",
        "\t<AntibodyDesignMover name=\"RAbD\" seq_design_cdrs=\"L1,L3,H1,H3\" mintype=\"relax light_chain=\"kappa\"/>"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "nbgrader": {
          "grade": true,
          "grade_id": "cell-a228ebfe14b610a7",
          "locked": false,
          "points": 0,
          "schema_version": 3,
          "solution": true
        },
        "id": "R3n6VrYw2LM6"
      },
      "outputs": [],
      "source": [
        "if not os.getenv(\"DEBUG\"):\n",
        "    ### BEGIN SOLUTION\n",
        "\n",
        "    pose = original_pose.clone()\n",
        "    rabd = XmlObjects.static_get_mover('<AntibodyDesignMover name=\"RAbD\" seq_design_cdrs=\"L1,L3,H1,H3\" mintype=\"relax\" light_chain=\"kappa\"/>')\n",
        "    rabd.apply(pose)\n",
        "\n",
        "    ### END SOLUTION"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "A_49L5js2LM6"
      },
      "source": [
        "### Tut D4. Framework Design\n",
        "\n",
        "Finally, we want to allow the framework residues AROUND the CDRs we will be designing and any interacting antigen residues to design as well here. In addition, we will disable conservative framework design as we want something funky (this is not typically recommended and is used here to indicate what you CAN do.  Note that we will also design the interface of the antigen using the `-design_antigen` option.  This can be useful for vaccine design. Note that these design options are cmd-line ony options currently (but will be available in a later version of Rosetta). Approx 900 second runtime.\n",
        "\n",
        "\n",
        "\tantibody_designer.linuxgccrelease -s my_ab.pdb -seq_design_cdrs L1 L3 H1 H3 \\\n",
        "\t\t\t    -light_chain kappa -resfile my_resfile.resfile -disallow_aa PRO CYS \\\n",
        "\t\t\t    -mintype relax -design_antigen -design_framework \\\n",
        "\t\t\t    -conservative_framework_design false -nstruct 1 -out:prefix tutD4_\n",
        "\n",
        "\t<AntibodyDesignMover name=\"RAbD\" seq_design_cdrs=\"L1,L3,H1,H3\" mintype=\"relax\" light_chain=\"kappa\"/>"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "k8HXgLf32LM6"
      },
      "source": [
        "### Tut D5.  H3 Stem, kT, and Sequence variablility.\n",
        "\n",
        "Finally, we want increased variability for our sequence designs.  So, we will increase number of sampling rounds for our lovely cluster profiles using the `-seq_design_profile_samples` option, increase kT, and allow H3 stem design.  \n",
        "\n",
        "We will enable H3 Stem design here, which can cause a flipping of the H3 stem type from bulged to non-bulged and vice-versa.  Typically, if you do this, you may want to run loop modeling on the top designs to confirm the H3 structure remains in-tact. Note that once again, these sequence-design specific options must be set on the  cmd-line.  \n",
        "\n",
        "Description of the `seq_design_profile_samples` option (default 1): \"If designing using profiles, this is the number of times the profile is sampled each time packing done.  Increase this number to increase variability of designs - especially if not using relax as the mintype.\"\n",
        "\n",
        "This tutorial takes approx 450 seconds.\n",
        "\n",
        "\tantibody_designer.linuxgccrelease -s my_ab.pdb -seq_design_cdrs L1 L2 H3 \\\n",
        "\t\t\t    -graft_design_cdrs L1 L2 -light_chain kappa -design_H3_stem -inner_kt 2.0 \\\n",
        "\t\t\t    -outer_kt 2.0 -seq_design_profile_samples 5 -nstruct 5 -out:prefix tutD5_\n",
        "\n",
        "\t<AntibodyDesignMover name=\"RAbD\" seq_design_cdrs=\"L1,L3,H1,H3\" mintype=\"relax\"\n",
        "\t\t\t                                                        inner_kt=\"2.0\" outer_kt=\"2.0\"/>\n",
        "\n",
        "How different is the sequence of L1,L2, and H3 from our starting antibody?"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "bTgLLWUS2LM6"
      },
      "source": [
        "__You should now be ready to explore and use RosettaAntibodyDesign on your own.  Congrats!  Thanks for going through this tutorial!__\n",
        "\n",
        "The full reference manual can be found here: https://www.rosettacommons.org/docs/latest/application_documentation/antibody/RosettaAntibodyDesign#antibody-design-cdr-instruction-file"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "3NIvQwqX2LM6"
      },
      "source": [
        "<!--NAVIGATION-->\n",
        "< [Side Chain Conformations and Dunbrack Energies](http://nbviewer.jupyter.org/github/RosettaCommons/PyRosetta.notebooks/blob/master/notebooks/06.01-Side-Chain-Conformations-and-Dunbrack-Energies.ipynb) | [Contents](toc.ipynb) | [Index](index.ipynb) | [Protein Design with a Resfile and FastRelax](http://nbviewer.jupyter.org/github/RosettaCommons/PyRosetta.notebooks/blob/master/notebooks/06.03-Design-with-a-resfile-and-relax.ipynb) ><p><a href=\"https://colab.research.google.com/github/RosettaCommons/PyRosetta.notebooks/blob/master/notebooks/06.02-Packing-design-and-regional-relax.ipynb\"><img align=\"left\" src=\"https://colab.research.google.com/assets/colab-badge.svg\" alt=\"Open in Colab\" title=\"Open in Google Colaboratory\"></a>"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "NpdMNxYw2LM7"
      },
      "outputs": [],
      "source": []
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "H1nDoT7z2LM7"
      },
      "source": [
        "<!--NAVIGATION-->\n",
        "< [RosettaAntibody Framework](http://nbviewer.jupyter.org/github/RosettaCommons/PyRosetta.notebooks/blob/master/notebooks/12.01-RosettaAntibody-Framework-and-SimpleMetrics.ipynb) | [Contents](toc.ipynb) | [Index](index.ipynb) | [RosettaCarbohydrates](http://nbviewer.jupyter.org/github/RosettaCommons/PyRosetta.notebooks/blob/master/notebooks/13.00-RosettaCarbohydrates-Working-with-Glycans.ipynb) ><p><a href=\"https://colab.research.google.com/github/RosettaCommons/PyRosetta.notebooks/blob/master/notebooks/12.02-RosettaAntibodyDesign-RAbD.ipynb\"><img align=\"left\" src=\"https://colab.research.google.com/assets/colab-badge.svg\" alt=\"Open in Colab\" title=\"Open in Google Colaboratory\"></a>"
      ]
    }
  ],
  "metadata": {
    "celltoolbar": "Create Assignment",
    "kernelspec": {
      "display_name": "Python 3",
      "language": "python",
      "name": "python3"
    },
    "language_info": {
      "codemirror_mode": {
        "name": "ipython",
        "version": 3
      },
      "file_extension": ".py",
      "mimetype": "text/x-python",
      "name": "python",
      "nbconvert_exporter": "python",
      "pygments_lexer": "ipython3",
      "version": "3.6.1"
    },
    "toc": {
      "base_numbering": 1,
      "nav_menu": {},
      "number_sections": true,
      "sideBar": true,
      "skip_h1_title": false,
      "title_cell": "Table of Contents",
      "title_sidebar": "Contents",
      "toc_cell": false,
      "toc_position": {},
      "toc_section_display": true,
      "toc_window_display": false
    },
    "colab": {
      "provenance": [],
      "include_colab_link": true
    }
  },
  "nbformat": 4,
  "nbformat_minor": 0
}